<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÁöÑ‰∏äÊ∏∏ÂàÜÊûêÊµÅÁ®ã„ÄÇ'>
<title>ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊµÅÁ®ãÔºàMetagenomic workflowÔºâ202408ÔΩúÊåÅÁª≠Êõ¥Êñ∞</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/metagenomic-workflow/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊµÅÁ®ãÔºàMetagenomic workflowÔºâ202408ÔΩúÊåÅÁª≠Êõ¥Êñ∞'>
<meta property='og:description' content='ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÁöÑ‰∏äÊ∏∏ÂàÜÊûêÊµÅÁ®ã„ÄÇ'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/metagenomic-workflow/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='microbiome' /><meta property='article:published_time' content='2024-05-28T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-05-28T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/metagenomic-workflow/images/workflow.png' />
<meta name="twitter:title" content="ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊµÅÁ®ãÔºàMetagenomic workflowÔºâ202408ÔΩúÊåÅÁª≠Êõ¥Êñ∞">
<meta name="twitter:description" content="ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÁöÑ‰∏äÊ∏∏ÂàÜÊûêÊµÅÁ®ã„ÄÇ"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/metagenomic-workflow/images/workflow.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">üêõüê∂</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/472862038'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/JB-Cao'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >‰∏≠Êñá</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#logs">Logs</a></li>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#preprocess">preprocess</a>
      <ol>
        <li><a href="#Ë¥®Êéßfastp">Ë¥®ÊéßÔºöfastp</a></li>
        <li><a href="#ÂéªÂÆø‰∏ª">ÂéªÂÆø‰∏ª</a>
          <ol>
            <li><a href="#bowtie2">bowtie2</a></li>
            <li><a href="#bwa-mem2">bwa-mem2</a></li>
          </ol>
        </li>
        <li><a href="#fastq‰ø°ÊÅØÁªüËÆ°">fastq‰ø°ÊÅØÁªüËÆ°</a></li>
      </ol>
    </li>
    <li><a href="#reads-based">reads-based</a>
      <ol>
        <li><a href="#Áâ©ÁßçÊ≥®Èáäkraken2">Áâ©ÁßçÊ≥®ÈáäÔºökraken2</a>
          <ol>
            <li><a href="#krakenËæìÂá∫Ê†ºÂºè">KrakenËæìÂá∫Ê†ºÂºè</a></li>
          </ol>
        </li>
        <li><a href="#Áâ©ÁßçÂäüËÉΩhumann">Áâ©Áßç+ÂäüËÉΩÔºöHUMAnN</a></li>
      </ol>
    </li>
    <li><a href="#contigs-based">contigs-based</a>
      <ol>
        <li><a href="#ÁªÑË£Öassembly">ÁªÑË£ÖÔºàassemblyÔºâ</a>
          <ol>
            <li><a href="#megahit">megahit</a></li>
            <li><a href="#metaspades">metaSPAdes</a></li>
          </ol>
        </li>
        <li><a href="#ÁªÑË£ÖËØÑ‰º∞quast">ÁªÑË£ÖËØÑ‰º∞ÔºöQUAST</a></li>
        <li><a href="#Âü∫Âõ†È¢ÑÊµãprodigal">Âü∫Âõ†È¢ÑÊµãÔºöProdigal</a></li>
        <li><a href="#ÂéªÂÜó‰Ωô">ÂéªÂÜó‰Ωô</a>
          <ol>
            <li><a href="#cd-hit">CD-HIT</a></li>
            <li><a href="#mmseqs2">Mmseqs2</a></li>
          </ol>
        </li>
        <li><a href="#fasta‰ø°ÊÅØÁªüËÆ°">fasta‰ø°ÊÅØÁªüËÆ°</a></li>
        <li><a href="#Âü∫Âõ†ÂÆöÈáèsalmon">Âü∫Âõ†ÂÆöÈáèÔºösalmon</a></li>
        <li><a href="#ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä">ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä</a>
          <ol>
            <li><a href="#eggnog-cogkeggcazy">eggNOG (COG/KEGG/CAZy)</a></li>
            <li><a href="#dbcan2Á¢≥Ê∞¥ÂåñÂêàÁâ©">dbCAN2ÔºàÁ¢≥Ê∞¥ÂåñÂêàÁâ©Ôºâ</a></li>
            <li><a href="#cardÊäóÁîüÁ¥†ÊäóÊÄßÂü∫Âõ†args">CARDÔºàÊäóÁîüÁ¥†ÊäóÊÄßÂü∫Âõ†ARGsÔºâ</a></li>
            <li><a href="#vfdbÊØíÂäõÂõ†Â≠ê">VFDBÔºàÊØíÂäõÂõ†Â≠êÔºâ</a></li>
            <li><a href="#ÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì">ÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì</a></li>
          </ol>
        </li>
        <li><a href="#ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂">ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂</a></li>
      </ol>
    </li>
    <li><a href="#ÂàÜÁÆ±binning">ÂàÜÁÆ±ÔºàbinningÔºâ</a></li>
    <li><a href="#reference">Reference</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/metagenomic-workflow/">
                <img src="/Cao-blog/p/metagenomic-workflow/images/workflow_hu_9d0727389e8f6507.png"
                        srcset="/Cao-blog/p/metagenomic-workflow/images/workflow_hu_9d0727389e8f6507.png 800w, /Cao-blog/p/metagenomic-workflow/images/workflow_hu_e31f75347f7b8ef2.png 1600w"
                        width="800" 
                        height="465" 
                        loading="lazy"
                        alt="Featured image of post ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊµÅÁ®ãÔºàMetagenomic workflowÔºâ202408ÔΩúÊåÅÁª≠Êõ¥Êñ∞" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/metagenomic-workflow/">ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊµÅÁ®ãÔºàMetagenomic workflowÔºâ202408ÔΩúÊåÅÁª≠Êõ¥Êñ∞</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÁöÑ‰∏äÊ∏∏ÂàÜÊûêÊµÅÁ®ã„ÄÇ
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">May 28, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    15 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                ÁâàÊùÉÂΩíCao's blogÊâÄÊúâÔºåËΩ¨ËΩΩËØ∑Ê≥®ÊòéÂá∫Â§Ñ
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="logs">Logs</h2>
<ul>
<li>Â¢ûÂä†‰ΩøÁî®bwa-mem2ÂéªÂÆø‰∏ªÔºåÂèØ‰ª•ÂéªÂæóÊõ¥Âπ≤ÂáÄ„ÄÇÈÄÇ‰∫éÂÆø‰∏ªÂ∫èÂàóÂæàÈ´òÔºåÂÆ≥ÊÄïÂÅáÈò≥ÊÄßÁªìÊûúÁöÑÊÉÖÂÜµ„ÄÇÔºà2024.07.15Ôºâ</li>
<li>Êõ¥Êñ∞ÂÖ®ÊñáÁªÜËäÇÔºà2024.05.29Ôºâ</li>
<li>Â¢ûÂä†Mmseqs2Áî®‰∫éÂéªÂÜó‰ΩôÔºåÂü∫Âõ†ËÅöÁ±ªÁöÑÈÄüÂ∫¶ÈùûÂ∏∏Âø´Ôºå‰∏îÈöèÂ∫èÂàóÈáèÁ∫øÊÄßÂ¢ûÈïøÔºà2024.03.12Ôºâ</li>
<li>Â¢ûÂä†RÂåÖ<code>pctax</code>ÂÜÖÁöÑ‰∏Ä‰∫õÂ∏ÆÂä©‰∏äÊ∏∏ÂàÜÊûêÁöÑÂ∞èËÑöÊú¨Ôºà2024.03.03Ôºâ</li>
</ul>
<p><strong>Ê≥®ÊÑèÔºö</strong></p>
<ul>
<li>
<p>ÁØáÂπÖÊúâÈôêÔºåÊú¨Êñá‰∏ÄËà¨‰∏ç‰ªãÁªçÂÖ∑‰ΩìÁöÑËΩØ‰ª∂ÂÆâË£ÖÈÖçÁΩÆÔºåÊï∞ÊçÆÂ∫ì‰∏ãËΩΩÁ≠âÔºå‰πü‰∏çÊèê‰æõÁ§∫‰æãÊï∞ÊçÆÔºå‰∏ªË¶ÅÂ∏ÆÂä©Âø´ÈÄü‰∫ÜËß£Êï¥‰∏™ÂÆèÂü∫Âõ†ÁªÑÁöÑ<strong>‰∏äÊ∏∏ÂàÜÊûê</strong>ÊµÅÁ®ã„ÄÇÊâÄ‰ª•ËΩØ‰ª∂ÂÆâË£ÖËØ∑ÂèÇËÄÉËΩØ‰ª∂Ëá™Ë∫´‰ªãÁªçÔºåËøêË°åÊó∂ÈÅáÂà∞ÂÖ∑ÈóÆÈ¢ò‰πüÊúÄÂ•ΩÂÖàGoogleÊàñËÄÖÂéªgithub issueÈáåÈù¢ÈóÆÈóÆ„ÄÇ</p>
</li>
<li>
<p>ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊäÄÊúØËøòÊòØÂú®‰∏çÂÅúÁöÑÊõ¥Êñ∞Ëø≠‰ª£ÔºåÂæàÂ§öÊõ¥Êñ∞Êõ¥Â•ΩÁî®ÁöÑËΩØ‰ª∂Âú®‰∏çÊñ≠Êé®Âá∫ÔºåÊâÄ‰ª•ÊàëÂú®Ëøô‰πü‰ºöÊåÅÁª≠Êõ¥Êñ∞ÔºàÂ¶ÇÊûúÊàëËøò‰∏ÄÁõ¥ÂÅöËøô‰∏™ÊñπÂêëÁöÑËØùÔºâ„ÄÇËøôÈáå‰ªãÁªçÁöÑ‰πüÂè™ÊòØÊØîËæÉÂü∫Êú¨ÁöÑÂàÜÊûêÊµÅÁ®ãÔºå‰ΩÜÊòØÊéåÊè°‰∫Ü‰πãÂêéÔºåËá™Â∑±Ëøõ‰∏ÄÊ≠•ÂéªÂÅö‰∏™ÊÄßÂåñÂàÜÊûê‰πü‰ºöÈ°∫ÊâãÂæàÂ§ö„ÄÇ</p>
</li>
<li>
<p>ÂÆåÊàê‰∏äÊ∏∏ÂàÜÊûêÂêéÊàë‰ª¨ÂèØ‰ª•ÂæóÂà∞ÂêÑÁßçÁâ©ÁßçÊàñËÄÖÂäüËÉΩÁöÑprofileÔºåÂêéÁª≠‰∏ÄËà¨Áî®pythonÔºåRËøõË°åÊï∞ÊçÆÂàÜÊûêÂíåÂèØËßÜÂåñÔºåÂèØÂèÇËÄÉÊàëÁöÑÂÖ∂‰ªñÂçöÊñá„ÄÇ</p>
</li>
<li>
<p>ÁªùÂ§ßÂ§öÊï∞ËøôÈáå‰ªãÁªçÁöÑËΩØ‰ª∂ÈÉΩÊòØ‰ªÖÊîØÊåÅlinuxÂπ≥Âè∞ÁöÑÔºåÊàë‰ª¨ÂÅöÊµãÂ∫èÊñá‰ª∂‰∏äÊ∏∏ÂàÜÊûê‰πüËÇØÂÆöÊòØÂú®ÊúçÂä°Âô®‰∏äÂÅöÔºå‰∏™‰∫∫PC‰∏ÄËà¨ÂæàÈöæÊª°Ë∂≥ÈúÄÊ±ÇÔºåÊâÄ‰ª•Âú®ÂÅöËøô‰∫õÂàÜÊûêÂâçÂøÖÈ°ªÂÖàÂ≠¶‰π†linuxÂü∫Á°ÄÁü•ËØÜÂ¶ÇÊñá‰ª∂Á≥ªÁªüÔºåshellËÑöÊú¨ÁºñÂÜôÔºåËΩØ‰ª∂ÂÆâË£ÖÁ≠â„ÄÇÂÆâË£ÖËΩØ‰ª∂Âª∫ËÆÆ‰ΩøÁî®condaÊàñmambaÔºàÊñ∞Âª∫ÁéØÂ¢ÉÂíåÁÆ°ÁêÜÔºâÔºåÊúâÂæàÂ§öÂèÇËÄÉÊñπÊ≥ï„ÄÇ</p>
</li>
<li>
<p>Êàë‰ΩøÁî®ÁöÑÈõÜÁæ§ÊòØslurm‰Ωú‰∏öÁÆ°ÁêÜÁ≥ªÁªüÔºå‰∏ãÈù¢ÁöÑÁ§∫‰æãËÑöÊú¨‰πüÊòØÈÄÇÁî®‰∏éslurmÁöÑÔºåÂ∞ΩÈáèÂÖàÂ≠¶‰π†‰∏Ä‰∏ã<a class="link" href="../slurm" >slurmÁöÑ‰ΩøÁî®</a>ÂÜçÂ∞ùËØïÊèê‰∫§‰Ωú‰∏ö„ÄÇÂ¶ÇÊûú‰∏çÁî®slurmÁöÑËØùÂèØ‰ª•Âè™ÂèÇËÄÉ<code>#############</code>‰∏≠Èó¥ÁöÑËΩØ‰ª∂ÈÉ®ÂàÜ„ÄÇ</p>
</li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇ</p>
<p>ÂÆèÂü∫Âõ†ÁªÑÁöÑÂ∫îÁî®ÈùûÂ∏∏ÂπøÊ≥õÔºåÂåÖÊã¨Ôºö</p>
<ul>
<li>ÁîüÁâ©Â§öÊ†∑ÊÄßÁ†îÁ©∂ÔºöÈÄöËøáÂØπÂÆèÂü∫Âõ†ÁªÑËøõË°åÂàÜÊûêÔºåÂèØ‰ª•‰∫ÜËß£‰∏çÂêåÁîüÊÄÅÁ≥ªÁªü‰∏≠ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄßÂíåÂàÜÂ∏ÉÊÉÖÂÜµ„ÄÇ</li>
<li>ÁîüÊÄÅÂ≠¶Á†îÁ©∂ÔºöÂÆèÂü∫Âõ†ÁªÑÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©Âú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÂäüËÉΩ„ÄÅ‰∫í‰ΩúÂÖ≥Á≥ªÂíåÁîüÊÄÅ‰ΩçÁ≠â„ÄÇ</li>
<li>ÁîüÁâ©ÊäÄÊúØÔºöÂÆèÂü∫Âõ†ÁªÑÂèØ‰ª•Áî®‰∫éÁ≠õÈÄâÂÖ∑ÊúâÁâπÂÆöÂäüËÉΩÁöÑÂæÆÁîüÁâ©Ôºå‰æãÂ¶ÇÔºåÂØªÊâæËÉΩÂ§üÈôçËß£ÊúâÂÆ≥Áâ©Ë¥®ÁöÑÂæÆÁîüÁâ©„ÄÇ</li>
</ul>
<p>ÂÆèÂü∫Âõ†ÁªÑÁöÑÂàÜÊûê‰∏ÄËà¨ÂåÖÊã¨‰ª•‰∏ãÊ≠•È™§Ôºö</p>
<ol>
<li>DNAÊèêÂèñ‰∏éÂª∫Â∫ì„ÄÇ</li>
<li>È´òÈÄöÈáèÊµãÂ∫èÔºö‰ΩøÁî®È´òÈÄöÈáèÊµãÂ∫èÊäÄÊúØÂØπÊâ©Â¢ûÂêéÁöÑDNAËøõË°åÊµãÂ∫èÔºåÂæóÂà∞ÂéüÂßãÂ∫èÂàóÊï∞ÊçÆ„ÄÇ</li>
<li>Êï∞ÊçÆÊ∏ÖÊ¥óÂíåÁªÑË£ÖÔºöÂØπÂéüÂßãÊï∞ÊçÆËøõË°åË¥®ÈáèÊéßÂà∂„ÄÅÂéªÈô§‰ΩéË¥®ÈáèÂ∫èÂàóÂíåÂÜó‰ΩôÂ∫èÂàóÔºåÂ∞ÜÂ∫èÂàóÊãºÊé•ÊàêËæÉÈïøÁöÑËøûÁª≠Â∫èÂàóÔºàcontigsÔºâ„ÄÇ</li>
<li>Âü∫Âõ†Ê≥®ÈáäÔºöÂ∞Ücontigs‰∏≠ÁöÑÂü∫Âõ†ËøõË°åÊ≥®ÈáäÔºåÂæóÂà∞Âü∫Âõ†ÂäüËÉΩ‰ø°ÊÅØ„ÄÇ</li>
<li>Êï∞ÊçÆÂàÜÊûêÔºö‰∫ÜËß£ÂæÆÁîüÁâ©Â§öÊ†∑ÊÄß„ÄÅÁæ§ËêΩÁªìÊûÑ„ÄÅÂäüËÉΩÁâπÂæÅÁ≠â‰ø°ÊÅØÔºàÊõ¥Â§öÊòØÊåáËé∑Âèñ‰∫ÜÁâ©Áßç‰∏∞Â∫¶Ë°®ÊàñÂäüËÉΩ‰∏∞Â∫¶Ë°®‰πãÂêéÁöÑËøõ‰∏ÄÊ≠•ÂàÜÊûêÔºâ„ÄÇ</li>
<li>MAGs binningÔºå ËøõÂåñÂä®ÊÄÅÁ≠âËøõ‰∏ÄÊ≠•ÂàÜÊûê</li>
</ol>
<p>‰∏ãÂõæÊòØÊàëÂ∏∏Áî®ÁöÑ‰∏ÄÂ•ó‰∏äÊ∏∏Âü∫Êú¨ÊµÅÁ®ãÔºåÂΩìÁÑ∂Âú®Èù¢ÂØπ‰∏çÂêåÈ°πÁõÆÊó∂Â∫îËØ•Êúâ‰∏çÂêåÁöÑ‰æßÈáçÁÇπÂíåÈÄÇÁî®ÁöÑÂàÜÊûêÊñπÊ≥ïÔºåÂèØ‰ª•Âú®Ê≠§Âü∫Á°Ä‰∏äÊ∑ªÂä†Êàñ‰øÆÊîπ„ÄÇ</p>
<p>ÊúÄÊó©ÁöÑÊó∂ÂÄôÔºåËøôÊñπÈù¢ÁöÑÂàÜÊûêÊàëÈÉΩÊòØÂèÇËÄÉÂàòÊ∞∏Èë´ËÄÅÂ∏àÁöÑ<a class="link" href="https://github.com/YongxinLiu/EasyMetagenome"  target="_blank" rel="noopener"
    ><strong>EasyMetagenome</strong></a>ÔºåÁé∞Âú®ËøôÂ•óÊµÅÁ®ã‰πüÂèëÊñáÁ´†‰∫Ü (<a class="link" href="#ref-liuPracticalGuideAmplicon2021" ><em>1</em></a>)ÔºåÂÄºÂæóÂèÇËÄÉÔºåÂØπ‰∏äÊâã16SÊµãÂ∫èÊï∞ÊçÆÊàñÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆÈÉΩÂæàÊúâÂ∏ÆÂä©„ÄÇ</p>
<img src="images/workflow.png" title=""/>
<p>ÊúÄËøëÔºà2024.3.3ÔºâÊàëÊï¥ÁêÜ‰∫Ü‰∏Ä‰∏ãÊµÅÁ®ãÊîæÂú®ÊàëÁöÑRÂåÖ<code>pctax</code>Èáå‰∫ÜÔºåÊñπ‰æøÊï¥‰∏™ÂÆèÂü∫Âõ†ÁªÑÁöÑ‰∏ä‰∏ãÊ∏∏ÊµÅÁ®ãÁªìÂêàÔºö</p>
<img src="images/pctax_workflow.png" title=""/>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;pctax&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#‰ΩøÁî®micro_sbatchÂèØ‰ª•Âø´ÈÄüËé∑ÂæóÊØè‰∏ÄÊ≠•ÁöÑslurmËÑöÊú¨Ôºö</span>
</span></span><span class="line"><span class="cl"><span class="n">pctax</span><span class="o">::</span><span class="nf">micro_sbatch</span><span class="p">(</span><span class="n">work_dir</span> <span class="o">=</span> <span class="s">&#34;~/work/&#34;</span><span class="p">,</span><span class="n">step</span> <span class="o">=</span> <span class="s">&#34;fastp&#34;</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="preprocess">preprocess</h2>
<p>‰∏ÄËà¨ÊääÊâÄÊúâÊ†∑Êú¨ÁöÑÊµãÂ∫èÂèåÁ´ØÊñá‰ª∂ÊîæÂú®‰∏Ä‰∏™dataÊñá‰ª∂Â§π‰∏ãÔºåÁÑ∂ÂêéÊääÊâÄÊúâÊ†∑Êú¨ÂêçÂÜôÂÖ•Âà∞‰∏Ä‰∏™<code>samplelist</code>Êñá‰ª∂‰∏≠ÔºåÊñπ‰æøÂêéÁª≠ÂêÑÁßçËΩØ‰ª∂ÁöÑÊâπÈáèÊìç‰Ωú„ÄÇ</p>
<h3 id="Ë¥®Êéßfastp">Ë¥®ÊéßÔºöfastp</h3>
<p>ÊµãÂ∫èÊñá‰ª∂Ë¥®ÊéßÊòØÊåáÂú®‰∏ãÊ∏∏ÂàÜÊûêÂâçÂØπÊµãÂ∫èÊï∞ÊçÆËøõË°åË¥®ÈáèÊéßÂà∂ÂíåÊ∏ÖÁêÜÔºå‰ª•Á°Æ‰øùÊï∞ÊçÆÁöÑÂáÜÁ°ÆÊÄßÂíåÂèØÈù†ÊÄß„ÄÇ</p>
<p>fastpÊòØ‰∏Ä‰∏™ÈùûÂ∏∏Âø´ÈÄü„ÄÅÂ§öÂäüËÉΩÁöÑÊµãÂ∫èÊï∞ÊçÆË¥®ÊéßÂ∑•ÂÖ∑ÔºåÊîØÊåÅËá™Âä®ÂåñÁöÑË¥®ÊéßÂ§ÑÁêÜÔºåÂåÖÊã¨ÂéªÈô§‰ΩéË¥®ÈáèÂ∫èÂàó„ÄÅÂâ™ÂàáÊé•Â§¥Â∫èÂàóÂíåÁîüÊàêË¥®ÊéßÊä•Âëä„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=fastp</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p data/fastp
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/fastp -w <span class="m">8</span> -i data/rawdata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_f1.fastq.gz -o data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_1.fq.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -I data/rawdata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_r2.fastq.gz -O data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_2.fq.gz -j data/fastp/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.json
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>fastpÁöÑjsonÊñá‰ª∂‰∏≠ÁªüËÆ°‰∫ÜÊµãÂ∫èÊï∞ÊçÆÁöÑÂü∫Êú¨ÊåáÊ†áÔºåÊàë‰ª¨ÂèØ‰ª•Â∞ÜÂÖ∂Êï¥ÁêÜ‰∏∫Ë°®Ê†ºÔºö
ÊääÊâÄÊúâÁöÑ.jsonÊñá‰ª∂ÁßªÂà∞‰∏Ä‰∏™Êñá‰ª∂Â§πÈáåreport/‰∏ãÔºå‰ΩøÁî®ÊàëÂÜôÁöÑRÂåÖ<code>pctax</code>ËØªÂèñjsonÊñá‰ª∂Âπ∂Êï¥ÁêÜÊàêË°®Ê†ºÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="n">pctax</span><span class="o">::</span><span class="nf">pre_fastp</span><span class="p">(</span><span class="s">&#34;report/&#34;</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>Âè¶Â§ñÔºåFastQCÔºåCutadaptÔºåTrimmomaticÁ≠â‰πüÊòØÂ∏∏Áî®ÁöÑÊµãÂ∫èË¥®ÊéßÂ∑•ÂÖ∑„ÄÇ</p>
<h3 id="ÂéªÂÆø‰∏ª">ÂéªÂÆø‰∏ª</h3>
<p>ÂéªÂÆø‰∏ªÁöÑËøáÁ®ãÂÖ∂ÂÆûÂ∞±ÊòØÂ∞ÜÂ∫èÂàóÊØîÂØπÂà∞ÂÆø‰∏ªÂü∫Âõ†ÁªÑ‰∏äÔºåÁÑ∂ÂêéÊ≤°ÊúâÊØîÂØπÂà∞ÁöÑÂ∫èÂàóÊï¥ÂêàÊàêÊñ∞Êñá‰ª∂Â∞±ÊòØÂéªÂÆø‰∏ªÂêéÁöÑ‰∫Ü„ÄÇ</p>
<h4 id="bowtie2">bowtie2</h4>
<p>ÂÆø‰∏ªÂü∫Âõ†ÁªÑÈúÄË¶ÅËá™Â∑±ÂÖà‰∏ãËΩΩÂ•ΩÂπ∂Áî® bowtie2-build Âª∫Á´ãÁ¥¢ÂºïÔºå‰ª•‰∫∫Á±ª‰∏∫‰æãÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rm_human</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p data/rm_human
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/bowtie2 -p <span class="m">8</span> -x ~/db/humangenome/hg38 -1 data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_1.fq.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_2.fq.gz -S data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sam <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --un-conc data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.fq --very-sensitive
</span></span><span class="line"><span class="cl">  rm data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sam
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÊçÆËØ¥bowtie2ÁöÑÊïèÊÑüÊÄßÊØîËæÉ‰ΩéÔºåÊâÄ‰ª•Êàë‰ª¨ËøôÈáåÂ∑≤ÁªèÊâìÂºÄ‰∫Ü<code>--very-sensitive</code>Ê®°Âºè„ÄÇ
ÊïèÊÑüÂ∫¶‰∏çË∂≥‰ºöÂØºËá¥Êúâ‰∫õÂèØËÉΩÊòØ‰∫∫Á±ªÂ∫èÂàó‰ΩÜ‰∏ç‰ºöÊØîÂØπÂà∞genome‰∏äËÄåÂØºËá¥ÁªìÊûúËøòÊúâ‰∫∫Á±ªÂ∫èÂàóÔºà<a class="link" href="https://mp.weixin.qq.com/s/Qxj0PvVn49ynrD5jVxucTw"  target="_blank" rel="noopener"
    >https://mp.weixin.qq.com/s/Qxj0PvVn49ynrD5jVxucTw</a>ÔºâÔºåÂèØ‰ª•Áî®‰ΩøÁî®bwa-memÔºåÊàñËÄÖÂú®ÂêéÁª≠kraken‰∏≠‰ΩøÁî®Â∏¶Êúâ‰∫∫Á±ªgenomeÁöÑÊï∞ÊçÆÂ∫ìËøõ‰∏ÄÊ≠•ÂéªÈô§ÂÆø‰∏ª„ÄÇ</p>
<h4 id="bwa-mem2">bwa-mem2</h4>
<p>bwa-mem2ÁöÑÊïèÊÑüÂ∫¶ËæÉÈ´òÔºà<a class="link" href="https://hc1023.github.io/2020/11/03/extract-paired-mapping-reads-and-flagstat/"  target="_blank" rel="noopener"
    >https://hc1023.github.io/2020/11/03/extract-paired-mapping-reads-and-flagstat/</a>Ôºâ„ÄÇ</p>
<p>ÊàëÊãø‰∫Ü‰∏Ä‰∏™Ê†∑Êú¨ËØï‰∫Ü‰∏ãÔºåÂéüÂßãreadsÊúâ10,260,663Êù°Ôºåbowtie2ËøáÊª§Êúâ9,441,462Êù°Ôºåbwa-memËøáÊª§ÊéâÂè™Êúâ6,520,171Êù°Ôºåbwa-memÊòØÂÆÅÂèØÈîôÊùÄ1000Ôºå‰∏çÊîæËøá‰∏Ä‰∏™„ÄÇ</p>
<p>ÂÆø‰∏ªÂü∫Âõ†ÁªÑÈúÄË¶ÅËá™Â∑±ÂÖà‰∏ãËΩΩÂ•ΩÂπ∂Áî® bwa-mem2 index Âª∫Á´ãÁ¥¢ÂºïÔºå‰ª•‰∫∫Á±ª‰∏∫‰æãÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span><span class="lnt">50
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rm_human</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># bwa-mem2 index ~/db/humangenome/hg38.fa #70GÂÜÖÂ≠òÔºå1300s</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p data/rm_human
</span></span><span class="line"><span class="cl">  bwa-mem2 mem -t <span class="m">8</span> ~/db/humangenome/hg38.fa data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_1.fq.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    data/fastp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_2.fq.gz &gt; data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_aligned_reads.sam <span class="c1"># 20GÂÜÖÂ≠òÔºå8Ê†∏Ôºå419s</span>
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/samtools view -Sb data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_aligned_reads.sam &gt; data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_reads.bam
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/samtools sort data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_reads.bam data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_sorted_reads
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/samtools view -b -f <span class="m">4</span> data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_sorted_reads.bam &gt; data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_non_host_reads.bam
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/bedtools bamtofastq -i data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_non_host_reads.bam -fq data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq -fq2 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  rm data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_aligned_reads.sam
</span></span><span class="line"><span class="cl">  rm data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_reads.bam
</span></span><span class="line"><span class="cl">  rm data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_sorted_reads.bam
</span></span><span class="line"><span class="cl">  rm data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_non_host_reads.bam
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="fastq‰ø°ÊÅØÁªüËÆ°">fastq‰ø°ÊÅØÁªüËÆ°</h3>
<p>ÂèØ‰ª•Áî®‰∏Ä‰∫õÂ∞èÂ∑•ÂÖ∑Â¶ÇFastqCountÔºåseqkitÁ≠âÊù•ÁªüËÆ°readsÊï∞ÔºåÁ¢±Âü∫Êï∞ÔºåGCÔºåQ20ÔºåQ30Á≠âÊåáÊ†áÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">~/biosoft/FastqCount-master/FastqCount_v0.5 xx.fastq.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">Total Reads     Total Bases     N Bases Q20     Q30     GC
</span></span><span class="line"><span class="cl"><span class="m">11568822</span> <span class="o">(</span>11.57 M<span class="o">)</span>      <span class="m">1702829127</span> <span class="o">(</span>1.70 G<span class="o">)</span>     0.00%   98.00%  94.00%  54.00%
</span></span></code></pre></td></tr></table>
</div>
</div><p>ËøòÂèØ‰ª•ÊääcleandataÂÜçË∑ë‰∏Ä‰∏ãfastpüòÇÔºå‰ΩÜÊòØÂè™ÁúãÂ§ÑÁêÜÂâçÁöÑÁªüËÆ°ÊåáÊ†á„ÄÇÂõ†‰∏∫fastpÁ°ÆÂÆûÈùûÂ∏∏Âø´ÔºåÁ∫ØÁ≤πÁî®Êù•ÁªüËÆ°‰πüË°å„ÄÇ</p>
<h2 id="reads-based">reads-based</h2>
<p>reads-basedÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊòØÊåáÁõ¥Êé•Âà©Áî®ÊµãÂ∫èËØªÈïøÔºàreadsÔºâËøõË°åÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆÂàÜÊûêÁöÑÊñπÊ≥ïÔºåËÄå‰∏çÊòØÂÖàËøõË°åÂü∫Âõ†ÁªÑÁªÑË£Ö„ÄÇËØ•ÊñπÊ≥ïÈÄöËøáÂØπÁü≠ËØªÈïøËøõË°åÊØîÂØπÂíåÊ≥®ÈáäÔºå‰ªé‰∏≠ÊèêÂèñÂäüËÉΩ‰ø°ÊÅØÂíåÁâ©ÁßçÁªÑÊàêÔºåÂ∏∏Áî®‰∫éÂø´ÈÄüËØÑ‰º∞ÁéØÂ¢ÉÊ†∑Êú¨‰∏≠ÁöÑÂæÆÁîüÁâ©Áæ§ËêΩÁªìÊûÑÂíåÂäüËÉΩÊΩúÂäõ„ÄÇ</p>
<ul>
<li><strong>‰ºòÁÇπ</strong>Ôºö
<ul>
<li><strong>Âø´ÈÄü</strong>ÔºöÊó†ÈúÄÁªÑË£ÖÔºåÂ§ÑÁêÜÈÄüÂ∫¶ËæÉÂø´„ÄÇ</li>
<li><strong>ÂÖ®Èù¢</strong>ÔºöËÉΩÂ§üÊ£ÄÊµãÂà∞‰Ωé‰∏∞Â∫¶ÁöÑÂæÆÁîüÁâ©ÂíåÂäüËÉΩÂü∫Âõ†ÔºåÈÄÇÂêàÂ§ßËßÑÊ®°Ê†∑Êú¨ÂàÜÊûê„ÄÇ</li>
</ul>
</li>
<li><strong>Â±ÄÈôêÊÄß</strong>Ôºö
<ul>
<li><strong>ÁâáÊÆµÂåñ</strong>ÔºöÁî±‰∫é‰∏çËøõË°åÁªÑË£ÖÔºåÂàÜÊûêÂü∫‰∫éÁü≠ËØªÈïøÔºåÂèØËÉΩ‰ºöÈîôËøáÈïøË∑ùÁ¶ªÁöÑÂü∫Âõ†ËÅîÁ≥ªÂíåÁªìÊûÑ‰ø°ÊÅØ„ÄÇ</li>
<li><strong>‰æùËµñÊï∞ÊçÆÂ∫ì</strong>ÔºöÂàÜÊûêÁªìÊûú‰æùËµñ‰∫éÂèÇËÄÉÊï∞ÊçÆÂ∫ìÁöÑÂÖ®Èù¢ÊÄßÂíåÂáÜÁ°ÆÊÄß„ÄÇ</li>
</ul>
</li>
</ul>
<h3 id="Áâ©ÁßçÊ≥®Èáäkraken2">Áâ©ÁßçÊ≥®ÈáäÔºökraken2</h3>
<p>Kraken2ÊòØ‰∏Ä‰∏™Áî®‰∫éÂØπÈ´òÈÄöÈáèÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ªÂíåÊ†áËØÜÁâ©ÁßçÁöÑËΩØ‰ª∂„ÄÇÂÆÉ‰ΩøÁî®ÂèÇËÄÉÊï∞ÊçÆÂ∫ì‰∏≠ÁöÑÂü∫Âõ†ÁªÑÂ∫èÂàóÊù•ËøõË°åÂàÜÁ±ªÔºåÂπ∂‰ΩøÁî®k-merÊñπÊ≥ïÊù•ÂÆûÁé∞Âø´ÈÄüÂíåÂáÜÁ°ÆÁöÑÂàÜÁ±ª„ÄÇ</p>
<p>‰ΩøÁî®Kraken2ËøõË°åÂü∫Êú¨ÂàÜÁ±ªÁöÑÁÆÄÂçïÊ≠•È™§Ôºö</p>
<ol>
<li>ÂáÜÂ§áÂèÇËÄÉÊï∞ÊçÆÂ∫ìÔºöKraken2ÈúÄË¶Å‰∏Ä‰∏™ÂèÇËÄÉÊï∞ÊçÆÂ∫ìÔºå‰ª•‰æøÂØπÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ª„ÄÇÂèØ‰ª•Áõ¥Êé•‰∏ãËΩΩÂÆòÊñπÊûÑÂª∫ÁöÑÊ†áÂáÜÂ∫ìÔºå‰πüÂèØ‰ª•‰ªéNCBI„ÄÅEnsemblÊàñÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì‰∏ãËΩΩÁõ∏Â∫îÁöÑÂü∫Âõ†ÁªÑÂ∫èÂàóÔºåÂπ∂‰ΩøÁî®Kraken2ÂÜÖÁΩÆÁöÑÂ∑•ÂÖ∑Êù•ÊûÑÂª∫Êï∞ÊçÆÂ∫ì„ÄÇ</li>
</ol>
<p><code>kraken2-build --standard --threads 24 --db ./</code></p>
<p><code>--standard</code>Ê†áÂáÜÊ®°Âºè‰∏ãÂè™‰∏ãËΩΩ5ÁßçÊï∞ÊçÆÂ∫ìÔºöÂè§Ëèåarchaea„ÄÅÁªÜËèåbacteria„ÄÅ‰∫∫Á±ªhuman„ÄÅËΩΩ‰ΩìUniVec_Core„ÄÅÁóÖÊØíviral„ÄÇ</p>
<ol start="2">
<li>
<p>ÂÆâË£ÖKraken2ÔºöÂèØ‰ª•‰ªéKraken2ÂÆòÊñπÁΩëÁ´ô‰∏ãËΩΩÂπ∂ÂÆâË£ÖKraken2ËΩØ‰ª∂„ÄÇ</p>
</li>
<li>
<p>ËøêË°åKraken2Ôºö‰ΩøÁî®Kraken2ÂØπÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ªÈúÄË¶Å‰ΩøÁî®‰ª•‰∏ãÂëΩ‰ª§Ôºö</p>
</li>
</ol>
<p><code>kraken2 --db &lt;path_to_database&gt; &lt;input_file&gt; --output &lt;output_file&gt;</code></p>
<p>ËøôÈáåÔºå<code>&lt;path_to_database&gt;</code>ÊòØÂèÇËÄÉÊï∞ÊçÆÂ∫ìÁöÑË∑ØÂæÑÔºå<code>&lt;input_file&gt;</code>ÊòØÈúÄË¶ÅËøõË°åÂàÜÁ±ªÁöÑËæìÂÖ•Êñá‰ª∂Ôºå<code>&lt;output_file&gt;</code>ÊòØËæìÂá∫Êñá‰ª∂ÁöÑÂêçÁß∞„ÄÇKraken2Â∞ÜËæìÂá∫‰∏Ä‰∏™ÂàÜÁ±ªÊä•ÂëäÊñá‰ª∂Âíå‰∏Ä‰∏™Â∫èÂàóÊñá‰ª∂„ÄÇ</p>
<p>ÈúÄË¶ÅÊ≥®ÊÑèÁöÑÊòØkrakenËøêË°åËá≥Â∞ëË¶ÅÊèê‰æõÊï∞ÊçÆÂ∫ìÂ§ßÂ∞èÁöÑÂÜÖÂ≠òÂ§ßÂ∞èÔºàËøêË°åÂÜÖÂ≠òÔºâÔºåÂõ†‰∏∫ÂÆÉ‰ºöÊääÊï¥‰∏™Êï∞ÊçÆÂ∫ìËΩΩÂÖ•ÂÜÖÂ≠òÂêéËøõË°åÂ∫èÂàóÁöÑÊ≥®ÈáäÔºåÊâÄ‰ª•Â¶ÇÊûúÂèëÁé∞Êó†Ê≥ïËΩΩÂÖ•Êï∞ÊçÆÂ∫ìÁöÑÊä•ÈîôÔºåÂèØ‰ª•Â∞ùËØïË∞ÉÂ§ßÂÜÖÂ≠òËµÑÊ∫ê„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=kraken2</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=40G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/kraken2
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/kraken2 --threads <span class="m">8</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --db ~/db/kraken2/stand_krakenDB <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --confidence 0.05 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --output result/kraken2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.output <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --report result/kraken2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.kreport <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --paired <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>Âè¶Â§ñÔºåkrakenÊï∞ÊçÆÂ∫ìÊòØÂèØ‰ª•Ëá™Â∑±ÊûÑÂª∫ÁöÑÔºåÊâÄ‰ª•ÈÄÇÁî®‰∫éÂêÑÁßçÈ°πÁõÆÁöÑÁâ©ÁßçÊ≥®ÈáäÔºåÊàëÂÅöÁöÑÊØîËæÉÂ§öÁöÑÊòØÁéØÂ¢ÉÊ†∑Êú¨ÁöÑÂÆèÂü∫Âõ†ÁªÑÔºåÂ∞±ÂèØËÉΩÈúÄË¶ÅÊõ¥ÂÖ®Èù¢ÁöÑÁâ©ÁßçÊï∞ÊçÆÂ∫ìÔºàÁîöËá≥Èô§‰∫ÜÂêÑÁßçÂæÆÁîüÁâ©ÔºåËøòË¶ÅÂä®Ê§çÁâ©Êï∞ÊçÆÁ≠âÔºâÔºåÊàë‰ª¨ÂÆûÈ™åÂÆ§ÁöÑWXÂ∏àÂßêÊî∂ÈõÜÊûÑÂª∫‰∫Ü‰∏Ä‰∏™Ë∂ÖÂ§ßÁöÑÁâ©ÁßçÂ∫ì„ÄÇ</p>
<p>krakenËΩØ‰ª∂ËøêË°åÊó∂ËΩΩÂÖ•Êï∞ÊçÆÂ∫ìÊòØ‰∏Ä‰∏™ÂçÅÂàÜËÄóÊó∂ÁöÑÊ≠•È™§ÔºåËÄåÊØèÊù°Â∫èÂàóÁöÑÈâ¥ÂÆöÊó∂Èó¥Â∑Æ‰∏çÂ§öÔºåÊâÄ‰ª•Êàë‰ª¨ÂèØ‰ª•Â∞ÜÂæàÂ§öÊ†∑Êú¨ÁöÑfastqÊñá‰ª∂ÂêàÂπ∂Êàê‰∏Ä‰∏™Â§ßÊñá‰ª∂ÂêéËæìÂÖ•krakenÊ≥®ÈáäÔºå‰πãÂêéÂÜçÊåâÁÖßÂ∫èÂàóÁöÑÊï∞ÈáèÊãÜÂàÜÁªìÊûúÊñá‰ª∂ÔºåËøôÊ†∑Â§ö‰∏™Ê†∑Êú¨‰πüÂè™ÈúÄË¶ÅËΩΩÂÖ•‰∏ÄÊ¨°Êï∞ÊçÆÂ∫ìÔºåËäÇÁúÅÊó∂Èó¥Ôºå‰ª•‰∏ãÊòØÁî®Ëá™ÂÆö‰πâÁöÑkraken2M.pyËÑöÊú¨ÂÆûÁé∞„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=kraken2M</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --time=14-00:00:00</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=mem</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=100G</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mkdir -p result/kraken
</span></span><span class="line"><span class="cl">  python /share/home/jianglab/shared/krakenDB/K2ols/kraken2M.py -t <span class="m">16</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -i data/rm_human <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -c 0.05 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -s .1.fq,.2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o result/kraken <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -d /share/home/jianglab/shared/krakenDB/mydb2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -k ~/miniconda3/envs/waste/bin/kraken2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -kt /share/home/jianglab/shared/krakenDB/K2ols/KrakenTools
</span></span><span class="line"><span class="cl">  mkdir -p result/kraken/kreport
</span></span><span class="line"><span class="cl">  mv result/kraken/*_kreport.txt result/kraken/kreport/
</span></span><span class="line"><span class="cl">  python ~/db/script/format_kreports.py -i result/kraken/kreport <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -kt /share/home/jianglab/shared/krakenDB/K2ols/KrakenTools --save-name2taxid
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="krakenËæìÂá∫Ê†ºÂºè">KrakenËæìÂá∫Ê†ºÂºè</h4>
<p>Ê†áÂáÜËæìÂá∫Ê†ºÂºèoutputÊñá‰ª∂(‰∫îÂàóË°®):</p>
<ul>
<li>C/U‰ª£Ë°®ÂàÜÁ±ªclassifiedÊàñÈùûÂàÜÁ±ªunclassifed</li>
<li>Â∫èÂàóID</li>
<li>Áâ©ÁßçÊ≥®Èáä</li>
<li>ÊØîÂ∫èÂàóÊ≥®ÈáäÁöÑÂå∫ÂüüÔºåÂ¶Ç98|94‰ª£Ë°®Â∑¶Á´Ø98bpÔºåÂè≥Á´Ø94bpÊØîÂØπËá≥Êï∞ÊçÆÂ∫ì</li>
<li>LCAÊØîÂØπÁªìÊûúÔºåÂ¶Ç‚Äù562:13 561:4‚Äù‰ª£Ë°®13 k-merÊØîÂØπËá≥Áâ©Áßç#562Ôºå4 k-merÊØîÂØπËá≥#561Áâ©Áßç</li>
</ul>
<p>Êä•ÂëäËæìÂá∫Ê†ºÂºèreportÊñá‰ª∂(ÂåÖÊã¨6ÂàóÔºåÊñπ‰æøÊï¥ÁêÜ‰∏ãÊ∏∏ÂàÜÊûê):</p>
<ol>
<li>ÁôæÂàÜÊØî</li>
<li>count</li>
<li>countÊúÄ‰ºò</li>
<li>(U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. ‚ÄúG2‚Äù‰ª£Ë°®‰Ωç‰∫éÂ±û‰∏ÄÁßçÈó¥</li>
<li>NCBIÁâ©ÁßçID</li>
<li>ÁßëÂ≠¶Áâ©ÁßçÂêç</li>
</ol>
<p>Â∏∏Áî®ÁöÑÁâ©Áßç‰∏∞Â∫¶Ë°®Ê†ºÂºèÈô§‰∫Ükraken reportÔºåËøòÊúâmpaÔºåspfÔºåkronaÁ≠âÊ†ºÂºèÔºåÂÖ≥‰∫ékrakenÁªìÊûúÁöÑÊï¥ÁêÜ‰ª•ÂèäÊ†ºÂºèËΩ¨Êç¢ÊñπÂºèÔºåÊúâ‰∏Ä‰∫õÁé∞ÊàêÁöÑËÑöÊú¨ÊàñËÄÖËá™Â∑±ÂÜô„ÄÇ</p>
<p><a class="link" href="https://ccb.jhu.edu/software/krakentools/"  target="_blank" rel="noopener"
    >KrakenTools (jhu.edu)</a> Â∞±ÊòØ‰∏ÄÂ•óÂæàÂ•ΩÁî®ÁöÑkrakenÂ∑•ÂÖ∑ÂåÖÔºåÂÖ∂‰∏≠Â∏∏Áî®ÁöÑÊúâÔºö</p>
<ol>
<li>
<p>extract_kraken_reads.py
Ê≠§Á®ãÂ∫èÊèêÂèñËØªÂèñÂú®‰ªª‰ΩïÁî®Êà∑ÊåáÂÆöÁöÑÂàÜÁ±ªidÂ§ÑÂàÜÁ±ªÁöÑÂÜÖÂÆπ„ÄÇÁî®Êà∑ÂøÖÈ°ªÊåáÂÆöKrakenËæìÂá∫Êñá‰ª∂„ÄÅÂ∫èÂàóÊñá‰ª∂ÂíåËá≥Â∞ë‰∏Ä‰∏™ÂàÜÁ±ªÊ≥ïID„ÄÇ‰∏ãÈù¢ÊåáÂÆö‰∫ÜÂÖ∂‰ªñÈÄâÈ°π„ÄÇÊà™Ëá≥2021Âπ¥4Êúà19Êó•ÔºåÊ≠§ËÑöÊú¨‰∏éKrakenUniq/Kraken2UniqÊä•ÂëäÂÖºÂÆπ„ÄÇ</p>
</li>
<li>
<p>combine_kreports.py
Ëøô‰∏™ËÑöÊú¨Â∞ÜÂ§ö‰∏™KrakenÊä•ÂëäÂêàÂπ∂Âà∞‰∏Ä‰∏™ÂêàÂπ∂ÁöÑÊä•ÂëäÊñá‰ª∂‰∏≠„ÄÇ</p>
</li>
<li>
<p>kreport2krona.py
Ëøô‰∏™Á®ãÂ∫èÈúÄË¶Å‰∏Ä‰∏™KrakenÊä•ÂëäÊñá‰ª∂ÔºåÂπ∂ÊâìÂç∞Âá∫‰∏Ä‰∏™kronaÂÖºÂÆπÁöÑÊñáÊú¨Êñá‰ª∂ÔºåÊç¢ÊàêkronaÊñá‰ª∂Â•ΩÁîªÂõæ„ÄÇ</p>
</li>
</ol>
<p>ËΩ¨Êç¢ÂêéÁöÑkrona‰ΩøÁî® <code>ktImportText MYSAMPLE.krona -o MYSAMPLE.krona.html</code>ÂæóÂà∞ÁΩëÈ°µÂèØËßÜÂåñÁªìÊûú„ÄÇ</p>
<ol start="4">
<li>
<p>kreport2mpa.py
Ëøô‰∏™Á®ãÂ∫èÈúÄË¶Å‰∏Ä‰∏™KrakenÊä•ÂëäÊñá‰ª∂ÔºåÂπ∂ÊâìÂç∞Âá∫‰∏Ä‰∏™mpa (MetaPhlAn)È£éÊ†ºÁöÑÊñáÊú¨Êñá‰ª∂„ÄÇ</p>
</li>
<li>
<p>combine_mpa.py
Ëøô‰∏™Á®ãÂ∫èÂêàÂπ∂Êù•Ëá™kreport2mpa.pyÁöÑÂ§ö‰∏™ËæìÂá∫„ÄÇ</p>
</li>
</ol>
<h3 id="Áâ©ÁßçÂäüËÉΩhumann">Áâ©Áßç+ÂäüËÉΩÔºöHUMAnN</h3>
<p>HUMAnNÔºàThe HMP Unified Metabolic Analysis NetworkÔºâÊòØ‰∏ÄÊ¨æÁî®‰∫éÂàÜÊûê‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑÁöÑÂäüËÉΩÂíå‰ª£Ë∞¢ËÉΩÂäõÁöÑÂ∑•ÂÖ∑„ÄÇ</p>
<p>ÂÆÉÈÄöËøáÂ∞ÜÂÆèÂü∫Âõ†ÁªÑÂ∫èÂàó‰∏éÂèÇËÄÉÂü∫Âõ†ÁªÑÊï∞ÊçÆÂ∫ìÊØîÂØπÔºåÂà©Áî®MetaCyc‰ª£Ë∞¢ÈÄöË∑ØÊï∞ÊçÆÂ∫ìÂíåUniRefËõãÁôΩË¥®Â∫èÂàóÊï∞ÊçÆÂ∫ìÔºåÂàÜÊûêÂæÆÁîüÁâ©ÁªÑÂú®ÂäüËÉΩÂíå‰ª£Ë∞¢ÈÄöË∑ØÊ∞¥Âπ≥‰∏äÁöÑÁªÑÊàêÂíåÊ¥ªÊÄß„ÄÇHUMAnN2ËøòÊèê‰æõ‰∫ÜÂ§öÊ†∑ÊÄßÂàÜÊûê„ÄÅÂÖ≥ËÅîÂàÜÊûêÂíåÂèØËßÜÂåñÂ∑•ÂÖ∑ÔºåÂèØÁî®‰∫éÊ∑±ÂÖ•Á†îÁ©∂‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑÂØπÂÆø‰∏ªÂÅ•Â∫∑ÁöÑÂΩ±ÂìçÂíåÊ≤ªÁñóÁ≠ñÁï•ÁöÑÂà∂ÂÆöÁ≠âÊñπÈù¢„ÄÇ</p>
<p>HUMAnNÊòØÁî±ÁæéÂõΩÂõΩÂÆ∂‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑËÆ°ÂàíÔºàHMPÔºâÂºÄÂèëÁöÑÔºåÁõÆÂâçÊúÄÊñ∞ÁâàÊú¨‰∏∫<a class="link" href="https://github.com/biobakery/humann"  target="_blank" rel="noopener"
    >HUMAnN3</a>Ôºå‰∫é2020Âπ¥ÂèëÂ∏É„ÄÇ‰∏éHUMAnN2Áõ∏ÊØîÔºåHUMAnN3ÊîπËøõ‰∫ÜÂü∫Âõ†ÂÆ∂ÊóèÊ≥®ÈáäÁöÑÊñπÊ≥ïÔºåÊèêÈ´ò‰∫ÜÊ≥®ÈáäÁ≤æÂ∫¶ÂíåÈÄüÂ∫¶ÔºåÂπ∂Êèê‰æõ‰∫ÜÊñ∞ÁöÑÂäüËÉΩÂíåÂ∑•ÂÖ∑ÔºåÂ¶ÇÂäüËÉΩÈüßÂ∫¶ÂàÜÊûê„ÄÅ‰ª£Ë∞¢ÊåáÁ∫πËØÜÂà´ÂíåÂ§öÊ†∑ÊÄßÂàÜÊûêÁ≠â„ÄÇ</p>
<p>‰ΩÜÊòØHUMAnNÁöÑÊï∞ÊçÆÂ∫ìÂü∫Êú¨ÈÉΩÊòØ‰∏é‰∫∫Áõ∏ÂÖ≥ÁöÑÂæÆÁîüÁâ©ÔºåÊØîËæÉÈÄÇÂêàÂÅöÂêÑÁßç‰∫∫‰ΩìÂæÆÁîüÁâ©ÁªÑÔºàËÇ†ÈÅìÔºåËÇ∫ÈÉ®ÔºåÂè£ËÖîÔºåÁöÆËÇ§Á≠âÁ≠âÔºâÔºåÂØπ‰∫éÁéØÂ¢ÉÊ†∑Êú¨ÂèØËÉΩunclassifiedÊØîËæÉÂ§ö„ÄÇ</p>
<p><strong>HUMAnNË¶ÅÊ±ÇÂèåÁ´ØÂ∫èÂàóÂêàÂπ∂ÁöÑÊñá‰ª∂‰Ωú‰∏∫ËæìÂÖ•</strong>ÔºåforÂæ™ÁéØÊ†πÊçÆÂÆûÈ™åËÆæËÆ°Ê†∑Êú¨ÂêçÊâπÈáèÂèåÁ´ØÂ∫èÂàóÂêàÂπ∂„ÄÇ</p>
<ul>
<li><strong>Áâ©ÁßçÁªÑÊàêË∞ÉÁî®MetaPhlAn3, bowtie2ÊØîÂØπËá≥Ê†∏ÈÖ∏Â∫èÂàóÂ∫ì</strong>ÔºåËß£ÂÜ≥ÊúâÂì™‰∫õÂæÆÁîüÁâ©Â≠òÂú®ÁöÑÈóÆÈ¢òÔºõ</li>
<li><strong>ÂäüËÉΩÁªÑÊàê‰∏∫humann3Ë∞ÉÁî®diamondÊØîÂØπËá≥ËõãÁôΩÂ∫ì</strong>ÔºåËß£ÂÜ≥Ëøô‰∫õÂæÆÁîüÁâ©ÂèÇ‰∏éÂì™‰∫õÂäüËÉΩÈÄöË∑ØÁöÑÈóÆÈ¢òÔºõ</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> data/rm_human
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="sb">`</span>cat ~/work/asthma/samplelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$i</span>
</span></span><span class="line"><span class="cl">cat <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_f1.fastq <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_r2.fastq &gt;<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_paired.fastq
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span><span class="lnt">50
</span><span class="lnt">51
</span><span class="lnt">52
</span><span class="lnt">53
</span><span class="lnt">54
</span><span class="lnt">55
</span><span class="lnt">56
</span><span class="lnt">57
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=humann</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --nodes=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --tasks-per-node=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#set the min and max</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> <span class="nv">$START</span> -lt <span class="m">1</span> <span class="o">]</span>
</span></span><span class="line"><span class="cl"><span class="k">then</span>
</span></span><span class="line"><span class="cl">  <span class="nv">START</span><span class="o">=</span><span class="m">1</span>
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> <span class="nv">$STOP</span> -gt <span class="m">40</span> <span class="o">]</span>
</span></span><span class="line"><span class="cl"><span class="k">then</span>
</span></span><span class="line"><span class="cl">  <span class="nv">STOP</span><span class="o">=</span><span class="m">40</span>
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p data/paired
</span></span><span class="line"><span class="cl">  cat data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq &gt; data/paired/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired.fq
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  mkdir -p result/humann
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/humann --input data/paired/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --output result/humann/ --threads <span class="m">24</span>
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">  ln result/humann/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann_temp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_metaphlan_bugs_list.tsv result/humann/
</span></span><span class="line"><span class="cl">  rm -rf result/humann/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann_temp
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p result/metaphlan2
</span></span><span class="line"><span class="cl"><span class="c1">## ÂêàÂπ∂„ÄÅ‰øÆÊ≠£Ê†∑Êú¨Âêç</span>
</span></span><span class="line"><span class="cl">merge_metaphlan_tables2.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  result/humann/*_metaphlan_bugs_list.tsv <span class="p">|</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  sed <span class="s1">&#39;s/_metaphlan_bugs_list//g&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  &gt; result/metaphlan2/taxonomy.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="contigs-based">contigs-based</h2>
<p>contigs-basedÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊòØÊåáÂÖàÂ∞ÜÊµãÂ∫èËØªÈïøÔºàreadsÔºâÁªÑË£ÖÊàêËæÉÈïøÁöÑÂ∫èÂàóÁâáÊÆµÔºàcontigsÔºâÔºåÁÑ∂ÂêéÂØπËøô‰∫õcontigsËøõË°åËøõ‰∏ÄÊ≠•ÁöÑÂàÜÊûê„ÄÇËøôÁßçÊñπÊ≥ïÂèØ‰ª•Êèê‰æõÊõ¥ÈïøÁöÑÂü∫Âõ†ÁªÑÁâáÊÆµÔºåÊúâÂä©‰∫éÊõ¥ÂáÜÁ°ÆÂú∞ËøõË°åÂü∫Âõ†Ê≥®ÈáäÂíåÂæÆÁîüÁâ©Áæ§ËêΩÁªìÊûÑÂàÜÊûê„ÄÇ</p>
<ul>
<li><strong>‰ºòÁÇπ</strong>Ôºö
<ul>
<li><strong>Êõ¥È´òÁöÑËß£ÊûêÂ∫¶</strong>ÔºöÈÄöËøáÁªÑË£ÖÁîüÊàêËæÉÈïøÁöÑÂ∫èÂàóÁâáÊÆµÔºåÂèØ‰ª•Êõ¥ÂáÜÁ°ÆÂú∞ËøõË°åÂü∫Âõ†ÂíåÂü∫Âõ†ÁªÑÊ≥®Èáä„ÄÇ</li>
<li><strong>ÁªìÊûÑ‰ø°ÊÅØ</strong>ÔºöËÉΩÂ§üËé∑ÂæóÈïøË∑ùÁ¶ªÁöÑÂü∫Âõ†ËÅîÁ≥ªÂíåÂü∫Âõ†ÁªÑÁªìÊûÑ‰ø°ÊÅØÔºåÊúâÂä©‰∫éÊõ¥ÂÖ®Èù¢ÁöÑÂäüËÉΩÂíåËøõÂåñÂàÜÊûê„ÄÇ</li>
</ul>
</li>
<li><strong>Â±ÄÈôêÊÄß</strong>Ôºö
<ul>
<li><strong>ËÆ°ÁÆóËµÑÊ∫êÈúÄÊ±ÇÈ´ò</strong>ÔºöÁªÑË£ÖËøáÁ®ãÈúÄË¶ÅËæÉÈ´òÁöÑËÆ°ÁÆóËµÑÊ∫êÂíåÊó∂Èó¥„ÄÇ</li>
<li><strong>Â§çÊùÇÊÄß</strong>ÔºöÁªÑË£ÖÂíåbinningÊ≠•È™§Â¢ûÂä†‰∫ÜÂàÜÊûêÁöÑÂ§çÊùÇÊÄßÔºåÈúÄË¶ÅÊõ¥Â§öÁöÑÊäÄÊúØÁªèÈ™åÂíåÂ∑•ÂÖ∑ÊîØÊåÅ„ÄÇ</li>
</ul>
</li>
</ul>
<h3 id="ÁªÑË£Öassembly">ÁªÑË£ÖÔºàassemblyÔºâ</h3>
<p>ÁªÑË£ÖÔºàassemblyÔºâÊòØÊåáÂ∞ÜÊµãÂ∫è‰∫ßÁîüÁöÑÁü≠ËØªÈïøÔºàreadsÔºâÊãºÊé•ÊàêÊõ¥ÈïøÁöÑËøûÁª≠Â∫èÂàóÔºàcontigsÔºâÁöÑËøáÁ®ã„ÄÇËøô‰∏™ËøáÁ®ãÂú®Âü∫Âõ†ÁªÑÂ≠¶ÂíåÂÆèÂü∫Âõ†ÁªÑÂ≠¶Á†îÁ©∂‰∏≠Ëá≥ÂÖ≥ÈáçË¶ÅÔºåÂõ†‰∏∫ÂÆÉËÉΩÂ§üÂ∞ÜÁâáÊÆµÂåñÁöÑ‰ø°ÊÅØÊï¥ÂêàÊàêÊõ¥ÂÆåÊï¥ÁöÑÂü∫Âõ†ÁªÑÂ∫èÂàóÔºå‰æø‰∫éÂêéÁª≠ÁöÑÂäüËÉΩÂíåÂàÜÁ±ªÂàÜÊûê„ÄÇ</p>
<p>Ê≥®ÊÑèassemblyÊúâÂçïÊ†∑Êú¨ÊãºË£ÖÂíåÊ∑∑ÊãºÁ≠âÁ≠ñÁï•ÔºåÂèØ‰ª•Ëá™Ë°åÊ†πÊçÆËÆ°ÁÆóËµÑÊ∫êÂíåÁ†îÁ©∂ÁõÆÁöÑÈÄâÊã©„ÄÇ</p>
<ul>
<li>ÂçïÊ†∑Êú¨ÊãºË£ÖÔºàSingle-sample AssemblyÔºâ</li>
</ul>
<p><strong>ÂÆö‰πâ</strong>ÔºöÂçïÊ†∑Êú¨ÊãºË£ÖÊòØÊåáÂØπÊù•Ëá™Âçï‰∏ÄÁéØÂ¢ÉÊàñÊù°‰ª∂‰∏ãÁöÑÊ†∑Êú¨ËøõË°åÁªÑË£Ö„ÄÇËøôÁßçÁ≠ñÁï•ÈÄÇÁî®‰∫éÁõ∏ÂØπÁÆÄÂçïÁöÑÊ†∑Êú¨ÔºåÊàñËÄÖÂ∏åÊúõÂçïÁã¨ÂàÜÊûêÊØè‰∏™Ê†∑Êú¨ÁöÑÂü∫Âõ†ÁªÑÁªÑÊàê„ÄÇ</p>
<p><strong>‰ºòÁÇπ</strong>Ôºö</p>
<ol>
<li><strong>Áã¨Á´ãÂàÜÊûê</strong>ÔºöÊØè‰∏™Ê†∑Êú¨ÂçïÁã¨ÁªÑË£ÖÔºåÊúâÂä©‰∫éÊ∑±ÂÖ•ÂàÜÊûêÊ†∑Êú¨ÁâπÂÆöÁöÑÂü∫Âõ†ÁªÑÁâπÂæÅ„ÄÇ</li>
<li><strong>Êï∞ÊçÆÁÆÄÂçï</strong>ÔºöÈÄÇÁî®‰∫éÊï∞ÊçÆÂ§çÊùÇÂ∫¶ËæÉ‰ΩéÁöÑÊ†∑Êú¨ÔºåÁªÑË£ÖÁªìÊûúËæÉ‰∏∫Ê∏ÖÊô∞„ÄÇ</li>
</ol>
<p><strong>Áº∫ÁÇπ</strong>Ôºö</p>
<ol>
<li><strong>ËµÑÊ∫êÈúÄÊ±ÇÈ´ò</strong>ÔºöÂØπÊØè‰∏™Ê†∑Êú¨ËøõË°åÁã¨Á´ãÁªÑË£ÖÔºåËÆ°ÁÆóËµÑÊ∫êÂíåÊó∂Èó¥ÈúÄÊ±ÇËæÉÈ´ò„ÄÇ</li>
<li><strong>ÊúâÈôêÁöÑË¶ÜÁõñÂ∫¶</strong>ÔºöÂØπ‰∫é‰Ωé‰∏∞Â∫¶ÂæÆÁîüÁâ©ÔºåÂçïÊ†∑Êú¨ÁªÑË£ÖÂèØËÉΩÊó†Ê≥ïÊèê‰æõË∂≥Â§üÁöÑË¶ÜÁõñÂ∫¶ÂíåÁªÑË£ÖË¥®Èáè„ÄÇ</li>
</ol>
<p><strong>ÈÄÇÁî®Âú∫ÊôØ</strong>Ôºö</p>
<ul>
<li>
<p>ÁéØÂ¢ÉÊ†∑Êú¨ËæÉ‰∏∫ÁÆÄÂçïÁöÑÁ†îÁ©∂„ÄÇ</p>
</li>
<li>
<p>ÈúÄË¶ÅÂØπÊØè‰∏™Ê†∑Êú¨ËøõË°åÁã¨Á´ãÊØîËæÉÂíåÂàÜÊûêÁöÑÁ†îÁ©∂„ÄÇ</p>
</li>
<li>
<p>Ê∑∑ÊãºÁ≠ñÁï•ÔºàCo-assemblyÔºâ</p>
</li>
</ul>
<p><strong>ÂÆö‰πâ</strong>ÔºöÊ∑∑ÊãºÊòØÊåáÂ∞ÜÂ§ö‰∏™Ê†∑Êú¨ÁöÑÊï∞ÊçÆÂêàÂπ∂Âú®‰∏ÄËµ∑ËøõË°åÁªÑË£Ö„ÄÇËøôÁßçÁ≠ñÁï•ÈÄÇÁî®‰∫éÂ§çÊùÇÁöÑÂÆèÂü∫Âõ†ÁªÑÊ†∑Êú¨ÔºåÈÄöËøáÊï¥ÂêàÂ§ö‰∏™Ê†∑Êú¨ÁöÑÊï∞ÊçÆÔºåÂèØ‰ª•ÊèêÈ´òÁªÑË£ÖÁöÑË¶ÜÁõñÂ∫¶ÂíåË¥®Èáè„ÄÇ</p>
<p><strong>‰ºòÁÇπ</strong>Ôºö</p>
<ol>
<li><strong>ÊèêÈ´òË¶ÜÁõñÂ∫¶</strong>ÔºöÂêàÂπ∂Â§ö‰∏™Ê†∑Êú¨ÁöÑÊï∞ÊçÆÔºåÂ¢ûÂä†‰∫ÜÂ∫èÂàóÁöÑË¶ÜÁõñÂ∫¶ÔºåÊúâÂä©‰∫éÊõ¥ÂÆåÊï¥ÁöÑÁªÑË£Ö„ÄÇ</li>
<li><strong>Â§ÑÁêÜÂ§çÊùÇÊ†∑Êú¨</strong>ÔºöÈÄÇÁî®‰∫éÂ§çÊùÇÁöÑÁéØÂ¢ÉÊ†∑Êú¨ÔºåËÉΩÂ§üÊçïÊçâÂà∞Êõ¥Â§öÁöÑ‰Ωé‰∏∞Â∫¶ÂæÆÁîüÁâ©ÂíåÂü∫Âõ†ÁªÑ‰ø°ÊÅØ„ÄÇ</li>
</ol>
<p><strong>Áº∫ÁÇπ</strong>Ôºö</p>
<ol>
<li><strong>Â§çÊùÇÊÄßÂ¢ûÂä†</strong>ÔºöÊ∑∑ÊãºÁöÑÊï∞ÊçÆÂ§çÊùÇÂ∫¶È´òÔºåÁªÑË£ÖÂíåÂêéÁª≠ÂàÜÊûêÊõ¥‰∏∫Â§çÊùÇ„ÄÇ</li>
<li><strong>Ê†∑Êú¨Èó¥Â∑ÆÂºÇÊ®°Á≥äÂåñ</strong>ÔºöÊ†∑Êú¨Èó¥ÁöÑÁã¨ÁâπÁâπÂæÅÂèØËÉΩÂú®Ê∑∑ÊãºËøáÁ®ã‰∏≠Ë¢´Ê®°Á≥äÂåñÔºåÂΩ±ÂìçÂØπ‰∏™‰ΩìÊ†∑Êú¨ÁöÑÂÖ∑‰ΩìÂàÜÊûê„ÄÇ</li>
</ol>
<p><strong>ÈÄÇÁî®Âú∫ÊôØ</strong>Ôºö</p>
<ul>
<li>ÁéØÂ¢ÉÊ†∑Êú¨Â§çÊùÇÔºåÂåÖÂê´Â§öÁßçÂæÆÁîüÁâ©Áæ§ËêΩ„ÄÇ</li>
<li>ÈúÄË¶ÅÊèêÈ´ò‰Ωé‰∏∞Â∫¶Áâ©ÁßçÁöÑÊ£ÄÊµãËÉΩÂäõ„ÄÇ</li>
<li>ÁõÆÊ†áÊòØËé∑ÂèñÊï¥‰ΩìÂæÆÁîüÁâ©Áæ§ËêΩÁöÑÁªºÂêà‰ø°ÊÅØ„ÄÇ</li>
</ul>
<h4 id="megahit">megahit</h4>
<p>MegahitÊòØ‰∏Ä‰∏™Áî®‰∫éÂØπÈ´òÈÄöÈáèÊµãÂ∫èÊï∞ÊçÆËøõË°åde novoÁªÑË£ÖÁöÑËΩØ‰ª∂„ÄÇ
ÂÆÉ‰ΩøÁî®‰∫Ü‰∏ÄÁßçÂü∫‰∫éÁü≠ËØªÊØîÂØπÂíåÂõæÂΩ¢ÊûÑÂª∫ÁöÑÁÆóÊ≥ïÊù•ÁªÑË£ÖÂü∫Âõ†ÁªÑÔºåÂèØ‰ª•È´òÊïàÂú∞Â§ÑÁêÜÂ§ßËßÑÊ®°ÁöÑÊï∞ÊçÆÈõÜ„ÄÇ
‰ª•‰∏ãÊòØMegahitÁöÑ‰∏Ä‰∫õ‰ºòÁÇπÂíåÈÄÇÁî®ÊÉÖÂÜµÔºö</p>
<ol>
<li>ÈÄüÂ∫¶Âø´ÔºöMegahitÁöÑÁÆóÊ≥ïÈùûÂ∏∏È´òÊïàÔºåÂèØ‰ª•Â§ÑÁêÜÂ§ßËßÑÊ®°ÁöÑÊï∞ÊçÆÈõÜÔºåÈÄöÂ∏∏ÊØîÂÖ∂‰ªñde novoÁªÑË£ÖÂ∑•ÂÖ∑Êõ¥Âø´„ÄÇ</li>
<li>È´òË¥®ÈáèÁöÑÁªÑË£ÖÔºöMegahitÂú®ÁªÑË£ÖÁªìÊûúÁöÑËøûÈÄöÊÄßÂíåÂáÜÁ°ÆÊÄßÊñπÈù¢Ë°®Áé∞‰ºòÂºÇÔºåÂ∞§ÂÖ∂Âú®Â§ÑÁêÜÈ´òGCÂê´ÈáèÂü∫Âõ†ÁªÑÊó∂ÊïàÊûúÊòæËëó„ÄÇ</li>
<li>ÈÄÇÁî®‰∫é‰∏çÂêåÁ±ªÂûãÁöÑÊµãÂ∫èÊï∞ÊçÆÔºöMegahitÊîØÊåÅÂ§öÁßç‰∏çÂêåÁ±ªÂûãÁöÑÊµãÂ∫èÊï∞ÊçÆÔºåÂåÖÊã¨ Illumina HiSeq/MiSeq„ÄÅIonTorrentÂíåPacBioÁ≠âÂπ≥Âè∞„ÄÇ</li>
<li>Êòì‰∫é‰ΩøÁî®ÔºöMegahitÂÖ∑ÊúâÁÆÄÂçïÁöÑÂëΩ‰ª§Ë°åËØ≠Ê≥ïÔºåÊñπ‰æøÁî®Êà∑ËøõË°åÁªÑË£ÖÊìç‰ΩúÔºå‰∏îÂÖ∑Êúâ‰∏≠Êñ≠ÁÇπÔºåÈÅøÂÖçÂ§±Ë¥•ÂêéÂÖ®ÈÉ®ÈáçË∑ë„ÄÇ</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=megahit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=10G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  <span class="c1">#single sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/megahit
</span></span><span class="line"><span class="cl">  mkdir -p result/megahit/contigs
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/megahit -t <span class="m">8</span> -1 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/megahit/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span> --out-prefix <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">  sed -i <span class="s2">&#34;/&gt;/s/&gt;/&gt;</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">_/&#34;</span> result/megahit/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.contigs.fa
</span></span><span class="line"><span class="cl">  mv result/megahit/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.contigs.fa result/megahit/contigs/
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>Ê∑∑ÊãºÁöÑËØùÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=megahit2</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=mem</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=100G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1">#multi-sampleÊ∑∑Êãº</span>
</span></span><span class="line"><span class="cl">  <span class="c1">#ÈúÄË¶ÅÂ§ßÂÜÖÂ≠òÔºåÂª∫ËÆÆ3ÂÄçfq.gzÂÜÖÂ≠ò</span>
</span></span><span class="line"><span class="cl">  mkdir -p data/com_read
</span></span><span class="line"><span class="cl">  mkdir -p result/megahit2
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="k">for</span> i in <span class="sb">`</span>cat samplelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl">  <span class="k">do</span>
</span></span><span class="line"><span class="cl">      <span class="nb">echo</span> <span class="si">${</span><span class="nv">i</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">      cat data/rm_human/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.1.fq &gt;&gt; data/com_read/R1.fq
</span></span><span class="line"><span class="cl">      cat data/rm_human/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.2.fq &gt;&gt; data/com_read/R2.fq
</span></span><span class="line"><span class="cl">  <span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/megahit -t <span class="m">8</span> -1 data/com_read/R1.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/com_read/R2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/megahit2/ --out-prefix multi_sample
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="metaspades">metaSPAdes</h4>
<p>SPAdes ÊòØ‰∏ÄÊ¨æÂ§öÂäüËÉΩÂ∑•ÂÖ∑ÂåÖÔºå‰∏ì‰∏∫ÊµãÂ∫èÊï∞ÊçÆÁöÑÁªÑË£ÖÂíåÂàÜÊûêËÄåËÆæËÆ°„ÄÇ SPAdes ‰∏ªË¶ÅÊòØ‰∏∫ Illumina ÊµãÂ∫èÊï∞ÊçÆËÄåÂºÄÂèëÁöÑÔºå‰ΩÜ‰πüÂèØÁî®‰∫é IonTorrent„ÄÇÂ§ßÂ§öÊï∞ SPAdes ÁÆ°ÈÅìÊîØÊåÅÊ∑∑ÂêàÊ®°ÂºèÔºåÂç≥ÂÖÅËÆ∏‰ΩøÁî®ÈïøËØªÂèñÔºàPacBio Âíå Oxford NanoporeÔºâ‰Ωú‰∏∫Ë°•ÂÖÖÊï∞ÊçÆ„ÄÇ</p>
<p>ËÄåmetaSPAdesÊòØÁõÆÂâçÂÆèÂü∫Âõ†ÁªÑÈ¢ÜÂüüÁªÑË£ÖÊåáÊ†áËæÉÂ•ΩÁöÑËΩØ‰ª∂ÔºåÂ∞§ÂÖ∂Âú®Ê†™Ê∞¥Âπ≥ÁªÑË£Ö‰ºòÂäøÊòéÊòæ„ÄÇÁõ∏ÊØîMegahitË°®Áé∞Âá∫Êõ¥Â•ΩÁöÑÂü∫Âõ†ÈïøÂ∫¶ÂíåËØªÈïøÊØîËæÉÁéáÔºå‰ΩÜÊòØÊõ¥Â•ΩÁöÑÁªÑË£ÖË¥®Èáè‰πü‰º¥ÈöèÁùÄÊõ¥ÈïøÊó∂Èó¥ÂíåÂÜÖÂ≠òÊ∂àËÄóÔºåÂêåÊó∂‰πüÊúâÈîôËØØÁªÑË£Ö‰∏äÂçáÁöÑÈ£éÈô©„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=asthma_megahit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>	
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  <span class="c1">#single sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/metaspades
</span></span><span class="line"><span class="cl">  mkdir -p result/metaspades/contigs
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/metawrap/bin/metaspades.py ‚Äìt <span class="m">8</span> -k 21,33,55,77,99,127 --careful <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -1 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/metaspades/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  sed -i <span class="s2">&#34;/&gt;/s/&gt;/&gt;</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">_/&#34;</span> result/metaspades/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>/scaffolds.fasta
</span></span><span class="line"><span class="cl">  mv result/metaspades/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>/scaffolds.fasta result/metaspades/contigs/
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÁªÑË£ÖËØÑ‰º∞quast">ÁªÑË£ÖËØÑ‰º∞ÔºöQUAST</h3>
<p><strong>QUAST</strong>ÊòØ‰∏Ä‰∏™Ë¥®ÈáèËØÑ‰º∞Â∑•ÂÖ∑„ÄÇ QUASTÂèØ‰ª•‰ΩøÁî®ÂèÇËÄÉÂü∫Âõ†ÁªÑ‰ª•Âèä‰∏ç‰ΩøÁî®ÂèÇËÄÉÂü∫Âõ†ÁªÑÊù•ËØÑ‰º∞Ë£ÖÈÖç„ÄÇ QUASTÁîüÊàêËØ¶ÁªÜÁöÑÊä•ÂëäÔºåË°®Ê†ºÂíåÂõæËß£Ôºå‰ª•ÊòæÁ§∫Ë£ÖÈÖçÁöÑ‰∏çÂêåÊñπÈù¢„ÄÇ</p>
<h3 id="Âü∫Âõ†È¢ÑÊµãprodigal">Âü∫Âõ†È¢ÑÊµãÔºöProdigal</h3>
<p>Âü∫Âõ†È¢ÑÊµãÊòØÊåáÂú®Âü∫Âõ†ÁªÑÂ∫èÂàóÔºàÂ¶ÇcontigsÔºâ‰∏≠ËØÜÂà´Âá∫ÁºñÁ†ÅÂå∫Â∫èÂàóÔºàCDS, Coding DNA SequencesÔºâÁöÑËøáÁ®ã„ÄÇËøô‰∏™ËøáÁ®ãÂØπ‰∫éÁêÜËß£Âü∫Âõ†ÁªÑÂäüËÉΩÂíåÊ≥®ÈáäÂü∫Âõ†ÁªÑ‰∏≠ÁöÑÂü∫Âõ†ÊòØËá≥ÂÖ≥ÈáçË¶ÅÁöÑ„ÄÇÂ∏∏Áî®ÁöÑËΩØ‰ª∂ÊúâProdigalÔºåGeneMarkÔºàGeneMarkSÁî®‰∫éÁªÜËèåÔºåGeneMark-ESÁî®‰∫éÁúüÊ†∏ÁîüÁâ©ÔºâÔºåGlimmerÁ≠â„ÄÇ</p>
<p>ËøôÈáå‰ª•Prodigal‰∏∫‰æãÔºö
ËæìÂÖ•Êñá‰ª∂ÔºöÊãºË£ÖÂ•ΩÁöÑÂ∫èÂàóÊñá‰ª∂ megahit/final.contigs.fa
ËæìÂá∫Êñá‰ª∂ÔºöprodigalÈ¢ÑÊµãÁöÑÂü∫Âõ†Â∫èÂàó prodigal/gene.fa</p>
<p>prodigal‰∏çÊîØÊåÅÂ§öÁ∫øÁ®ãËøêË°åÔºåÊâÄ‰ª•Êàë‰ª¨ÂèØ‰ª•Ëá™Ë°åÂàÜÂâ≤Â∫èÂàóÊñá‰ª∂Ë∞ÉÁî®Â§ö‰∏™prodigalÁ®ãÂ∫èÂàÜÂà´Ë∑ëÂÆûÁé∞‰º™Â§öÁ∫øÁ®ã„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=prodigal</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/prodigal
</span></span><span class="line"><span class="cl">  mkdir -p result/prodigal/gene_fa
</span></span><span class="line"><span class="cl">  mkdir -p result/prodigal/gene_gff
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/prodigal -i result/megahit/contigs/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.contigs.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -d result/prodigal/gene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.gene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o result/prodigal/gene_gff/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.gene.gff <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -p meta -f gff
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mkdir -p result/prodigal/fullgene_fa
</span></span><span class="line"><span class="cl">  <span class="c1">## ÊèêÂèñÂÆåÊï¥Âü∫Âõ†(ÂÆåÊï¥ÁâáÊÆµËé∑ÂæóÁöÑÂü∫Âõ†ÂÖ®‰∏∫ÂÆåÊï¥ÔºåÂ¶ÇÊàêÁéØÁöÑÁªÜËèåÂü∫Âõ†ÁªÑ)</span>
</span></span><span class="line"><span class="cl">  grep <span class="s1">&#39;partial=00&#39;</span> result/prodigal/gene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.gene.fa <span class="p">|</span> cut -f1 -d <span class="s1">&#39; &#39;</span><span class="p">|</span> sed <span class="s1">&#39;s/&gt;//&#39;</span> &gt; result/prodigal/gene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.fullid
</span></span><span class="line"><span class="cl">  seqkit grep -f result/prodigal/gene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.fullid result/prodigal/gene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.gene.fa &gt; result/prodigal/fullgene_fa/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.gene.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÂéªÂÜó‰Ωô">ÂéªÂÜó‰Ωô</h3>
<p>Âü∫Âõ†ÈõÜÂéªÂÜó‰ΩôÊòØÊåáÂú®È¢ÑÊµãÂà∞ÁöÑÂü∫Âõ†ÈõÜ‰∏≠ÂéªÈô§ÈáçÂ§çÁöÑÂü∫Âõ†ÊàñÈ´òÂ∫¶Áõ∏‰ººÁöÑÂü∫Âõ†Ôºå‰ª•Ëé∑Âæó‰∏Ä‰∏™ÈùûÂÜó‰ΩôÁöÑÂü∫Âõ†ÈõÜÂêà„ÄÇËøô‰∏ÄÊ≠•È™§Âú®ÂÆèÂü∫Âõ†ÁªÑÂàÜÊûê‰∏≠ÈùûÂ∏∏ÈáçË¶ÅÔºåÂõ†‰∏∫ÂÆèÂü∫Âõ†ÁªÑÊ†∑Êú¨‰∏≠ÈÄöÂ∏∏ÂåÖÂê´Êù•Ëá™Â§öÁßçÂæÆÁîüÁâ©ÁöÑÈáçÂ§çÊàñÁõ∏‰ººÂ∫èÂàóÔºåÂéªÂÜó‰ΩôÂèØ‰ª•ÁÆÄÂåñÊï∞ÊçÆÈõÜÂπ∂ÊèêÈ´òÂêéÁª≠ÂàÜÊûêÁöÑÊïàÁéáÂíåÂáÜÁ°ÆÊÄß„ÄÇÂ∏∏Áî®ËΩØ‰ª∂ÊúâCD-HITÔºåMMseqs2ÔºåUCLUSTÔºàÈõÜÊàêÂú®QIIMEÂíåUSEARCH‰∏≠ÁöÑÂ∑•ÂÖ∑ÔºâÁ≠â„ÄÇ</p>
<p>‰∏äÈù¢‰∫ßÁîü‰∫Ün‰∏™Ê†∑Êú¨ÁöÑÂü∫Âõ†È¢ÑÊµãÁªìÊûúÊñá‰ª∂Ôºågene.faÊñá‰ª∂Ë¶ÅÂÖàÊÉ≥ÂäûÊ≥ïÊï¥Âêà‰∏∫‰∏Ä‰∏™Êñá‰ª∂ÂÜçÂéªÂéªÂÜó‰Ωô„ÄÇ</p>
<h4 id="cd-hit">CD-HIT</h4>
<p>‰πãÂâçÂ§ßÂÆ∂Â∏∏Áî®ÁöÑÊòØCD-HITÊù•ÂéªÂÜó‰ΩôÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=cluster</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --nodes=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --tasks-per-node=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;start merge&#34;</span>
</span></span><span class="line"><span class="cl">  cat result/prodigal/gene_fa/*.gene.fa &gt; result/prodigal/all.gene.fa
</span></span><span class="line"><span class="cl">  cat result/prodigal/fullgene_fa/*.gene.fa &gt; result/prodigal/all.fullgene.fa
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;done&#34;</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/cd-hit-est -i result/prodigal/all.gene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -aS 0.9 -c 0.9 -G <span class="m">0</span> -g <span class="m">0</span> -T <span class="m">0</span> -M <span class="m">0</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mv result/NR/nucleotide_rep_seq.fasta result/NR/nucleotide.fa
</span></span><span class="line"><span class="cl">  rm result/NR/nucleotide_all_seqs.fasta
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/seqkit translate result/NR/nucleotide.fa &gt; result/NR/protein.fa
</span></span><span class="line"><span class="cl">  sed <span class="s1">&#39;s/\*//g&#39;</span> result/NR/protein.fa &gt; result/NR/protein_no_end.fa
</span></span><span class="line"><span class="cl">  rm result/NR/protein.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="mmseqs2">Mmseqs2</h4>
<p>ÊúÄËøëÂèëÁé∞‰∫ÜMmseqs2Ëøô‰∏™Á•ûÂô®ÔºåËøô‰∏™ËÅöÁ±ªË¶ÅÊØîCD-HITÂø´ÈùûÂ∏∏Â§öÔºåÂÖ∂‰ªñÂ∫îÁî®ÂèÇËßÅ<a class="link" href="../mmseqs2" >Mmseqs2ÁöÑÂü∫Á°Ä‰ΩøÁî®</a>Ôºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=cluster</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --nodes=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --tasks-per-node=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;start merge&#34;</span>
</span></span><span class="line"><span class="cl">  cat result/prodigal/gene_fa/*.gene.fa &gt; result/prodigal/all.gene.fa
</span></span><span class="line"><span class="cl">  cat result/prodigal/fullgene_fa/*.gene.fa &gt; result/prodigal/all.fullgene.fa
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;done&#34;</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">  mkdir -p result/NR
</span></span><span class="line"><span class="cl">  mmseqs easy-linclust result/prodigal/all.gene.fa result/NR/nucleotide mmseqs_tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mv result/NR/nucleotide_rep_seq.fasta result/NR/nucleotide.fa
</span></span><span class="line"><span class="cl">  rm result/NR/nucleotide_all_seqs.fasta
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/bin/seqkit translate result/NR/nucleotide.fa &gt; result/NR/protein.fa
</span></span><span class="line"><span class="cl">  sed <span class="s1">&#39;s/\*//g&#39;</span> result/NR/protein.fa &gt; result/NR/protein_no_end.fa
</span></span><span class="line"><span class="cl">  rm result/NR/protein.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="fasta‰ø°ÊÅØÁªüËÆ°">fasta‰ø°ÊÅØÁªüËÆ°</h3>
<p>‰ΩøÁî®seqkitÁªüËÆ°ÂàöÂàöÂæóÂà∞ÁöÑ‰∏Ä‰∫õÂÖ≥ÈîÆfastaÊñá‰ª∂ÁöÑÂü∫Á°Ä‰ø°ÊÅØ„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nv">test_file</span><span class="o">=</span><span class="sb">`</span>head -n1 <span class="nv">$samplelist</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -f result/megahit/contigs/<span class="si">${</span><span class="nv">test_file</span><span class="si">}</span>.contigs.fa <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">~/miniconda3/envs/waste/bin/seqkit stats result/megahit/contigs/*.contigs.fa &gt;result/megahit/contig_stats
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -f result/prodigal/gene_fa/<span class="si">${</span><span class="nv">test_file</span><span class="si">}</span>.gene.fa <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">~/miniconda3/envs/waste/bin/seqkit stats result/prodigal/gene_fa/*.gene.fa &gt;result/prodigal/gene_fa_stats
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -f result/prodigal/fullgene_fa/<span class="si">${</span><span class="nv">test_file</span><span class="si">}</span>.gene.fa <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">~/miniconda3/envs/waste/bin/seqkit stats result/prodigal/fullgene_fa/*.gene.fa &gt;result/prodigal/fullgene_fa_stats
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -f result/NR/nucleotide.fa <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">~/miniconda3/envs/waste/bin/seqkit stats result/NR/nucleotide.fa &gt;result/NR/nucleotide_stat
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="Âü∫Âõ†ÂÆöÈáèsalmon">Âü∫Âõ†ÂÆöÈáèÔºösalmon</h3>
<ol>
<li>Âª∫Á´ãÁ¥¢Âºï</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon-index</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=4</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=10G</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1">## Âª∫Á¥¢Âºï, -tÂ∫èÂàó, -i Á¥¢Âºï</span>
</span></span><span class="line"><span class="cl">  <span class="c1"># Â§ßÁÇπÂÜÖÂ≠ò</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/salmon
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/share/salmon/bin/salmon index <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -t result/NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -p <span class="m">4</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i result/salmon/index
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>ÂØπÊØè‰∏™Ê†∑Êú¨ÂÆöÈáè</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon-quant</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=4</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">NUMLINES</span><span class="o">=</span><span class="m">40</span> <span class="c1">#how many sample in one array</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">STOP</span><span class="o">=</span><span class="k">$((</span>SLURM_ARRAY_TASK_ID*NUMLINES<span class="k">))</span>
</span></span><span class="line"><span class="cl"><span class="nv">START</span><span class="o">=</span><span class="s2">&#34;</span><span class="k">$((</span><span class="nv">$STOP</span> <span class="o">-</span> <span class="k">$((</span><span class="nv">$NUMLINES</span> <span class="o">-</span> <span class="m">1</span><span class="k">))))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;START=</span><span class="nv">$START</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;STOP=</span><span class="nv">$STOP</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂÖ•Êñá‰ª∂ÔºöÂéªÂÜó‰ΩôÂêéÁöÑÂü∫Âõ†ÂíåËõãÁôΩÂ∫èÂàóÔºöNR/nucleotide.fa</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂá∫Êñá‰ª∂ÔºöSalmonÂÆöÈáèÂêéÁöÑÁªìÊûúÔºösalmon/gene.count; salmon/gene.TPM</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÂÆöÈáèÔºålÊñáÂ∫ìÁ±ªÂûãËá™Âä®ÈÄâÊã©ÔºåpÁ∫øÁ®ãÔºå--metaÂÆèÂü∫Âõ†ÁªÑÊ®°Âºè</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> <span class="o">((</span> <span class="nv">N</span> <span class="o">=</span> <span class="nv">$START</span><span class="p">;</span> N &lt;<span class="o">=</span> <span class="nv">$STOP</span><span class="p">;</span> N++ <span class="o">))</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="s2">&#34;</span><span class="nv">$N</span><span class="s2">&#34;</span> <span class="nv">$samplelist</span> <span class="p">|</span> tail -n 1<span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/salmon/quant
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/share/salmon/bin/salmon quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --validateMappings <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -i result/salmon/index -l A -p <span class="m">4</span> --meta <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -1 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.1.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -2 data/rm_human/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o result/salmon/quant/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.quant
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>ÂêàÂπ∂ÂêÑÊ†∑Êú¨ÁªìÊûú</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon-merge</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nv">samplelist</span><span class="o">=</span>/share/home/jianglab/pengchen/work/asthma/samplelist
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  ls result/salmon/quant/*.quant/quant.sf <span class="p">|</span>awk -F<span class="s1">&#39;/&#39;</span> <span class="s1">&#39;{print $(NF-1)}&#39;</span> <span class="p">|</span>sed <span class="s1">&#39;s/.quant//&#39;</span> &gt;tmp_finished
</span></span><span class="line"><span class="cl">  diff_rows -f samplelist -s tmp_finished &gt;tmp_diff
</span></span><span class="line"><span class="cl">  <span class="c1"># ËÆ°ÁÆóÁªìÊûúÁöÑË°åÊï∞</span>
</span></span><span class="line"><span class="cl">  <span class="nv">line_count</span><span class="o">=</span><span class="k">$(</span> cat tmp_diff<span class="p">|</span> wc -l<span class="k">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1"># Ê£ÄÊü•Ë°åÊï∞ÊòØÂê¶Â§ß‰∫é1ÔºåÂ¶ÇÊûúÊòØÂàôÁªàÊ≠¢ËÑöÊú¨</span>
</span></span><span class="line"><span class="cl">  <span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$line_count</span><span class="s2">&#34;</span> -gt <span class="m">1</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s1">&#39;sfÊñá‰ª∂ÂíåsamplelistÊï∞Èáè‰∏ç‰∏ÄËá¥&#39;</span>
</span></span><span class="line"><span class="cl">    <span class="nb">exit</span> <span class="m">1</span>
</span></span><span class="line"><span class="cl">  <span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1">##mapping rates</span>
</span></span><span class="line"><span class="cl">  cat result/salmon/quant/*.quant/logs/salmon_quant.log <span class="p">|</span>grep <span class="s1">&#39;Mapping rate = &#39;</span><span class="p">|</span>awk <span class="s1">&#39;{print $NF}&#39;</span>&gt; tmp
</span></span><span class="line"><span class="cl">  paste samplelist tmp &gt; result/salmon/mapping_rates
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1">## combine</span>
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/share/salmon/bin/salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --quants result/salmon/quant/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o result/salmon/gene.TPM
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/waste/share/salmon/bin/salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --quants result/salmon/quant/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --column NumReads -o result/salmon/gene.count
</span></span><span class="line"><span class="cl">  sed -i <span class="s1">&#39;1 s/.quant//g&#39;</span> result/salmon/gene.*
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä">ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä</h3>
<p>ÂäüËÉΩÂü∫Âõ†Ê≥®ÈáäÊòØÂØπÈùûÂÜó‰ΩôÂü∫Âõ†ÈõÜËøõË°åÂàÜÁ±ªÂíåÂäüËÉΩÊèèËø∞ÁöÑËøáÁ®ãÔºå‰ª•‰æøÁêÜËß£Ëøô‰∫õÂü∫Âõ†Âú®ÁîüÁâ©‰ΩìÂÜÖÂèØËÉΩÁöÑËßíËâ≤Âíå‰ΩúÁî®„ÄÇ</p>
<p>‰∏ä‰∏ÄÊ≠•Â∑≤ÁªèÊúâ‰∫ÜÊâÄÊúâÁöÑÂü∫Âõ†ÂíåÊØè‰∏™Ê†∑Êú¨ÊâÄÊúâÂü∫Âõ†ÁöÑread countÂÆöÈáèÁªìÊûúÔºå
Êàë‰ª¨Âè™ÈúÄË¶ÅÂØπ‰∏ä‰∏ÄÊ≠•ÁöÑÂü∫Âõ†Â∫èÂàóÔºàÊàñËõãÁôΩË¥®Â∫èÂàóÔºâËøõË°å‰∏çÂêåÊï∞ÊçÆÂ∫ìÁöÑÊ≥®ÈáäÔºå
ÁÑ∂ÂêéÂêàÂπ∂Ê≥®ÈáäÁªìÊûúÂæóÂà∞ÁöÑÂ∞±ÊòØÂäüËÉΩ‰∏∞Â∫¶Ë°®„ÄÇ</p>
<p>ÂæàÂ§öÊï∞ÊçÆÂ∫ìËá™Â∏¶ËΩØ‰ª∂ÈÉΩÊòØÁî®diamondÊØîÂØπÔºåÂ¶ÇÊûúÊ≤°Êúâ‰∏ìÁî®ËΩØ‰ª∂ÁöÑÊï∞ÊçÆÂ∫ìÊàë‰ª¨‰πüÂèØ‰ª•Ëá™Â∑±Áî®diamondÊØîÂØπ„ÄÇ</p>
<p>diamondÈÄâÊã©&ndash;outfmt 6ÁöÑËæìÂá∫ÁªìÊûúÂíåblastpÊ†ºÂºè‰∏ÄÊ†∑„ÄÇ</p>
<h4 id="eggnog-cogkeggcazy">eggNOG (COG/KEGG/CAZy)</h4>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOGÊï∞ÊçÆÂ∫ì</a>Êî∂ÈõÜ‰∫ÜCOGÔºàClusters of Orthologous Groups of proteinsÔºåÁõ¥Á≥ªÂêåÊ∫êËõãÁôΩÁ∞áÔºâ,ÊûÑÊàêÊØè‰∏™COGÁöÑËõãÁôΩÈÉΩÊòØË¢´ÂÅáÂÆö‰∏∫Êù•Ëá™‰∫é‰∏Ä‰∏™Á•ñÂÖàËõãÁôΩÔºåÂõ†Ê≠§ÊòØorthologsÊàñËÄÖÊòØparalogs„ÄÇÈÄöËøáÊääÊâÄÊúâÂÆåÊï¥Âü∫Âõ†ÁªÑÁöÑÁºñÁ†ÅËõãÁôΩ‰∏Ä‰∏™‰∏Ä‰∏™ÁöÑ‰∫íÁõ∏ÊØîËæÉÁ°ÆÂÆöÁöÑ„ÄÇÂú®ËÄÉËôëÊù•Ëá™‰∏Ä‰∏™ÁªôÂÆöÂü∫Âõ†ÁªÑÁöÑËõãÁôΩÊó∂ÔºåËøôÁßçÊØîËæÉÂ∞ÜÁªôÂá∫ÊØè‰∏™ÂÖ∂‰ªñÂü∫Âõ†ÁªÑÁöÑ‰∏Ä‰∏™ÊúÄÁõ∏‰ººÁöÑËõãÁôΩÔºàÂõ†Ê≠§ÈúÄË¶ÅÁî®ÂÆåÊï¥ÁöÑÂü∫Âõ†ÁªÑÊù•ÂÆö‰πâCOGÔºâÔºåËøô‰∫õÂü∫Âõ†ÁöÑÊØè‰∏Ä‰∏™ÈÉΩËΩÆÁï™Âú∞Ë¢´ËÄÉËôë„ÄÇÂ¶ÇÊûúÂú®Ëøô‰∫õËõãÁôΩÔºàÊàñÂ≠êÈõÜÔºâ‰πãÈó¥‰∏Ä‰∏™Áõ∏‰∫íÁöÑÊúÄ‰Ω≥ÂåπÈÖçÂÖ≥Á≥ªË¢´ÂèëÁé∞ÔºåÈÇ£‰πàÈÇ£‰∫õÁõ∏‰∫íÁöÑÊúÄ‰Ω≥ÂåπÈÖçÂ∞ÜÂΩ¢Êàê‰∏Ä‰∏™COG„ÄÇËøôÊ†∑Ôºå‰∏Ä‰∏™COG‰∏≠ÁöÑÊàêÂëòÂ∞Ü‰∏éËøô‰∏™COG‰∏≠ÁöÑÂÖ∂‰ªñÊàêÂëòÊØîËµ∑Ë¢´ÊØîËæÉÁöÑÂü∫Âõ†ÁªÑ‰∏≠ÁöÑÂÖ∂‰ªñËõãÁôΩÊõ¥Áõ∏ÂÉè„ÄÇ</p>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOG</a>ÈáåÈù¢Â∑≤ÁªèÂåÖÂê´‰∫ÜGOÔºåKEGGÔºåCAZyÁ≠â„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÂ∏∏Áî®Êï∞ÊçÆÂ∫ìÔºåÊ≥®ÊÑèËÆæÁΩÆ‰∏ãËΩΩ‰ΩçÁΩÆ</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5
</span></span><span class="line"><span class="cl"><span class="c1">## -yÈªòËÆ§ÂêåÊÑèÔºå-fÂº∫Âà∂‰∏ãËΩΩÔºåeggnog.db.gz 7.9G+4.9G</span>
</span></span><span class="line"><span class="cl">download_eggnog_data.py -y -f --data_dir ./
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÊñπÂºè2(ÂèØÈÄâ)ÔºöÈìæÊé•Áõ¥Êé•‰∏ãËΩΩ</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog.db.gz <span class="c1">## 7.9G</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog_proteins.dmnd.gz <span class="c1">## 4.9G</span>
</span></span><span class="line"><span class="cl">gunzip *.gz
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=eggnog</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=10G</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">  <span class="c1">## Â§ßÁÇπÂÜÖÂ≠ò, Êï∞ÊçÆÂ∫ìÊúâ50GÂ∑¶Âè≥</span>
</span></span><span class="line"><span class="cl">  <span class="nb">source</span> ~/miniconda3/etc/profile.d/conda.sh
</span></span><span class="line"><span class="cl">  conda activate func
</span></span><span class="line"><span class="cl">  <span class="c1">## diamondÊØîÂØπÂü∫Âõ†Ëá≥eggNOG 5.0Êï∞ÊçÆÂ∫ì, 1~9hÔºåÈªòËÆ§diamond 1e-3</span>
</span></span><span class="line"><span class="cl">  mkdir -p result/eggnog
</span></span><span class="line"><span class="cl">  emapper.py --no_annot --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i result/NR/protein_no_end.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --cpu <span class="m">8</span> -m diamond <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/eggnog/protein
</span></span><span class="line"><span class="cl">  <span class="c1">## ÊØîÂØπÁªìÊûúÂäüËÉΩÊ≥®Èáä, 1h</span>
</span></span><span class="line"><span class="cl">  emapper.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --annotate_hits_table result/eggnog/protein.emapper.seed_orthologs <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --cpu <span class="m">8</span> --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o result/eggnog/output
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1">## Ê∑ªË°®Â§¥, 1Âàó‰∏∫IDÔºå9ÂàóKOÔºå16ÂàóCAZyÔºå21ÂàóCOGÔºå22ÂàóÊèèËø∞</span>
</span></span><span class="line"><span class="cl">  sed <span class="s1">&#39;1 i Name\tortholog\tevalue\tscore\ttaxonomic\tprotein\tGO\tEC\tKO\tPathway\tModule\tReaction\trclass\tBRITE\tTC\tCAZy\tBiGG\ttax_scope\tOG\tbestOG\tCOG\tdescription&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    result/eggnog/output.emapper.annotations <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    &gt; result/eggnog/eggnog_anno_output
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="dbcan2Á¢≥Ê∞¥ÂåñÂêàÁâ©">dbCAN2ÔºàÁ¢≥Ê∞¥ÂåñÂêàÁâ©Ôºâ</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## dbCAN2 http://bcb.unl.edu/dbCAN2</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÂàõÂª∫Êï∞ÊçÆÂ∫ìÂ≠òÊîæÁõÆÂΩïÂπ∂ËøõÂÖ•</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2
</span></span><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÂ∫èÂàóÂíåÊèèËø∞</span>
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/CAZyDB.07312020.fa
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/Databases/CAZyDB.07302020.fam-activities.txt
</span></span><span class="line"><span class="cl"><span class="c1">## Â§áÁî®Êï∞ÊçÆÂ∫ì‰∏ãËΩΩÂú∞ÂùÄÂπ∂Ëß£Âéã </span>
</span></span><span class="line"><span class="cl"><span class="c1">#wget -c http://210.75.224.110/db/dbcan2/CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">#gunzip CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">## diamondÂª∫Á¥¢Âºï</span>
</span></span><span class="line"><span class="cl"><span class="nb">time</span> diamond makedb <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --in CAZyDB.07312020.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --db CAZyDB.07312020
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=cazy</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mkdir -p result/dbcan2
</span></span><span class="line"><span class="cl">  diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--db ~/db/dbcan2/CAZyDB.07312020  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--query result/NR/protein_no_end.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--threads <span class="m">8</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--out result/dbcan2/gene_diamond.f6
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="cardÊäóÁîüÁ¥†ÊäóÊÄßÂü∫Âõ†args">CARDÔºàÊäóÁîüÁ¥†ÊäóÊÄßÂü∫Âõ†ARGsÔºâ</h4>
<p>RGIËΩØ‰ª∂Github‰∏ªÈ°µ: <a class="link" href="https://github.com/arpcard/rgi"  target="_blank" rel="noopener"
    >https://github.com/arpcard/rgi</a>ÔºåÂèØ‰ª•ÂèÇËÄÉ‰πãÂâçÂÜôÁöÑ<a class="link" href="../antibiotics-resistance" >Antibiotics resistance gene</a>„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rgi</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="nb">source</span> ~/miniconda3/etc/profile.d/conda.sh
</span></span><span class="line"><span class="cl">  conda activate rgi
</span></span><span class="line"><span class="cl">  mkdir -p result/card
</span></span><span class="line"><span class="cl">  ~/miniconda3/envs/rgi/bin/rgi main <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --input_sequence result/NR/protein_no_end.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --output_file result/card/protein <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --input_type protein --num_threads <span class="m">8</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --clean --alignment_tool DIAMOND <span class="c1"># --low_quality #partial genes</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="vfdbÊØíÂäõÂõ†Â≠ê">VFDBÔºàÊØíÂäõÂõ†Â≠êÔºâ</h4>
<p>ÂÆòÁΩëÂú∞ÂùÄÔºö<a class="link" href="http://www.mgc.ac.cn/VFs/"  target="_blank" rel="noopener"
    >http://www.mgc.ac.cn/VFs/</a></p>
<p>Âú®ÂÆòÁΩë‰∏ãËΩΩÊï∞ÊçÆÂ∫ìÊó∂ÔºåÂ∏¶ÊúâsetA ÁöÑÂ∫ì‰∏∫VFDBÊï∞ÊçÆÂ∫ìÊ†∏ÂøÉÂ∫ì(set A)ÔºåËÄåsetB‰∏∫ÂÖ®Â∫ì(setB), ÂÖ∂‰∏≠setA‰ªÖÂåÖÂê´ÁªèÂÆûÈ™åÈ™åËØÅËøáÁöÑÊØíÂäõÂü∫Âõ†ÔºåËÄåsetBÂàôÂú®setAÁöÑÂü∫Á°Ä‰∏äÂ¢ûÂä†‰∫ÜÈ¢ÑÊµãÁöÑÊØíÂäõÂü∫Âõ†ÔºåÈÄâÊã©Â•ΩÊï∞ÊçÆÂ∫ìÂêéÔºåÁõ¥Êé•Áî®blast/diamondÂç≥ÂèØÂÆåÊàêÊ≥®Èáä„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=vfdb</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/log/%x_%A_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=2G</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  mkdir -p result/vfdb
</span></span><span class="line"><span class="cl">  diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--db ~/db/VFDB/VFDB_setB_pro <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--query result/NR/protein_no_end.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--threads <span class="m">8</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  	--out result/vfdb/gene_diamond.f6
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s2">&#34;%Y-%m-%d %T&#34;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="ÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì">ÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì</h4>
<ul>
<li>NCBI NRÔºàNon-redundantÔºâ:ÂåÖÂê´ÂπøÊ≥õÁöÑÂ∑≤Áü•ËõãÁôΩÂ∫èÂàóÔºåÊòØÂ∏∏Áî®ÁöÑÂèÇËÄÉÊï∞ÊçÆÂ∫ì„ÄÇ</li>
<li>PfamÔºöÂåÖÂê´ËõãÁôΩË¥®ÂÆ∂ÊóèÂíåÂäüËÉΩÂüüÁöÑ‰ø°ÊÅØÔºåÈÄöËøáHidden Markov Models (HMMs) ÊèèËø∞ËõãÁôΩË¥®ÂäüËÉΩÂüü„ÄÇ</li>
<li>ÂÖÉÁ¥†Âæ™ÁéØÊï∞ÊçÆÂ∫ìÔºö‰∏ìÊ≥®‰∫éÁéØÂ¢É‰∏≠ÁöÑÂÖÉÁ¥†ÔºàÂ¶ÇÁ¢≥„ÄÅÊ∞Æ„ÄÅÁ°´„ÄÅÁ£∑Á≠âÔºâÁöÑÁîüÁâ©Âú∞ÁêÉÂåñÂ≠¶Âæ™ÁéØÁõ∏ÂÖ≥Âü∫Âõ†ÂíåÈÄîÂæÑÊ≥®Èáä„ÄÇÂ¶ÇNCycDBÔºåSCycDBÔºåPCycDBÔºåMCycDBÁ≠â</li>
</ul>
<p>Ëøô‰∫õÊï∞ÊçÆÂ∫ìÊàë‰ª¨ÂèØ‰ª•Ëá™Ë°å‰∏ãËΩΩËõãÁôΩÊï∞ÊçÆÂ∫ìÔºåÁÑ∂ÂêéÁî®blastÂª∫Â∫ìÊØîÂØπ„ÄÇ</p>
<h3 id="ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂">ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂</h3>
<p>ÂÜô‰∏Ä‰∏™pythonËÑöÊú¨ÔºåÂ∞ÜË°®1ÔºàÂü∫Âõ†-ÂäüËÉΩÁöÑÂØπÂ∫îË°®Ôºâ‰∏éË°®2ÔºàÂü∫Âõ†‰∏∞Â∫¶Ë°®ÔºâÂêàÂπ∂ÔºåÂç≥‰∏çÂêåÂü∫Âõ†ÂèØËÉΩÊ≥®ÈáäÂà∞Áõ∏ÂêåÂäüËÉΩÔºåÊääÂÆÉ‰ª¨ÁöÑ‰∏∞Â∫¶Âä†Âú®‰∏ÄËµ∑ÂæóÂà∞Êñ∞Ë°®3ÔºàÂäüËÉΩ‰∏∞Â∫¶Ë°®Ôºâ„ÄÇ</p>
<p>ÂèØ‰ª•ÂèÇËÄÉ<a class="link" href="https://github.com/YongxinLiu/EasyMicrobiome/blob/master/script/summarizeAbundance.py"  target="_blank" rel="noopener"
    >https://github.com/YongxinLiu/EasyMicrobiome/blob/master/script/summarizeAbundance.py</a>„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p result/summ_table
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -f result/eggnog/eggnog_anno_output <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl"><span class="c1"># Ê±áÊÄªÔºå9ÂàóKOÔºå16ÂàóCAZyÊåâÈÄóÂè∑ÂàÜÈöîÔºå21ÂàóCOGÊåâÂ≠óÊØçÂàÜÈöîÔºåÂéüÂßãÂÄºÁ¥ØÂä†</span>
</span></span><span class="line"><span class="cl">~/db/script/summarizeAbundance.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -i result/salmon/gene.count <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -m result/eggnog/eggnog_anno_output <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -c <span class="s1">&#39;9,16,21&#39;</span> -s <span class="s1">&#39;,+,+*&#39;</span> -n raw <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -o result/summ_table/eggnog
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;s/^ko://&#39;</span> summ_table/eggnog.KO.raw.txt
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="ÂàÜÁÆ±binning">ÂàÜÁÆ±ÔºàbinningÔºâ</h2>
<p>ÂÆèÂü∫Âõ†ÁªÑbinningÊòØÊåáÂ∞Ü‰∏çÂêåÁöÑÂ∫èÂàóÈõÜÂêàÔºàÂ¶ÇmetagenomeÂ∫èÂàóÈõÜÂêàÔºâÊ†πÊçÆÂÆÉ‰ª¨ÁöÑÁâ©ÁßçÂΩíÁ±ªÂà∞‰∏çÂêåÁöÑbins‰∏≠Ôºå‰ª•‰æøËøõ‰∏ÄÊ≠•Á†îÁ©∂ÂÆÉ‰ª¨ÁöÑÁªÑÊàêÂíåÂäüËÉΩ„ÄÇËøô‰∏™ËøáÁ®ãÂèØ‰ª•Â∞ÜÁ±ª‰ººÁöÑÂ∫èÂàóÁªÑÂêàÂú®‰∏ÄËµ∑ÔºåÂΩ¢Êàê‰ª£Ë°®‰∏çÂêåÁâ©ÁßçÊàñÂü∫Âõ†ÁªÑÁöÑbinsÔºå‰ª•‰æøËøõË°åÂêéÁª≠ÂàÜÊûêÔºåÂ¶ÇÁâ©ÁßçÊ≥®Èáä„ÄÅÂü∫Âõ†ÁªÑÁªÑË£ÖÁ≠â„ÄÇ</p>
<p>‰ª•‰∏ãÊòØÂ∏∏Áî®ÁöÑÂÆèÂü∫Âõ†ÁªÑbinningÊñπÊ≥ïÔºö</p>
<ol>
<li>
<p>Âü∫‰∫éËÅöÁ±ªÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®Â∫èÂàóËÅöÁ±ªÂ∞ÜÁõ∏‰ººÂ∫èÂàóÂàÜÂà∞Âêå‰∏Ä‰∏™bin‰∏≠„ÄÇ‰∏ÄËà¨Êù•ËØ¥ÔºåËÅöÁ±ªÁÆóÊ≥ïÂèØÂàÜ‰∏∫‰∏§Á±ªÔºöÊó†ÁõëÁù£ËÅöÁ±ªÔºàÂ¶Çk-means„ÄÅDBSCANÁ≠âÔºâÂíåÊúâÁõëÁù£ËÅöÁ±ªÔºàÂ¶ÇCAMI„ÄÅMyCCÁ≠âÔºâ„ÄÇ</p>
</li>
<li>
<p>Âü∫‰∫éÁªÑË£ÖÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®de novoÁªÑË£ÖÊù•Â∞ÜÁõ∏‰ººÂ∫èÂàóÁªÑË£ÖÊàêËøûÁª≠ÁöÑÂ∫èÂàóÔºåÂÜçÊ†πÊçÆËøô‰∫õÂ∫èÂàóÁöÑÂü∫Âõ†ÁªÑ‰ø°ÊÅØÊù•Â∞ÜÂÖ∂ÂàÜÁ±ªÂà∞‰∏çÂêåÁöÑbins‰∏≠„ÄÇËøôÁßçÊñπÊ≥ïÁöÑ‰ºòÁÇπÊòØÂèØ‰ª•Êõ¥Â•ΩÂú∞Â§ÑÁêÜÈáçÂ§çÂ∫èÂàóÔºåÁº∫ÁÇπÊòØÈúÄË¶ÅÂ§ßÈáèÁöÑËÆ°ÁÆóËµÑÊ∫êÂíåÊó∂Èó¥„ÄÇ</p>
</li>
<li>
<p>Âü∫‰∫éÂàÜÁ±ªÂô®ÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®Êú∫Âô®Â≠¶‰π†ÂàÜÁ±ªÂô®Êù•Â∞ÜÂ∫èÂàóÂàÜÈÖçÂà∞‰∏çÂêåÁöÑbins‰∏≠„ÄÇËøôÁßçÊñπÊ≥ïÁöÑ‰ºòÁÇπÊòØÂèØ‰ª•Ëá™Âä®Â≠¶‰π†ÁâπÂæÅÂπ∂Âú®Â§ÑÁêÜÂ§ßËßÑÊ®°Êï∞ÊçÆÊó∂ÊïàÁéáÈ´òÔºåÁº∫ÁÇπÊòØÈúÄË¶ÅÂÖàÂª∫Á´ã‰∏Ä‰∏™ÂàÜÁ±ªÂô®Âπ∂ËøõË°åËÆ≠ÁªÉ„ÄÇ</p>
</li>
</ol>
<p>Âú®ËøõË°åÂÆèÂü∫Âõ†ÁªÑbinningÊó∂ÔºåÈÄöÂ∏∏ÈúÄË¶Å‰ΩøÁî®Â§ö‰∏™ÊñπÊ≥ïËøõË°åÊØîËæÉÔºå‰ª•ÈÄâÊã©ÊúÄÈÄÇÂêàÊï∞ÊçÆÈõÜÁöÑÊñπÊ≥ï„ÄÇÂèØ‰ª•‰ΩøÁî®‰∏Ä‰∫õÊµÅË°åÁöÑÂ∑•ÂÖ∑Êù•ËøõË°åbinningÔºåÂ¶ÇMetaBAT„ÄÅMaxBin„ÄÅCONCOCTÂíåMEGANÁ≠â„ÄÇËøô‰∫õÂ∑•ÂÖ∑ÈÄöÂ∏∏ÂåÖÂê´ÂêÑÁßçbinningÊñπÊ≥ïÔºåÂèØ‰ª•Ê†πÊçÆÊï∞ÊçÆÈõÜÂíåÂàÜÊûêÁõÆÁöÑÈÄâÊã©ÈÄÇÂêàÁöÑÊñπÊ≥ï„ÄÇ</p>
<p>ÁØáÂπÖÈôêÂà∂ÔºåÂÖ∑‰ΩìÁöÑÊñπÊ≥ïÊîæÂú®Âè¶‰∏ÄÁØáÈáåÈù¢ËÆ≤Ëß£ÂêßÔºåÊü•Áúã<a class="link" href="../binning" >ÂÆèÂü∫Âõ†ÁªÑÂàÜÁÆ±ÊµÅÁ®ã</a>„ÄÇ</p>
<h2 id="reference">Reference</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-liuPracticalGuideAmplicon2021" class="csl-entry">
<p><span class="csl-left-margin">1. </span><span class="csl-right-inline">Y.-X. Liu, Y. Qin, T. Chen, M. Lu, <em>et al.</em>, <a class="link" href="https://doi.org/10.1007/s13238-020-00724-8"  target="_blank" rel="noopener"
    >A practical guide to amplicon and metagenomic analysis of microbiome data</a>. <em>Protein &amp; Cell</em>. <strong>12</strong>, 315‚Äì330 (2021).</span></p>
</div>
</div>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/microbiome/">Microbiome</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà4Ôºâ"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà4Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-3/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-3/images/fig1.20d52f738dfd506b0754bda8d4b0ead0_hu_d6f6d5c8a568f26d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà3Ôºâ"
                        data-key="25-nature-3" 
                        data-hash="md5-INUvc439UGsHVL2o1LDq0A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà3Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-2/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-2/images/2-1-2.f13dbba185acaee28c931cc543f77916_hu_94453c680a4ffc60.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà2Ôºâ"
                        data-key="25-nature-2" 
                        data-hash="md5-8T27oYWsruKMkxzFQ/d5Fg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà2Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature/images/1-1-1.7ec2e930cb5bff1118bfef6c33cf66bd_hu_e11798ac155eaa9a.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà1Ôºâ"
                        data-key="25-nature" 
                        data-hash="md5-fsLpMMtb/xEYv&#43;9sM89mvQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà1Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/indoor/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/indoor/images/ind3.f88b22cb27b61a7f362968c78b715b40_hu_3690543364814c81.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ÂÆ§ÂÜÖÂæÆÁîüÁâ©ÁªÑÁõ∏ÂÖ≥Á†îÁ©∂"
                        data-key="indoor" 
                        data-hash="md5-&#43;Isiyye2Gn82KWjHi3FbQA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ÂÆ§ÂÜÖÂæÆÁîüÁâ©ÁªÑÁõ∏ÂÖ≥Á†îÁ©∂</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
