<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='GTDBï¼ˆGenome Taxonomy Databaseï¼‰æ˜¯ä¸€ä¸ªåŸºå› ç»„åˆ†ç±»å­¦æ•°æ®åº“ï¼Œæ—¨åœ¨æä¾›é«˜è´¨é‡çš„ç»†èŒå’Œå¤ç»†èŒç­‰åˆ†ç±»ä¿¡æ¯ã€‚GTDB-Tk æ˜¯ä¸€ä¸ªè½¯ä»¶å·¥å…·åŒ…ï¼Œç”¨äºæ ¹æ®GTDBå¯¹ç»†èŒå’Œå¤ç»†èŒåŸºå› ç»„è¿›è¡Œå®¢è§‚åˆ†ç±»ã€‚'>
<title>GTDBåŸºå› ç»„åˆ†ç±»æ•°æ®åº“åŠGTDB-Tkå·¥å…·</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/gtdb/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='GTDBåŸºå› ç»„åˆ†ç±»æ•°æ®åº“åŠGTDB-Tkå·¥å…·'>
<meta property='og:description' content='GTDBï¼ˆGenome Taxonomy Databaseï¼‰æ˜¯ä¸€ä¸ªåŸºå› ç»„åˆ†ç±»å­¦æ•°æ®åº“ï¼Œæ—¨åœ¨æä¾›é«˜è´¨é‡çš„ç»†èŒå’Œå¤ç»†èŒç­‰åˆ†ç±»ä¿¡æ¯ã€‚GTDB-Tk æ˜¯ä¸€ä¸ªè½¯ä»¶å·¥å…·åŒ…ï¼Œç”¨äºæ ¹æ®GTDBå¯¹ç»†èŒå’Œå¤ç»†èŒåŸºå› ç»„è¿›è¡Œå®¢è§‚åˆ†ç±»ã€‚'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/gtdb/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='genome' /><meta property='article:tag' content='database' /><meta property='article:published_time' content='2024-05-20T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-05-20T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/gtdb/images/gtdb.png' />
<meta name="twitter:title" content="GTDBåŸºå› ç»„åˆ†ç±»æ•°æ®åº“åŠGTDB-Tkå·¥å…·">
<meta name="twitter:description" content="GTDBï¼ˆGenome Taxonomy Databaseï¼‰æ˜¯ä¸€ä¸ªåŸºå› ç»„åˆ†ç±»å­¦æ•°æ®åº“ï¼Œæ—¨åœ¨æä¾›é«˜è´¨é‡çš„ç»†èŒå’Œå¤ç»†èŒç­‰åˆ†ç±»ä¿¡æ¯ã€‚GTDB-Tk æ˜¯ä¸€ä¸ªè½¯ä»¶å·¥å…·åŒ…ï¼Œç”¨äºæ ¹æ®GTDBå¯¹ç»†èŒå’Œå¤ç»†èŒåŸºå› ç»„è¿›è¡Œå®¢è§‚åˆ†ç±»ã€‚"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/gtdb/images/gtdb.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">ğŸ›ğŸ¶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/472862038'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/JB-Cao'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >ä¸­æ–‡</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#gtdbæ•°æ®åº“">GTDBæ•°æ®åº“</a>
      <ol>
        <li><a href="#browsers">Browsers</a></li>
        <li><a href="#tools">Tools</a></li>
        <li><a href="#downloads">Downloads</a></li>
        <li><a href="#statistics">Statistics</a></li>
        <li><a href="#forum">Forum</a></li>
        <li><a href="#help">Help</a></li>
      </ol>
    </li>
    <li><a href="#gtdb-tk">GTDB-Tk</a>
      <ol>
        <li><a href="#install">Install</a>
          <ol>
            <li><a href="#1conda">1.conda</a></li>
            <li><a href="#2pip">2.pip</a></li>
            <li><a href="#3docker">3.docker</a></li>
            <li><a href="#reference-data">Reference data</a></li>
            <li><a href="#check">Check</a></li>
          </ol>
        </li>
        <li><a href="#ä½¿ç”¨æ–¹æ³•">ä½¿ç”¨æ–¹æ³•</a>
          <ol>
            <li><a href="#classify_wf-æ³¨é‡Šç‰©ç§">classify_wf æ³¨é‡Šç‰©ç§</a></li>
            <li><a href="#output">output</a></li>
          </ol>
        </li>
        <li><a href="#åˆ†æ­¥ç¤ºä¾‹">åˆ†æ­¥ç¤ºä¾‹</a>
          <ol>
            <li><a href="#è·å–æ•°æ®">è·å–æ•°æ®</a></li>
            <li><a href="#åŸºå› é‰´å®š-identify">åŸºå› é‰´å®š (identify)</a></li>
            <li><a href="#åŸºå› æ¯”å¯¹-align">åŸºå› æ¯”å¯¹ (align)</a></li>
            <li><a href="#ç‰©ç§åˆ†ç±»-classify">ç‰©ç§åˆ†ç±» (classify)</a></li>
          </ol>
        </li>
        <li><a href="#æ³¨æ„äº‹é¡¹">æ³¨æ„äº‹é¡¹</a></li>
      </ol>
    </li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/gtdb/">
                <img src="/Cao-blog/p/gtdb/images/gtdb_hu_db9b99b2154e9c0f.png"
                        srcset="/Cao-blog/p/gtdb/images/gtdb_hu_db9b99b2154e9c0f.png 800w, /Cao-blog/p/gtdb/images/gtdb_hu_7152a73598cda3f2.png 1600w"
                        width="800" 
                        height="408" 
                        loading="lazy"
                        alt="Featured image of post GTDBåŸºå› ç»„åˆ†ç±»æ•°æ®åº“åŠGTDB-Tkå·¥å…·" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/gtdb/">GTDBåŸºå› ç»„åˆ†ç±»æ•°æ®åº“åŠGTDB-Tkå·¥å…·</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            GTDBï¼ˆGenome Taxonomy Databaseï¼‰æ˜¯ä¸€ä¸ªåŸºå› ç»„åˆ†ç±»å­¦æ•°æ®åº“ï¼Œæ—¨åœ¨æä¾›é«˜è´¨é‡çš„ç»†èŒå’Œå¤ç»†èŒç­‰åˆ†ç±»ä¿¡æ¯ã€‚GTDB-Tk æ˜¯ä¸€ä¸ªè½¯ä»¶å·¥å…·åŒ…ï¼Œç”¨äºæ ¹æ®GTDBå¯¹ç»†èŒå’Œå¤ç»†èŒåŸºå› ç»„è¿›è¡Œå®¢è§‚åˆ†ç±»ã€‚
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">May 20, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    8 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                ç‰ˆæƒå½’Cao's blogæ‰€æœ‰ï¼Œè½¬è½½è¯·æ³¨æ˜å‡ºå¤„
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>GTDBï¼ˆGenome Taxonomy Databaseï¼‰æ˜¯ä¸€ä¸ªä¸“æ³¨äºåŸºå› ç»„åˆ†ç±»å­¦çš„æ•°æ®åº“ï¼Œæ—¨åœ¨æä¾›é«˜è´¨é‡çš„ç»†èŒã€å¤ç»†èŒã€å¶ç»¿ä½“å’Œçº¿ç²’ä½“çš„åˆ†ç±»ä¿¡æ¯ã€‚é€šè¿‡ä½¿ç”¨å…¨åŸºå› ç»„ä¿¡æ¯ï¼ŒGTDBæä¾›äº†å¤šçº§åˆ«çš„åˆ†ç±»ï¼ŒåŒ…æ‹¬åŸŸã€ç•Œã€çº²ã€ç›®ã€ç§‘ã€å±ç­‰çº§åˆ«ã€‚å…¶ç³»ç»Ÿå‘è‚²åˆ†ç±»ç³»ç»Ÿä½¿å¾—ç ”ç©¶äººå‘˜èƒ½å¤Ÿæ·±å…¥äº†è§£ç”Ÿç‰©çš„æ¼”åŒ–å’ŒåŠŸèƒ½ã€‚GTDBä¸»è¦ä»¥ç»†èŒä¸­æ™®éå­˜åœ¨çš„120ä¸ªå•æ‹·è´è›‹ç™½è´¨ï¼ˆbac120ï¼‰å’Œå¤èŒä¸­53ä¸ªï¼ˆarc53ï¼Œä»R07-RS207å¼€å§‹ï¼‰æ ‡è®°è›‹ç™½ä¸ºåŸºç¡€ï¼Œåœ¨å¯¹å¤šåˆ†ç»„ç±»åˆ«æ¶ˆæ­§åï¼Œæ ¹æ®ç›¸å¯¹æ¼”åŒ–æ•£åº¦æ ‡å‡†åŒ–å’Œåˆ†çº§ï¼Œå¾—åˆ°åŸºå› ç»„åˆ†ç±»ç»“æœã€‚</p>
<p>GTDBå®šæœŸæ›´æ–°ï¼Œä¸å…¶ä»–ç”Ÿç‰©ä¿¡æ¯æ•°æ®åº“æ•´åˆï¼Œä¸ºåŸºå› ç»„å­¦ç ”ç©¶å’Œå¾®ç”Ÿç‰©ç”Ÿæ€å­¦ç­‰é¢†åŸŸçš„ç ”ç©¶æä¾›äº†å…¨é¢ä¸”å¯é çš„åˆ†ç±»ä¿¡æ¯ã€‚GTDBæ•°æ®åº“å·²ç»äº2018/2020è¿ç»­å‘è¡¨ä¸¤ç¯‡Nature Biotechnologyï¼Œ2021å¹´åˆå‘è¡¨åœ¨Nature Microbiologyå’ŒNucleic Acids
Researchä¸Šï¼Œæ˜¯ç›®å‰æœ€æƒå¨çš„å¾®ç”Ÿç‰©åˆ†ç±»æ•°æ®åº“ä¹‹ä¸€ã€‚
GTDBé…å¥—è½¯ä»¶åœ¨2019/2022å‘è¡¨äº†ä¸¤ç¯‡Bioinformatcsï¼Œæ—¶æä¾›äº†ç‰©ç§åˆ†ç±»ã€è¿›åŒ–æ ‘æ„å»ºçš„å…¨å¥—å·¥å…·ã€‚</p>
<p>å®˜ç½‘ï¼š<a class="link" href="https://gtdb.ecogenomic.org/"  target="_blank" rel="noopener"
    >https://gtdb.ecogenomic.org/</a></p>
<img src="images/gtdb.png" style="width:100.0%" />
<h2 id="gtdbæ•°æ®åº“">GTDBæ•°æ®åº“</h2>
<p>å¯ä»¥çœ‹åˆ°ä¸Šé¢çš„å®˜ç½‘ä¸»é¡µï¼Œç›®å‰æœ€æ–°ç‰ˆçš„GTDBæ•°æ®åº“åŒ…å«äº†584,382ä¸ªç»†èŒåŸºå› ç»„å’Œ12,477ä¸ªå¤èŒåŸºå› ç»„ï¼ˆ2024.04.24ï¼ŒRelease 09-RS220ï¼‰ï¼Œåœ¨å±ã€ç§åˆ†è¾¨ç‡æ°´å¹³ä¸Šæè¿°äº†190å¤šä¸ªé—¨ã€‚</p>
<h3 id="browsers">Browsers</h3>
<p>GTDBæ”¯æŒä»¥Alphabetical
tableå’ŒTreeä¸¤ç§å½¢å¼æ¥æ¢ç´¢æˆ–æŸ¥è¯¢æ•´ä¸ªGTDBæ•°æ®åº“ï¼ŒåŒ…æ‹¬åˆ†ç±»ä¿¡æ¯å’ŒåŸºå› ç»„ä¿¡æ¯ã€‚</p>
<p>æ­¤å¤–ï¼Œå¯ä»¥åœ¨Taxon Historyè¾“å…¥ GTDB æˆ– NCBI
åˆ†ç±»å•å…ƒï¼Œä»¥å¯è§†åŒ–åˆ†é…ç»™è¯¥åˆ†ç±»å•å…ƒçš„åŸºå› ç»„åœ¨ç‰ˆæœ¬ä¹‹é—´å¦‚ä½•å˜åŒ–ï¼Œæ¯”å¦‚p__Proteobacteriaä»Release
214å¼€å§‹æ”¹åä¸ºp__Pseudomonadotaï¼Œåœ¨ä¸æ–­å£®å¤§çš„åŒæ—¶ä¹Ÿæœ‰ä¸€éƒ¨åˆ†genomeè¢«åˆ†é…ç»™äº†å…¶ä»–é—¨ã€‚</p>
<h3 id="tools">Tools</h3>
<p>GTDBæä¾›äº†ä¸€äº›é…å¥—çš„å·¥å…·ï¼š</p>
<ul>
<li>GTDB-Tkï¼šä¸‹ä¸€ç« èŠ‚è¯¦ç»†ä»‹ç»</li>
<li>Third-party toolsï¼šAnnoTreeï¼ŒIDTAXAï¼ŒSingleMï¼Œsourmash</li>
<li>Third-party GTDB
databasesï¼šHUMANn2ï¼ŒKraken2ï¼ŒBracken2ï¼ŒCentrifugeï¼ŒKrakenï¼ŒDADA2ç­‰</li>
<li>Third-party
scriptsï¼štax_from_gtdb.pyï¼Œgenome_updaterï¼Œgtdb_to_taxdump</li>
<li>FastANI calculatorï¼šå¯ç”¨äºåŸæ ¸ç”Ÿç‰© NCBI
åŸºå› ç»„çš„ANIæ¯”è¾ƒï¼Œä¸€æ¬¡æœ€å¤šå¯ä»¥è¿›è¡Œ 1,000 æ¬¡æˆå¯¹æ¯”è¾ƒï¼Œä¾‹å¦‚ä¸‹å›¾ã€‚</li>
</ul>
<img src="images/ani.png" title=""/>
<h3 id="downloads">Downloads</h3>
<p>è¯¥é¡µé¢æä¾›äº†GTDBæ•°æ®åº“å’Œé…å¥—è½¯ä»¶çš„ä¸‹è½½é“¾æ¥ã€‚</p>
<h3 id="statistics">Statistics</h3>
<p>ç»Ÿè®¡é¡µé¢æä¾›äº†å„ä¸ªç‰ˆæœ¬çš„GTDBæ•°æ®åº“çš„å„ç±»åŸºç¡€ä¿¡æ¯ï¼š</p>
<ol>
<li>
<p>Taxon overviewï¼šGTDB R220 åŒ…å« 596,859 ä¸ªåŸºå› ç»„ï¼Œç»„ç»‡æˆ 113,104
ä¸ªç‰©ç§ç°‡ã€‚</p>
</li>
<li>
<p>Species overviewï¼šGTDB R220 ç”± 584,382 ä¸ªç»†èŒå’Œ 12,477
ä¸ªå¤ç»†èŒåŸºå› ç»„ç»„æˆï¼Œåˆ†ä¸º 107,235 ä¸ªç»†èŒå’Œ 5,869 ä¸ªå¤ç»†èŒç‰©ç§ç°‡ã€‚</p>
</li>
<li>
<p>Genome categoriesï¼šGTDB
åˆ†ç±»ç¾¤ç”±åˆ†ç¦»åŸºå› ç»„ï¼ˆisolateï¼‰ã€å®åŸºå› ç»„ç»„è£…åŸºå› ç»„ (MAG)
å’Œå•æ‰©å¢åŸºå› ç»„ (SAG) ç»„æˆã€‚</p>
</li>
<li>
<p>GTDB species representativesï¼šæ¯ä¸ª GTDB ç‰©ç§ç°‡éƒ½ç”±ä¸€ä¸ªåŸºå› ç»„è¡¨ç¤ºã€‚</p>
</li>
<li>
<p>Quality of GTDB representative genomesï¼šè¢«é€‰ä¸º GTDB
ç‰©ç§ä»£è¡¨çš„åŸºå› ç»„çš„è´¨é‡ã€‚ä½¿ç”¨ CheckM ä¼°è®¡åŸºå› ç»„å®Œæ•´æ€§å’Œæ±¡æŸ“ï¼Œå¹¶æ ¹æ®
MIMAG
åŸºå› ç»„æ ‡å‡†è¿›è¡Œç€è‰²ã€‚ä¸€èˆ¬æ¥è¯´ï¼Œä»£è¡¨æ€§åŸºå› ç»„ä»…é™äºæ»¡è¶³<code>completeness - 5*contamination &gt;50</code>ï¼Œå­˜åœ¨ä¸€äº›ä¾‹å¤–æƒ…å†µï¼Œå¦‚ä¸‹é¢æ‰€ç¤ºã€‚
<img src="images/genomes.png" title=""/></p>
</li>
<li>
<p>Taxa with the largest number of speciesï¼šç»™å‡ºäº†æ¯ä¸ªåˆ†ç±»ç­‰çº§åŒ…å«æœ€å¤š
GTDB
ç‰©ç§ç°‡çš„åˆ†ç±»å•å…ƒï¼Œé—¨æ°´å¹³æ˜¯å¸¸è§çš„Pseudomonadotaï¼ŒBacillotaï¼ŒBacteroidotaï¼ŒActinomycetotaç­‰ã€‚</p>
</li>
<li>
<p>Taxa with the largest number of sequenced genomesï¼Œä¸6ç±»ä¼¼ã€‚</p>
</li>
<li>
<p>Relative evolutionary
divergenceï¼ˆREDï¼‰ï¼šæä¾›äº†ä»é—¨åˆ°å±çš„æ¯ä¸ªåˆ†ç±»ç­‰çº§çš„ç±»ç¾¤çš„ç›¸å¯¹è¿›åŒ–åˆ†æ­§ï¼ˆREDï¼‰ã€‚
RED å€¼æä¾›äº†ç›¸å¯¹æ—¶é—´çš„æ“ä½œè¿‘ä¼¼å€¼ï¼Œå…¶ä¸­ç°æœ‰åˆ†ç±»ç¾¤å­˜åœ¨äºå½“å‰
(RED=1)ï¼Œæœ€åä¸€ä¸ªå…±åŒç¥–å…ˆå‡ºç°åœ¨è¿‡å»çš„å›ºå®šæ—¶é—´
(RED=0)ï¼Œå†…éƒ¨èŠ‚ç‚¹æ ¹æ®ä»¥ä¸‹å€¼åœ¨è¿™äº›å€¼ä¹‹é—´çº¿æ€§æ’å€¼è°±ç³»ç‰¹å®šçš„è¿›åŒ–é€Ÿç‡ã€‚</p>
</li>
<li>
<p>Comparison of GTDB and NCBI taxaï¼šGTDB
ç‰©ç§ä»£è¡¨æ€§åŸºå› ç»„å’Œæ‰€æœ‰å…·æœ‰åˆ†é…çš„ NCBI åˆ†ç±»æ³•çš„ GTDB åŸºå› ç»„çš„ GTDB å’Œ
NCBI
åˆ†ç±»å­¦åˆ†é…çš„æ¯”è¾ƒã€‚å¯¹äºæ¯ä¸ªåˆ†ç±»ç­‰çº§ï¼Œå¦‚æœä¸€ä¸ªåˆ†ç±»å•å…ƒåœ¨ä¸¤ä¸ªåˆ†ç±»æ ‡å‡†ä¸­åç§°ç›¸åŒï¼Œåˆ™è¯¥åˆ†ç±»å•å…ƒè¢«åˆ†ç±»ä¸º<code>unchange</code>ï¼›å¦‚æœ
GTDB åˆ†ç±»æ ‡å‡†æä¾›äº† NCBI
åˆ†ç±»æ ‡å‡†ä¸­ç¼ºå°‘çš„åç§°ä¿¡æ¯ï¼Œåˆ™è¯¥åˆ†ç±»å•å…ƒè¢«åˆ†ç±»ä¸º<code>passive change</code>ï¼›å¦‚æœä¸¤ä¸ªåˆ†ç±»æ ‡å‡†ä¹‹é—´çš„åç§°ä¸åŒï¼Œåˆ™è¯¥åˆ†ç±»å•å…ƒè¢«åˆ†ç±»ä¸º<code>actively changed</code>ã€‚å¯ä»¥çœ‹åˆ°ï¼Œåœ¨genomeæ°´å¹³ä¸ŠGTDBå’ŒNCBIè¿˜æ˜¯æœ‰å¾ˆå¤šåç§°ä¸å¯¹åº”çš„ï¼Œå¯ä»¥è€ƒè™‘ä½¿ç”¨gtdb_to_taxdumpè¿›è¡Œè½¬åŒ–ã€‚
<img src="images/gtdb2ncbi.png" title=""/></p>
</li>
<li>
<p>Genomic statisticsï¼šGTDB ç‰©ç§ä»£è¡¨æ€§åŸºå› ç»„å’Œ GTDB
ä¸­æ‰€æœ‰åŸºå› ç»„çš„å…³é”®åŸºå› ç»„ç»Ÿè®¡æ•°æ®ã€‚ <img src="images/genome_stat.png" title=""/></p>
</li>
<li>
<p>Nomenclatural types per
rankï¼šæ˜¾ç¤ºäº†æ¯ä¸ªåˆ†ç±»ç­‰çº§çš„å ä½ç¬¦ä¸æ‹‰ä¸åŒ–åç§°çš„ç»†åˆ†ã€‚</p>
</li>
</ol>
<h3 id="forum">Forum</h3>
<p>GTDBè¿˜ä¸“é—¨æ­å»ºäº†ä¸€ä¸ªç”¨æˆ·ç¤¾åŒºå’Œè®¨è®ºåŒºï¼š</p>
<ul>
<li>Announcementï¼šç”¨äºå…¬å¸ƒæœ€æ–° GTDB
æ–°é—»ã€å³å°†ä¸¾è¡Œçš„ä¼šè®®å’Œä»¥åˆ†ç±»å­¦ä¸ºé‡ç‚¹çš„æ‰‹ç¨¿çš„å¼€æ”¾è®ºå›ã€‚</li>
<li>Taxonomy
(Bacteria/Archaea)ï¼šè¿™ä¸ªåˆ†ç±»ä¸“æ³¨äºè®¨è®ºä¸GTDBåˆ†ç±»æ³•ç›¸å…³çš„å†…å®¹ã€‚ç”¨æˆ·åœ¨è¿™é‡Œè®¨è®ºåˆ†ç±»æ³•çš„å…·ä½“ç»†èŠ‚ã€ä¸å…¶ä»–æ•°æ®åº“ï¼ˆå¦‚NCBIï¼‰çš„æ¯”è¾ƒã€åˆ†ç±»æ ‡å‡†çš„æ›´æ–°ç­‰ã€‚</li>
<li>GTDB-Tkï¼šä¸“æ³¨äºGTDB-Tkå·¥å…·åŒ…çš„ä½¿ç”¨å’Œè®¨è®ºã€‚ç”¨æˆ·å¯ä»¥åœ¨è¿™é‡Œæå‡ºå…³äºGTDB-Tkçš„ä½¿ç”¨é—®é¢˜ã€åˆ†äº«ä½¿ç”¨ç»éªŒã€è®¨è®ºå·¥å…·åŒ…çš„åŠŸèƒ½å’Œæ€§èƒ½ç­‰ã€‚</li>
<li>GTDBï¼šä¸åŸºå› ç»„åˆ†ç±»æ•°æ®åº“ (GTDB) ç›¸å…³çš„ä¸€èˆ¬é—®é¢˜ã€‚</li>
<li>Website feedbackï¼šè®¨è®ºè¯¥ç½‘ç«™ã€å…¶ç»„ç»‡ã€å…¶è¿ä½œæ–¹å¼ä»¥åŠå¦‚ä½•æ”¹è¿›å®ƒã€‚</li>
</ul>
<h3 id="help">Help</h3>
<p>å¸®åŠ©é¡µé¢ä¹Ÿæä¾›äº†ä¸€äº›éå¸¸æœ‰ç”¨çš„ä¿¡æ¯ã€‚</p>
<ul>
<li>
<p>Aboutï¼šGTDBç®€è¦ä»‹ç»ï¼Œç®¡ç†å›¢é˜Ÿä»¥åŠå‘è¡¨è®ºæ–‡</p>
</li>
<li>
<p>FAQï¼šåˆ—å‡ºäº†ä¸€äº›å¸¸è§é—®é¢˜ï¼Œä¾‹å¦‚ï¼š</p>
<ol>
<li>ä¸ºä»€ä¹ˆé—¨åç§°çš„åç¼€æ›´æ”¹ä¸º-otaï¼ŸWhitman et al.Â (2018)
å»ºè®®åƒå…¶ä»–ç­‰çº§ä¸€æ ·è§„èŒƒé—¨ç­‰çº§çš„åç¼€ã€‚</li>
<li>ä¸ºä»€ä¹ˆä¸€äº›familyå’Œæ›´é«˜çº§åˆ«çš„åå­—ä»¥å­—æ¯åç¼€ç»“å°¾ï¼Ÿå±ç­‰çº§ä»¥ä¸Šçš„åˆ†ç±»å•å…ƒåç§°é™„åŠ å­—æ¯åç¼€è¡¨ç¤ºå±äºä»¥ä¸‹ç±»åˆ«çš„ç±»ç¾¤ï¼š
<ol>
<li>GTDB å‚è€ƒæ ‘ä¸­ä¸æ˜¯å•ç³»ç±»ç¾¤ï¼Œä½†å­˜åœ¨å…¶ä»–è¯æ®è¡¨æ˜å®ƒä»¬æ˜¯å•ç³»ç±»ç¾¤ï¼›</li>
<li>åœ¨ç‰ˆæœ¬ä¹‹é—´ä½ç½®ä¸ç¨³å®šçš„ç»„ã€‚</li>
</ol>
</li>
</ol>
</li>
<li>
<p>Methodsï¼šå¯¹GTDBæ•°æ®åº“æ„å»ºè¿‡ç¨‹ä¸­çš„æ–¹æ³•è¿›è¡Œå…·ä½“æè¿°ï¼Œä¾‹å¦‚ï¼š</p>
<ol>
<li>Gene
identificationï¼Œä½¿ç”¨Prodigalé‰´å®šåŸºå› ï¼ŒHMMERé‰´å®šæ ‡è®°åŸºå› ï¼ˆæ¥è‡ªPfamå’ŒTIGRFAMï¼‰å¹¶åºåˆ—æ¯”å¯¹ã€‚</li>
<li>Multiple sequence alignmentï¼Œç»†èŒå’Œå¤èŒå¤šé‡åºåˆ—æ¯”å¯¹ (MSA) åˆ†åˆ«ç”±
120 ä¸ª (bac120) æˆ– 53 ä¸ª (arc53) ç³»ç»Ÿå‘è‚²ä¿¡æ¯æ ‡è®°ä¸²è”è€Œæˆã€‚</li>
<li>Tree inferenceï¼Œç»†èŒå‚è€ƒæ ‘æ˜¯åœ¨ WAG æ¨¡å‹ä¸‹ä½¿ç”¨
FastTreeã€‚å¤èŒå‚è€ƒæ ‘æ˜¯æ ¹æ® PMSF æ¨¡å‹ä¸‹çš„ IQ-Treeï¼Œå¹¶ä½¿ç”¨
FastTreeæ¨æ–­åˆå§‹æ ‘ä»¥æ¢å¤å•ç³»è°±ç³»ã€‚</li>
<li>Identifying 16S rRNA sequencesï¼Œä½¿ç”¨ nhmmerä»¥åŠæ¥è‡ª RFAM
æ•°æ®åº“çš„ 16S rRNA æ¨¡å‹ï¼ˆRF00177ï¼‰ã€‚</li>
<li>Average nucleotide identityï¼Œskaniï¼ˆä»ç‰ˆæœ¬ 220
å¼€å§‹ï¼‰ï¼Œä¹‹å‰æ˜¯FastANIã€‚</li>
<li>RED
normalizationï¼šç›¸å¯¹è¿›åŒ–åˆ†æ­§ç”¨äºå¸®åŠ©æç»˜ç‰©ç§ç­‰çº§ä¹‹ä¸Šçš„åˆ†ç±»ç¾¤ï¼ŒREDçš„æ¦‚å¿µæ˜¯ç›¸åŒç­‰çº§çš„ç±»ç¾¤åœ¨è¿‡å»åº”è¯¥åŒæ—¶å‡ºç°ã€‚RED
æä¾›äº†ç›¸å¯¹æ—¶é—´çš„æ“ä½œè¿‘ä¼¼å€¼ï¼Œå…¶ä¸­ç°æœ‰åˆ†ç±»ç¾¤å­˜åœ¨äºå½“å‰ (RED =
1)ï¼Œæœ€åä¸€ä¸ªå…±åŒç¥–å…ˆå‡ºç°åœ¨è¿‡å»çš„å›ºå®šæ—¶é—´ (RED =
0)ï¼Œå†…éƒ¨èŠ‚ç‚¹æ ¹æ®è°±ç³»åœ¨è¿™äº›å€¼ä¹‹é—´çº¿æ€§æ’å€¼-ç‰¹å®šçš„è¿›åŒ–é€Ÿç‡ã€‚</li>
</ol>
</li>
<li>
<p>APIï¼šåŠŸèƒ½è¾ƒå¤šï¼Œä½†æ–‡æ¡£ä¸æ˜¯å¾ˆè¯¦ç»†ï¼Œæœªæ¥ä¹Ÿè®¸ä¼šç”¨ä¸Š</p>
</li>
</ul>
<h2 id="gtdb-tk">GTDB-Tk</h2>
<p>GTDB-Tk æ˜¯ä¸€ä¸ªè½¯ä»¶å·¥å…·åŒ…ï¼Œç”¨äºæ ¹æ®åŸºå› ç»„æ•°æ®åº“åˆ†ç±»æ³• (GTDB)
å¯¹ç»†èŒå’Œå¤ç»†èŒåŸºå› ç»„è¿›è¡Œå®¢è§‚åˆ†ç±»ã€‚
å®ƒæ—¨åœ¨ä¸æœ€æ–°è¿›å±•ç›¸ç»“åˆï¼Œå…è®¸ç›´æ¥ä»ç¯å¢ƒæ ·æœ¬ä¸­è·å¾—æ•°ç™¾æˆ–æ•°åƒä¸ªå®åŸºå› ç»„ç»„è£…åŸºå› ç»„
(MAG)ï¼Œè¿˜å¯ä»¥åº”ç”¨äºåˆ†ç¦»å’Œå•ç»†èƒåŸºå› ç»„ã€‚</p>
<p>åœ°å€ï¼š<a class="link" href="https://github.com/Ecogenomics/GTDBTk"  target="_blank" rel="noopener"
    >https://github.com/Ecogenomics/GTDBTk</a></p>
<p>å®˜æ–¹æ–‡æ¡£ï¼š<a class="link" href="https://ecogenomics.github.io/GTDBTk/"  target="_blank" rel="noopener"
    >https://ecogenomics.github.io/GTDBTk/</a></p>
<h3 id="install">Install</h3>
<p>é¦–å…ˆè¦å®‰è£…è½¯ä»¶å’Œä¸‹è½½å¯¹åº”çš„Reference
dataï¼Œæœ€æ–°çš„R220æ•°æ®åº“éœ€è¦ä½¿ç”¨v2.4.0çš„GTDB-Tkï¼ˆ2024.04.24ï¼‰ã€‚å¦‚æœè¦å’Œä¹‹å‰é¡¹ç›®ä¸€è‡´ï¼Œå¯ä»¥ä½¿ç”¨R214ç‰ˆæœ¬æ•°æ®åº“å¯¹åº”v2.3.2çš„GTDB-Tkã€‚</p>
<h4 id="1conda">1.conda</h4>
<p>condaå®‰è£…æ¯”è¾ƒç®€å•ï¼Œä½†æœ€å¥½æ–°å»ºä¸€ä¸ªç¯å¢ƒï¼Œå¯èƒ½ä¼šæ¯”è¾ƒæ…¢ï¼ˆmambaä»£æ›¿condaä¼šå¿«ä¸€ç‚¹ï¼‰ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># using conda</span>
</span></span><span class="line"><span class="cl">conda create -n gtdbtk-2.3.2 -c conda-forge -c bioconda <span class="nv">gtdbtk</span><span class="o">=</span>2.3.2
</span></span><span class="line"><span class="cl"><span class="c1"># ä¸‹è½½ GTDB-Tk å‚è€ƒæ•°æ®å¹¶ä¸ºå…¶å‘½å</span>
</span></span><span class="line"><span class="cl">download-db.sh
</span></span><span class="line"><span class="cl"><span class="c1"># æ¿€æ´»GTDB-Tk condaç¯å¢ƒ</span>
</span></span><span class="line"><span class="cl">conda æ¿€æ´» gtdbtk-2.3.2
</span></span><span class="line"><span class="cl"><span class="c1"># å°†ç¯å¢ƒå˜é‡è®¾ç½®ä¸ºåŒ…å«GTDB-Tkå‚è€ƒæ•°æ®çš„ç›®å½•</span>
</span></span><span class="line"><span class="cl">conda env config vars <span class="nb">set</span> <span class="nv">GTDBTK_DATA_PATH</span><span class="o">=</span><span class="s2">&#34;/path/to/release/package/&#34;</span><span class="p">;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="2pip">2.pip</h4>
<p>pipå¯ä»¥å¿«é€Ÿå®‰è£…gtdbtkåŒ…ï¼Œä½†æ˜¯å®ƒçš„ç¬¬ä¸‰æ–¹ä¾èµ–é¡¹éœ€è¦æ‰‹åŠ¨å®‰è£…ä¸€ä¸‹
ï¼ˆä¸æ˜¯å¾ˆå¤æ‚ï¼Œæˆ‘ç”¨çš„pipå®‰è£…ï¼Œå› ä¸ºä¸æƒ³è¦å¤ªå¤šçš„condaç¯å¢ƒï¼Œå¤ªå†—ä½™äº†ï¼‰ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">python -m pip install <span class="nv">gtdbtk</span><span class="o">==</span>2.3.2
</span></span></code></pre></td></tr></table>
</div>
</div><p>ç¬¬ä¸‰æ–¹ä¾èµ–é¡¹ï¼š</p>
<ul>
<li>Prodigalï¼ˆåŸæ ¸åŸºå› è¯†åˆ«å’Œç¿»è¯‘èµ·å§‹ä½ç‚¹é‰´å®šï¼‰</li>
<li>HMMERï¼ˆåŠ é€Ÿæ—HMM æœç´¢ï¼‰</li>
<li>pplacerï¼ˆçº¿æ€§æ—¶é—´æœ€å¤§ä¼¼ç„¶å’Œè´å¶æ–¯ç³»ç»Ÿå‘è‚²å°†åºåˆ—æ”¾ç½®åˆ°å›ºå®šå‚è€ƒæ ‘ä¸Šï¼‰</li>
<li>skani (v2.4.0å¼€å§‹ä¾èµ–ï¼Œé€šè¿‡ skani
çš„ç¨€ç–é“¾æ¥è¿›è¡Œå¿«é€Ÿã€ç¨³å¥çš„å®åŸºå› ç»„åºåˆ—æ¯”è¾ƒï¼Œæ¯”fastANIè¿˜å¿«ï¼Œä¸‹æ¬¡æœ‰æœºä¼šè®²ä¸€ä¸‹è¿™ä¸ª)</li>
<li></li>
<li>FastTreeï¼ˆç”¨äºå¤§å¯¹é½çš„è¿‘ä¼¼æœ€å¤§ä¼¼ç„¶æ ‘ï¼‰</li>
<li>Mashï¼ˆå¿«é€Ÿä¼°è®¡åŸºå› ç»„å’Œå®åŸºå› ç»„è·ç¦»ï¼‰</li>
</ul>
<p>è¿™äº›éƒ½æ˜¯å¾ˆå¸¸ç”¨çš„è½¯ä»¶ï¼Œæ²¡æœ‰çš„è¯ç”¨condaå®‰è£…ä¸€ä¸‹ã€‚</p>
<p>GTDB-Tk éœ€è¦å°† GTDBTK_DATA_PATH ç¯å¢ƒå˜é‡è®¾ç½®ä¸ºåŒ…å«æœªå­˜æ¡£çš„ GTDB-Tk
å‚è€ƒæ•°æ®çš„ç›®å½•ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export GTDBTK_DATA_PATH=/path/to/release/package/&#39;</span> &gt;&gt; ~/.bashrc
</span></span><span class="line"><span class="cl"><span class="nb">source</span> ~/.bashrc
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="3docker">3.docker</h4>
<p>å¦‚æœç”¨è¿‡dockerçš„è¯ï¼Œè¿™ä¸ªä¹Ÿéå¸¸ç®€å•ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#ä¸‹è½½è§£å‹çš„ GTDB-Tk å‚è€ƒæ•°æ®åˆ°æ‚¨é€‰æ‹©çš„ç›®å½•ï¼Œä¾‹å¦‚ï¼š/host/release_data/</span>
</span></span><span class="line"><span class="cl">mv /path/to/release/package/ /host/release_data/
</span></span><span class="line"><span class="cl"><span class="c1">#ä¸º GTDB-Tk åˆ›å»ºæœ¬åœ° I/O ç›®å½•</span>
</span></span><span class="line"><span class="cl">mkdir /host/gtdbtk_io/
</span></span><span class="line"><span class="cl"><span class="c1"># è¿è¡Œå®¹å™¨</span>
</span></span><span class="line"><span class="cl">docker run -v /host/gtdbtk_io:/data -v /host/release_data:/refdata ecogenomic/gtdbtk classify_wf --genome_dir /data/genomes --out_dir /data/output
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="reference-data">Reference data</h4>
<p>GTDB-Tk éœ€è¦çº¦ 110G
çš„å¤–éƒ¨æ•°æ®ï¼Œéœ€è¦ä¸‹è½½å’Œè§£æ¡£ã€‚ç›´æ¥wgetå¾ˆæ…¢çš„è¯ï¼Œä¹Ÿå¯ä»¥å…ˆä¸‹è½½åˆ°æœ¬åœ°ï¼Œå†é€šè¿‡FTPè½¯ä»¶æˆ–scpä¸Šä¼ åˆ°æœåŠ¡å™¨ï¼š</p>
<p>æ•°æ®åœ°å€ï¼š<a class="link" href="https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/gtdbtk_package/"  target="_blank" rel="noopener"
    >https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/gtdbtk_package/</a></p>
<p>å¯¹äºå®Œæ•´åŒ…(full package)ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">wget https://data.ace.uq.edu.au/public/gtdb/data/releases/latest/auxillary_files/gtdbtk_package/full_package/gtdbtk_data.tar.gz
</span></span><span class="line"><span class="cl">tar xvzf gtdbtk_data.tar.gz
</span></span></code></pre></td></tr></table>
</div>
</div><p>å¯¹äºæ‹†åˆ†åŒ…(split package)ï¼š</p>
<ol>
<li>ç¡®ä¿ GTDB-Tk å‘å¸ƒæ•°æ®çš„æ‰€æœ‰éƒ¨åˆ†éƒ½ä½äºåŒä¸€ç›®å½•ä¸­ã€‚</li>
<li>æ‰“å¼€ç»ˆç«¯æˆ–å‘½ä»¤æç¤ºç¬¦ã€‚</li>
<li>å¯¼èˆªåˆ°åŒ…å« GTDB-Tk å‘å¸ƒæ•°æ®éƒ¨åˆ†çš„ç›®å½•ã€‚</li>
<li>ä½¿ç”¨ä»¥ä¸‹å‘½ä»¤å°†æ‰€æœ‰éƒ¨åˆ†è¿æ¥åˆ°ä¸€ä¸ªå­˜æ¡£ä¸­ï¼š
<code>cat gtdbtk_r220_data.tar.gz.part_* &gt; gtdbtk_r220_data.tar.gz</code></li>
<li>å‘½ä»¤æ‰§è¡Œå®Œæˆåï¼Œå°†åœ¨åŒä¸€ç›®å½•ä¸­æ‹¥æœ‰ä¸€ä¸ªåä¸ºâ€œgtdbtk_r220_data.tar.gzâ€çš„å­˜æ¡£æ–‡ä»¶ã€‚</li>
</ol>
<h4 id="check">Check</h4>
<p>æ£€æŸ¥æ•°æ®åº“å’Œè½¯ä»¶ï¼Œç»“æœæ˜¾ç¤ºOKå’ŒDoneè¡¨ç¤ºæ­£å¸¸ï¼ˆæ£€æŸ¥fastaniè¿™ä¸€æ­¥è¦å¾ˆä¹…ï¼Œæ–‡ä»¶éå¸¸å¤šã€‚æ‰€ä»¥å¦‚æœä¸‰æ–¹è½¯ä»¶éƒ½OKçš„è¯ä¹Ÿå¯ä»¥ç›´æ¥è¿è¡Œä¸‹ä¸€ä¸ªç¤ºä¾‹æµ‹è¯•ï¼‰:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk check_install
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 14:10:37] INFO: GTDB-Tk v2.3.2
[2024-05-20 14:10:37] INFO: gtdbtk check_install
[2024-05-20 14:10:37] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:10:37] INFO: Running install verification
[2024-05-20 14:10:37] INFO: Checking that all third-party software are on the system path:
[2024-05-20 14:10:37] INFO:          |-- FastTree         OK
[2024-05-20 14:10:37] INFO:          |-- FastTreeMP       OK
[2024-05-20 14:10:37] INFO:          |-- fastANI          OK
[2024-05-20 14:10:37] INFO:          |-- guppy            OK
[2024-05-20 14:10:37] INFO:          |-- hmmalign         OK
[2024-05-20 14:10:37] INFO:          |-- hmmsearch        OK
[2024-05-20 14:10:37] INFO:          |-- mash             OK
[2024-05-20 14:10:37] INFO:          |-- pplacer          OK
[2024-05-20 14:10:37] INFO:          |-- prodigal         OK
[2024-05-20 14:10:37] INFO: Checking integrity of reference package: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:10:38] INFO:          |-- pplacer          OK
[2024-05-20 14:10:38] INFO:          |-- masks            OK
[2024-05-20 14:10:39] INFO:          |-- markers          OK
[2024-05-20 14:10:39] INFO:          |-- radii            OK
[2024-05-20 14:10:41] INFO:          |-- msa              OK
[2024-05-20 14:10:41] INFO:          |-- metadata         OK
[2024-05-20 14:10:41] INFO:          |-- taxonomy         OK
[2024-05-20 15:13:45] INFO:          |-- fastani          OK
[2024-05-20 15:13:45] INFO:          |-- mrca_red         OK
[2024-05-20 15:13:45] INFO: Done.
</code></pre>
<p>æ³¨æ„GTDB-Tkçš„ç¡¬ä»¶è¦æ±‚ï¼Œå…³é”®æ˜¯Memoryä¸€å®šè¦ç»™è¶³å¤Ÿï¼Œä¸ç„¶ä¼šç¨‹åºä¼šä¸­é€”ç»ˆæ­¢ï¼š</p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th><strong>Domain</strong></th>
          <th><strong>Memory</strong></th>
          <th><strong>Storage</strong></th>
          <th><strong>Time</strong></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>Archaea</td>
          <td>~60 GB</td>
          <td>~106 GB</td>
          <td>~90 minutes / 1,000 genomes @ 64 CPUs</td>
      </tr>
      <tr>
          <td>Bacteria</td>
          <td>~90GB (545 GB when using â€“full_tree)</td>
          <td>~106 GB</td>
          <td>~90 minutes / 1,000 genomes @ 64 CPUs</td>
      </tr>
  </tbody>
</table></div>
<p>æµ‹è¯•æµç¨‹ï¼Œæ— æŠ¥é”™ç¡®å®šç¨‹åºçš„è½¯ä»¶å’Œæ•°æ®åº“æ­£å¸¸ï¼ˆæ­¤ç¤ºä¾‹æ˜¯3ä¸ªå¤èŒgenomeï¼Œå»ºè®®åˆ†é…60GBå†…å­˜æ¥è¿è¡Œï¼Œå¦å¤–gtdbtk_testå¿…é¡»æ˜¯ä¸å­˜åœ¨çš„æˆ–ç©ºæ–‡ä»¶å¤¹ï¼‰:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk <span class="nb">test</span> --out_dir gtdbtk_test
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 14:34:51] INFO: GTDB-Tk v2.3.2
[2024-05-20 14:34:51] INFO: gtdbtk test --out_dir gtdbtk_test
[2024-05-20 14:34:51] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:34:51] INFO: Command: gtdbtk classify_wf --skip_ani_screen --genome_dir gtdbtk_test/genomes --out_dir gtdbtk_test/output --cpus 1 -f
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:34:52] INFO: gtdbtk classify_wf --skip_ani_screen --genome_dir gtdbtk_test/genomes --out_dir gtdbtk_test/    
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:34:52] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/      
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:42:06] INFO: Note that Tk classification mode is insufficient for publication of new taxonomic designatio    
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:42:06] INFO: Done.     
[2024-05-20 14:42:09] INFO: Test has successfully finished.
</code></pre>
<p>æˆ‘åˆ†é…äº†4æ ¸å…±60GBå†…å­˜è¿è¡Œ<code>gtdbtk test</code>ï¼Œçº¦439så®Œæˆï¼Œæ¶ˆè€—äº†~45GBå†…å­˜ã€‚</p>
<h3 id="ä½¿ç”¨æ–¹æ³•">ä½¿ç”¨æ–¹æ³•</h3>
<p>å…ˆçœ‹çœ‹helpæ–‡æ¡£ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ gtdbtk -h
</span></span><span class="line"><span class="cl"><span class="c1">## GTDB-Tk v2.4.0 å¸®åŠ©æ–‡æ¡£</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Workflows:
</span></span><span class="line"><span class="cl">    classify_wf -&gt; é€šè¿‡åœ¨ GTDB å‚è€ƒæ ‘ä¸­çš„ä½ç½®å¯¹åŸºå› ç»„è¿›è¡Œåˆ†ç±»
</span></span><span class="line"><span class="cl">                     <span class="o">(</span>ani_screening -&gt; identify -&gt; align -&gt; classify<span class="o">)</span>
</span></span><span class="line"><span class="cl">    de_novo_wf  -&gt; æ¨æ–­æ–°çš„æ ‘å¹¶ä½¿ç”¨ GTDB åˆ†ç±»è¿›è¡Œä¿®é¥°
</span></span><span class="line"><span class="cl">                     <span class="o">(</span>identify -&gt; align -&gt; infer -&gt; root -&gt; decorate<span class="o">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Methods:
</span></span><span class="line"><span class="cl">    identify -&gt; è¯†åˆ«åŸºå› ç»„ä¸­çš„æ ‡è®°åŸºå› 
</span></span><span class="line"><span class="cl">    align    -&gt; åˆ›å»ºå¤šåºåˆ—æ¯”å¯¹
</span></span><span class="line"><span class="cl">    classify -&gt; ç¡®å®šåŸºå› ç»„çš„åˆ†ç±»
</span></span><span class="line"><span class="cl">    infer    -&gt; ä»å¤šåºåˆ—æ¯”å¯¹ä¸­æ¨æ–­æ ‘
</span></span><span class="line"><span class="cl">    root     -&gt; ä½¿ç”¨å¤–ç¾¤å¯¹æ ‘è¿›è¡Œæ ¹å®š
</span></span><span class="line"><span class="cl">    decorate -&gt; ç”¨ GTDB åˆ†ç±»ä¿®é¥°æ ‘
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Tools:
</span></span><span class="line"><span class="cl">    infer_ranks        -&gt; åˆ©ç”¨REDå»ºç«‹å†…éƒ¨èŠ‚ç‚¹çš„åˆ†ç±»ç­‰çº§
</span></span><span class="line"><span class="cl">    ani_rep            -&gt; è®¡ç®—ANIåˆ°GTDBçš„ä»£è¡¨æ€§åŸºå› ç»„
</span></span><span class="line"><span class="cl">    trim_msa           -&gt; åŸºäºæ©ç ä¿®å‰ªæœªä¿®å‰ªçš„MSAæ–‡ä»¶
</span></span><span class="line"><span class="cl">    export_msa         -&gt; å¯¼å‡ºæœªä¿®å‰ªçš„å¤ç”ŸèŒæˆ–ç»†èŒMSAæ–‡ä»¶
</span></span><span class="line"><span class="cl">    remove_labels      -&gt; ä»Newickæ ‘ä¸­åˆ é™¤æ ‡ç­¾<span class="o">(</span>å¼•å¯¼å€¼ï¼ŒèŠ‚ç‚¹æ ‡ç­¾<span class="o">)</span>
</span></span><span class="line"><span class="cl">    convert_to_itol    -&gt; å°†GTDB-Tk Newickæ ‘è½¬æ¢ä¸ºiTOLæ ‘
</span></span><span class="line"><span class="cl">    convert_to_species -&gt; å°†GTDBåŸºå› ç»„idè½¬æ¢ä¸ºGTDBç‰©ç§åç§°
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Testing:
</span></span><span class="line"><span class="cl">    <span class="nb">test</span>          -&gt; ç”¨3ä¸ªå¤ç»†èŒåŸºå› ç»„éªŒè¯classify_wfç®¡é“
</span></span><span class="line"><span class="cl">    check_install -&gt; éªŒè¯ç¬¬ä¸‰æ–¹ç¨‹åºå’ŒGTDBå‚è€ƒåŒ…
</span></span></code></pre></td></tr></table>
</div>
</div><p>å¯¹äºæ¯ä¸€ä¸ªæ¨¡å—å¯ä»¥å•ç‹¬æŸ¥çœ‹helpï¼Œä¾‹å¦‚æœ€é‡è¦çš„<code>classify_wf</code>å·¥ä½œæµï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ gtdbtk classify_wf -h
</span></span><span class="line"><span class="cl">usage: gtdbtk classify_wf <span class="o">(</span>--genome_dir GENOME_DIR <span class="p">|</span> --batchfile BATCHFILE<span class="o">)</span> --out_dir OUT_DIR
</span></span><span class="line"><span class="cl">                          <span class="o">(</span>--skip_ani_screen <span class="p">|</span> --mash_db MASH_DB<span class="o">)</span> <span class="o">[</span>--no_mash<span class="o">]</span> <span class="o">[</span>--mash_k MASH_K<span class="o">]</span> <span class="o">[</span>--mash_s MASH_S<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--mash_v MASH_V<span class="o">]</span> <span class="o">[</span>--mash_max_distance MASH_MAX_DISTANCE<span class="o">]</span> <span class="o">[</span>-f<span class="o">]</span> <span class="o">[</span>-x EXTENSION<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--min_perc_aa MIN_PERC_AA<span class="o">]</span> <span class="o">[</span>--prefix PREFIX<span class="o">]</span> <span class="o">[</span>--genes<span class="o">]</span> <span class="o">[</span>--cpus CPUS<span class="o">]</span> <span class="o">[</span>--pplacer_cpus PPLACER_CPUS<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--force<span class="o">]</span> <span class="o">[</span>--scratch_dir SCRATCH_DIR<span class="o">]</span> <span class="o">[</span>--write_single_copy_genes<span class="o">]</span> <span class="o">[</span>--keep_intermediates<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--min_af MIN_AF<span class="o">]</span> <span class="o">[</span>--tmpdir TMPDIR<span class="o">]</span> <span class="o">[</span>--debug<span class="o">]</span> <span class="o">[</span>-h<span class="o">]</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mutually exclusive required arguments:
</span></span><span class="line"><span class="cl">  --genome_dir GENOME_DIR åŒ…å«FASTAæ ¼å¼åŸºå› ç»„æ–‡ä»¶çš„ç›®å½•
</span></span><span class="line"><span class="cl">  --batchfile BATCHFILE æè¿°åŸºå› ç»„çš„æ–‡ä»¶è·¯å¾„ - 2æˆ–3åˆ—çš„åˆ¶è¡¨ç¬¦åˆ†éš”æ–‡ä»¶ï¼ˆFASTAæ–‡ä»¶ã€åŸºå› ç»„IDã€ç¿»è¯‘è¡¨<span class="o">[</span>å¯é€‰<span class="o">]</span>ï¼‰
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">required named arguments:
</span></span><span class="line"><span class="cl">  --out_dir OUT_DIR     è¾“å‡ºæ–‡ä»¶çš„ç›®å½•
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mutually exclusive required arguments:
</span></span><span class="line"><span class="cl">  --skip_ani_screen     è·³è¿‡aniç­›é€‰æ­¥éª¤ï¼Œä½¿ç”¨mashå’Œskaniå¯¹åŸºå› ç»„è¿›è¡Œåˆ†ç±»ã€‚ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --mash_db MASH_DB     ä¿å­˜/è¯»å–Mashå‚è€ƒè‰å›¾æ•°æ®åº“ï¼ˆ.mshï¼‰çš„è·¯å¾„
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional Mash arguments:
</span></span><span class="line"><span class="cl">  --no_mash             è·³è¿‡ä½¿ç”¨Mashè¿›è¡Œçš„åŸºå› ç»„é¢„è¿‡æ»¤ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --mash_k MASH_K       k-merå¤§å°<span class="o">[</span>1-32<span class="o">]</span>ï¼ˆé»˜è®¤ï¼š16ï¼‰
</span></span><span class="line"><span class="cl">  --mash_s MASH_S       æœ€å¤§éå†—ä½™å“ˆå¸Œæ•°ï¼ˆé»˜è®¤ï¼š5000ï¼‰
</span></span><span class="line"><span class="cl">  --mash_v MASH_V       ä¿ç•™çš„æœ€å¤§på€¼<span class="o">[</span>0-1<span class="o">]</span>ï¼ˆé»˜è®¤ï¼š1.0ï¼‰
</span></span><span class="line"><span class="cl">  --mash_max_distance MASH_MAX_DISTANCE é€‰æ‹©æ½œåœ¨GTDBåŸºå› ç»„ä½œä¸ºç”¨æˆ·åŸºå› ç»„ä»£è¡¨çš„æœ€å¤§Mashè·ç¦»ã€‚ï¼ˆé»˜è®¤ï¼š0.15ï¼‰
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional arguments:
</span></span><span class="line"><span class="cl">  -fï¼Œ--full_tree       åœ¨åˆ†ç±»æ­¥éª¤ä¸­ä½¿ç”¨æœªåˆ†å‰²çš„ç»†èŒæ ‘ï¼›è¿™æ˜¯åŸå§‹çš„GTDB-Tkæ–¹æ³•ï¼ˆç‰ˆæœ¬&lt;2ï¼‰ï¼Œ
</span></span><span class="line"><span class="cl">                        éœ€è¦è¶…è¿‡320 GBçš„RAMæ¥åŠ è½½å‚è€ƒæ ‘ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  -xï¼Œ--extension EXTENSION è¦å¤„ç†çš„æ–‡ä»¶æ‰©å±•åï¼Œgz <span class="o">=</span> gzippedï¼ˆé»˜è®¤ï¼šfnaï¼‰
</span></span><span class="line"><span class="cl">  --min_perc_aa MIN_PERC_AA 
</span></span><span class="line"><span class="cl">                        æ’é™¤åœ¨MSAä¸­AAç™¾åˆ†æ¯”ä¸è¶³æ­¤å€¼çš„åŸºå› ç»„ï¼ˆåŒ…å«ç•Œé™ï¼‰ï¼ˆé»˜è®¤ï¼š10ï¼‰
</span></span><span class="line"><span class="cl">  --prefix PREFIX       æ‰€æœ‰è¾“å‡ºæ–‡ä»¶çš„å‰ç¼€ï¼ˆé»˜è®¤ï¼šgtdbtkï¼‰
</span></span><span class="line"><span class="cl">  --genes               è¡¨ç¤ºè¾“å…¥æ–‡ä»¶åŒ…å«é¢„æµ‹çš„è›‹ç™½è´¨ï¼ˆä½œä¸ºæ°¨åŸºé…¸ï¼‰ï¼Œè·³è¿‡åŸºå› è°ƒç”¨ã€‚
</span></span><span class="line"><span class="cl">                        è­¦å‘Šï¼šæ­¤æ ‡å¿—å°†è·³è¿‡ANIæ¯”è¾ƒæ­¥éª¤ï¼ˆaniç­›é€‰å’Œåˆ†ç±»ï¼‰ã€‚ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --cpus CPUS           è¦ä½¿ç”¨çš„CPUæ•°é‡ï¼ˆé»˜è®¤ï¼š1ï¼‰
</span></span><span class="line"><span class="cl">  --pplacer_cpus PPLACER_CPUS pplaceræ”¾ç½®è¿‡ç¨‹ä¸­ä½¿ç”¨çš„CPUæ•°é‡
</span></span><span class="line"><span class="cl">  --force               å¦‚æœå•ä¸ªåŸºå› ç»„å‘ç”Ÿé”™è¯¯ï¼Œåˆ™ç»§ç»­å¤„ç†ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --scratch_dir SCRATCH_DIR
</span></span><span class="line"><span class="cl">                        é€šè¿‡å†™å…¥ç£ç›˜æ¥å‡å°‘pplacerå†…å­˜ä½¿ç”¨ï¼ˆè¾ƒæ…¢ï¼‰
</span></span><span class="line"><span class="cl">  --write_single_copy_genes
</span></span><span class="line"><span class="cl">                        è¾“å‡ºæœªå¯¹é½çš„å•æ‹·è´æ ‡è®°åŸºå› ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --keep_intermediates  åœ¨æœ€ç»ˆç›®å½•ä¸­ä¿ç•™ä¸­é—´æ–‡ä»¶ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span><span class="line"><span class="cl">  --min_af MIN_AF       å°†åŸºå› ç»„åˆ†é…åˆ°ç‰©ç§ç°‡çš„æœ€å°å¯¹é½æ¯”ä¾‹ï¼ˆé»˜è®¤ï¼š0.5ï¼‰
</span></span><span class="line"><span class="cl">  --tmpdir TMPDIR       æŒ‡å®šä¸´æ—¶æ–‡ä»¶çš„å¤‡ç”¨ç›®å½•ï¼ˆé»˜è®¤ï¼š/tmpï¼‰
</span></span><span class="line"><span class="cl">  --debug               åˆ›å»ºç”¨äºè°ƒè¯•ç›®çš„çš„ä¸­é—´æ–‡ä»¶ï¼ˆé»˜è®¤ï¼šFalseï¼‰
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="classify_wf-æ³¨é‡Šç‰©ç§">classify_wf æ³¨é‡Šç‰©ç§</h4>
<p>classify_wfå…¨ç§°ä¸º Classify workflowï¼Œå³åˆ†ç±»å·¥ä½œæµç¨‹ã€‚</p>
<p>åˆ†ç±»å·¥ä½œæµç¨‹ç”±å››ä¸ªæ­¥éª¤ç»„æˆï¼šani_screenï¼ˆv2.4.0å¼€å§‹åŠ å…¥ï¼‰, identify, align, classify.</p>
<ol>
<li>ani_screen æ­¥éª¤å°†ç”¨æˆ·åŸºå› ç»„ä¸ç”±æ‰€æœ‰ GTDB ä»£è¡¨æ€§åŸºå› ç»„ç»„æˆçš„ Mash æ•°æ®åº“è¿›è¡Œæ¯”è¾ƒï¼Œç„¶åä½¿ç”¨ skani éªŒè¯æœ€ä½³çš„ mash å‘½ä¸­ã€‚ä½¿ç”¨ skani åˆ†ç±»çš„ç”¨æˆ·åŸºå› ç»„ä¸ä¼šè¿è¡Œç®¡é“çš„å…¶ä½™éƒ¨åˆ†ï¼ˆè¯†åˆ«ã€å¯¹é½ã€åˆ†ç±»ï¼‰ï¼Œå¹¶åœ¨æ‘˜è¦æ–‡ä»¶ä¸­æŠ¥å‘Šã€‚</li>
<li>identifyæ­¥éª¤ä½¿ç”¨ Prodigal è°ƒç”¨åŸºå› ï¼Œå¹¶ä½¿ç”¨ HMM æ¨¡å‹å’Œ HMMER åŒ…æ¥è¯†åˆ«ç”¨äºç³»ç»Ÿå‘è‚²æ¨æ–­çš„ 120 ä¸ªç»†èŒå’Œ 53 ä¸ªå¤ç»†èŒæ ‡è®°åŸºå›  (Parks et al., 2018)ã€‚é€šè¿‡å°†æ ‡è®°åŸºå› ä¸å…¶å„è‡ªçš„ HMM æ¨¡å‹è¿›è¡Œæ¯”å¯¹æ¥è·å¾—å¤šé‡åºåˆ—æ¯”å¯¹ (MSA)ã€‚</li>
<li>alignæ­¥éª¤è¿æ¥æ¯”å¯¹çš„æ ‡è®°åŸºå› ï¼Œå¹¶å°†è¿æ¥çš„ MSA è¿‡æ»¤ä¸ºå¤§çº¦ 5,000 ä¸ªæ°¨åŸºé…¸ã€‚</li>
<li>æœ€åï¼Œclassifyæ­¥éª¤ä½¿ç”¨ pplacer æŸ¥æ‰¾ GTDB-Tk å‚è€ƒæ ‘ä¸­æ¯ä¸ªåŸºå› ç»„çš„æœ€å¤§ä¼¼ç„¶ä½ç½®ã€‚ GTDB-Tk æ ¹æ®æ¯ä¸ªåŸºå› ç»„åœ¨å‚è€ƒæ ‘ä¸­çš„ä½ç½®ã€ç›¸å¯¹è¿›åŒ–åˆ†æ­§å’Œ/æˆ–ä¸å‚è€ƒåŸºå› ç»„çš„å¹³å‡æ ¸è‹·é…¸åŒä¸€æ€§ (ANI) å¯¹æ¯ä¸ªåŸºå› ç»„è¿›è¡Œåˆ†ç±»ã€‚</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># è¾“å…¥åŸºå› ç»„ç›®å½•</span>
</span></span><span class="line"><span class="cl">gtdbtk classify_wf --genome_dir genomes/ --out_dir classify_wf_out --cpus <span class="m">16</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>å¯ä»¥æ ¹æ®ä¸‹å›¾ä¼°è®¡è‡ªå·±çš„ä»»åŠ¡è¿è¡Œæ—¶é—´ï¼Œè¯„ä¼°èµ„æºåˆ†é…ï¼š
<img src="images/gtdb_perf.png" title=""/></p>
<h4 id="output">output</h4>
<p>GTDB-Tk åœ¨classify_wfç®¡é“çš„å„ä¸ªé˜¶æ®µè¾“å‡ºè®¸å¤šæ–‡ä»¶ï¼Œå…·ä½“æ ¼å¼å’Œç¤ºä¾‹å¯ä»¥åœ¨å®˜ç½‘æŸ¥çœ‹ï¼š
<a class="link" href="https://ecogenomics.github.io/GTDBTk/files/index.html"  target="_blank" rel="noopener"
    >https://ecogenomics.github.io/GTDBTk/files/index.html</a></p>
<p>ç»“æœç›®å½•(classify_wf_out/output)ä¸­ä¸»è¦ç»“æœæ–‡ä»¶ (ar53ä¸ºå¤èŒç›¸å…³æ–‡ä»¶ï¼Œbac120ä¸ºç»†èŒç›¸å…³æ–‡ä»¶)ï¼š</p>
<ul>
<li>classify/gtdbtk.ar53.classify.tree æ ‘æ–‡ä»¶ï¼ŒNewickæ ¼å¼ï¼Œå¯ä»¥ç”¨äºç³»ç»Ÿå‘è‚²æ ‘çš„ç»˜åˆ¶ã€‚</li>
<li>classify/gtdbtk.ar53.summary.tsv åˆ†ç±»ç»“æœæ–‡ä»¶ï¼Œä¸»è¦ç»“æœï¼Œå¯ä»¥ç”¨äºç³»ç»Ÿå‘è‚²æ ‘çš„æ³¨é‡Šã€‚</li>
<li>identify/gtdbtk.ar53.markers_summary.tsv ä½¿ç”¨æ ‡è®°åŸºå› æƒ…å†µ</li>
<li>align/gtdbtk.ar53.msa.fasta.gz å¤šåºåˆ—å¯¹é½æ–‡ä»¶(è¾“å…¥+å‚è€ƒ)</li>
<li>align/gtdbtk.ar53.user_msa.fasta.gz å¤šåºåˆ—å¯¹é½æ–‡ä»¶(è¾“å…¥)</li>
</ul>
<p>ç»˜åˆ¶ç³»ç»Ÿå‘è‚²æ ‘å¯ä»¥å‚è€ƒæˆ‘ä¹‹å‰å†™çš„<a class="link" href="../r-tree" >Rç»˜åˆ¶ä¼˜ç¾çš„è¿›åŒ–æ ‘ï¼ˆåŸºç¡€ï¼‰</a>å’Œ<a class="link" href="../r-tree2" >Rç»˜åˆ¶ä¼˜ç¾çš„è¿›åŒ–æ ‘ï¼ˆè¿›é˜¶ï¼‰</a>ï¼Œæˆ–è€…ä½¿ç”¨iPhyloç½‘ç«™æ¥äº¤äº’å¼ç»˜å›¾ï¼š<a class="link" href="../iphylo" >iPhylo ç”Ÿæˆå¹¶ç»˜åˆ¶ä¼˜ç¾çš„åˆ†ç±»æ ‘</a>ã€‚</p>
<h3 id="åˆ†æ­¥ç¤ºä¾‹">åˆ†æ­¥ç¤ºä¾‹</h3>
<h4 id="è·å–æ•°æ®">è·å–æ•°æ®</h4>
<p>æ­¤ç¤ºä¾‹å°†ä½¿ç”¨ä»¥ä¸‹ä¸¤ä¸ªåŸºå› ç»„ï¼šåŸºå› ç»„Aï¼šGCF_003947435.1 [GTDB / NCBI]ï¼ŒåŸºå› ç»„Bï¼šGCA_002011125.1 [GTDB / NCBI]</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># è·å–åŸºå› ç»„ã€‚</span>
</span></span><span class="line"><span class="cl">mkdir -p gtdbtk_test_steps/genomes
</span></span><span class="line"><span class="cl">wget -q https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/947/435/GCF_003947435.1_ASM394743v1/GCF_003947435.1_ASM394743v1_genomic.fna.gz -O gtdbtk_test_steps/genomes/genome_a.fna.gz
</span></span><span class="line"><span class="cl">wget -q https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002/011/125/GCA_002011125.1_ASM201112v1/GCA_002011125.1_ASM201112v1_genomic.fna.gz -O gtdbtk_test_steps/genomes/genome_b.fna.gz
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="åŸºå› é‰´å®š-identify">åŸºå› é‰´å®š (identify)</h4>
<p>å¯ä»¥é€šè¿‡<code>classify_wf</code>å•æ­¥å®Œæˆï¼Œä½†æ˜¯æœ‰æ—¶å€™åˆ†æ­¥æµç¨‹æ›´é€‚åˆå¤„ç†å¤§æ•°æ®ï¼ˆå¯ä»¥è‡ªè¡Œåˆ†é…æ¯ä¸€æ­¥çš„èµ„æºï¼‰ï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls -l gtdbtk_test_steps/genomes
</span></span><span class="line"><span class="cl"><span class="c1"># â€“extensionï¼Œgzæ˜¯åŸºå› ç»„æ–‡ä»¶çš„åç¼€å</span>
</span></span><span class="line"><span class="cl">gtdbtk identify --genome_dir gtdbtk_test_steps/genomes --out_dir gtdbtk_test_steps/identify --extension gz --cpus <span class="m">2</span>
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:25:20] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:25:20] INFO: gtdbtk identify --genome_dir gtdbtk_test_steps/genomes --out_dir gtdbtk_test_steps/identify --extension gz --cpus 2
[2024-05-20 15:25:20] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:25:20] INFO: Identifying markers in 2 genomes with 2 threads.
[2024-05-20 15:25:20] TASK: Running Prodigal V2.6.3 to identify genes.
[2024-05-20 15:25:30] INFO: Completed 2 genomes in 9.31 seconds (4.65 seconds/genome).
[2024-05-20 15:25:30] TASK: Identifying TIGRFAM protein families.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 7.16 seconds (3.58 seconds/genome).
[2024-05-20 15:25:38] TASK: Identifying Pfam protein families.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 0.59 seconds (3.37 genomes/second).
[2024-05-20 15:25:38] INFO: Annotations done using HMMER 3.3.2 (Nov 2020).
[2024-05-20 15:25:38] TASK: Summarising identified marker genes.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 0.03 seconds (75.34 genomes/second).
[2024-05-20 15:25:38] INFO: Done.
</code></pre>
<p>è€—æ—¶40sï¼Œå†…å­˜æ¶ˆè€—131.76MBã€‚</p>
<p>äº§ç”Ÿ<code>idetity/</code>æ–‡ä»¶å¤¹ï¼Œå…³æ³¨æ‘˜è¦æ–‡ä»¶gtdbtk_test_steps/identify/identify/gtdbtk.ar53.markers_summary.tsvï¼Œå…¶ä¸­è¯¦ç»†ä»‹ç»äº†ä»å¤ç»†èŒ 53 æˆ–ç»†èŒ 120 æ ‡è®°ç»„ä¸­è¯†åˆ«å‡ºçš„æ ‡è®°ã€‚</p>
<p>è¿˜å¯ä»¥åœ¨æ¯ä¸ªåŸºå› ç»„å¯¹åº”çš„ä¸­é—´æ–‡ä»¶ç›®å½•ä¸‹æ‰¾åˆ°è¢«è°ƒç”¨çš„åŸºå› å’Œæ ‡è®°ä¿¡æ¯:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls gtdbtk_test_steps/identify/identify/intermediate_results/marker_genes/genome_a.fna/
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="åŸºå› æ¯”å¯¹-align">åŸºå› æ¯”å¯¹ (align)</h4>
<p>æ¯”å¯¹æ­¥éª¤å°†å¯¹é½æ‰€æœ‰å·²è¯†åˆ«çš„æ ‡è®°ï¼Œç¡®å®šæœ€å¯èƒ½çš„åŸŸå¹¶è¾“å‡ºä¸²è”çš„MSA:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk align --identify_dir gtdbtk_test_steps/identify --out_dir gtdbtk_test_steps/align --cpus <span class="m">2</span>
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:32:01] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:32:01] INFO: gtdbtk align --identify_dir gtdbtk_test_steps/identify --out_dir gtdbtk_test_steps/align --cpus 2
[2024-05-20 15:32:01] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:32:03] INFO: Aligning markers in 2 genomes with 2 CPUs.
[2024-05-20 15:32:04] INFO: Processing 2 genomes identified as archaeal.
[2024-05-20 15:32:06] INFO: Read concatenated alignment for 4,416 GTDB genomes.
[2024-05-20 15:32:06] TASK: Generating concatenated alignment for each marker.
[2024-05-20 15:32:06] INFO: Completed 2 genomes in 0.02 seconds (130.19 genomes/second).
[2024-05-20 15:32:06] TASK: Aligning 52 identified markers using hmmalign 3.3.2 (Nov 2020).
[2024-05-20 15:32:06] INFO: Completed 52 markers in 0.43 seconds (121.60 markers/second).
[2024-05-20 15:32:06] TASK: Masking columns of archaeal multiple sequence alignment using canonical mask.
[2024-05-20 15:32:10] INFO: Completed 4,418 sequences in 3.59 seconds (1,231.58 sequences/second).
[2024-05-20 15:32:10] INFO: Masked archaeal alignment from 13,540 to 10,135 AAs.
[2024-05-20 15:32:10] INFO: 0 archaeal user genomes have amino acids in &lt;10.0% of columns in filtered MSA.
[2024-05-20 15:32:10] INFO: Creating concatenated alignment for 4,418 archaeal GTDB and user genomes.
[2024-05-20 15:32:12] INFO: Creating concatenated alignment for 2 archaeal user genomes.
[2024-05-20 15:32:12] INFO: Done.
</code></pre>
<p>è€—æ—¶10sï¼Œå†…å­˜å¿½ç•¥ä¸è®¡ï¼ˆå› ä¸ºè¿™é‡Œåªæœ‰ä¸¤ä¸ªgenomeï¼Œè¾“å…¥çš„å¤šäº†è¿˜æ˜¯è¦è€ƒè™‘è¿™æ­¥çš„å†…å­˜çš„ï¼‰</p>
<p>äº§ç”Ÿ<code>align/</code>æ–‡ä»¶å¤¹ï¼Œå¦‚æœåŸºå› ç»„è¯†åˆ«å‡ºçš„æ ‡è®°æ•°é‡è¾ƒå°‘ï¼Œå®ƒå°†è¢«æ’é™¤åœ¨è¿™ä¸€æ­¥çš„åˆ†æä¹‹å¤–ã€‚å¦‚æœæ˜¯è¿™ç§æƒ…å†µï¼Œå°†ä¼šå‡ºç°Warningã€‚</p>
<p>æ ¹æ®åŸºå› ç»„Domainçš„ä¸åŒï¼Œå°†å‡ºç° ar53 æˆ– bac120 å‰ç¼€æ–‡ä»¶ã€‚</p>
<ul>
<li>gtdbtk.ar53.msa.fasta.gzï¼šå…¶ä¸­åŒ…å«ç”¨æˆ·åŸºå› ç»„å’Œ GTDBåŸºå› ç»„çš„ MSAã€‚</li>
<li>gtdbtk.ar53.user_msa.fasta.gzï¼šä»…åŒ…å«ç”¨æˆ·åŸºå› ç»„</li>
</ul>
<h4 id="ç‰©ç§åˆ†ç±»-classify">ç‰©ç§åˆ†ç±» (classify)</h4>
<p>åˆ†ç±»æ­¥éª¤ä¼šå°†åŸºå› ç»„æ”¾å…¥å‚è€ƒæ ‘ä¸­ï¼Œç„¶åç¡®å®šå…¶æœ€å¯èƒ½çš„åˆ†ç±»ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk classify --genome_dir gtdbtk_test_steps/genomes --align_dir gtdbtk_test_steps/align --out_dir gtdbtk_test_steps/classify -x gz --cpus <span class="m">2</span> --skip_ani_screen
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:42:48] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:42:48] INFO: gtdbtk classify --genome_dir gtdbtk_test_steps/genomes --align_dir gtdbtk_test_steps/align --out_dir gtdbtk_test_steps/classify -x gz --cpus 2 --skip_ani_screen
[2024-05-20 15:42:48] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:42:49] TASK: Placing 2 archaeal genomes into reference tree with pplacer using 2 CPUs (be patient).
[2024-05-20 15:42:49] INFO: pplacer version: v1.1.alpha19-0-g807f6f3
[2024-05-20 15:48:46] INFO: Calculating RED values based on reference tree.
[2024-05-20 15:48:48] TASK: Traversing tree to determine classification method.
[2024-05-20 15:48:48] INFO: Completed 2 genomes in 0.00 seconds (8,621.39 genomes/second).
[2024-05-20 15:48:48] TASK: Calculating average nucleotide identity using FastANI (v1.33).
[2024-05-20 15:48:52] INFO: Completed 6 comparisons in 2.69 seconds (2.23 comparisons/second).
[2024-05-20 15:48:52] INFO: 2 genome(s) have been classified using FastANI and pplacer.
[2024-05-20 15:48:52] INFO: Note that Tk classification mode is insufficient for publication of new taxonomic designations. New designations should be based on one or more de novo trees, an example of which can be produced by Tk in de novo mode.
[2024-05-20 15:48:52] INFO: Done.
</code></pre>
<p>è€—æ—¶368sï¼Œ<strong>å†…å­˜æ¶ˆè€—45GB</strong>ï¼ˆå› ä¸ºè¿™é‡Œåªæœ‰ä¸¤ä¸ªå¤èŒgenomeï¼Œå¦‚æœè¿˜æœ‰ç»†èŒçš„è¯åº”è¯¥åˆ†é…100GBå·¦å³ï¼‰</p>
<p>äº§ç”Ÿ<code>align/</code>æ–‡ä»¶å¤¹ï¼Œæ‘˜è¦æ–‡ä»¶gtdbtk.ar53.summary.tsvå’ŒåŒ…å«è¿™äº›åŸºå› ç»„çš„å‚è€ƒæ ‘gtdbtk.ar53.classify.treeã€‚</p>
<h3 id="æ³¨æ„äº‹é¡¹">æ³¨æ„äº‹é¡¹</h3>
<ul>
<li>ä¸ºä»€ä¹ˆ GTDB-Tk å’Œ NCBI æˆ– Silva åˆ†ç±»åç§°ä¹‹é—´çš„å‘½åç³»ç»Ÿå­˜åœ¨å·®å¼‚ï¼Ÿ</li>
</ul>
<p>GTDB-Tk ä½¿ç”¨ GTDB åˆ†ç±»æ³• (<a class="link" href="https://gtdb.ecogenomic.org/"  target="_blank" rel="noopener"
    >https://gtdb.ecogenomic.org/</a>)ã€‚æ­¤åˆ†ç±»æ³•ä¸NCBI å’Œ Silva ç±»ä¼¼ï¼Œä½†ä¸å®Œå…¨ç›¸åŒã€‚
åœ¨è®¸å¤šæƒ…å†µä¸‹ï¼ŒGTDBåˆ†ç±»æ³•æ›´ä¸¥æ ¼åœ°éµå¾ªç­‰çº§åç¼€çš„å‘½åè§„åˆ™ï¼Œæ¯”å¦‚GTDBæœ‰ Nitrospirota è€Œæ²¡æœ‰Nitrospiraeã€‚</p>
<ul>
<li>å¯ä»¥å°†ç»†èŒæ ‘å’Œå¤ç»†èŒæ ‘åˆå¹¶æˆä¸€æ£µæ ‘å—ï¼Ÿ</li>
</ul>
<p>ç»†èŒæ ‘å’Œå¤ç»†èŒæ ‘æ˜¯ä»ä¸åŒçš„æ ‡è®°åŸºå› æ¨æ–­å‡ºæ¥çš„ã€‚ç›®å‰ï¼Œè¿™äº›æ ‘çš„æ­£ç¡®ç”Ÿæ ¹ä»ç„¶æ˜¯ä¸€ä¸ªå¼€æ”¾çš„ç ”ç©¶é¢†åŸŸã€‚
GTDB-Tk ä¸æä¾›åˆå¹¶æ ‘çš„å·¥å…·ï¼Œä½†å¯ä»¥é€šè¿‡æ“ä½œ Newickæ–‡ä»¶æ¥äººä¸ºåœ°ç»„åˆå®ƒä»¬ã€‚</p>
<ul>
<li>GTDB-Tk å¦‚ä½•ä½¿ç”¨å¹³å‡æ ¸è‹·é…¸åŒä¸€æ€§ï¼ˆANIï¼‰éªŒè¯ç‰©ç§åˆ†é…ï¼Ÿ</li>
</ul>
<p>GTDB-Tk ä½¿ç”¨ skaniï¼ˆåœ¨ v2.3.2 ä¹‹å‰ä¸€ç›´ä½¿ç”¨ fastANIï¼‰æ¥ä¼°è®¡åŸºå› ç»„ä¹‹é—´çš„ANIã€‚
ä»…å½“åŸºå› ç»„ä¹‹é—´çš„ ANI åœ¨ç‰©ç§ ANI é™åˆ¶åŠå¾„å†…ï¼ˆé€šå¸¸ä¸º 95%ï¼‰ä¸”æ¯”å¯¹åˆ†æ•°(AF)&gt;=0.5æ—¶ï¼ŒæŸ¥è¯¢åŸºå› ç»„æ‰è¢«åˆ†ç±»ä¸ºä¸å‚è€ƒåŸºå› ç»„å±äºåŒä¸€ç‰©ç§ã€‚
åœ¨æŸäº›æƒ…å†µä¸‹ï¼ŒæŸ¥è¯¢åŸºå› ç»„çš„ç³»ç»Ÿå‘è‚²ä½ç½®å¯èƒ½ä¸æ”¯æŒç‰©ç§åˆ†é…ã€‚</p>
<ul>
<li>äº’æ–¥é€‰é¡¹ <code>--mash_db</code> å’Œ <code>--skip_ani_screen</code> ä¹‹é—´æœ‰ä»€ä¹ˆåŒºåˆ«ï¼Ÿ</li>
</ul>
<p>ä» GTDB-Tk v2.2+ å¼€å§‹ï¼Œclassify_wfå’Œåˆ†ç±»å‡½æ•°éœ€è¦é¢å¤–çš„å‚æ•°æ‰èƒ½è¿è¡Œï¼š<code>--mash_db</code> æˆ– <code>--skip_ani_screen</code>ã€‚</p>
<p>ä½¿ç”¨æ–°ç‰ˆæœ¬çš„ GTDB-Tk v2.4.0ï¼Œåˆ†ç±»ç®¡é“ï¼ˆclassify_wf å’Œ classifyï¼‰çš„ç¬¬ä¸€é˜¶æ®µæ˜¯å°†æ‰€æœ‰ç”¨æˆ·åŸºå› ç»„ä¸æ‰€æœ‰å‚è€ƒåŸºå› ç»„è¿›è¡Œæ¯”è¾ƒï¼Œå¹¶åœ¨å¯èƒ½çš„æƒ…å†µä¸‹æ ¹æ®ANI åŒ¹é…å¯¹å®ƒä»¬è¿›è¡Œæ³¨é‡Šã€‚
ä½¿ç”¨ <code>--mash_db</code> é€‰é¡¹å°†å‘ GTDB-Tk æŒ‡ç¤º ANIç­›é€‰æ‰€éœ€çš„è‰å›¾ Mash æ•°æ®åº“çš„è·¯å¾„ã€‚
å¦‚æœæ²¡æœ‰å¯ç”¨çš„æ•°æ®åº“ï¼ˆå³è¿™æ˜¯ç¬¬ä¸€æ¬¡è¿è¡Œåˆ†ç±»ï¼‰ï¼Œåˆ™ <code>--mash_db</code>é€‰é¡¹å°†ç»˜åˆ¶ä¸€ä¸ªå¯ç”¨äºåç»­è°ƒç”¨çš„æ–° Mash æ•°æ®åº“ã€‚ <code>--skip_ani_screen</code> é€‰é¡¹å°†è·³è¿‡é¢„ç­›é€‰æ­¥éª¤å¹¶å¯¹æ‰€æœ‰åŸºå› ç»„è¿›è¡Œåˆ†ç±»ï¼Œç±»ä¼¼äºä¹‹å‰ç‰ˆæœ¬çš„ GTDB-Tkã€‚</p>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/genome/">Genome</a>
        
            <a href="/Cao-blog/tags/database/">Database</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/gcf-big-fam/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/gcf-big-fam/images/Screenshot%202025-01-08%20at%2011.53.28.0fbb5b0c2b31df564edb14075af07c92_hu_b51a4a4f8db8bb65.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ç”Ÿç‰©åˆæˆåŸºå› ç°‡å®¶æ—ï¼ˆGCFï¼‰æ•°æ®åº“ BiG-FAM"
                        data-key="gcf-big-fam" 
                        data-hash="md5-D7tbDCsx31ZO2xQHWvB8kg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ç”Ÿç‰©åˆæˆåŸºå› ç°‡å®¶æ—ï¼ˆGCFï¼‰æ•°æ®åº“ BiG-FAM</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/orthofinder/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/orthofinder/images/Workflow.b06f2895b33f76fcddcd78da78c36ab1_hu_54a1bd9442f615be.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ä½¿ç”¨Orthofinderè¿›è¡Œç³»ç»Ÿå‘è‚²ç›´ç³»åŒæºæ¨æ–­"
                        data-key="orthofinder" 
                        data-hash="md5-sG8olbM/dvzdzXjaeMNqsQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ä½¿ç”¨Orthofinderè¿›è¡Œç³»ç»Ÿå‘è‚²ç›´ç³»åŒæºæ¨æ–­</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/ani/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/ani/images/ani3.68da08c94ea4c007938042e49c0e07d5_hu_fe3e6b2cf8eeb457.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ä½¿ç”¨FastANIä¸Skaniè®¡ç®—å¹³å‡æ ¸è‹·é…¸ä¸€è‡´æ€§(ANI)"
                        data-key="ani" 
                        data-hash="md5-aNoIyU6kwAeTgELknA4H1Q==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ä½¿ç”¨FastANIä¸Skaniè®¡ç®—å¹³å‡æ ¸è‹·é…¸ä¸€è‡´æ€§(ANI)</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/amplicon/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/amplicon/images/nbt1.18bd9e2218108348ce7845ad52e631ec_hu_7a8b1c11bb1d244d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post å¸¸ç”¨æ‰©å¢å­æ•°æ®åº“"
                        data-key="amplicon" 
                        data-hash="md5-GL2eIhgQg0jOeEWtUuYx7A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">å¸¸ç”¨æ‰©å¢å­æ•°æ®åº“</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/emp/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/emp/images/emp1.87b75213fcbf279ba4a02c0d17fd6823_hu_44af3bd2e29e0581.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post åœ°çƒå¾®ç”Ÿç‰©ç»„è®¡åˆ’ä»‹ç»ï¼ˆEMPï¼‰"
                        data-key="emp" 
                        data-hash="md5-h7dSE/y/J5ukoCwNF/1oIw==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">åœ°çƒå¾®ç”Ÿç‰©ç»„è®¡åˆ’ä»‹ç»ï¼ˆEMPï¼‰</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
