<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='GTDBÔºàGenome Taxonomy DatabaseÔºâÊòØ‰∏Ä‰∏™Âü∫Âõ†ÁªÑÂàÜÁ±ªÂ≠¶Êï∞ÊçÆÂ∫ìÔºåÊó®Âú®Êèê‰æõÈ´òË¥®ÈáèÁöÑÁªÜËèåÂíåÂè§ÁªÜËèåÁ≠âÂàÜÁ±ª‰ø°ÊÅØ„ÄÇGTDB-Tk ÊòØ‰∏Ä‰∏™ËΩØ‰ª∂Â∑•ÂÖ∑ÂåÖÔºåÁî®‰∫éÊ†πÊçÆGTDBÂØπÁªÜËèåÂíåÂè§ÁªÜËèåÂü∫Âõ†ÁªÑËøõË°åÂÆ¢ËßÇÂàÜÁ±ª„ÄÇ'>
<title>GTDBÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ìÂèäGTDB-TkÂ∑•ÂÖ∑</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/gtdb/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='GTDBÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ìÂèäGTDB-TkÂ∑•ÂÖ∑'>
<meta property='og:description' content='GTDBÔºàGenome Taxonomy DatabaseÔºâÊòØ‰∏Ä‰∏™Âü∫Âõ†ÁªÑÂàÜÁ±ªÂ≠¶Êï∞ÊçÆÂ∫ìÔºåÊó®Âú®Êèê‰æõÈ´òË¥®ÈáèÁöÑÁªÜËèåÂíåÂè§ÁªÜËèåÁ≠âÂàÜÁ±ª‰ø°ÊÅØ„ÄÇGTDB-Tk ÊòØ‰∏Ä‰∏™ËΩØ‰ª∂Â∑•ÂÖ∑ÂåÖÔºåÁî®‰∫éÊ†πÊçÆGTDBÂØπÁªÜËèåÂíåÂè§ÁªÜËèåÂü∫Âõ†ÁªÑËøõË°åÂÆ¢ËßÇÂàÜÁ±ª„ÄÇ'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/gtdb/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='genome' /><meta property='article:tag' content='database' /><meta property='article:published_time' content='2024-05-20T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-05-20T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/gtdb/images/gtdb.png' />
<meta name="twitter:title" content="GTDBÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ìÂèäGTDB-TkÂ∑•ÂÖ∑">
<meta name="twitter:description" content="GTDBÔºàGenome Taxonomy DatabaseÔºâÊòØ‰∏Ä‰∏™Âü∫Âõ†ÁªÑÂàÜÁ±ªÂ≠¶Êï∞ÊçÆÂ∫ìÔºåÊó®Âú®Êèê‰æõÈ´òË¥®ÈáèÁöÑÁªÜËèåÂíåÂè§ÁªÜËèåÁ≠âÂàÜÁ±ª‰ø°ÊÅØ„ÄÇGTDB-Tk ÊòØ‰∏Ä‰∏™ËΩØ‰ª∂Â∑•ÂÖ∑ÂåÖÔºåÁî®‰∫éÊ†πÊçÆGTDBÂØπÁªÜËèåÂíåÂè§ÁªÜËèåÂü∫Âõ†ÁªÑËøõË°åÂÆ¢ËßÇÂàÜÁ±ª„ÄÇ"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/gtdb/images/gtdb.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">üêõüê∂</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/472862038'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/JB-Cao'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >‰∏≠Êñá</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#gtdbÊï∞ÊçÆÂ∫ì">GTDBÊï∞ÊçÆÂ∫ì</a>
      <ol>
        <li><a href="#browsers">Browsers</a></li>
        <li><a href="#tools">Tools</a></li>
        <li><a href="#downloads">Downloads</a></li>
        <li><a href="#statistics">Statistics</a></li>
        <li><a href="#forum">Forum</a></li>
        <li><a href="#help">Help</a></li>
      </ol>
    </li>
    <li><a href="#gtdb-tk">GTDB-Tk</a>
      <ol>
        <li><a href="#install">Install</a>
          <ol>
            <li><a href="#1conda">1.conda</a></li>
            <li><a href="#2pip">2.pip</a></li>
            <li><a href="#3docker">3.docker</a></li>
            <li><a href="#reference-data">Reference data</a></li>
            <li><a href="#check">Check</a></li>
          </ol>
        </li>
        <li><a href="#‰ΩøÁî®ÊñπÊ≥ï">‰ΩøÁî®ÊñπÊ≥ï</a>
          <ol>
            <li><a href="#classify_wf-Ê≥®ÈáäÁâ©Áßç">classify_wf Ê≥®ÈáäÁâ©Áßç</a></li>
            <li><a href="#output">output</a></li>
          </ol>
        </li>
        <li><a href="#ÂàÜÊ≠•Á§∫‰æã">ÂàÜÊ≠•Á§∫‰æã</a>
          <ol>
            <li><a href="#Ëé∑ÂèñÊï∞ÊçÆ">Ëé∑ÂèñÊï∞ÊçÆ</a></li>
            <li><a href="#Âü∫Âõ†Èâ¥ÂÆö-identify">Âü∫Âõ†Èâ¥ÂÆö (identify)</a></li>
            <li><a href="#Âü∫Âõ†ÊØîÂØπ-align">Âü∫Âõ†ÊØîÂØπ (align)</a></li>
            <li><a href="#Áâ©ÁßçÂàÜÁ±ª-classify">Áâ©ÁßçÂàÜÁ±ª (classify)</a></li>
          </ol>
        </li>
        <li><a href="#Ê≥®ÊÑè‰∫ãÈ°π">Ê≥®ÊÑè‰∫ãÈ°π</a></li>
      </ol>
    </li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/gtdb/">
                <img src="/Cao-blog/p/gtdb/images/gtdb_hu_db9b99b2154e9c0f.png"
                        srcset="/Cao-blog/p/gtdb/images/gtdb_hu_db9b99b2154e9c0f.png 800w, /Cao-blog/p/gtdb/images/gtdb_hu_7152a73598cda3f2.png 1600w"
                        width="800" 
                        height="408" 
                        loading="lazy"
                        alt="Featured image of post GTDBÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ìÂèäGTDB-TkÂ∑•ÂÖ∑" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/gtdb/">GTDBÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ìÂèäGTDB-TkÂ∑•ÂÖ∑</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            GTDBÔºàGenome Taxonomy DatabaseÔºâÊòØ‰∏Ä‰∏™Âü∫Âõ†ÁªÑÂàÜÁ±ªÂ≠¶Êï∞ÊçÆÂ∫ìÔºåÊó®Âú®Êèê‰æõÈ´òË¥®ÈáèÁöÑÁªÜËèåÂíåÂè§ÁªÜËèåÁ≠âÂàÜÁ±ª‰ø°ÊÅØ„ÄÇGTDB-Tk ÊòØ‰∏Ä‰∏™ËΩØ‰ª∂Â∑•ÂÖ∑ÂåÖÔºåÁî®‰∫éÊ†πÊçÆGTDBÂØπÁªÜËèåÂíåÂè§ÁªÜËèåÂü∫Âõ†ÁªÑËøõË°åÂÆ¢ËßÇÂàÜÁ±ª„ÄÇ
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">May 20, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    8 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                ÁâàÊùÉÂΩíCao's blogÊâÄÊúâÔºåËΩ¨ËΩΩËØ∑Ê≥®ÊòéÂá∫Â§Ñ
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>GTDBÔºàGenome Taxonomy DatabaseÔºâÊòØ‰∏Ä‰∏™‰∏ìÊ≥®‰∫éÂü∫Âõ†ÁªÑÂàÜÁ±ªÂ≠¶ÁöÑÊï∞ÊçÆÂ∫ìÔºåÊó®Âú®Êèê‰æõÈ´òË¥®ÈáèÁöÑÁªÜËèå„ÄÅÂè§ÁªÜËèå„ÄÅÂè∂Áªø‰ΩìÂíåÁ∫øÁ≤í‰ΩìÁöÑÂàÜÁ±ª‰ø°ÊÅØ„ÄÇÈÄöËøá‰ΩøÁî®ÂÖ®Âü∫Âõ†ÁªÑ‰ø°ÊÅØÔºåGTDBÊèê‰æõ‰∫ÜÂ§öÁ∫ßÂà´ÁöÑÂàÜÁ±ªÔºåÂåÖÊã¨Âüü„ÄÅÁïå„ÄÅÁ∫≤„ÄÅÁõÆ„ÄÅÁßë„ÄÅÂ±ûÁ≠âÁ∫ßÂà´„ÄÇÂÖ∂Á≥ªÁªüÂèëËÇ≤ÂàÜÁ±ªÁ≥ªÁªü‰ΩøÂæóÁ†îÁ©∂‰∫∫ÂëòËÉΩÂ§üÊ∑±ÂÖ•‰∫ÜËß£ÁîüÁâ©ÁöÑÊºîÂåñÂíåÂäüËÉΩ„ÄÇGTDB‰∏ªË¶Å‰ª•ÁªÜËèå‰∏≠ÊôÆÈÅçÂ≠òÂú®ÁöÑ120‰∏™ÂçïÊã∑Ë¥ùËõãÁôΩË¥®Ôºàbac120ÔºâÂíåÂè§Ëèå‰∏≠53‰∏™Ôºàarc53Ôºå‰ªéR07-RS207ÂºÄÂßãÔºâÊ†áËÆ∞ËõãÁôΩ‰∏∫Âü∫Á°ÄÔºåÂú®ÂØπÂ§öÂàÜÁªÑÁ±ªÂà´Ê∂àÊ≠ßÂêéÔºåÊ†πÊçÆÁõ∏ÂØπÊºîÂåñÊï£Â∫¶Ê†áÂáÜÂåñÂíåÂàÜÁ∫ßÔºåÂæóÂà∞Âü∫Âõ†ÁªÑÂàÜÁ±ªÁªìÊûú„ÄÇ</p>
<p>GTDBÂÆöÊúüÊõ¥Êñ∞Ôºå‰∏éÂÖ∂‰ªñÁîüÁâ©‰ø°ÊÅØÊï∞ÊçÆÂ∫ìÊï¥ÂêàÔºå‰∏∫Âü∫Âõ†ÁªÑÂ≠¶Á†îÁ©∂ÂíåÂæÆÁîüÁâ©ÁîüÊÄÅÂ≠¶Á≠âÈ¢ÜÂüüÁöÑÁ†îÁ©∂Êèê‰æõ‰∫ÜÂÖ®Èù¢‰∏îÂèØÈù†ÁöÑÂàÜÁ±ª‰ø°ÊÅØ„ÄÇGTDBÊï∞ÊçÆÂ∫ìÂ∑≤Áªè‰∫é2018/2020ËøûÁª≠ÂèëË°®‰∏§ÁØáNature BiotechnologyÔºå2021Âπ¥ÂèàÂèëË°®Âú®Nature MicrobiologyÂíåNucleic Acids
Research‰∏äÔºåÊòØÁõÆÂâçÊúÄÊùÉÂ®ÅÁöÑÂæÆÁîüÁâ©ÂàÜÁ±ªÊï∞ÊçÆÂ∫ì‰πã‰∏Ä„ÄÇ
GTDBÈÖçÂ•óËΩØ‰ª∂Âú®2019/2022ÂèëË°®‰∫Ü‰∏§ÁØáBioinformatcsÔºåÊó∂Êèê‰æõ‰∫ÜÁâ©ÁßçÂàÜÁ±ª„ÄÅËøõÂåñÊ†ëÊûÑÂª∫ÁöÑÂÖ®Â•óÂ∑•ÂÖ∑„ÄÇ</p>
<p>ÂÆòÁΩëÔºö<a class="link" href="https://gtdb.ecogenomic.org/"  target="_blank" rel="noopener"
    >https://gtdb.ecogenomic.org/</a></p>
<img src="images/gtdb.png" style="width:100.0%" />
<h2 id="gtdbÊï∞ÊçÆÂ∫ì">GTDBÊï∞ÊçÆÂ∫ì</h2>
<p>ÂèØ‰ª•ÁúãÂà∞‰∏äÈù¢ÁöÑÂÆòÁΩë‰∏ªÈ°µÔºåÁõÆÂâçÊúÄÊñ∞ÁâàÁöÑGTDBÊï∞ÊçÆÂ∫ìÂåÖÂê´‰∫Ü584,382‰∏™ÁªÜËèåÂü∫Âõ†ÁªÑÂíå12,477‰∏™Âè§ËèåÂü∫Âõ†ÁªÑÔºà2024.04.24ÔºåRelease 09-RS220ÔºâÔºåÂú®Â±û„ÄÅÁßçÂàÜËæ®ÁéáÊ∞¥Âπ≥‰∏äÊèèËø∞‰∫Ü190Â§ö‰∏™Èó®„ÄÇ</p>
<h3 id="browsers">Browsers</h3>
<p>GTDBÊîØÊåÅ‰ª•Alphabetical
tableÂíåTree‰∏§ÁßçÂΩ¢ÂºèÊù•Êé¢Á¥¢ÊàñÊü•ËØ¢Êï¥‰∏™GTDBÊï∞ÊçÆÂ∫ìÔºåÂåÖÊã¨ÂàÜÁ±ª‰ø°ÊÅØÂíåÂü∫Âõ†ÁªÑ‰ø°ÊÅØ„ÄÇ</p>
<p>Ê≠§Â§ñÔºåÂèØ‰ª•Âú®Taxon HistoryËæìÂÖ• GTDB Êàñ NCBI
ÂàÜÁ±ªÂçïÂÖÉÔºå‰ª•ÂèØËßÜÂåñÂàÜÈÖçÁªôËØ•ÂàÜÁ±ªÂçïÂÖÉÁöÑÂü∫Âõ†ÁªÑÂú®ÁâàÊú¨‰πãÈó¥Â¶Ç‰ΩïÂèòÂåñÔºåÊØîÂ¶Çp__Proteobacteria‰ªéRelease
214ÂºÄÂßãÊîπÂêç‰∏∫p__PseudomonadotaÔºåÂú®‰∏çÊñ≠Â£ÆÂ§ßÁöÑÂêåÊó∂‰πüÊúâ‰∏ÄÈÉ®ÂàÜgenomeË¢´ÂàÜÈÖçÁªô‰∫ÜÂÖ∂‰ªñÈó®„ÄÇ</p>
<h3 id="tools">Tools</h3>
<p>GTDBÊèê‰æõ‰∫Ü‰∏Ä‰∫õÈÖçÂ•óÁöÑÂ∑•ÂÖ∑Ôºö</p>
<ul>
<li>GTDB-TkÔºö‰∏ã‰∏ÄÁ´†ËäÇËØ¶ÁªÜ‰ªãÁªç</li>
<li>Third-party toolsÔºöAnnoTreeÔºåIDTAXAÔºåSingleMÔºåsourmash</li>
<li>Third-party GTDB
databasesÔºöHUMANn2ÔºåKraken2ÔºåBracken2ÔºåCentrifugeÔºåKrakenÔºåDADA2Á≠â</li>
<li>Third-party
scriptsÔºötax_from_gtdb.pyÔºågenome_updaterÔºågtdb_to_taxdump</li>
<li>FastANI calculatorÔºöÂèØÁî®‰∫éÂéüÊ†∏ÁîüÁâ© NCBI
Âü∫Âõ†ÁªÑÁöÑANIÊØîËæÉÔºå‰∏ÄÊ¨°ÊúÄÂ§öÂèØ‰ª•ËøõË°å 1,000 Ê¨°ÊàêÂØπÊØîËæÉÔºå‰æãÂ¶Ç‰∏ãÂõæ„ÄÇ</li>
</ul>
<img src="images/ani.png" title=""/>
<h3 id="downloads">Downloads</h3>
<p>ËØ•È°µÈù¢Êèê‰æõ‰∫ÜGTDBÊï∞ÊçÆÂ∫ìÂíåÈÖçÂ•óËΩØ‰ª∂ÁöÑ‰∏ãËΩΩÈìæÊé•„ÄÇ</p>
<h3 id="statistics">Statistics</h3>
<p>ÁªüËÆ°È°µÈù¢Êèê‰æõ‰∫ÜÂêÑ‰∏™ÁâàÊú¨ÁöÑGTDBÊï∞ÊçÆÂ∫ìÁöÑÂêÑÁ±ªÂü∫Á°Ä‰ø°ÊÅØÔºö</p>
<ol>
<li>
<p>Taxon overviewÔºöGTDB R220 ÂåÖÂê´ 596,859 ‰∏™Âü∫Âõ†ÁªÑÔºåÁªÑÁªáÊàê 113,104
‰∏™Áâ©ÁßçÁ∞á„ÄÇ</p>
</li>
<li>
<p>Species overviewÔºöGTDB R220 Áî± 584,382 ‰∏™ÁªÜËèåÂíå 12,477
‰∏™Âè§ÁªÜËèåÂü∫Âõ†ÁªÑÁªÑÊàêÔºåÂàÜ‰∏∫ 107,235 ‰∏™ÁªÜËèåÂíå 5,869 ‰∏™Âè§ÁªÜËèåÁâ©ÁßçÁ∞á„ÄÇ</p>
</li>
<li>
<p>Genome categoriesÔºöGTDB
ÂàÜÁ±ªÁæ§Áî±ÂàÜÁ¶ªÂü∫Âõ†ÁªÑÔºàisolateÔºâ„ÄÅÂÆèÂü∫Âõ†ÁªÑÁªÑË£ÖÂü∫Âõ†ÁªÑ (MAG)
ÂíåÂçïÊâ©Â¢ûÂü∫Âõ†ÁªÑ (SAG) ÁªÑÊàê„ÄÇ</p>
</li>
<li>
<p>GTDB species representativesÔºöÊØè‰∏™ GTDB Áâ©ÁßçÁ∞áÈÉΩÁî±‰∏Ä‰∏™Âü∫Âõ†ÁªÑË°®Á§∫„ÄÇ</p>
</li>
<li>
<p>Quality of GTDB representative genomesÔºöË¢´ÈÄâ‰∏∫ GTDB
Áâ©Áßç‰ª£Ë°®ÁöÑÂü∫Âõ†ÁªÑÁöÑË¥®Èáè„ÄÇ‰ΩøÁî® CheckM ‰º∞ËÆ°Âü∫Âõ†ÁªÑÂÆåÊï¥ÊÄßÂíåÊ±°ÊüìÔºåÂπ∂Ê†πÊçÆ
MIMAG
Âü∫Âõ†ÁªÑÊ†áÂáÜËøõË°åÁùÄËâ≤„ÄÇ‰∏ÄËà¨Êù•ËØ¥Ôºå‰ª£Ë°®ÊÄßÂü∫Âõ†ÁªÑ‰ªÖÈôê‰∫éÊª°Ë∂≥<code>completeness - 5*contamination &gt;50</code>ÔºåÂ≠òÂú®‰∏Ä‰∫õ‰æãÂ§ñÊÉÖÂÜµÔºåÂ¶Ç‰∏ãÈù¢ÊâÄÁ§∫„ÄÇ
<img src="images/genomes.png" title=""/></p>
</li>
<li>
<p>Taxa with the largest number of speciesÔºöÁªôÂá∫‰∫ÜÊØè‰∏™ÂàÜÁ±ªÁ≠âÁ∫ßÂåÖÂê´ÊúÄÂ§ö
GTDB
Áâ©ÁßçÁ∞áÁöÑÂàÜÁ±ªÂçïÂÖÉÔºåÈó®Ê∞¥Âπ≥ÊòØÂ∏∏ËßÅÁöÑPseudomonadotaÔºåBacillotaÔºåBacteroidotaÔºåActinomycetotaÁ≠â„ÄÇ</p>
</li>
<li>
<p>Taxa with the largest number of sequenced genomesÔºå‰∏é6Á±ª‰ºº„ÄÇ</p>
</li>
<li>
<p>Relative evolutionary
divergenceÔºàREDÔºâÔºöÊèê‰æõ‰∫Ü‰ªéÈó®Âà∞Â±ûÁöÑÊØè‰∏™ÂàÜÁ±ªÁ≠âÁ∫ßÁöÑÁ±ªÁæ§ÁöÑÁõ∏ÂØπËøõÂåñÂàÜÊ≠ßÔºàREDÔºâ„ÄÇ
RED ÂÄºÊèê‰æõ‰∫ÜÁõ∏ÂØπÊó∂Èó¥ÁöÑÊìç‰ΩúËøë‰ººÂÄºÔºåÂÖ∂‰∏≠Áé∞ÊúâÂàÜÁ±ªÁæ§Â≠òÂú®‰∫éÂΩìÂâç
(RED=1)ÔºåÊúÄÂêé‰∏Ä‰∏™ÂÖ±ÂêåÁ•ñÂÖàÂá∫Áé∞Âú®ËøáÂéªÁöÑÂõ∫ÂÆöÊó∂Èó¥
(RED=0)ÔºåÂÜÖÈÉ®ËäÇÁÇπÊ†πÊçÆ‰ª•‰∏ãÂÄºÂú®Ëøô‰∫õÂÄº‰πãÈó¥Á∫øÊÄßÊèíÂÄºË∞±Á≥ªÁâπÂÆöÁöÑËøõÂåñÈÄüÁéá„ÄÇ</p>
</li>
<li>
<p>Comparison of GTDB and NCBI taxaÔºöGTDB
Áâ©Áßç‰ª£Ë°®ÊÄßÂü∫Âõ†ÁªÑÂíåÊâÄÊúâÂÖ∑ÊúâÂàÜÈÖçÁöÑ NCBI ÂàÜÁ±ªÊ≥ïÁöÑ GTDB Âü∫Âõ†ÁªÑÁöÑ GTDB Âíå
NCBI
ÂàÜÁ±ªÂ≠¶ÂàÜÈÖçÁöÑÊØîËæÉ„ÄÇÂØπ‰∫éÊØè‰∏™ÂàÜÁ±ªÁ≠âÁ∫ßÔºåÂ¶ÇÊûú‰∏Ä‰∏™ÂàÜÁ±ªÂçïÂÖÉÂú®‰∏§‰∏™ÂàÜÁ±ªÊ†áÂáÜ‰∏≠ÂêçÁß∞Áõ∏ÂêåÔºåÂàôËØ•ÂàÜÁ±ªÂçïÂÖÉË¢´ÂàÜÁ±ª‰∏∫<code>unchange</code>ÔºõÂ¶ÇÊûú
GTDB ÂàÜÁ±ªÊ†áÂáÜÊèê‰æõ‰∫Ü NCBI
ÂàÜÁ±ªÊ†áÂáÜ‰∏≠Áº∫Â∞ëÁöÑÂêçÁß∞‰ø°ÊÅØÔºåÂàôËØ•ÂàÜÁ±ªÂçïÂÖÉË¢´ÂàÜÁ±ª‰∏∫<code>passive change</code>ÔºõÂ¶ÇÊûú‰∏§‰∏™ÂàÜÁ±ªÊ†áÂáÜ‰πãÈó¥ÁöÑÂêçÁß∞‰∏çÂêåÔºåÂàôËØ•ÂàÜÁ±ªÂçïÂÖÉË¢´ÂàÜÁ±ª‰∏∫<code>actively changed</code>„ÄÇÂèØ‰ª•ÁúãÂà∞ÔºåÂú®genomeÊ∞¥Âπ≥‰∏äGTDBÂíåNCBIËøòÊòØÊúâÂæàÂ§öÂêçÁß∞‰∏çÂØπÂ∫îÁöÑÔºåÂèØ‰ª•ËÄÉËôë‰ΩøÁî®gtdb_to_taxdumpËøõË°åËΩ¨Âåñ„ÄÇ
<img src="images/gtdb2ncbi.png" title=""/></p>
</li>
<li>
<p>Genomic statisticsÔºöGTDB Áâ©Áßç‰ª£Ë°®ÊÄßÂü∫Âõ†ÁªÑÂíå GTDB
‰∏≠ÊâÄÊúâÂü∫Âõ†ÁªÑÁöÑÂÖ≥ÈîÆÂü∫Âõ†ÁªÑÁªüËÆ°Êï∞ÊçÆ„ÄÇ <img src="images/genome_stat.png" title=""/></p>
</li>
<li>
<p>Nomenclatural types per
rankÔºöÊòæÁ§∫‰∫ÜÊØè‰∏™ÂàÜÁ±ªÁ≠âÁ∫ßÁöÑÂç†‰ΩçÁ¨¶‰∏éÊãâ‰∏ÅÂåñÂêçÁß∞ÁöÑÁªÜÂàÜ„ÄÇ</p>
</li>
</ol>
<h3 id="forum">Forum</h3>
<p>GTDBËøò‰∏ìÈó®Êê≠Âª∫‰∫Ü‰∏Ä‰∏™Áî®Êà∑Á§æÂå∫ÂíåËÆ®ËÆ∫Âå∫Ôºö</p>
<ul>
<li>AnnouncementÔºöÁî®‰∫éÂÖ¨Â∏ÉÊúÄÊñ∞ GTDB
Êñ∞Èóª„ÄÅÂç≥Â∞Ü‰∏æË°åÁöÑ‰ºöËÆÆÂíå‰ª•ÂàÜÁ±ªÂ≠¶‰∏∫ÈáçÁÇπÁöÑÊâãÁ®øÁöÑÂºÄÊîæËÆ∫Âùõ„ÄÇ</li>
<li>Taxonomy
(Bacteria/Archaea)ÔºöËøô‰∏™ÂàÜÁ±ª‰∏ìÊ≥®‰∫éËÆ®ËÆ∫‰∏éGTDBÂàÜÁ±ªÊ≥ïÁõ∏ÂÖ≥ÁöÑÂÜÖÂÆπ„ÄÇÁî®Êà∑Âú®ËøôÈáåËÆ®ËÆ∫ÂàÜÁ±ªÊ≥ïÁöÑÂÖ∑‰ΩìÁªÜËäÇ„ÄÅ‰∏éÂÖ∂‰ªñÊï∞ÊçÆÂ∫ìÔºàÂ¶ÇNCBIÔºâÁöÑÊØîËæÉ„ÄÅÂàÜÁ±ªÊ†áÂáÜÁöÑÊõ¥Êñ∞Á≠â„ÄÇ</li>
<li>GTDB-TkÔºö‰∏ìÊ≥®‰∫éGTDB-TkÂ∑•ÂÖ∑ÂåÖÁöÑ‰ΩøÁî®ÂíåËÆ®ËÆ∫„ÄÇÁî®Êà∑ÂèØ‰ª•Âú®ËøôÈáåÊèêÂá∫ÂÖ≥‰∫éGTDB-TkÁöÑ‰ΩøÁî®ÈóÆÈ¢ò„ÄÅÂàÜ‰∫´‰ΩøÁî®ÁªèÈ™å„ÄÅËÆ®ËÆ∫Â∑•ÂÖ∑ÂåÖÁöÑÂäüËÉΩÂíåÊÄßËÉΩÁ≠â„ÄÇ</li>
<li>GTDBÔºö‰∏éÂü∫Âõ†ÁªÑÂàÜÁ±ªÊï∞ÊçÆÂ∫ì (GTDB) Áõ∏ÂÖ≥ÁöÑ‰∏ÄËà¨ÈóÆÈ¢ò„ÄÇ</li>
<li>Website feedbackÔºöËÆ®ËÆ∫ËØ•ÁΩëÁ´ô„ÄÅÂÖ∂ÁªÑÁªá„ÄÅÂÖ∂Ëøê‰ΩúÊñπÂºè‰ª•ÂèäÂ¶Ç‰ΩïÊîπËøõÂÆÉ„ÄÇ</li>
</ul>
<h3 id="help">Help</h3>
<p>Â∏ÆÂä©È°µÈù¢‰πüÊèê‰æõ‰∫Ü‰∏Ä‰∫õÈùûÂ∏∏ÊúâÁî®ÁöÑ‰ø°ÊÅØ„ÄÇ</p>
<ul>
<li>
<p>AboutÔºöGTDBÁÆÄË¶Å‰ªãÁªçÔºåÁÆ°ÁêÜÂõ¢Èòü‰ª•ÂèäÂèëË°®ËÆ∫Êñá</p>
</li>
<li>
<p>FAQÔºöÂàóÂá∫‰∫Ü‰∏Ä‰∫õÂ∏∏ËßÅÈóÆÈ¢òÔºå‰æãÂ¶ÇÔºö</p>
<ol>
<li>‰∏∫‰ªÄ‰πàÈó®ÂêçÁß∞ÁöÑÂêéÁºÄÊõ¥Êîπ‰∏∫-otaÔºüWhitman et al.¬†(2018)
Âª∫ËÆÆÂÉèÂÖ∂‰ªñÁ≠âÁ∫ß‰∏ÄÊ†∑ËßÑËåÉÈó®Á≠âÁ∫ßÁöÑÂêéÁºÄ„ÄÇ</li>
<li>‰∏∫‰ªÄ‰πà‰∏Ä‰∫õfamilyÂíåÊõ¥È´òÁ∫ßÂà´ÁöÑÂêçÂ≠ó‰ª•Â≠óÊØçÂêéÁºÄÁªìÂ∞æÔºüÂ±ûÁ≠âÁ∫ß‰ª•‰∏äÁöÑÂàÜÁ±ªÂçïÂÖÉÂêçÁß∞ÈôÑÂä†Â≠óÊØçÂêéÁºÄË°®Á§∫Â±û‰∫é‰ª•‰∏ãÁ±ªÂà´ÁöÑÁ±ªÁæ§Ôºö
<ol>
<li>GTDB ÂèÇËÄÉÊ†ë‰∏≠‰∏çÊòØÂçïÁ≥ªÁ±ªÁæ§Ôºå‰ΩÜÂ≠òÂú®ÂÖ∂‰ªñËØÅÊçÆË°®ÊòéÂÆÉ‰ª¨ÊòØÂçïÁ≥ªÁ±ªÁæ§Ôºõ</li>
<li>Âú®ÁâàÊú¨‰πãÈó¥‰ΩçÁΩÆ‰∏çÁ®≥ÂÆöÁöÑÁªÑ„ÄÇ</li>
</ol>
</li>
</ol>
</li>
<li>
<p>MethodsÔºöÂØπGTDBÊï∞ÊçÆÂ∫ìÊûÑÂª∫ËøáÁ®ã‰∏≠ÁöÑÊñπÊ≥ïËøõË°åÂÖ∑‰ΩìÊèèËø∞Ôºå‰æãÂ¶ÇÔºö</p>
<ol>
<li>Gene
identificationÔºå‰ΩøÁî®ProdigalÈâ¥ÂÆöÂü∫Âõ†ÔºåHMMERÈâ¥ÂÆöÊ†áËÆ∞Âü∫Âõ†ÔºàÊù•Ëá™PfamÂíåTIGRFAMÔºâÂπ∂Â∫èÂàóÊØîÂØπ„ÄÇ</li>
<li>Multiple sequence alignmentÔºåÁªÜËèåÂíåÂè§ËèåÂ§öÈáçÂ∫èÂàóÊØîÂØπ (MSA) ÂàÜÂà´Áî±
120 ‰∏™ (bac120) Êàñ 53 ‰∏™ (arc53) Á≥ªÁªüÂèëËÇ≤‰ø°ÊÅØÊ†áËÆ∞‰∏≤ËÅîËÄåÊàê„ÄÇ</li>
<li>Tree inferenceÔºåÁªÜËèåÂèÇËÄÉÊ†ëÊòØÂú® WAG Ê®°Âûã‰∏ã‰ΩøÁî®
FastTree„ÄÇÂè§ËèåÂèÇËÄÉÊ†ëÊòØÊ†πÊçÆ PMSF Ê®°Âûã‰∏ãÁöÑ IQ-TreeÔºåÂπ∂‰ΩøÁî®
FastTreeÊé®Êñ≠ÂàùÂßãÊ†ë‰ª•ÊÅ¢Â§çÂçïÁ≥ªË∞±Á≥ª„ÄÇ</li>
<li>Identifying 16S rRNA sequencesÔºå‰ΩøÁî® nhmmer‰ª•ÂèäÊù•Ëá™ RFAM
Êï∞ÊçÆÂ∫ìÁöÑ 16S rRNA Ê®°ÂûãÔºàRF00177Ôºâ„ÄÇ</li>
<li>Average nucleotide identityÔºåskaniÔºà‰ªéÁâàÊú¨ 220
ÂºÄÂßãÔºâÔºå‰πãÂâçÊòØFastANI„ÄÇ</li>
<li>RED
normalizationÔºöÁõ∏ÂØπËøõÂåñÂàÜÊ≠ßÁî®‰∫éÂ∏ÆÂä©ÊèèÁªòÁâ©ÁßçÁ≠âÁ∫ß‰πã‰∏äÁöÑÂàÜÁ±ªÁæ§ÔºåREDÁöÑÊ¶ÇÂøµÊòØÁõ∏ÂêåÁ≠âÁ∫ßÁöÑÁ±ªÁæ§Âú®ËøáÂéªÂ∫îËØ•ÂêåÊó∂Âá∫Áé∞„ÄÇRED
Êèê‰æõ‰∫ÜÁõ∏ÂØπÊó∂Èó¥ÁöÑÊìç‰ΩúËøë‰ººÂÄºÔºåÂÖ∂‰∏≠Áé∞ÊúâÂàÜÁ±ªÁæ§Â≠òÂú®‰∫éÂΩìÂâç (RED =
1)ÔºåÊúÄÂêé‰∏Ä‰∏™ÂÖ±ÂêåÁ•ñÂÖàÂá∫Áé∞Âú®ËøáÂéªÁöÑÂõ∫ÂÆöÊó∂Èó¥ (RED =
0)ÔºåÂÜÖÈÉ®ËäÇÁÇπÊ†πÊçÆË∞±Á≥ªÂú®Ëøô‰∫õÂÄº‰πãÈó¥Á∫øÊÄßÊèíÂÄº-ÁâπÂÆöÁöÑËøõÂåñÈÄüÁéá„ÄÇ</li>
</ol>
</li>
<li>
<p>APIÔºöÂäüËÉΩËæÉÂ§öÔºå‰ΩÜÊñáÊ°£‰∏çÊòØÂæàËØ¶ÁªÜÔºåÊú™Êù•‰πüËÆ∏‰ºöÁî®‰∏ä</p>
</li>
</ul>
<h2 id="gtdb-tk">GTDB-Tk</h2>
<p>GTDB-Tk ÊòØ‰∏Ä‰∏™ËΩØ‰ª∂Â∑•ÂÖ∑ÂåÖÔºåÁî®‰∫éÊ†πÊçÆÂü∫Âõ†ÁªÑÊï∞ÊçÆÂ∫ìÂàÜÁ±ªÊ≥ï (GTDB)
ÂØπÁªÜËèåÂíåÂè§ÁªÜËèåÂü∫Âõ†ÁªÑËøõË°åÂÆ¢ËßÇÂàÜÁ±ª„ÄÇ
ÂÆÉÊó®Âú®‰∏éÊúÄÊñ∞ËøõÂ±ïÁõ∏ÁªìÂêàÔºåÂÖÅËÆ∏Áõ¥Êé•‰ªéÁéØÂ¢ÉÊ†∑Êú¨‰∏≠Ëé∑ÂæóÊï∞ÁôæÊàñÊï∞ÂçÉ‰∏™ÂÆèÂü∫Âõ†ÁªÑÁªÑË£ÖÂü∫Âõ†ÁªÑ
(MAG)ÔºåËøòÂèØ‰ª•Â∫îÁî®‰∫éÂàÜÁ¶ªÂíåÂçïÁªÜËÉûÂü∫Âõ†ÁªÑ„ÄÇ</p>
<p>Âú∞ÂùÄÔºö<a class="link" href="https://github.com/Ecogenomics/GTDBTk"  target="_blank" rel="noopener"
    >https://github.com/Ecogenomics/GTDBTk</a></p>
<p>ÂÆòÊñπÊñáÊ°£Ôºö<a class="link" href="https://ecogenomics.github.io/GTDBTk/"  target="_blank" rel="noopener"
    >https://ecogenomics.github.io/GTDBTk/</a></p>
<h3 id="install">Install</h3>
<p>È¶ñÂÖàË¶ÅÂÆâË£ÖËΩØ‰ª∂Âíå‰∏ãËΩΩÂØπÂ∫îÁöÑReference
dataÔºåÊúÄÊñ∞ÁöÑR220Êï∞ÊçÆÂ∫ìÈúÄË¶Å‰ΩøÁî®v2.4.0ÁöÑGTDB-TkÔºà2024.04.24Ôºâ„ÄÇÂ¶ÇÊûúË¶ÅÂíå‰πãÂâçÈ°πÁõÆ‰∏ÄËá¥ÔºåÂèØ‰ª•‰ΩøÁî®R214ÁâàÊú¨Êï∞ÊçÆÂ∫ìÂØπÂ∫îv2.3.2ÁöÑGTDB-Tk„ÄÇ</p>
<h4 id="1conda">1.conda</h4>
<p>condaÂÆâË£ÖÊØîËæÉÁÆÄÂçïÔºå‰ΩÜÊúÄÂ•ΩÊñ∞Âª∫‰∏Ä‰∏™ÁéØÂ¢ÉÔºåÂèØËÉΩ‰ºöÊØîËæÉÊÖ¢Ôºàmamba‰ª£Êõøconda‰ºöÂø´‰∏ÄÁÇπÔºâÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># using conda</span>
</span></span><span class="line"><span class="cl">conda create -n gtdbtk-2.3.2 -c conda-forge -c bioconda <span class="nv">gtdbtk</span><span class="o">=</span>2.3.2
</span></span><span class="line"><span class="cl"><span class="c1"># ‰∏ãËΩΩ GTDB-Tk ÂèÇËÄÉÊï∞ÊçÆÂπ∂‰∏∫ÂÖ∂ÂëΩÂêç</span>
</span></span><span class="line"><span class="cl">download-db.sh
</span></span><span class="line"><span class="cl"><span class="c1"># ÊøÄÊ¥ªGTDB-Tk condaÁéØÂ¢É</span>
</span></span><span class="line"><span class="cl">conda ÊøÄÊ¥ª gtdbtk-2.3.2
</span></span><span class="line"><span class="cl"><span class="c1"># Â∞ÜÁéØÂ¢ÉÂèòÈáèËÆæÁΩÆ‰∏∫ÂåÖÂê´GTDB-TkÂèÇËÄÉÊï∞ÊçÆÁöÑÁõÆÂΩï</span>
</span></span><span class="line"><span class="cl">conda env config vars <span class="nb">set</span> <span class="nv">GTDBTK_DATA_PATH</span><span class="o">=</span><span class="s2">&#34;/path/to/release/package/&#34;</span><span class="p">;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="2pip">2.pip</h4>
<p>pipÂèØ‰ª•Âø´ÈÄüÂÆâË£ÖgtdbtkÂåÖÔºå‰ΩÜÊòØÂÆÉÁöÑÁ¨¨‰∏âÊñπ‰æùËµñÈ°πÈúÄË¶ÅÊâãÂä®ÂÆâË£Ö‰∏Ä‰∏ã
Ôºà‰∏çÊòØÂæàÂ§çÊùÇÔºåÊàëÁî®ÁöÑpipÂÆâË£ÖÔºåÂõ†‰∏∫‰∏çÊÉ≥Ë¶ÅÂ§™Â§öÁöÑcondaÁéØÂ¢ÉÔºåÂ§™ÂÜó‰Ωô‰∫ÜÔºâÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">python -m pip install <span class="nv">gtdbtk</span><span class="o">==</span>2.3.2
</span></span></code></pre></td></tr></table>
</div>
</div><p>Á¨¨‰∏âÊñπ‰æùËµñÈ°πÔºö</p>
<ul>
<li>ProdigalÔºàÂéüÊ†∏Âü∫Âõ†ËØÜÂà´ÂíåÁøªËØëËµ∑Âßã‰ΩçÁÇπÈâ¥ÂÆöÔºâ</li>
<li>HMMERÔºàÂä†ÈÄüÊóèHMM ÊêúÁ¥¢Ôºâ</li>
<li>pplacerÔºàÁ∫øÊÄßÊó∂Èó¥ÊúÄÂ§ß‰ººÁÑ∂ÂíåË¥ùÂè∂ÊñØÁ≥ªÁªüÂèëËÇ≤Â∞ÜÂ∫èÂàóÊîæÁΩÆÂà∞Âõ∫ÂÆöÂèÇËÄÉÊ†ë‰∏äÔºâ</li>
<li>skani (v2.4.0ÂºÄÂßã‰æùËµñÔºåÈÄöËøá skani
ÁöÑÁ®ÄÁñèÈìæÊé•ËøõË°åÂø´ÈÄü„ÄÅÁ®≥ÂÅ•ÁöÑÂÆèÂü∫Âõ†ÁªÑÂ∫èÂàóÊØîËæÉÔºåÊØîfastANIËøòÂø´Ôºå‰∏ãÊ¨°ÊúâÊú∫‰ºöËÆ≤‰∏Ä‰∏ãËøô‰∏™)</li>
<li></li>
<li>FastTreeÔºàÁî®‰∫éÂ§ßÂØπÈΩêÁöÑËøë‰ººÊúÄÂ§ß‰ººÁÑ∂Ê†ëÔºâ</li>
<li>MashÔºàÂø´ÈÄü‰º∞ËÆ°Âü∫Âõ†ÁªÑÂíåÂÆèÂü∫Âõ†ÁªÑË∑ùÁ¶ªÔºâ</li>
</ul>
<p>Ëøô‰∫õÈÉΩÊòØÂæàÂ∏∏Áî®ÁöÑËΩØ‰ª∂ÔºåÊ≤°ÊúâÁöÑËØùÁî®condaÂÆâË£Ö‰∏Ä‰∏ã„ÄÇ</p>
<p>GTDB-Tk ÈúÄË¶ÅÂ∞Ü GTDBTK_DATA_PATH ÁéØÂ¢ÉÂèòÈáèËÆæÁΩÆ‰∏∫ÂåÖÂê´Êú™Â≠òÊ°£ÁöÑ GTDB-Tk
ÂèÇËÄÉÊï∞ÊçÆÁöÑÁõÆÂΩï„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export GTDBTK_DATA_PATH=/path/to/release/package/&#39;</span> &gt;&gt; ~/.bashrc
</span></span><span class="line"><span class="cl"><span class="nb">source</span> ~/.bashrc
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="3docker">3.docker</h4>
<p>Â¶ÇÊûúÁî®ËøádockerÁöÑËØùÔºåËøô‰∏™‰πüÈùûÂ∏∏ÁÆÄÂçïÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#‰∏ãËΩΩËß£ÂéãÁöÑ GTDB-Tk ÂèÇËÄÉÊï∞ÊçÆÂà∞ÊÇ®ÈÄâÊã©ÁöÑÁõÆÂΩïÔºå‰æãÂ¶ÇÔºö/host/release_data/</span>
</span></span><span class="line"><span class="cl">mv /path/to/release/package/ /host/release_data/
</span></span><span class="line"><span class="cl"><span class="c1">#‰∏∫ GTDB-Tk ÂàõÂª∫Êú¨Âú∞ I/O ÁõÆÂΩï</span>
</span></span><span class="line"><span class="cl">mkdir /host/gtdbtk_io/
</span></span><span class="line"><span class="cl"><span class="c1"># ËøêË°åÂÆπÂô®</span>
</span></span><span class="line"><span class="cl">docker run -v /host/gtdbtk_io:/data -v /host/release_data:/refdata ecogenomic/gtdbtk classify_wf --genome_dir /data/genomes --out_dir /data/output
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="reference-data">Reference data</h4>
<p>GTDB-Tk ÈúÄË¶ÅÁ∫¶ 110G
ÁöÑÂ§ñÈÉ®Êï∞ÊçÆÔºåÈúÄË¶Å‰∏ãËΩΩÂíåËß£Ê°£„ÄÇÁõ¥Êé•wgetÂæàÊÖ¢ÁöÑËØùÔºå‰πüÂèØ‰ª•ÂÖà‰∏ãËΩΩÂà∞Êú¨Âú∞ÔºåÂÜçÈÄöËøáFTPËΩØ‰ª∂Êàñscp‰∏ä‰º†Âà∞ÊúçÂä°Âô®Ôºö</p>
<p>Êï∞ÊçÆÂú∞ÂùÄÔºö<a class="link" href="https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/gtdbtk_package/"  target="_blank" rel="noopener"
    >https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/gtdbtk_package/</a></p>
<p>ÂØπ‰∫éÂÆåÊï¥ÂåÖ(full package)Ôºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">wget https://data.ace.uq.edu.au/public/gtdb/data/releases/latest/auxillary_files/gtdbtk_package/full_package/gtdbtk_data.tar.gz
</span></span><span class="line"><span class="cl">tar xvzf gtdbtk_data.tar.gz
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÂØπ‰∫éÊãÜÂàÜÂåÖ(split package)Ôºö</p>
<ol>
<li>Á°Æ‰øù GTDB-Tk ÂèëÂ∏ÉÊï∞ÊçÆÁöÑÊâÄÊúâÈÉ®ÂàÜÈÉΩ‰Ωç‰∫éÂêå‰∏ÄÁõÆÂΩï‰∏≠„ÄÇ</li>
<li>ÊâìÂºÄÁªàÁ´ØÊàñÂëΩ‰ª§ÊèêÁ§∫Á¨¶„ÄÇ</li>
<li>ÂØºËà™Âà∞ÂåÖÂê´ GTDB-Tk ÂèëÂ∏ÉÊï∞ÊçÆÈÉ®ÂàÜÁöÑÁõÆÂΩï„ÄÇ</li>
<li>‰ΩøÁî®‰ª•‰∏ãÂëΩ‰ª§Â∞ÜÊâÄÊúâÈÉ®ÂàÜËøûÊé•Âà∞‰∏Ä‰∏™Â≠òÊ°£‰∏≠Ôºö
<code>cat gtdbtk_r220_data.tar.gz.part_* &gt; gtdbtk_r220_data.tar.gz</code></li>
<li>ÂëΩ‰ª§ÊâßË°åÂÆåÊàêÂêéÔºåÂ∞ÜÂú®Âêå‰∏ÄÁõÆÂΩï‰∏≠Êã•Êúâ‰∏Ä‰∏™Âêç‰∏∫‚Äúgtdbtk_r220_data.tar.gz‚ÄùÁöÑÂ≠òÊ°£Êñá‰ª∂„ÄÇ</li>
</ol>
<h4 id="check">Check</h4>
<p>Ê£ÄÊü•Êï∞ÊçÆÂ∫ìÂíåËΩØ‰ª∂ÔºåÁªìÊûúÊòæÁ§∫OKÂíåDoneË°®Á§∫Ê≠£Â∏∏ÔºàÊ£ÄÊü•fastaniËøô‰∏ÄÊ≠•Ë¶ÅÂæà‰πÖÔºåÊñá‰ª∂ÈùûÂ∏∏Â§ö„ÄÇÊâÄ‰ª•Â¶ÇÊûú‰∏âÊñπËΩØ‰ª∂ÈÉΩOKÁöÑËØù‰πüÂèØ‰ª•Áõ¥Êé•ËøêË°å‰∏ã‰∏Ä‰∏™Á§∫‰æãÊµãËØïÔºâ:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk check_install
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 14:10:37] INFO: GTDB-Tk v2.3.2
[2024-05-20 14:10:37] INFO: gtdbtk check_install
[2024-05-20 14:10:37] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:10:37] INFO: Running install verification
[2024-05-20 14:10:37] INFO: Checking that all third-party software are on the system path:
[2024-05-20 14:10:37] INFO:          |-- FastTree         OK
[2024-05-20 14:10:37] INFO:          |-- FastTreeMP       OK
[2024-05-20 14:10:37] INFO:          |-- fastANI          OK
[2024-05-20 14:10:37] INFO:          |-- guppy            OK
[2024-05-20 14:10:37] INFO:          |-- hmmalign         OK
[2024-05-20 14:10:37] INFO:          |-- hmmsearch        OK
[2024-05-20 14:10:37] INFO:          |-- mash             OK
[2024-05-20 14:10:37] INFO:          |-- pplacer          OK
[2024-05-20 14:10:37] INFO:          |-- prodigal         OK
[2024-05-20 14:10:37] INFO: Checking integrity of reference package: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:10:38] INFO:          |-- pplacer          OK
[2024-05-20 14:10:38] INFO:          |-- masks            OK
[2024-05-20 14:10:39] INFO:          |-- markers          OK
[2024-05-20 14:10:39] INFO:          |-- radii            OK
[2024-05-20 14:10:41] INFO:          |-- msa              OK
[2024-05-20 14:10:41] INFO:          |-- metadata         OK
[2024-05-20 14:10:41] INFO:          |-- taxonomy         OK
[2024-05-20 15:13:45] INFO:          |-- fastani          OK
[2024-05-20 15:13:45] INFO:          |-- mrca_red         OK
[2024-05-20 15:13:45] INFO: Done.
</code></pre>
<p>Ê≥®ÊÑèGTDB-TkÁöÑÁ°¨‰ª∂Ë¶ÅÊ±ÇÔºåÂÖ≥ÈîÆÊòØMemory‰∏ÄÂÆöË¶ÅÁªôË∂≥Â§üÔºå‰∏çÁÑ∂‰ºöÁ®ãÂ∫è‰ºö‰∏≠ÈÄîÁªàÊ≠¢Ôºö</p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th><strong>Domain</strong></th>
          <th><strong>Memory</strong></th>
          <th><strong>Storage</strong></th>
          <th><strong>Time</strong></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td>Archaea</td>
          <td>~60 GB</td>
          <td>~106 GB</td>
          <td>~90 minutes / 1,000 genomes @ 64 CPUs</td>
      </tr>
      <tr>
          <td>Bacteria</td>
          <td>~90GB (545 GB when using ‚Äìfull_tree)</td>
          <td>~106 GB</td>
          <td>~90 minutes / 1,000 genomes @ 64 CPUs</td>
      </tr>
  </tbody>
</table></div>
<p>ÊµãËØïÊµÅÁ®ãÔºåÊó†Êä•ÈîôÁ°ÆÂÆöÁ®ãÂ∫èÁöÑËΩØ‰ª∂ÂíåÊï∞ÊçÆÂ∫ìÊ≠£Â∏∏ÔºàÊ≠§Á§∫‰æãÊòØ3‰∏™Âè§ËèågenomeÔºåÂª∫ËÆÆÂàÜÈÖç60GBÂÜÖÂ≠òÊù•ËøêË°åÔºåÂè¶Â§ñgtdbtk_testÂøÖÈ°ªÊòØ‰∏çÂ≠òÂú®ÁöÑÊàñÁ©∫Êñá‰ª∂Â§πÔºâ:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk <span class="nb">test</span> --out_dir gtdbtk_test
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 14:34:51] INFO: GTDB-Tk v2.3.2
[2024-05-20 14:34:51] INFO: gtdbtk test --out_dir gtdbtk_test
[2024-05-20 14:34:51] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 14:34:51] INFO: Command: gtdbtk classify_wf --skip_ani_screen --genome_dir gtdbtk_test/genomes --out_dir gtdbtk_test/output --cpus 1 -f
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:34:52] INFO: gtdbtk classify_wf --skip_ani_screen --genome_dir gtdbtk_test/genomes --out_dir gtdbtk_test/    
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:34:52] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/      
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:42:06] INFO: Note that Tk classification mode is insufficient for publication of new taxonomic designatio    
    &lt;TEST OUTPUT&gt;     [2024-05-20 14:42:06] INFO: Done.     
[2024-05-20 14:42:09] INFO: Test has successfully finished.
</code></pre>
<p>ÊàëÂàÜÈÖç‰∫Ü4Ê†∏ÂÖ±60GBÂÜÖÂ≠òËøêË°å<code>gtdbtk test</code>ÔºåÁ∫¶439sÂÆåÊàêÔºåÊ∂àËÄó‰∫Ü~45GBÂÜÖÂ≠ò„ÄÇ</p>
<h3 id="‰ΩøÁî®ÊñπÊ≥ï">‰ΩøÁî®ÊñπÊ≥ï</h3>
<p>ÂÖàÁúãÁúãhelpÊñáÊ°£Ôºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ gtdbtk -h
</span></span><span class="line"><span class="cl"><span class="c1">## GTDB-Tk v2.4.0 Â∏ÆÂä©ÊñáÊ°£</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Workflows:
</span></span><span class="line"><span class="cl">    classify_wf -&gt; ÈÄöËøáÂú® GTDB ÂèÇËÄÉÊ†ë‰∏≠ÁöÑ‰ΩçÁΩÆÂØπÂü∫Âõ†ÁªÑËøõË°åÂàÜÁ±ª
</span></span><span class="line"><span class="cl">                     <span class="o">(</span>ani_screening -&gt; identify -&gt; align -&gt; classify<span class="o">)</span>
</span></span><span class="line"><span class="cl">    de_novo_wf  -&gt; Êé®Êñ≠Êñ∞ÁöÑÊ†ëÂπ∂‰ΩøÁî® GTDB ÂàÜÁ±ªËøõË°å‰øÆÈ•∞
</span></span><span class="line"><span class="cl">                     <span class="o">(</span>identify -&gt; align -&gt; infer -&gt; root -&gt; decorate<span class="o">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Methods:
</span></span><span class="line"><span class="cl">    identify -&gt; ËØÜÂà´Âü∫Âõ†ÁªÑ‰∏≠ÁöÑÊ†áËÆ∞Âü∫Âõ†
</span></span><span class="line"><span class="cl">    align    -&gt; ÂàõÂª∫Â§öÂ∫èÂàóÊØîÂØπ
</span></span><span class="line"><span class="cl">    classify -&gt; Á°ÆÂÆöÂü∫Âõ†ÁªÑÁöÑÂàÜÁ±ª
</span></span><span class="line"><span class="cl">    infer    -&gt; ‰ªéÂ§öÂ∫èÂàóÊØîÂØπ‰∏≠Êé®Êñ≠Ê†ë
</span></span><span class="line"><span class="cl">    root     -&gt; ‰ΩøÁî®Â§ñÁæ§ÂØπÊ†ëËøõË°åÊ†πÂÆö
</span></span><span class="line"><span class="cl">    decorate -&gt; Áî® GTDB ÂàÜÁ±ª‰øÆÈ•∞Ê†ë
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Tools:
</span></span><span class="line"><span class="cl">    infer_ranks        -&gt; Âà©Áî®REDÂª∫Á´ãÂÜÖÈÉ®ËäÇÁÇπÁöÑÂàÜÁ±ªÁ≠âÁ∫ß
</span></span><span class="line"><span class="cl">    ani_rep            -&gt; ËÆ°ÁÆóANIÂà∞GTDBÁöÑ‰ª£Ë°®ÊÄßÂü∫Âõ†ÁªÑ
</span></span><span class="line"><span class="cl">    trim_msa           -&gt; Âü∫‰∫éÊé©Á†Å‰øÆÂâ™Êú™‰øÆÂâ™ÁöÑMSAÊñá‰ª∂
</span></span><span class="line"><span class="cl">    export_msa         -&gt; ÂØºÂá∫Êú™‰øÆÂâ™ÁöÑÂè§ÁîüËèåÊàñÁªÜËèåMSAÊñá‰ª∂
</span></span><span class="line"><span class="cl">    remove_labels      -&gt; ‰ªéNewickÊ†ë‰∏≠Âà†Èô§Ê†áÁ≠æ<span class="o">(</span>ÂºïÂØºÂÄºÔºåËäÇÁÇπÊ†áÁ≠æ<span class="o">)</span>
</span></span><span class="line"><span class="cl">    convert_to_itol    -&gt; Â∞ÜGTDB-Tk NewickÊ†ëËΩ¨Êç¢‰∏∫iTOLÊ†ë
</span></span><span class="line"><span class="cl">    convert_to_species -&gt; Â∞ÜGTDBÂü∫Âõ†ÁªÑidËΩ¨Êç¢‰∏∫GTDBÁâ©ÁßçÂêçÁß∞
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  Testing:
</span></span><span class="line"><span class="cl">    <span class="nb">test</span>          -&gt; Áî®3‰∏™Âè§ÁªÜËèåÂü∫Âõ†ÁªÑÈ™åËØÅclassify_wfÁÆ°ÈÅì
</span></span><span class="line"><span class="cl">    check_install -&gt; È™åËØÅÁ¨¨‰∏âÊñπÁ®ãÂ∫èÂíåGTDBÂèÇËÄÉÂåÖ
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÂØπ‰∫éÊØè‰∏Ä‰∏™Ê®°ÂùóÂèØ‰ª•ÂçïÁã¨Êü•ÁúãhelpÔºå‰æãÂ¶ÇÊúÄÈáçË¶ÅÁöÑ<code>classify_wf</code>Â∑•‰ΩúÊµÅÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ gtdbtk classify_wf -h
</span></span><span class="line"><span class="cl">usage: gtdbtk classify_wf <span class="o">(</span>--genome_dir GENOME_DIR <span class="p">|</span> --batchfile BATCHFILE<span class="o">)</span> --out_dir OUT_DIR
</span></span><span class="line"><span class="cl">                          <span class="o">(</span>--skip_ani_screen <span class="p">|</span> --mash_db MASH_DB<span class="o">)</span> <span class="o">[</span>--no_mash<span class="o">]</span> <span class="o">[</span>--mash_k MASH_K<span class="o">]</span> <span class="o">[</span>--mash_s MASH_S<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--mash_v MASH_V<span class="o">]</span> <span class="o">[</span>--mash_max_distance MASH_MAX_DISTANCE<span class="o">]</span> <span class="o">[</span>-f<span class="o">]</span> <span class="o">[</span>-x EXTENSION<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--min_perc_aa MIN_PERC_AA<span class="o">]</span> <span class="o">[</span>--prefix PREFIX<span class="o">]</span> <span class="o">[</span>--genes<span class="o">]</span> <span class="o">[</span>--cpus CPUS<span class="o">]</span> <span class="o">[</span>--pplacer_cpus PPLACER_CPUS<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--force<span class="o">]</span> <span class="o">[</span>--scratch_dir SCRATCH_DIR<span class="o">]</span> <span class="o">[</span>--write_single_copy_genes<span class="o">]</span> <span class="o">[</span>--keep_intermediates<span class="o">]</span>
</span></span><span class="line"><span class="cl">                          <span class="o">[</span>--min_af MIN_AF<span class="o">]</span> <span class="o">[</span>--tmpdir TMPDIR<span class="o">]</span> <span class="o">[</span>--debug<span class="o">]</span> <span class="o">[</span>-h<span class="o">]</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mutually exclusive required arguments:
</span></span><span class="line"><span class="cl">  --genome_dir GENOME_DIR ÂåÖÂê´FASTAÊ†ºÂºèÂü∫Âõ†ÁªÑÊñá‰ª∂ÁöÑÁõÆÂΩï
</span></span><span class="line"><span class="cl">  --batchfile BATCHFILE ÊèèËø∞Âü∫Âõ†ÁªÑÁöÑÊñá‰ª∂Ë∑ØÂæÑ - 2Êàñ3ÂàóÁöÑÂà∂Ë°®Á¨¶ÂàÜÈöîÊñá‰ª∂ÔºàFASTAÊñá‰ª∂„ÄÅÂü∫Âõ†ÁªÑID„ÄÅÁøªËØëË°®<span class="o">[</span>ÂèØÈÄâ<span class="o">]</span>Ôºâ
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">required named arguments:
</span></span><span class="line"><span class="cl">  --out_dir OUT_DIR     ËæìÂá∫Êñá‰ª∂ÁöÑÁõÆÂΩï
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">mutually exclusive required arguments:
</span></span><span class="line"><span class="cl">  --skip_ani_screen     Ë∑≥ËøáaniÁ≠õÈÄâÊ≠•È™§Ôºå‰ΩøÁî®mashÂíåskaniÂØπÂü∫Âõ†ÁªÑËøõË°åÂàÜÁ±ª„ÄÇÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --mash_db MASH_DB     ‰øùÂ≠ò/ËØªÂèñMashÂèÇËÄÉËçâÂõæÊï∞ÊçÆÂ∫ìÔºà.mshÔºâÁöÑË∑ØÂæÑ
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional Mash arguments:
</span></span><span class="line"><span class="cl">  --no_mash             Ë∑≥Ëøá‰ΩøÁî®MashËøõË°åÁöÑÂü∫Âõ†ÁªÑÈ¢ÑËøáÊª§ÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --mash_k MASH_K       k-merÂ§ßÂ∞è<span class="o">[</span>1-32<span class="o">]</span>ÔºàÈªòËÆ§Ôºö16Ôºâ
</span></span><span class="line"><span class="cl">  --mash_s MASH_S       ÊúÄÂ§ßÈùûÂÜó‰ΩôÂìàÂ∏åÊï∞ÔºàÈªòËÆ§Ôºö5000Ôºâ
</span></span><span class="line"><span class="cl">  --mash_v MASH_V       ‰øùÁïôÁöÑÊúÄÂ§ßpÂÄº<span class="o">[</span>0-1<span class="o">]</span>ÔºàÈªòËÆ§Ôºö1.0Ôºâ
</span></span><span class="line"><span class="cl">  --mash_max_distance MASH_MAX_DISTANCE ÈÄâÊã©ÊΩúÂú®GTDBÂü∫Âõ†ÁªÑ‰Ωú‰∏∫Áî®Êà∑Âü∫Âõ†ÁªÑ‰ª£Ë°®ÁöÑÊúÄÂ§ßMashË∑ùÁ¶ª„ÄÇÔºàÈªòËÆ§Ôºö0.15Ôºâ
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional arguments:
</span></span><span class="line"><span class="cl">  -fÔºå--full_tree       Âú®ÂàÜÁ±ªÊ≠•È™§‰∏≠‰ΩøÁî®Êú™ÂàÜÂâ≤ÁöÑÁªÜËèåÊ†ëÔºõËøôÊòØÂéüÂßãÁöÑGTDB-TkÊñπÊ≥ïÔºàÁâàÊú¨&lt;2ÔºâÔºå
</span></span><span class="line"><span class="cl">                        ÈúÄË¶ÅË∂ÖËøá320 GBÁöÑRAMÊù•Âä†ËΩΩÂèÇËÄÉÊ†ëÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  -xÔºå--extension EXTENSION Ë¶ÅÂ§ÑÁêÜÁöÑÊñá‰ª∂Êâ©Â±ïÂêçÔºågz <span class="o">=</span> gzippedÔºàÈªòËÆ§ÔºöfnaÔºâ
</span></span><span class="line"><span class="cl">  --min_perc_aa MIN_PERC_AA 
</span></span><span class="line"><span class="cl">                        ÊéíÈô§Âú®MSA‰∏≠AAÁôæÂàÜÊØî‰∏çË∂≥Ê≠§ÂÄºÁöÑÂü∫Âõ†ÁªÑÔºàÂåÖÂê´ÁïåÈôêÔºâÔºàÈªòËÆ§Ôºö10Ôºâ
</span></span><span class="line"><span class="cl">  --prefix PREFIX       ÊâÄÊúâËæìÂá∫Êñá‰ª∂ÁöÑÂâçÁºÄÔºàÈªòËÆ§ÔºögtdbtkÔºâ
</span></span><span class="line"><span class="cl">  --genes               Ë°®Á§∫ËæìÂÖ•Êñá‰ª∂ÂåÖÂê´È¢ÑÊµãÁöÑËõãÁôΩË¥®Ôºà‰Ωú‰∏∫Ê∞®Âü∫ÈÖ∏ÔºâÔºåË∑≥ËøáÂü∫Âõ†Ë∞ÉÁî®„ÄÇ
</span></span><span class="line"><span class="cl">                        Ë≠¶ÂëäÔºöÊ≠§Ê†áÂøóÂ∞ÜË∑≥ËøáANIÊØîËæÉÊ≠•È™§ÔºàaniÁ≠õÈÄâÂíåÂàÜÁ±ªÔºâ„ÄÇÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --cpus CPUS           Ë¶Å‰ΩøÁî®ÁöÑCPUÊï∞ÈáèÔºàÈªòËÆ§Ôºö1Ôºâ
</span></span><span class="line"><span class="cl">  --pplacer_cpus PPLACER_CPUS pplacerÊîæÁΩÆËøáÁ®ã‰∏≠‰ΩøÁî®ÁöÑCPUÊï∞Èáè
</span></span><span class="line"><span class="cl">  --force               Â¶ÇÊûúÂçï‰∏™Âü∫Âõ†ÁªÑÂèëÁîüÈîôËØØÔºåÂàôÁªßÁª≠Â§ÑÁêÜÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --scratch_dir SCRATCH_DIR
</span></span><span class="line"><span class="cl">                        ÈÄöËøáÂÜôÂÖ•Á£ÅÁõòÊù•ÂáèÂ∞ëpplacerÂÜÖÂ≠ò‰ΩøÁî®ÔºàËæÉÊÖ¢Ôºâ
</span></span><span class="line"><span class="cl">  --write_single_copy_genes
</span></span><span class="line"><span class="cl">                        ËæìÂá∫Êú™ÂØπÈΩêÁöÑÂçïÊã∑Ë¥ùÊ†áËÆ∞Âü∫Âõ†ÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --keep_intermediates  Âú®ÊúÄÁªàÁõÆÂΩï‰∏≠‰øùÁïô‰∏≠Èó¥Êñá‰ª∂ÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span><span class="line"><span class="cl">  --min_af MIN_AF       Â∞ÜÂü∫Âõ†ÁªÑÂàÜÈÖçÂà∞Áâ©ÁßçÁ∞áÁöÑÊúÄÂ∞èÂØπÈΩêÊØî‰æãÔºàÈªòËÆ§Ôºö0.5Ôºâ
</span></span><span class="line"><span class="cl">  --tmpdir TMPDIR       ÊåáÂÆö‰∏¥Êó∂Êñá‰ª∂ÁöÑÂ§áÁî®ÁõÆÂΩïÔºàÈªòËÆ§Ôºö/tmpÔºâ
</span></span><span class="line"><span class="cl">  --debug               ÂàõÂª∫Áî®‰∫éË∞ÉËØïÁõÆÁöÑÁöÑ‰∏≠Èó¥Êñá‰ª∂ÔºàÈªòËÆ§ÔºöFalseÔºâ
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="classify_wf-Ê≥®ÈáäÁâ©Áßç">classify_wf Ê≥®ÈáäÁâ©Áßç</h4>
<p>classify_wfÂÖ®Áß∞‰∏∫ Classify workflowÔºåÂç≥ÂàÜÁ±ªÂ∑•‰ΩúÊµÅÁ®ã„ÄÇ</p>
<p>ÂàÜÁ±ªÂ∑•‰ΩúÊµÅÁ®ãÁî±Âõõ‰∏™Ê≠•È™§ÁªÑÊàêÔºöani_screenÔºàv2.4.0ÂºÄÂßãÂä†ÂÖ•Ôºâ, identify, align, classify.</p>
<ol>
<li>ani_screen Ê≠•È™§Â∞ÜÁî®Êà∑Âü∫Âõ†ÁªÑ‰∏éÁî±ÊâÄÊúâ GTDB ‰ª£Ë°®ÊÄßÂü∫Âõ†ÁªÑÁªÑÊàêÁöÑ Mash Êï∞ÊçÆÂ∫ìËøõË°åÊØîËæÉÔºåÁÑ∂Âêé‰ΩøÁî® skani È™åËØÅÊúÄ‰Ω≥ÁöÑ mash ÂëΩ‰∏≠„ÄÇ‰ΩøÁî® skani ÂàÜÁ±ªÁöÑÁî®Êà∑Âü∫Âõ†ÁªÑ‰∏ç‰ºöËøêË°åÁÆ°ÈÅìÁöÑÂÖ∂‰ΩôÈÉ®ÂàÜÔºàËØÜÂà´„ÄÅÂØπÈΩê„ÄÅÂàÜÁ±ªÔºâÔºåÂπ∂Âú®ÊëòË¶ÅÊñá‰ª∂‰∏≠Êä•Âëä„ÄÇ</li>
<li>identifyÊ≠•È™§‰ΩøÁî® Prodigal Ë∞ÉÁî®Âü∫Âõ†ÔºåÂπ∂‰ΩøÁî® HMM Ê®°ÂûãÂíå HMMER ÂåÖÊù•ËØÜÂà´Áî®‰∫éÁ≥ªÁªüÂèëËÇ≤Êé®Êñ≠ÁöÑ 120 ‰∏™ÁªÜËèåÂíå 53 ‰∏™Âè§ÁªÜËèåÊ†áËÆ∞Âü∫Âõ† (Parks et al., 2018)„ÄÇÈÄöËøáÂ∞ÜÊ†áËÆ∞Âü∫Âõ†‰∏éÂÖ∂ÂêÑËá™ÁöÑ HMM Ê®°ÂûãËøõË°åÊØîÂØπÊù•Ëé∑ÂæóÂ§öÈáçÂ∫èÂàóÊØîÂØπ (MSA)„ÄÇ</li>
<li>alignÊ≠•È™§ËøûÊé•ÊØîÂØπÁöÑÊ†áËÆ∞Âü∫Âõ†ÔºåÂπ∂Â∞ÜËøûÊé•ÁöÑ MSA ËøáÊª§‰∏∫Â§ßÁ∫¶ 5,000 ‰∏™Ê∞®Âü∫ÈÖ∏„ÄÇ</li>
<li>ÊúÄÂêéÔºåclassifyÊ≠•È™§‰ΩøÁî® pplacer Êü•Êâæ GTDB-Tk ÂèÇËÄÉÊ†ë‰∏≠ÊØè‰∏™Âü∫Âõ†ÁªÑÁöÑÊúÄÂ§ß‰ººÁÑ∂‰ΩçÁΩÆ„ÄÇ GTDB-Tk Ê†πÊçÆÊØè‰∏™Âü∫Âõ†ÁªÑÂú®ÂèÇËÄÉÊ†ë‰∏≠ÁöÑ‰ΩçÁΩÆ„ÄÅÁõ∏ÂØπËøõÂåñÂàÜÊ≠ßÂíå/Êàñ‰∏éÂèÇËÄÉÂü∫Âõ†ÁªÑÁöÑÂπ≥ÂùáÊ†∏Ëã∑ÈÖ∏Âêå‰∏ÄÊÄß (ANI) ÂØπÊØè‰∏™Âü∫Âõ†ÁªÑËøõË°åÂàÜÁ±ª„ÄÇ</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># ËæìÂÖ•Âü∫Âõ†ÁªÑÁõÆÂΩï</span>
</span></span><span class="line"><span class="cl">gtdbtk classify_wf --genome_dir genomes/ --out_dir classify_wf_out --cpus <span class="m">16</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÂèØ‰ª•Ê†πÊçÆ‰∏ãÂõæ‰º∞ËÆ°Ëá™Â∑±ÁöÑ‰ªªÂä°ËøêË°åÊó∂Èó¥ÔºåËØÑ‰º∞ËµÑÊ∫êÂàÜÈÖçÔºö
<img src="images/gtdb_perf.png" title=""/></p>
<h4 id="output">output</h4>
<p>GTDB-Tk Âú®classify_wfÁÆ°ÈÅìÁöÑÂêÑ‰∏™Èò∂ÊÆµËæìÂá∫ËÆ∏Â§öÊñá‰ª∂ÔºåÂÖ∑‰ΩìÊ†ºÂºèÂíåÁ§∫‰æãÂèØ‰ª•Âú®ÂÆòÁΩëÊü•ÁúãÔºö
<a class="link" href="https://ecogenomics.github.io/GTDBTk/files/index.html"  target="_blank" rel="noopener"
    >https://ecogenomics.github.io/GTDBTk/files/index.html</a></p>
<p>ÁªìÊûúÁõÆÂΩï(classify_wf_out/output)‰∏≠‰∏ªË¶ÅÁªìÊûúÊñá‰ª∂ (ar53‰∏∫Âè§ËèåÁõ∏ÂÖ≥Êñá‰ª∂Ôºåbac120‰∏∫ÁªÜËèåÁõ∏ÂÖ≥Êñá‰ª∂)Ôºö</p>
<ul>
<li>classify/gtdbtk.ar53.classify.tree Ê†ëÊñá‰ª∂ÔºåNewickÊ†ºÂºèÔºåÂèØ‰ª•Áî®‰∫éÁ≥ªÁªüÂèëËÇ≤Ê†ëÁöÑÁªòÂà∂„ÄÇ</li>
<li>classify/gtdbtk.ar53.summary.tsv ÂàÜÁ±ªÁªìÊûúÊñá‰ª∂Ôºå‰∏ªË¶ÅÁªìÊûúÔºåÂèØ‰ª•Áî®‰∫éÁ≥ªÁªüÂèëËÇ≤Ê†ëÁöÑÊ≥®Èáä„ÄÇ</li>
<li>identify/gtdbtk.ar53.markers_summary.tsv ‰ΩøÁî®Ê†áËÆ∞Âü∫Âõ†ÊÉÖÂÜµ</li>
<li>align/gtdbtk.ar53.msa.fasta.gz Â§öÂ∫èÂàóÂØπÈΩêÊñá‰ª∂(ËæìÂÖ•+ÂèÇËÄÉ)</li>
<li>align/gtdbtk.ar53.user_msa.fasta.gz Â§öÂ∫èÂàóÂØπÈΩêÊñá‰ª∂(ËæìÂÖ•)</li>
</ul>
<p>ÁªòÂà∂Á≥ªÁªüÂèëËÇ≤Ê†ëÂèØ‰ª•ÂèÇËÄÉÊàë‰πãÂâçÂÜôÁöÑ<a class="link" href="../r-tree" >RÁªòÂà∂‰ºòÁæéÁöÑËøõÂåñÊ†ëÔºàÂü∫Á°ÄÔºâ</a>Âíå<a class="link" href="../r-tree2" >RÁªòÂà∂‰ºòÁæéÁöÑËøõÂåñÊ†ëÔºàËøõÈò∂Ôºâ</a>ÔºåÊàñËÄÖ‰ΩøÁî®iPhyloÁΩëÁ´ôÊù•‰∫§‰∫íÂºèÁªòÂõæÔºö<a class="link" href="../iphylo" >iPhylo ÁîüÊàêÂπ∂ÁªòÂà∂‰ºòÁæéÁöÑÂàÜÁ±ªÊ†ë</a>„ÄÇ</p>
<h3 id="ÂàÜÊ≠•Á§∫‰æã">ÂàÜÊ≠•Á§∫‰æã</h3>
<h4 id="Ëé∑ÂèñÊï∞ÊçÆ">Ëé∑ÂèñÊï∞ÊçÆ</h4>
<p>Ê≠§Á§∫‰æãÂ∞Ü‰ΩøÁî®‰ª•‰∏ã‰∏§‰∏™Âü∫Âõ†ÁªÑÔºöÂü∫Âõ†ÁªÑAÔºöGCF_003947435.1 [GTDB / NCBI]ÔºåÂü∫Âõ†ÁªÑBÔºöGCA_002011125.1 [GTDB / NCBI]</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># Ëé∑ÂèñÂü∫Âõ†ÁªÑ„ÄÇ</span>
</span></span><span class="line"><span class="cl">mkdir -p gtdbtk_test_steps/genomes
</span></span><span class="line"><span class="cl">wget -q https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/947/435/GCF_003947435.1_ASM394743v1/GCF_003947435.1_ASM394743v1_genomic.fna.gz -O gtdbtk_test_steps/genomes/genome_a.fna.gz
</span></span><span class="line"><span class="cl">wget -q https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002/011/125/GCA_002011125.1_ASM201112v1/GCA_002011125.1_ASM201112v1_genomic.fna.gz -O gtdbtk_test_steps/genomes/genome_b.fna.gz
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="Âü∫Âõ†Èâ¥ÂÆö-identify">Âü∫Âõ†Èâ¥ÂÆö (identify)</h4>
<p>ÂèØ‰ª•ÈÄöËøá<code>classify_wf</code>ÂçïÊ≠•ÂÆåÊàêÔºå‰ΩÜÊòØÊúâÊó∂ÂÄôÂàÜÊ≠•ÊµÅÁ®ãÊõ¥ÈÄÇÂêàÂ§ÑÁêÜÂ§ßÊï∞ÊçÆÔºàÂèØ‰ª•Ëá™Ë°åÂàÜÈÖçÊØè‰∏ÄÊ≠•ÁöÑËµÑÊ∫êÔºâÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls -l gtdbtk_test_steps/genomes
</span></span><span class="line"><span class="cl"><span class="c1"># ‚ÄìextensionÔºågzÊòØÂü∫Âõ†ÁªÑÊñá‰ª∂ÁöÑÂêéÁºÄÂêç</span>
</span></span><span class="line"><span class="cl">gtdbtk identify --genome_dir gtdbtk_test_steps/genomes --out_dir gtdbtk_test_steps/identify --extension gz --cpus <span class="m">2</span>
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:25:20] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:25:20] INFO: gtdbtk identify --genome_dir gtdbtk_test_steps/genomes --out_dir gtdbtk_test_steps/identify --extension gz --cpus 2
[2024-05-20 15:25:20] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:25:20] INFO: Identifying markers in 2 genomes with 2 threads.
[2024-05-20 15:25:20] TASK: Running Prodigal V2.6.3 to identify genes.
[2024-05-20 15:25:30] INFO: Completed 2 genomes in 9.31 seconds (4.65 seconds/genome).
[2024-05-20 15:25:30] TASK: Identifying TIGRFAM protein families.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 7.16 seconds (3.58 seconds/genome).
[2024-05-20 15:25:38] TASK: Identifying Pfam protein families.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 0.59 seconds (3.37 genomes/second).
[2024-05-20 15:25:38] INFO: Annotations done using HMMER 3.3.2 (Nov 2020).
[2024-05-20 15:25:38] TASK: Summarising identified marker genes.
[2024-05-20 15:25:38] INFO: Completed 2 genomes in 0.03 seconds (75.34 genomes/second).
[2024-05-20 15:25:38] INFO: Done.
</code></pre>
<p>ËÄóÊó∂40sÔºåÂÜÖÂ≠òÊ∂àËÄó131.76MB„ÄÇ</p>
<p>‰∫ßÁîü<code>idetity/</code>Êñá‰ª∂Â§πÔºåÂÖ≥Ê≥®ÊëòË¶ÅÊñá‰ª∂gtdbtk_test_steps/identify/identify/gtdbtk.ar53.markers_summary.tsvÔºåÂÖ∂‰∏≠ËØ¶ÁªÜ‰ªãÁªç‰∫Ü‰ªéÂè§ÁªÜËèå 53 ÊàñÁªÜËèå 120 Ê†áËÆ∞ÁªÑ‰∏≠ËØÜÂà´Âá∫ÁöÑÊ†áËÆ∞„ÄÇ</p>
<p>ËøòÂèØ‰ª•Âú®ÊØè‰∏™Âü∫Âõ†ÁªÑÂØπÂ∫îÁöÑ‰∏≠Èó¥Êñá‰ª∂ÁõÆÂΩï‰∏ãÊâæÂà∞Ë¢´Ë∞ÉÁî®ÁöÑÂü∫Âõ†ÂíåÊ†áËÆ∞‰ø°ÊÅØ:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls gtdbtk_test_steps/identify/identify/intermediate_results/marker_genes/genome_a.fna/
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="Âü∫Âõ†ÊØîÂØπ-align">Âü∫Âõ†ÊØîÂØπ (align)</h4>
<p>ÊØîÂØπÊ≠•È™§Â∞ÜÂØπÈΩêÊâÄÊúâÂ∑≤ËØÜÂà´ÁöÑÊ†áËÆ∞ÔºåÁ°ÆÂÆöÊúÄÂèØËÉΩÁöÑÂüüÂπ∂ËæìÂá∫‰∏≤ËÅîÁöÑMSA:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk align --identify_dir gtdbtk_test_steps/identify --out_dir gtdbtk_test_steps/align --cpus <span class="m">2</span>
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:32:01] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:32:01] INFO: gtdbtk align --identify_dir gtdbtk_test_steps/identify --out_dir gtdbtk_test_steps/align --cpus 2
[2024-05-20 15:32:01] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:32:03] INFO: Aligning markers in 2 genomes with 2 CPUs.
[2024-05-20 15:32:04] INFO: Processing 2 genomes identified as archaeal.
[2024-05-20 15:32:06] INFO: Read concatenated alignment for 4,416 GTDB genomes.
[2024-05-20 15:32:06] TASK: Generating concatenated alignment for each marker.
[2024-05-20 15:32:06] INFO: Completed 2 genomes in 0.02 seconds (130.19 genomes/second).
[2024-05-20 15:32:06] TASK: Aligning 52 identified markers using hmmalign 3.3.2 (Nov 2020).
[2024-05-20 15:32:06] INFO: Completed 52 markers in 0.43 seconds (121.60 markers/second).
[2024-05-20 15:32:06] TASK: Masking columns of archaeal multiple sequence alignment using canonical mask.
[2024-05-20 15:32:10] INFO: Completed 4,418 sequences in 3.59 seconds (1,231.58 sequences/second).
[2024-05-20 15:32:10] INFO: Masked archaeal alignment from 13,540 to 10,135 AAs.
[2024-05-20 15:32:10] INFO: 0 archaeal user genomes have amino acids in &lt;10.0% of columns in filtered MSA.
[2024-05-20 15:32:10] INFO: Creating concatenated alignment for 4,418 archaeal GTDB and user genomes.
[2024-05-20 15:32:12] INFO: Creating concatenated alignment for 2 archaeal user genomes.
[2024-05-20 15:32:12] INFO: Done.
</code></pre>
<p>ËÄóÊó∂10sÔºåÂÜÖÂ≠òÂøΩÁï•‰∏çËÆ°ÔºàÂõ†‰∏∫ËøôÈáåÂè™Êúâ‰∏§‰∏™genomeÔºåËæìÂÖ•ÁöÑÂ§ö‰∫ÜËøòÊòØË¶ÅËÄÉËôëËøôÊ≠•ÁöÑÂÜÖÂ≠òÁöÑÔºâ</p>
<p>‰∫ßÁîü<code>align/</code>Êñá‰ª∂Â§πÔºåÂ¶ÇÊûúÂü∫Âõ†ÁªÑËØÜÂà´Âá∫ÁöÑÊ†áËÆ∞Êï∞ÈáèËæÉÂ∞ëÔºåÂÆÉÂ∞ÜË¢´ÊéíÈô§Âú®Ëøô‰∏ÄÊ≠•ÁöÑÂàÜÊûê‰πãÂ§ñ„ÄÇÂ¶ÇÊûúÊòØËøôÁßçÊÉÖÂÜµÔºåÂ∞Ü‰ºöÂá∫Áé∞Warning„ÄÇ</p>
<p>Ê†πÊçÆÂü∫Âõ†ÁªÑDomainÁöÑ‰∏çÂêåÔºåÂ∞ÜÂá∫Áé∞ ar53 Êàñ bac120 ÂâçÁºÄÊñá‰ª∂„ÄÇ</p>
<ul>
<li>gtdbtk.ar53.msa.fasta.gzÔºöÂÖ∂‰∏≠ÂåÖÂê´Áî®Êà∑Âü∫Âõ†ÁªÑÂíå GTDBÂü∫Âõ†ÁªÑÁöÑ MSA„ÄÇ</li>
<li>gtdbtk.ar53.user_msa.fasta.gzÔºö‰ªÖÂåÖÂê´Áî®Êà∑Âü∫Âõ†ÁªÑ</li>
</ul>
<h4 id="Áâ©ÁßçÂàÜÁ±ª-classify">Áâ©ÁßçÂàÜÁ±ª (classify)</h4>
<p>ÂàÜÁ±ªÊ≠•È™§‰ºöÂ∞ÜÂü∫Âõ†ÁªÑÊîæÂÖ•ÂèÇËÄÉÊ†ë‰∏≠ÔºåÁÑ∂ÂêéÁ°ÆÂÆöÂÖ∂ÊúÄÂèØËÉΩÁöÑÂàÜÁ±ª„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gtdbtk classify --genome_dir gtdbtk_test_steps/genomes --align_dir gtdbtk_test_steps/align --out_dir gtdbtk_test_steps/classify -x gz --cpus <span class="m">2</span> --skip_ani_screen
</span></span></code></pre></td></tr></table>
</div>
</div><pre><code>[2024-05-20 15:42:48] INFO: GTDB-Tk v2.3.2
[2024-05-20 15:42:48] INFO: gtdbtk classify --genome_dir gtdbtk_test_steps/genomes --align_dir gtdbtk_test_steps/align --out_dir gtdbtk_test_steps/classify -x gz --cpus 2 --skip_ani_screen
[2024-05-20 15:42:48] INFO: Using GTDB-Tk reference data version r214: /data/home/jianglab/share/GTDB_r214/release214/
[2024-05-20 15:42:49] TASK: Placing 2 archaeal genomes into reference tree with pplacer using 2 CPUs (be patient).
[2024-05-20 15:42:49] INFO: pplacer version: v1.1.alpha19-0-g807f6f3
[2024-05-20 15:48:46] INFO: Calculating RED values based on reference tree.
[2024-05-20 15:48:48] TASK: Traversing tree to determine classification method.
[2024-05-20 15:48:48] INFO: Completed 2 genomes in 0.00 seconds (8,621.39 genomes/second).
[2024-05-20 15:48:48] TASK: Calculating average nucleotide identity using FastANI (v1.33).
[2024-05-20 15:48:52] INFO: Completed 6 comparisons in 2.69 seconds (2.23 comparisons/second).
[2024-05-20 15:48:52] INFO: 2 genome(s) have been classified using FastANI and pplacer.
[2024-05-20 15:48:52] INFO: Note that Tk classification mode is insufficient for publication of new taxonomic designations. New designations should be based on one or more de novo trees, an example of which can be produced by Tk in de novo mode.
[2024-05-20 15:48:52] INFO: Done.
</code></pre>
<p>ËÄóÊó∂368sÔºå<strong>ÂÜÖÂ≠òÊ∂àËÄó45GB</strong>ÔºàÂõ†‰∏∫ËøôÈáåÂè™Êúâ‰∏§‰∏™Âè§ËèågenomeÔºåÂ¶ÇÊûúËøòÊúâÁªÜËèåÁöÑËØùÂ∫îËØ•ÂàÜÈÖç100GBÂ∑¶Âè≥Ôºâ</p>
<p>‰∫ßÁîü<code>align/</code>Êñá‰ª∂Â§πÔºåÊëòË¶ÅÊñá‰ª∂gtdbtk.ar53.summary.tsvÂíåÂåÖÂê´Ëøô‰∫õÂü∫Âõ†ÁªÑÁöÑÂèÇËÄÉÊ†ëgtdbtk.ar53.classify.tree„ÄÇ</p>
<h3 id="Ê≥®ÊÑè‰∫ãÈ°π">Ê≥®ÊÑè‰∫ãÈ°π</h3>
<ul>
<li>‰∏∫‰ªÄ‰πà GTDB-Tk Âíå NCBI Êàñ Silva ÂàÜÁ±ªÂêçÁß∞‰πãÈó¥ÁöÑÂëΩÂêçÁ≥ªÁªüÂ≠òÂú®Â∑ÆÂºÇÔºü</li>
</ul>
<p>GTDB-Tk ‰ΩøÁî® GTDB ÂàÜÁ±ªÊ≥ï (<a class="link" href="https://gtdb.ecogenomic.org/"  target="_blank" rel="noopener"
    >https://gtdb.ecogenomic.org/</a>)„ÄÇÊ≠§ÂàÜÁ±ªÊ≥ï‰∏éNCBI Âíå Silva Á±ª‰ººÔºå‰ΩÜ‰∏çÂÆåÂÖ®Áõ∏Âêå„ÄÇ
Âú®ËÆ∏Â§öÊÉÖÂÜµ‰∏ãÔºåGTDBÂàÜÁ±ªÊ≥ïÊõ¥‰∏•Ê†ºÂú∞ÈÅµÂæ™Á≠âÁ∫ßÂêéÁºÄÁöÑÂëΩÂêçËßÑÂàôÔºåÊØîÂ¶ÇGTDBÊúâ Nitrospirota ËÄåÊ≤°ÊúâNitrospirae„ÄÇ</p>
<ul>
<li>ÂèØ‰ª•Â∞ÜÁªÜËèåÊ†ëÂíåÂè§ÁªÜËèåÊ†ëÂêàÂπ∂Êàê‰∏ÄÊ£µÊ†ëÂêóÔºü</li>
</ul>
<p>ÁªÜËèåÊ†ëÂíåÂè§ÁªÜËèåÊ†ëÊòØ‰ªé‰∏çÂêåÁöÑÊ†áËÆ∞Âü∫Âõ†Êé®Êñ≠Âá∫Êù•ÁöÑ„ÄÇÁõÆÂâçÔºåËøô‰∫õÊ†ëÁöÑÊ≠£Á°ÆÁîüÊ†π‰ªçÁÑ∂ÊòØ‰∏Ä‰∏™ÂºÄÊîæÁöÑÁ†îÁ©∂È¢ÜÂüü„ÄÇ
GTDB-Tk ‰∏çÊèê‰æõÂêàÂπ∂Ê†ëÁöÑÂ∑•ÂÖ∑Ôºå‰ΩÜÂèØ‰ª•ÈÄöËøáÊìç‰Ωú NewickÊñá‰ª∂Êù•‰∫∫‰∏∫Âú∞ÁªÑÂêàÂÆÉ‰ª¨„ÄÇ</p>
<ul>
<li>GTDB-Tk Â¶Ç‰Ωï‰ΩøÁî®Âπ≥ÂùáÊ†∏Ëã∑ÈÖ∏Âêå‰∏ÄÊÄßÔºàANIÔºâÈ™åËØÅÁâ©ÁßçÂàÜÈÖçÔºü</li>
</ul>
<p>GTDB-Tk ‰ΩøÁî® skaniÔºàÂú® v2.3.2 ‰πãÂâç‰∏ÄÁõ¥‰ΩøÁî® fastANIÔºâÊù•‰º∞ËÆ°Âü∫Âõ†ÁªÑ‰πãÈó¥ÁöÑANI„ÄÇ
‰ªÖÂΩìÂü∫Âõ†ÁªÑ‰πãÈó¥ÁöÑ ANI Âú®Áâ©Áßç ANI ÈôêÂà∂ÂçäÂæÑÂÜÖÔºàÈÄöÂ∏∏‰∏∫ 95%Ôºâ‰∏îÊØîÂØπÂàÜÊï∞(AF)&gt;=0.5Êó∂ÔºåÊü•ËØ¢Âü∫Âõ†ÁªÑÊâçË¢´ÂàÜÁ±ª‰∏∫‰∏éÂèÇËÄÉÂü∫Âõ†ÁªÑÂ±û‰∫éÂêå‰∏ÄÁâ©Áßç„ÄÇ
Âú®Êüê‰∫õÊÉÖÂÜµ‰∏ãÔºåÊü•ËØ¢Âü∫Âõ†ÁªÑÁöÑÁ≥ªÁªüÂèëËÇ≤‰ΩçÁΩÆÂèØËÉΩ‰∏çÊîØÊåÅÁâ©ÁßçÂàÜÈÖç„ÄÇ</p>
<ul>
<li>‰∫íÊñ•ÈÄâÈ°π <code>--mash_db</code> Âíå <code>--skip_ani_screen</code> ‰πãÈó¥Êúâ‰ªÄ‰πàÂå∫Âà´Ôºü</li>
</ul>
<p>‰ªé GTDB-Tk v2.2+ ÂºÄÂßãÔºåclassify_wfÂíåÂàÜÁ±ªÂáΩÊï∞ÈúÄË¶ÅÈ¢ùÂ§ñÁöÑÂèÇÊï∞ÊâçËÉΩËøêË°åÔºö<code>--mash_db</code> Êàñ <code>--skip_ani_screen</code>„ÄÇ</p>
<p>‰ΩøÁî®Êñ∞ÁâàÊú¨ÁöÑ GTDB-Tk v2.4.0ÔºåÂàÜÁ±ªÁÆ°ÈÅìÔºàclassify_wf Âíå classifyÔºâÁöÑÁ¨¨‰∏ÄÈò∂ÊÆµÊòØÂ∞ÜÊâÄÊúâÁî®Êà∑Âü∫Âõ†ÁªÑ‰∏éÊâÄÊúâÂèÇËÄÉÂü∫Âõ†ÁªÑËøõË°åÊØîËæÉÔºåÂπ∂Âú®ÂèØËÉΩÁöÑÊÉÖÂÜµ‰∏ãÊ†πÊçÆANI ÂåπÈÖçÂØπÂÆÉ‰ª¨ËøõË°åÊ≥®Èáä„ÄÇ
‰ΩøÁî® <code>--mash_db</code> ÈÄâÈ°πÂ∞ÜÂêë GTDB-Tk ÊåáÁ§∫ ANIÁ≠õÈÄâÊâÄÈúÄÁöÑËçâÂõæ Mash Êï∞ÊçÆÂ∫ìÁöÑË∑ØÂæÑ„ÄÇ
Â¶ÇÊûúÊ≤°ÊúâÂèØÁî®ÁöÑÊï∞ÊçÆÂ∫ìÔºàÂç≥ËøôÊòØÁ¨¨‰∏ÄÊ¨°ËøêË°åÂàÜÁ±ªÔºâÔºåÂàô <code>--mash_db</code>ÈÄâÈ°πÂ∞ÜÁªòÂà∂‰∏Ä‰∏™ÂèØÁî®‰∫éÂêéÁª≠Ë∞ÉÁî®ÁöÑÊñ∞ Mash Êï∞ÊçÆÂ∫ì„ÄÇ <code>--skip_ani_screen</code> ÈÄâÈ°πÂ∞ÜË∑≥ËøáÈ¢ÑÁ≠õÈÄâÊ≠•È™§Âπ∂ÂØπÊâÄÊúâÂü∫Âõ†ÁªÑËøõË°åÂàÜÁ±ªÔºåÁ±ª‰ºº‰∫é‰πãÂâçÁâàÊú¨ÁöÑ GTDB-Tk„ÄÇ</p>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/genome/">Genome</a>
        
            <a href="/Cao-blog/tags/database/">Database</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/gcf-big-fam/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/gcf-big-fam/images/Screenshot%202025-01-08%20at%2011.53.28.0fbb5b0c2b31df564edb14075af07c92_hu_b51a4a4f8db8bb65.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ÁîüÁâ©ÂêàÊàêÂü∫Âõ†Á∞áÂÆ∂ÊóèÔºàGCFÔºâÊï∞ÊçÆÂ∫ì BiG-FAM"
                        data-key="gcf-big-fam" 
                        data-hash="md5-D7tbDCsx31ZO2xQHWvB8kg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ÁîüÁâ©ÂêàÊàêÂü∫Âõ†Á∞áÂÆ∂ÊóèÔºàGCFÔºâÊï∞ÊçÆÂ∫ì BiG-FAM</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/orthofinder/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/orthofinder/images/Workflow.b06f2895b33f76fcddcd78da78c36ab1_hu_54a1bd9442f615be.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ‰ΩøÁî®OrthofinderËøõË°åÁ≥ªÁªüÂèëËÇ≤Áõ¥Á≥ªÂêåÊ∫êÊé®Êñ≠"
                        data-key="orthofinder" 
                        data-hash="md5-sG8olbM/dvzdzXjaeMNqsQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">‰ΩøÁî®OrthofinderËøõË°åÁ≥ªÁªüÂèëËÇ≤Áõ¥Á≥ªÂêåÊ∫êÊé®Êñ≠</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/ani/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/ani/images/ani3.68da08c94ea4c007938042e49c0e07d5_hu_fe3e6b2cf8eeb457.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ‰ΩøÁî®FastANI‰∏éSkaniËÆ°ÁÆóÂπ≥ÂùáÊ†∏Ëã∑ÈÖ∏‰∏ÄËá¥ÊÄß(ANI)"
                        data-key="ani" 
                        data-hash="md5-aNoIyU6kwAeTgELknA4H1Q==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">‰ΩøÁî®FastANI‰∏éSkaniËÆ°ÁÆóÂπ≥ÂùáÊ†∏Ëã∑ÈÖ∏‰∏ÄËá¥ÊÄß(ANI)</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/amplicon/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/amplicon/images/nbt1.18bd9e2218108348ce7845ad52e631ec_hu_7a8b1c11bb1d244d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post Â∏∏Áî®Êâ©Â¢ûÂ≠êÊï∞ÊçÆÂ∫ì"
                        data-key="amplicon" 
                        data-hash="md5-GL2eIhgQg0jOeEWtUuYx7A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">Â∏∏Áî®Êâ©Â¢ûÂ≠êÊï∞ÊçÆÂ∫ì</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/emp/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/emp/images/emp1.87b75213fcbf279ba4a02c0d17fd6823_hu_44af3bd2e29e0581.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post Âú∞ÁêÉÂæÆÁîüÁâ©ÁªÑËÆ°Âàí‰ªãÁªçÔºàEMPÔºâ"
                        data-key="emp" 
                        data-hash="md5-h7dSE/y/J5ukoCwNF/1oIw==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">Âú∞ÁêÉÂæÆÁîüÁâ©ÁªÑËÆ°Âàí‰ªãÁªçÔºàEMPÔºâ</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
