<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='抗生素抗性基因（ARGs）是指存在于细菌或其他微生物基因组中的基因，使它们具有对抗生素的抵抗能力。ARG被视为是一种新出现的生物污染物，是宏基因组研究的重要方面，这里介绍常用的rgi软件进行分析。'>
<title>抗生素抗性基因（ARG）基础学习</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='抗生素抗性基因（ARG）基础学习'>
<meta property='og:description' content='抗生素抗性基因（ARGs）是指存在于细菌或其他微生物基因组中的基因，使它们具有对抗生素的抵抗能力。ARG被视为是一种新出现的生物污染物，是宏基因组研究的重要方面，这里介绍常用的rgi软件进行分析。'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='antibiotics' /><meta property='article:tag' content='ARGs' /><meta property='article:published_time' content='2023-05-16T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2023-05-16T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/images/ARG_mechanism.png' />
<meta name="twitter:title" content="抗生素抗性基因（ARG）基础学习">
<meta name="twitter:description" content="抗生素抗性基因（ARGs）是指存在于细菌或其他微生物基因组中的基因，使它们具有对抗生素的抵抗能力。ARG被视为是一种新出现的生物污染物，是宏基因组研究的重要方面，这里介绍常用的rgi软件进行分析。"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/images/ARG_mechanism.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">🐛🐶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/142776729'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://asa12138.github.io/'
                        target="_blank"
                        title="Homepage"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-world-pin"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M20.972 11.291a9 9 0 1 0 -8.322 9.686" /><path d="M3.6 9h16.8" /><path d="M3.6 15h8.9" /><path d="M11.5 3a17 17 0 0 0 0 18" /><path d="M12.5 3a16.986 16.986 0 0 1 2.578 9.018" /><path d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z" /><path d="M19 18v.01" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com/Asa12138_'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >中文</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a>
      <ol>
        <li><a href="#抗生素">抗生素</a></li>
        <li><a href="#抗生素抗性">抗生素抗性</a></li>
      </ol>
    </li>
    <li><a href="#methods--tools">Methods &amp; Tools</a>
      <ol>
        <li><a href="#tools">Tools</a></li>
        <li><a href="#database">Database</a></li>
        <li><a href="#rgi">RGI</a></li>
      </ol>
    </li>
    <li><a href="#reference">Reference</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/antibiotics-resistance/">
                <img src="/Cao-blog/p/antibiotics-resistance/images/ARG_mechanism_hu_eb674ac26c7d5b94.png"
                        srcset="/Cao-blog/p/antibiotics-resistance/images/ARG_mechanism_hu_eb674ac26c7d5b94.png 800w, /Cao-blog/p/antibiotics-resistance/images/ARG_mechanism_hu_ff3a1905a84a7ec0.png 1600w"
                        width="800" 
                        height="440" 
                        loading="lazy"
                        alt="Featured image of post 抗生素抗性基因（ARG）基础学习" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/microbial-ecology/" style="background-color: #fb8072; color: #ffffff;">
                microbial-ecology
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/antibiotics-resistance/">抗生素抗性基因（ARG）基础学习</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            抗生素抗性基因（ARGs）是指存在于细菌或其他微生物基因组中的基因，使它们具有对抗生素的抵抗能力。ARG被视为是一种新出现的生物污染物，是宏基因组研究的重要方面，这里介绍常用的rgi软件进行分析。
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">May 16, 2023</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    5 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                版权归Cao's blog所有，转载请注明出处
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <script src="https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/index_files/kePrint/kePrint.js"></script>
<link href="https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/index_files/lightable/lightable.css" rel="stylesheet" />
<script src="https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/index_files/kePrint/kePrint.js"></script>
<link href="https://jb-cao.github.io/Cao-blog/p/antibiotics-resistance/index_files/lightable/lightable.css" rel="stylesheet" />
<h2 id="introduction">Introduction</h2>
<h3 id="抗生素">抗生素</h3>
<p>抗生素是一类药物，用于治疗或预防由细菌引起的感染。它们通过抑制细菌的生长或杀死细菌，帮助人体抵抗细菌感染。</p>
<blockquote>
<p>An antibiotic is a type of antimicrobial substance active against bacteria.</p></blockquote>
<figure>
<img src="images/antibiotic_mechanism.png" style="width:90.0%" alt="图1: Mechanisms of Antibiotics (YouTube (Antibiotics - Lecture 3))" />
<figcaption aria-hidden="true">图1: Mechanisms of Antibiotics (YouTube (Antibiotics - Lecture 3))</figcaption>
</figure>
<p>抗生素可以通过不同的机制对细菌产生作用。以下是几种常见的抗生素类型及其作用机制（图1）：</p>
<ol>
<li>
<p>青霉素类抗生素：作用于细菌细胞壁的合成，干扰其结构，导致细菌细胞壁破裂，细菌死亡。</p>
</li>
<li>
<p>大环内酯类抗生素：通过阻断细菌蛋白质的合成过程，阻碍细菌的生长和复制。</p>
</li>
<li>
<p>氨基糖苷类抗生素：抑制细菌蛋白质的合成，通过与细菌的核糖体结合，阻碍蛋白质合成的进行。</p>
</li>
<li>
<p>四环素类抗生素：抑制细菌的蛋白质合成，通过与细菌的核糖体结合，阻碍蛋白质链的延伸。</p>
</li>
<li>
<p>氟喹诺酮类抗生素：抑制细菌DNA的复制和转录过程，阻断细菌的生长。</p>
</li>
</ol>
<h3 id="抗生素抗性">抗生素抗性</h3>
<figure>
<img src="images/ARG_mechanism.png" style="width:90.0%" alt="图2: Mechanisms of Antibiotic Resistance (Courtesy of E. Wistrand-Yuen.)" />
<figcaption aria-hidden="true">图2: Mechanisms of Antibiotic Resistance (Courtesy of E. Wistrand-Yuen.)</figcaption>
</figure>
<p><strong>抗生素抗性</strong>是指细菌或其他微生物对抗生素的治疗效果降低或完全失效的现象。当细菌暴露在抗生素的作用下，一些细菌可能具有抗药性，能够存活和继续繁殖，导致感染的持续存在或进一步扩散。</p>
<blockquote>
<p>Antibiotic resistance is the ability of a microorganism to withstand the effects of antibiotic(s).</p></blockquote>
<p>抗生素抗性可以是天然的，即细菌天生具有抗药性。然而，更为严重的是通过基因突变或水平基因转移等机制，细菌可以获得抗药性。</p>
<p>同样，为了对付抗生素的作用机制有多种耐药机制，<a class="link" href="https://card.mcmaster.ca/"  target="_blank" rel="noopener"
    ><strong>CARD数据库</strong></a> (<a class="link" href="#ref-alcockCARD2020Antibiotic2020" ><em>1</em></a>) 将其分为7类：</p>
<ol>
<li>
<p>导致抗生素耐药性的抗生素靶标的突变改变或酶促修饰。</p>
</li>
<li>
<p>抗生素作用靶点的更换或替代，该过程会导致抗生素耐药性。</p>
</li>
<li>
<p>保护抗生素作用靶点免受抗生素结合，该过程将导致抗生素耐药性。</p>
</li>
<li>
<p>抗生素的酶促灭活赋予耐药性。</p>
</li>
<li>
<p>通过将抗生素转运出细胞而产生的抗生素耐药性。</p>
</li>
<li>
<p>通常通过减少孔蛋白的产生来降低对抗生素的渗透性，可以提供耐药性。</p>
</li>
<li>
<p>基因缺失（通常是孔蛋白）赋予抗生素耐药性的机制。</p>
</li>
</ol>
<p><strong>抗生素抗性基因（Antibiotics resistance genes，ARGs）</strong> 是指存在于细菌或其他微生物基因组中的基因，使它们具有对抗生素的抵抗能力。 这些基因编码了一系列的蛋白质或其他分子机制，使细菌能够对抗生素产生耐药性。</p>
<p>ARG被视为是一种新出现的生物污染物，由ARG引起的抗生素耐药性被认为是当代医学最重要的挑战之一，也是严重的公共卫生问题。抗生素耐药性阻碍了临床上对细菌感染的有效治疗，并成为流行病威胁和高死亡率的原因。</p>
<p>根据获得的欧盟和欧洲经济活动国家的数据，抗生素抗性细菌（ARB）在2015年造成671,689人感染，每年导致超过33,000人死亡 (<a class="link" href="#ref-cassiniAttributableDeathsDisabilityadjusted2019" ><em>2</em></a>)。</p>
<p>ARGs在环境中的广泛传播提高了其作为污染物的危害性。 抗生素在人类、兽医和农业用途中的大量使用导致它们不断释放到环境中，同时，ARG也被引入各种环境中，如临床（医院、诊所）和兽医环境、人体（胃肠道微生物组）以及废水处理厂和自然环境（土壤、水、空气）等。</p>
<p>移动遗传元件（MGE）如质粒、转座子、整合子被证明与ARG的转移传播具有很强的相关性 (<a class="link" href="#ref-wangEnhancedRemovalAntibiotic2021" ><em>3</em></a>)，将 ARG 与移动遗传元件联系起来是十分重要的。</p>
<h2 id="methods--tools">Methods &amp; Tools</h2>
<p>生物信息学在ARG研究上的主要目标是开发分析流程，用于准确检测抗性组（抗性基因的补充延伸）并随后根据基因组和宏基因组数据准确预测抗菌谱（AMR 的表型范围和易感性）以及ARG宿主等信息。</p>
<p>目前已开发出很多的ARG相关的生物信息学软件工具（表1)和数据库（表2），注释管理不一致，每个工具和数据库都有不同的重点领域和不同的预测范围。</p>
<h3 id="tools">Tools</h3>
<figure>
<img src="images/ARG_workflow.png" style="width:90.0%" alt="图3: 宏基因组测序数据ARG分析的工作流程" />
<figcaption aria-hidden="true">图3: 宏基因组测序数据ARG分析的工作流程</figcaption>
</figure>
<p>目前使用宏基因组测序数据对存在于微生物群落中的ARG进行识别鉴定的计算工作流程主要有两种：基于组装的contigs的分析和基于原始reads的比对分析（图3）。</p>
<p>且大多数方法从宏基因组数据中获得的序列(原始reads或contigs)通常不能明确地确定其属于物种起源，分析的范围通常仅限于基于同源性就能识别的ARG家族，而缺乏高度依赖环境的抗性决定因素，如点突变的分析。</p>
<table class=" lightable-classic" style="font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Table 1: 用于鉴定、预测ARGs的生信软件
</caption>
<thead>
<tr>
<th style="text-align:left;">
软件名
</th>
<th style="text-align:left;">
描述
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ARG-ANNOT
</td>
<td style="text-align:left;">
使用 BLAST 针对抗菌素耐药性 (AMR) 参考序列和 SNP 的精选数据库检测细菌基因组中现有的和假定的新抗生素耐药性基因。
</td>
</tr>
<tr>
<td style="text-align:left;">
ARGs-OAP
</td>
<td style="text-align:left;">
用于从宏基因组序列中快速注释和分类抗生素抗性基因样序列（使用 BLASTX 针对 SARG 数据库）的在线管道。
</td>
</tr>
<tr>
<td style="text-align:left;">
BacAnt
</td>
<td style="text-align:left;">
允许同时注释 ARG、整合子和转座子的，可用于比较基因组分析的应用程序
</td>
</tr>
<tr>
<td style="text-align:left;">
DeepARG
</td>
<td style="text-align:left;">
通过深度学习方法，分别为短读序列和全基因长度序列构建了两个深度学习模型 DeepARG-SS 和 DeepARG-LS，提供了准确的抗菌素耐药性注释。
</td>
</tr>
<tr>
<td style="text-align:left;">
Mykrobe predictor
</td>
<td style="text-align:left;">
快速筛选金黄色葡萄球菌和结核分枝杆菌原始 FASTQ 测序结果中的 AMR 基因和 SNP，并对 12 种抗菌药物进行抗生素谱预测。
</td>
</tr>
<tr>
<td style="text-align:left;">
ResFinder
</td>
<td style="text-align:left;">
通过 BLAST 针对 AMR 参考序列的精选数据库，在全基因组数据中识别水平获得的 AMR 基因。
</td>
</tr>
<tr>
<td style="text-align:left;">
RGI
</td>
<td style="text-align:left;">
根据综合抗生素耐药性数据库 (CARD) 中的同源性和 SNP 模型，从蛋白质或核苷酸数据中预测耐药性组。
</td>
</tr>
<tr>
<td style="text-align:left;">
SEAR
</td>
<td style="text-align:left;">
使用 ARG-ANNOT 数据库检测基因组或宏基因组测序数据中水平获得的 AMR 基因。
</td>
</tr>
<tr>
<td style="text-align:left;">
SRST
</td>
<td style="text-align:left;">
一种快速测序读取映射工具，用于快速准确地检测 MLST、毒力、AMR 或其他标记。包括 ResFinder 和 ARG-ANNOT 参考数据库。
</td>
</tr>
<tr>
<td style="text-align:left;">
AMRFinder
</td>
<td style="text-align:left;">
NCBI开发的一种使用高质量精选 AMR 基因参考数据库识别 AMR 基因的工具。
</td>
</tr>
</tbody>
</table>
<h3 id="database">Database</h3>
<p>近年来随着ARG数据的快速增长，数据管理、分析和访问需要更完善的数据库来承载。</p>
<p>ARG命名本身就是一个难题，同义词经常使用，名称冲突，基因名称有时基于核苷酸序列有时基于蛋白质序列，一些基因家族使用通用名称，而另一些则命名每个等位基因，加上各种ARG数据库非正式地交换有关 AMR 管理、命名和分类的信息，导致管理较为混乱。</p>
<p>当然，没有一个ARG数据库是完整的，ARG在各种病原体间移动与突变会增加大量数据，突变数据的管理是最具挑战性的任务。</p>
<table class=" lightable-classic" style="font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Table 2: ARGs综合信息数据库
</caption>
<thead>
<tr>
<th style="text-align:left;">
数据库
</th>
<th style="text-align:left;">
描述
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ARDB
</td>
<td style="text-align:left;">
一个手动管理的数据库，其中每个基因和抗性类型都用抗性概况、作用机制、本体论和序列和蛋白质数据库的外部链接进行注释。自 2009 年以来未更新，所有数据均已整理到 CARD 中。
</td>
</tr>
<tr>
<td style="text-align:left;">
ARG-ANNOT
</td>
<td style="text-align:left;">
AMR 参考序列和 SNP 的精选数据库。
</td>
</tr>
<tr>
<td style="text-align:left;">
CARD
</td>
<td style="text-align:left;">
手动管理的抗性基因和突变、其产物和相关表型数据库，涵盖 AMR 的所有机制。由抗生素耐药性本体论 (ARO) 组织管理。
</td>
</tr>
<tr>
<td style="text-align:left;">
CBMAR
</td>
<td style="text-align:left;">
提供对 β-内酰胺酶家族的分子和生化表征有用的信息。
</td>
</tr>
<tr>
<td style="text-align:left;">
MvirDB
</td>
<td style="text-align:left;">
通过整合来自多个来源的数据，专注于对生物防御应用至关重要的毒素、毒力因子和抗生素抗性基因。
</td>
</tr>
<tr>
<td style="text-align:left;">
NCBI BioProject PRJNA313047
</td>
<td style="text-align:left;">
以抗性为重点的 AMR 基因序列整理。
</td>
</tr>
<tr>
<td style="text-align:left;">
PATRIC
</td>
<td style="text-align:left;">
细菌感染性疾病信息系统，以ARDB和CARD为基础，辅以AMR元数据，对完整的病原体基因组进行注释。
</td>
</tr>
<tr>
<td style="text-align:left;">
Resfams
</td>
<td style="text-align:left;">
蛋白质家族和相关配置文件隐藏马尔可夫模型 (HMM) 的精选数据库，确认具有抗生素抗性功能并按本体组织。
</td>
</tr>
<tr>
<td style="text-align:left;">
ResFinder
</td>
<td style="text-align:left;">
水平获得的 AMR 基因数据库。
</td>
</tr>
<tr>
<td style="text-align:left;">
SARG
</td>
<td style="text-align:left;">
抗生素抗性基因、亚型和参考序列，整合来自ARDB和CARD的信息。
</td>
</tr>
</tbody>
</table>
<p>很多ARG数据库已经很久没有管理升级了，目前CARD是注释比较全面，管理完善，很多ARG条目经过实验验证的数据库。</p>
<p>CARD提供参考 DNA 和蛋白质序列、检测模型和基于细菌抗菌素耐药性(AMR)分子基础的生物信息学工具，设计了抗生素抗性本体论 (ARO)。</p>
<p>ARO 分为三个主要分支：抗生素耐药性决定因素(ARO:3000000)、抗生素分子(ARO:1000003)和抗生素耐药机制(ARO:1000002)，最新的CARD现在已经有6000多个本体术语，这些还得到了很多研究论文的支持。</p>
<p>CARD 开发了抗性基因标识符（RGI) 软件（v5.0）。</p>
<p>RGI软件利用四种CARD模型类型来预测抵抗组：蛋白质同源模型（使用BLASTP或DIAMOND检测AMR基因的功能同源物）、蛋白质变异模型（用于准确区分易感内在基因和获得赋予AMR的突变的内在基因，基CARD的精选SNP矩阵）、rRNA突变模型（用于检测抗药性rRNA靶序列）和蛋白质过度表达模型（检测与AMR相关的外排亚基，但也突出显示存在时赋予过度表达的突变）。</p>
<p>所以我一般用的也是RGI+CARD来做宏基因组的ARG鉴定。当然鉴定后的下游分析就跟其他的功能基因类似，有很多可以做的，最好跟实验设计紧紧联系说明问题。</p>
<h3 id="rgi">RGI</h3>
<p>RGI（Resistance Gene Identifier）是一个用于检测和注释细菌基因组中抗生素抗性基因的工具。</p>
<p>地址：<a class="link" href="https://github.com/arpcard/rgi"  target="_blank" rel="noopener"
    >https://github.com/arpcard/rgi</a></p>
<p>RGI有网页版应用<a class="link" href="https://card.mcmaster.ca/analyze/rgi"  target="_blank" rel="noopener"
    >https://card.mcmaster.ca/analyze/rgi</a>，可以选择两种输入，右侧调节参数：</p>
<ol>
<li>Enter a GenBank accession(s):</li>
<li>Upload FASTA sequence file(s):</li>
</ol>
<p>GenBank: JN420336.1，试试这个 Klebsiella pneumoniae plasmid pNDM-MAR, complete sequence，肺炎克雷伯菌质粒 pNDM-MAR，完整序列， 267242 bp。</p>
<p>结果会返回一个表，多个旭日图： <img src="images/rig_web.png" style="width:90.0%" /></p>
<p>当然我们要做宏基因组分析的话肯定要用命令行版本的rgi软件：</p>
<p>安装方法：</p>
<p>最简单的方式就是conda新建环境后直接安装：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># searches rgi package and show available versions</span>
</span></span><span class="line"><span class="cl">$ conda search --channel bioconda rgi
</span></span><span class="line"><span class="cl"><span class="c1"># install rgi package</span>
</span></span><span class="line"><span class="cl">$ conda install --channel bioconda rgi
</span></span><span class="line"><span class="cl"><span class="c1"># install rgi specific version</span>
</span></span><span class="line"><span class="cl">$ conda install --channel bioconda <span class="nv">rgi</span><span class="o">=</span>3.1.1
</span></span><span class="line"><span class="cl"><span class="c1"># remove rgi package</span>
</span></span><span class="line"><span class="cl">$ conda remove --channel bioconda rgi
</span></span></code></pre></td></tr></table>
</div>
</div><p>如果上述方法有问题的话（比如我当时的环境），可以考虑源码编译：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 克隆仓库，</span>
</span></span><span class="line"><span class="cl">git clone https://github.com/arpcard/rgi
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 在rgi文件夹内新建环境</span>
</span></span><span class="line"><span class="cl">conda env create -f conda_env.yml
</span></span><span class="line"><span class="cl">conda activate rgi
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 在rgi文件夹内build</span>
</span></span><span class="line"><span class="cl">python setup.py build
</span></span><span class="line"><span class="cl">python setup.py <span class="nb">test</span>
</span></span><span class="line"><span class="cl">python setup.py install
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#test</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> tests
</span></span><span class="line"><span class="cl">pytest -v -rxs
</span></span><span class="line"><span class="cl"><span class="c1">#这里可能会有几个问题，需要把card.json，和由card_database_v3.1.4.fasta创建的card_reference.fasta放在合适位置</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#查看安装情况</span>
</span></span><span class="line"><span class="cl">rgi main -h
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#成功的话会输出帮助文件</span>
</span></span><span class="line"><span class="cl">usage: rgi main <span class="o">[</span>-h<span class="o">]</span> -i INPUT_SEQUENCE -o OUTPUT_FILE <span class="o">[</span>-t <span class="o">{</span>contig,protein<span class="o">}]</span>
</span></span><span class="line"><span class="cl">                <span class="o">[</span>-a <span class="o">{</span>DIAMOND,BLAST<span class="o">}]</span> <span class="o">[</span>-n THREADS<span class="o">]</span> <span class="o">[</span>--include_loose<span class="o">]</span>
</span></span><span class="line"><span class="cl">                <span class="o">[</span>--include_nudge<span class="o">]</span> <span class="o">[</span>--local<span class="o">]</span> <span class="o">[</span>--clean<span class="o">]</span> <span class="o">[</span>--keep<span class="o">]</span> <span class="o">[</span>--debug<span class="o">]</span>
</span></span><span class="line"><span class="cl">                <span class="o">[</span>--low_quality<span class="o">]</span> <span class="o">[</span>-d <span class="o">{</span>wgs,plasmid,chromosome,NA<span class="o">}]</span> <span class="o">[</span>-v<span class="o">]</span>
</span></span><span class="line"><span class="cl">                <span class="o">[</span>-g <span class="o">{</span>PRODIGAL,PYRODIGAL<span class="o">}]</span> <span class="o">[</span>--split_prodigal_jobs<span class="o">]</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">Resistance Gene Identifier - 6.0.2 - Main
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#下载card数据库</span>
</span></span><span class="line"><span class="cl">wget https://card.mcmaster.ca/latest/data
</span></span><span class="line"><span class="cl">tar -xvf data ./card.json
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#在本地或工作目录中加载card数据库</span>
</span></span><span class="line"><span class="cl">rgi load --card_json /path/to/card.json --local
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#查看本地数据库版本</span>
</span></span><span class="line"><span class="cl">rgi database --version --local
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#注意不加--local的话数据库会加载到全局环境中，在任意目录都可以运行</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>安装成功后，就可以来对我们的序列进行鉴定了，常用的有两种模式：</p>
<ol>
<li>基因组或组装序列（DNA/蛋白质）</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">rgi main --input_sequence /path/to/nucleotide_input.fasta
</span></span><span class="line"><span class="cl">  --output_file /path/to/output_file --local --clean
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">部分可选参数：
</span></span><span class="line"><span class="cl">    -h, --help 显示此帮助信息并退出
</span></span><span class="line"><span class="cl">    -i 输入序列, --input_sequence 输入序列
</span></span><span class="line"><span class="cl">        输入文件必须在 FASTA（重叠群和蛋白质）或 gzip 格式！例如 myFile.fasta
</span></span><span class="line"><span class="cl">    -o 输出文件, --output_file 输出文件，输出文件夹和文件名
</span></span><span class="line"><span class="cl">    -t <span class="o">{</span>contig,protein<span class="o">}</span>, --input_type <span class="o">{</span>contig,protein<span class="o">}</span>
</span></span><span class="line"><span class="cl">        指定数据输入类型（默认 <span class="o">=</span> contig）
</span></span><span class="line"><span class="cl">    -a <span class="o">{</span>DIAMOND,BLAST<span class="o">}</span>, --alignment_tool <span class="o">{</span>DIAMOND,BLAST<span class="o">}</span>
</span></span><span class="line"><span class="cl">        指定比对工具（默认 <span class="o">=</span> BLAST）
</span></span><span class="line"><span class="cl">    -n 线程，--num_threads 线程
</span></span><span class="line"><span class="cl">        BLAST 搜索中使用的线程数 <span class="o">(</span>CPU<span class="o">)</span>（默认值<span class="o">=</span>16）
</span></span><span class="line"><span class="cl">    --include_loose 除了严格和完美之外还包括宽松的命中点击（默认值：False）
</span></span><span class="line"><span class="cl">    --include_nudge 包括从宽松到严格命中的命中（默认值：False）
</span></span><span class="line"><span class="cl">    --local 使用本地数据库（默认：使用数据库可执行目录）
</span></span><span class="line"><span class="cl">    --clean 删除临时文件（默认值：False）
</span></span></code></pre></td></tr></table>
</div>
</div><p>结果表格每列的内容如下：</p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th></th>
          <th></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td><strong>Field</strong></td>
          <td><strong>Content</strong></td>
      </tr>
      <tr>
          <td>ORF_ID</td>
          <td>开放阅读框架标识符（RGI 内部）</td>
      </tr>
      <tr>
          <td>Contig</td>
          <td>源序列</td>
      </tr>
      <tr>
          <td>Start</td>
          <td>ORF起始坐标</td>
      </tr>
      <tr>
          <td>Stop</td>
          <td>ORF的结束坐标</td>
      </tr>
      <tr>
          <td>Orientation</td>
          <td>ORF链</td>
      </tr>
      <tr>
          <td>Cut_Off</td>
          <td>RGI 检测范式（完美、严格、松散）</td>
      </tr>
      <tr>
          <td>Pass_Bitscore</td>
          <td>严格检测模型 bitscore 截断</td>
      </tr>
      <tr>
          <td>Best_Hit_Bitscore</td>
          <td>与 CARD 中的最高命中匹配的 Bitscore 值</td>
      </tr>
      <tr>
          <td>Best_Hit_ARO</td>
          <td>CARD 中的最高命中匹配的 ARO 术语</td>
      </tr>
      <tr>
          <td>Best_Identities</td>
          <td>与 CARD 中的最高命中匹配的同一性百分比</td>
      </tr>
      <tr>
          <td>ARO</td>
          <td>ARO 匹配到 CARD 中的最高命中的Accession号</td>
      </tr>
      <tr>
          <td>Model_type</td>
          <td>CARD检测模型类型</td>
      </tr>
      <tr>
          <td>SNPs_in_Best_Hit_ARO</td>
          <td>在 CARD 中最高命中的 ARO 术语中观察到的突变</td>
      </tr>
      <tr>
          <td>Other_SNPs</td>
          <td>以模型 ID 指示的其他匹配项的 ARO 术语观察到的突变</td>
      </tr>
      <tr>
          <td>Drug Class</td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td>Resistance Mechanism</td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td>AMR Gene Family</td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td>Predicted_DNA</td>
          <td>ORF预测核苷酸序列</td>
      </tr>
      <tr>
          <td>Predicted_Protein</td>
          <td>ORF预测的蛋白质序列</td>
      </tr>
      <tr>
          <td>CARD_Protein_Sequence</td>
          <td>CARD中top hit的蛋白质序列</td>
      </tr>
      <tr>
          <td>Percentage Length of Reference Sequence</td>
          <td>ORF蛋白长度/CARD参考蛋白长度</td>
      </tr>
      <tr>
          <td>ID</td>
          <td>HSP 标识符（RGI 内部）</td>
      </tr>
      <tr>
          <td>Model_id</td>
          <td>CARD检测型号id</td>
      </tr>
      <tr>
          <td>Nudged</td>
          <td>TRUE = 命中从松散微调到严格</td>
      </tr>
      <tr>
          <td>Note</td>
          <td>其他注意事项的原因</td>
      </tr>
  </tbody>
</table></div>
<ol start="2">
<li>宏基因组reads，基因组reads</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">rgi bwt --read_one /path/to/fastq/R1.fastq.gz
</span></span><span class="line"><span class="cl">  --read_two /path/to/fastq/R2.fastq.gz --output_file output_prefix
</span></span><span class="line"><span class="cl">  --local
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">部分可选参数：
</span></span><span class="line"><span class="cl">    -h, --help 显示此帮助信息并退出
</span></span><span class="line"><span class="cl">    -1 READ_ONE, --read_one READ_ONE
</span></span><span class="line"><span class="cl">    -2 READ_TWO，--read_two READ_TWO
</span></span><span class="line"><span class="cl">    -a <span class="o">{</span>kma,bowtie2,bwa<span class="o">}</span>, --aligner <span class="o">{</span>kma,bowtie2,bwa<span class="o">}</span>
</span></span><span class="line"><span class="cl">        选择读取对齐器（默认<span class="o">=</span>kma）
</span></span><span class="line"><span class="cl">    -n 线程，--threads 线程
</span></span><span class="line"><span class="cl">        要使用的线程 <span class="o">(</span>CPU<span class="o">)</span> 数（默认值<span class="o">=</span>16）
</span></span><span class="line"><span class="cl">    -o 输出文件, --output_file 输出文件
</span></span><span class="line"><span class="cl">        输出文件名的名称
</span></span><span class="line"><span class="cl">    --debug 调试模式（默认<span class="o">=</span>False）
</span></span><span class="line"><span class="cl">    --clean 删除临时文件（默认<span class="o">=</span>False）
</span></span><span class="line"><span class="cl">    --local 使用本地数据库（默认：使用可执行目录中的数据库）
</span></span><span class="line"><span class="cl">    --include_wildcard 包含wild数据库（默认<span class="o">=</span>False）
</span></span></code></pre></td></tr></table>
</div>
</div><p>最终结果有5个文件：</p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th></th>
          <th></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td><strong>File</strong></td>
          <td><strong>Contents</strong></td>
      </tr>
      <tr>
          <td><strong>output_prefix.allele_mapping_data.txt</strong></td>
          <td>RGI bwt read mapping results at allele level</td>
      </tr>
      <tr>
          <td><strong>output_prefix.gene_mapping_data.txt</strong></td>
          <td>RGI bwt read mapping results at gene level</td>
      </tr>
      <tr>
          <td>output_prefix.artifacts_mapping_stats.txt</td>
          <td>Statistics for read mapping artifacts</td>
      </tr>
      <tr>
          <td>output_prefix.overall_mapping_stats.txt</td>
          <td>Statistics for overall read mapping results</td>
      </tr>
      <tr>
          <td>output_prefix.reference_mapping_stats.txt</td>
          <td>Statistics for reference matches</td>
      </tr>
  </tbody>
</table></div>
<p>常用的进一步分析的是 at allele level，这个表格每列的内容如下：</p>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th></th>
          <th></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td><strong>Field</strong></td>
          <td><strong>Contents</strong></td>
      </tr>
      <tr>
          <td>Reference Sequence</td>
          <td>read映射到的参考等位基因</td>
      </tr>
      <tr>
          <td><strong>ARO Term</strong></td>
          <td>ARO Term</td>
      </tr>
      <tr>
          <td><strong>ARO Accession</strong></td>
          <td>ARO Accession</td>
      </tr>
      <tr>
          <td>Reference Model Type</td>
          <td>CARD 检测模型类型</td>
      </tr>
      <tr>
          <td>Reference DB</td>
          <td>参考等位基因来自 CARD 或 WildCARD</td>
      </tr>
      <tr>
          <td>Reference Allele Source</td>
          <td>See below</td>
      </tr>
      <tr>
          <td>Resistomes &amp; Variants: Observed in Genome(s)</td>
          <td>这个等位基因序列是否在 CARD 患病率基因组序列中被观察到？</td>
      </tr>
      <tr>
          <td>Resistomes &amp; Variants: Observed in Plasmid(s)</td>
          <td>是否已在 CARD Prevalence 质粒序列中观察到该等位基因序列？</td>
      </tr>
      <tr>
          <td>Resistomes &amp; Variants: Observed Pathogen(s)</td>
          <td>CARD 携带此等位基因序列的流行病原体。 如果 Reference DB 是 CARD，将显示在 CARD 检测模型中用作参考的病原体。 使用 k-mers 验证病原体来源。</td>
      </tr>
      <tr>
          <td><strong>Completely Mapped Reads</strong></td>
          <td>完全映射到等位基因的read数量</td>
      </tr>
      <tr>
          <td><strong>Mapped Reads with Flanking Sequence</strong></td>
          <td>未完全映射到等位基因的read数量</td>
      </tr>
      <tr>
          <td><strong>All Mapped Reads</strong></td>
          <td>前两列的总和</td>
      </tr>
      <tr>
          <td>Percent Coverage</td>
          <td>读数覆盖的参考等位基因百分比</td>
      </tr>
      <tr>
          <td>Length Coverage (bp)</td>
          <td>Base pairs of reference allele covered by reads</td>
      </tr>
      <tr>
          <td>Average MAPQ (Completely Mapped Reads)</td>
          <td>平均 MAPQ 值，映射质量得分量化了读取错位的可能性。 Heng Li 和 Richard Durbin 在他们描述 MAQ 的论文中介绍了它们，并且通常以 Phred 量表进行报告。</td>
      </tr>
      <tr>
          <td>Mate Pair Linkage</td>
          <td>对于配对双端测序，如果姐妹读取映射到不同的 AMR 基因，则会列出</td>
      </tr>
      <tr>
          <td>Reference Length</td>
          <td>参考等位基因的长度 (bp)</td>
      </tr>
      <tr>
          <td><strong>AMR Gene Family</strong></td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td><strong>Drug Class</strong></td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td><strong>Resistance Mechanism</strong></td>
          <td>ARO分类</td>
      </tr>
      <tr>
          <td>Depth</td>
          <td>覆盖深度（仅在使用 kma 时报告）</td>
      </tr>
      <tr>
          <td>SNPs</td>
          <td>从映射读取中观察到的单核苷酸多态性（仅在使用 kma 时报告）</td>
      </tr>
      <tr>
          <td>Consensus Sequence DNA</td>
          <td>使用映射读取的核苷酸一致序列（仅在使用 kma 时报告）</td>
      </tr>
      <tr>
          <td>Consensus Sequence Protein</td>
          <td>从 DNA 翻译的蛋白质共有序列（仅在使用 kma 时报告）</td>
      </tr>
  </tbody>
</table></div>
<h2 id="reference">Reference</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-alcockCARD2020Antibiotic2020" class="csl-entry">
<p><span class="csl-left-margin">1. </span><span class="csl-right-inline">B. P. Alcock, A. R. Raphenya, T. T. Y. Lau, K. K. Tsang, <em>et al.</em>, <a class="link" href="https://doi.org/10.1093/nar/gkz935"  target="_blank" rel="noopener"
    >CARD 2020: Antibiotic resistome surveillance with the comprehensive antibiotic resistance database</a>. <em>Nucleic Acids Research</em>. <strong>48</strong>, D517–D525 (2020).</span></p>
</div>
<div id="ref-cassiniAttributableDeathsDisabilityadjusted2019" class="csl-entry">
<p><span class="csl-left-margin">2. </span><span class="csl-right-inline">A. Cassini, L. D. Högberg, D. Plachouras, A. Quattrocchi, <em>et al.</em>, <a class="link" href="https://doi.org/10.1016/S1473-3099%2818%2930605-4"  target="_blank" rel="noopener"
    >Attributable deaths and disability-adjusted life-years caused by infections with antibiotic-resistant bacteria in the EU and the european economic area in 2015: A population-level modelling analysis</a>. <em>The Lancet. Infectious Diseases</em>. <strong>19</strong>, 56–66 (2019).</span></p>
</div>
<div id="ref-wangEnhancedRemovalAntibiotic2021" class="csl-entry">
<p><span class="csl-left-margin">3. </span><span class="csl-right-inline">J. Wang, J. Gu, X. Wang, Z. Song, <em>et al.</em>, <a class="link" href="https://doi.org/10.1016/j.jhazmat.2021.125135"  target="_blank" rel="noopener"
    >Enhanced removal of antibiotic resistance genes and mobile genetic elements during swine manure composting inoculated with mature compost</a>. <em>Journal of Hazardous materials</em>. <strong>411</strong>, 125135 (2021).</span></p>
</div>
</div>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/antibiotics/">Antibiotics</a>
        
            <a href="/Cao-blog/tags/args/">ARGs</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/global-arg/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/global-arg/images/global1.76903c71775de5690292660a4936b869_hu_a39006653f9d1429.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 全球尺度的抗生素耐药基因（ARG）研究"
                        data-key="global-arg" 
                        data-hash="md5-dpA8cXdd5WkCkmYKSTa4aQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">全球尺度的抗生素耐药基因（ARG）研究</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/lignocellulose/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/lignocellulose/images/fig2.59af298d5f9d6988af95b9c999b78bc6_hu_4319374f044a70ec.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 功能基因探索：木质纤维素（lignocellulose）降解"
                        data-key="lignocellulose" 
                        data-hash="md5-Wa8pjV&#43;daYivlbnJmbeLxg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">功能基因探索：木质纤维素（lignocellulose）降解</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/acid/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/acid/images/fig1.de95313758fafea536f71fc59ed49352_hu_ce6b739f5be49f54.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 微生物的耐酸机制研究"
                        data-key="acid" 
                        data-hash="md5-3pUxN1j6/qU29x/FntSTUg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">微生物的耐酸机制研究</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/giant-virus-res/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/giant-virus-res/images/gvmags2.a3ac8242b44f18b05f2bba7c449dda8c_hu_a831f89e1165abe9.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 一些基于宏基因组的巨型病毒研究"
                        data-key="giant-virus-res" 
                        data-hash="md5-o6yCQrRPGLBfK7p8RJ3ajA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">一些基于宏基因组的巨型病毒研究</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/giant-virus/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/giant-virus/images/gv2.73515dbbbf9bea756beb1dc235f3c58f_hu_628db13019d7d2b1.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 巨型病毒(Giant virus)生物学和多样性研究"
                        data-key="giant-virus" 
                        data-hash="md5-c1Fdu7&#43;b6nVr6x3CNfPFjw==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">巨型病毒(Giant virus)生物学和多样性研究</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
