<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÊâÆÊºîÁùÄÂÖ≥ÈîÆËßíËâ≤ÔºåÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆöÂçÅÂàÜÈáçË¶Å„ÄÇÊú¨Êñá‰ªãÁªç‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂíåÂ∑•ÂÖ∑„ÄÇ'>
<title>‰ªéÂÆèÂü∫Âõ†ÁªÑ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/virus/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='‰ªéÂÆèÂü∫Âõ†ÁªÑ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó'>
<meta property='og:description' content='ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÊâÆÊºîÁùÄÂÖ≥ÈîÆËßíËâ≤ÔºåÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆöÂçÅÂàÜÈáçË¶Å„ÄÇÊú¨Êñá‰ªãÁªç‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂíåÂ∑•ÂÖ∑„ÄÇ'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/virus/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='microbiome' /><meta property='article:tag' content='virus' /><meta property='article:published_time' content='2024-06-20T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-06-20T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/virus/images/genomad.png' />
<meta name="twitter:title" content="‰ªéÂÆèÂü∫Âõ†ÁªÑ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó">
<meta name="twitter:description" content="ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÊâÆÊºîÁùÄÂÖ≥ÈîÆËßíËâ≤ÔºåÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆöÂçÅÂàÜÈáçË¶Å„ÄÇÊú¨Êñá‰ªãÁªç‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂíåÂ∑•ÂÖ∑„ÄÇ"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/virus/images/genomad.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">üêõüê∂</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/142776729'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://asa12138.github.io/'
                        target="_blank"
                        title="Homepage"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-world-pin"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M20.972 11.291a9 9 0 1 0 -8.322 9.686" /><path d="M3.6 9h16.8" /><path d="M3.6 15h8.9" /><path d="M11.5 3a17 17 0 0 0 0 18" /><path d="M12.5 3a16.986 16.986 0 0 1 2.578 9.018" /><path d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z" /><path d="M19 18v.01" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com/Asa12138_'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >‰∏≠Êñá</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#tools">Tools</a>
      <ol>
        <li><a href="#genomad">geNomad</a>
          <ol>
            <li><a href="#installation">Installation</a></li>
            <li><a href="#example">Example</a></li>
            <li><a href="#pipeline">Pipeline</a></li>
          </ol>
        </li>
        <li><a href="#deepvirfinder">DeepVirFinder</a>
          <ol>
            <li><a href="#installation-1">Installation</a></li>
            <li><a href="#example-1">Example</a></li>
          </ol>
        </li>
        <li><a href="#virsorter2">VirSorter2</a>
          <ol>
            <li><a href="#installation-2">Installation</a></li>
            <li><a href="#example-2">Example</a></li>
          </ol>
        </li>
        <li><a href="#vibrant">VIBRANT</a></li>
        <li><a href="#others">Others</a></li>
      </ol>
    </li>
    <li><a href="#quality-assessment">Quality assessment</a>
      <ol>
        <li><a href="#checkv">CheckV</a>
          <ol>
            <li><a href="#installation-3">Installation</a></li>
            <li><a href="#example-3">Example</a></li>
            <li><a href="#pipeline-1">Pipeline</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#application">Application</a>
      <ol>
        <li><a href="#biogeographic-patterns-and-drivers-of-soil-viromes">Biogeographic patterns and drivers of soil viromes</a></li>
        <li><a href="#a-global-atlas-of-soil-viruses">A global atlas of soil viruses</a></li>
        <li><a href="#bacteriophage-in-1300-year-old-human-palaeofaeces">Bacteriophage in 1300-year-old human palaeofaeces</a></li>
      </ol>
    </li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/virus/">
                <img src="/Cao-blog/p/virus/images/genomad_hu_2656be63aec98381.png"
                        srcset="/Cao-blog/p/virus/images/genomad_hu_2656be63aec98381.png 800w, /Cao-blog/p/virus/images/genomad_hu_8f6dcb0940689a4.png 1600w"
                        width="800" 
                        height="426" 
                        loading="lazy"
                        alt="Featured image of post ‰ªéÂÆèÂü∫Âõ†ÁªÑ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/virus/">‰ªéÂÆèÂü∫Âõ†ÁªÑ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÊâÆÊºîÁùÄÂÖ≥ÈîÆËßíËâ≤ÔºåÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆöÂçÅÂàÜÈáçË¶Å„ÄÇÊú¨Êñá‰ªãÁªç‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂíåÂ∑•ÂÖ∑„ÄÇ
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Jun 20, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    9 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                ÁâàÊùÉÂΩíCao's blogÊâÄÊúâÔºåËΩ¨ËΩΩËØ∑Ê≥®ÊòéÂá∫Â§Ñ
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>Âú®ÁéØÂ¢ÉÂæÆÁîüÁâ©Â≠¶ÂíåÁîüÊÄÅÂ≠¶Á†îÁ©∂‰∏≠ÔºåÂÆèÂü∫Âõ†ÁªÑÂ≠¶ÔºàMetagenomicsÔºâÊäÄÊúØÁöÑÂ∫îÁî®Â∑≤ÁªèÂΩªÂ∫ïÊîπÂèò‰∫ÜÊàë‰ª¨ÂØπÂæÆÁîüÁâ©Áæ§ËêΩÁöÑÁêÜËß£„ÄÇÂÆèÂü∫Âõ†ÁªÑÂ≠¶ÈÄöËøáÂØπÁéØÂ¢ÉÊ†∑Êú¨‰∏≠ÁöÑÂÖ®ÈÉ®ÈÅó‰º†Áâ©Ë¥®ËøõË°åÊµãÂ∫èÂíåÂàÜÊûêÔºåÂèØ‰ª•ÂÖ®Èù¢Êè≠Á§∫ÂæÆÁîüÁâ©Áæ§ËêΩÁöÑÁªÑÊàê„ÄÅÂäüËÉΩÂíåÁõ∏‰∫í‰ΩúÁî®„ÄÇÁÑ∂ËÄåÔºåÂú®Ëøô‰∫õÂ§çÊùÇÁöÑÂæÆÁîüÁâ©Áæ§ËêΩ‰∏≠ÔºåÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆö‰ªçÁÑ∂ÊòØ‰∏Ä‰∏™Â∑®Â§ßÁöÑÊåëÊàò„ÄÇ</p>
<p>ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÊâÆÊºîÁùÄÂÖ≥ÈîÆËßíËâ≤ÔºåÂÆÉ‰ª¨ÈÄöËøáÊÑüÊüìÂÆø‰∏ªÂæÆÁîüÁâ©Ë∞ÉÊéßÂæÆÁîüÁâ©Áæ§ËêΩÁöÑÂä®ÊÄÅÂíåÂäüËÉΩ„ÄÇÁóÖÊØíÁöÑÂ≠òÂú®ÂíåÊ¥ªÂä®ÂØπÂÆø‰∏ªÁöÑËøõÂåñ„ÄÅÁîüÁâ©Âú∞ÁêÉÂåñÂ≠¶Âæ™ÁéØ‰ª•ÂèäÁéØÂ¢ÉÂÅ•Â∫∑ÊúâÁùÄÊ∑±ËøúÁöÑÂΩ±Âìç„ÄÇÂ∞ΩÁÆ°ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠Ëá≥ÂÖ≥ÈáçË¶ÅÔºå‰ΩÜÁî±‰∫éÂÆÉ‰ª¨ÁöÑÂü∫Âõ†ÁªÑÂ§öÊ†∑ÊÄßÈ´ò„ÄÅÊ≤°ÊúâÈÄöÁî®ÁöÑ‰øùÂÆàÂü∫Âõ†„ÄÅ‰ª•ÂèäÂü∫Âõ†ÁªÑÁâáÊÆµÂåñ‰∏•ÈáçÔºå‰º†ÁªüÁöÑÂÆèÂü∫Âõ†ÁªÑÂàÜÊûêÊñπÊ≥ïÂ∏∏Â∏∏Èöæ‰ª•ÊúâÊïàÂú∞ËØÜÂà´ÂíåËß£ÊûêÁóÖÊØíÂ∫èÂàó„ÄÇ</p>
<p>‰∏∫‰∫ÜÂÖãÊúçËøô‰∫õÊåëÊàòÔºåÁ†îÁ©∂‰∫∫ÂëòÂºÄÂèë‰∫Ü‰∏ÄÁ≥ªÂàó‰∏ìÈó®ÁöÑÂ∑•ÂÖ∑ÂíåÊñπÊ≥ïÔºåÁî®‰∫é‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠ÊèêÂèñÂíåÈâ¥ÂÆöÁóÖÊØíÂ∫èÂàó„ÄÇËøô‰∫õÊñπÊ≥ïÁªìÂêà‰∫ÜÂ∫èÂàóÊØîÂØπ„ÄÅÁªìÊûÑÁâπÂæÅÂàÜÊûêÂíåÊú∫Âô®Â≠¶‰π†Á≠âÊäÄÊúØÔºåÊòæËëóÊèêÈ´ò‰∫ÜÁóÖÊØíÂ∫èÂàóÁöÑÈâ¥ÂÆöÊïàÁéáÂíåÂáÜÁ°ÆÊÄß„ÄÇÈÄöËøáËøô‰∫õÊñπÊ≥ïÔºåÁßëÂ≠¶ÂÆ∂‰ª¨‰∏ç‰ªÖËÉΩÂ§üÊõ¥ÂÖ®Èù¢Âú∞‰∫ÜËß£ÁéØÂ¢ÉÁóÖÊØíÁöÑÂ§öÊ†∑ÊÄßÔºåËøòËÉΩÊè≠Á§∫ÁóÖÊØí‰∏éÂÖ∂ÂÆø‰∏ª‰πãÈó¥ÁöÑÂ§çÊùÇÂÖ≥Á≥ªÔºå‰∏∫ÁîüÊÄÅÂ≠¶„ÄÅËøõÂåñÁîüÁâ©Â≠¶ÂíåÂÖ¨ÂÖ±Âç´ÁîüÁ≠âÈ¢ÜÂüüÁöÑÁ†îÁ©∂Êèê‰æõÈáçË¶ÅÁöÑÂèÇËÄÉ„ÄÇ</p>
<p>Êú¨ÊñáÊàë‰ª¨Â∞Ü‰ªãÁªç‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂíåÂ∑•ÂÖ∑„ÄÇ</p>
<p>‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑ‰∏ªË¶ÅÊñπÊ≥ïÂèØ‰ª•ÂàÜ‰∏∫Âá†Á±ªÔºö</p>
<ol>
<li>Â∫èÂàóÊØîÂØπÊñπÊ≥ï</li>
</ol>
<p>Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÁöÑÊúÄÁõ¥Êé•ÊñπÊ≥ï„ÄÇËøô‰∫õÊñπÊ≥ï‰æùËµñ‰∫éÂ∞ÜÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠ÁöÑÂ∫èÂàó‰∏éÂ∑≤Áü•ÁóÖÊØíÊï∞ÊçÆÂ∫ìËøõË°åÊØîÂØπÔºå‰ª•ÊâæÂà∞Áõ∏‰ººÁöÑÂ∫èÂàó„ÄÇÂ∏∏Áî®ÁöÑÂ∫èÂàóÊØîÂØπÂ∑•ÂÖ∑ÂíåÊï∞ÊçÆÂ∫ìÂåÖÊã¨Ôºö</p>
<ul>
<li><strong>BLAST</strong>ÔºàBasic Local Alignment Search ToolÔºâÔºöËøôÊòØÊúÄÂ∏∏Áî®ÁöÑÂ∫èÂàóÊØîÂØπÂ∑•ÂÖ∑ÔºåÂèØ‰ª•Áî®‰∫éÂ∞ÜÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏éÁóÖÊØíÊï∞ÊçÆÂ∫ìËøõË°åÊØîÂØπ„ÄÇ</li>
<li><strong>Diamond</strong>ÔºöÊØîBLASTÊõ¥Âø´ÈÄüÁöÑÂ∫èÂàóÊØîÂØπÂ∑•ÂÖ∑ÔºåÈÄÇÁî®‰∫éÂ§ßËßÑÊ®°ÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆÁöÑÂ§ÑÁêÜ„ÄÇ</li>
<li><strong>NCBI RefSeq Viral Database</strong>Ôºö‰∏Ä‰∏™ÂπøÊ≥õ‰ΩøÁî®ÁöÑÁóÖÊØíÂ∫èÂàóÊï∞ÊçÆÂ∫ì„ÄÇ</li>
</ul>
<ol start="2">
<li>ÁâπÂæÅÂàÜÊûêÊñπÊ≥ï</li>
</ol>
<p>ÈÄöËøáÂàÜÊûêÂ∫èÂàóÁöÑÁâπÂÆöÁâπÂæÅÊù•Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó„ÄÇÂ∏∏Áî®ÁöÑÁâπÂæÅÂàÜÊûêÂåÖÊã¨Ôºö</p>
<ul>
<li><strong>k-mer ÂàÜÊûê</strong>ÔºöÂàÜÊûêÂ∫èÂàó‰∏≠Âõ∫ÂÆöÈïøÂ∫¶ÁöÑÂ≠êÂ∫èÂàóÔºàk-mersÔºâÁöÑÈ¢ëÁéáÂàÜÂ∏ÉÔºåÁî®‰∫éÂå∫ÂàÜÁóÖÊØíÂíåÂÆø‰∏ªÂ∫èÂàó„ÄÇ</li>
<li><strong>GCÂê´ÈáèÂàÜÊûê</strong>ÔºöÁóÖÊØíÂíåÂÆø‰∏ªÁöÑGCÂê´ÈáèÂæÄÂæÄÊúâÊâÄ‰∏çÂêåÔºåÈÄöËøáÂàÜÊûêGCÂê´ÈáèÂèØ‰ª•ËæÖÂä©Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàó„ÄÇ</li>
<li><strong>Â∫èÂàóÊ∑±Â∫¶ÂàÜÊûê</strong>ÔºöÁóÖÊØíÂ∫èÂàóÂú®ÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠ÁöÑË¶ÜÁõñÊ∑±Â∫¶ÂèØËÉΩ‰∏éÂÆø‰∏ªÂ∫èÂàó‰∏çÂêåÔºåÈÄöËøáÊØîËæÉË¶ÜÁõñÊ∑±Â∫¶ÂèØ‰ª•ËæÖÂä©ËØÜÂà´ÁóÖÊØíÂ∫èÂàó„ÄÇ</li>
</ul>
<ol start="3">
<li>Êú∫Âô®Â≠¶‰π†ÊñπÊ≥ï</li>
</ol>
<p>Êú∫Âô®Â≠¶‰π†ÊñπÊ≥ïÈÄöËøáËÆ≠ÁªÉÊ®°ÂûãÊù•Ëá™Âä®Èâ¥ÂÆöÁóÖÊØíÂ∫èÂàóÔºåÂ¶Ç<strong>ÊîØÊåÅÂêëÈáèÊú∫ÔºàSVMÔºâ</strong>Ôºå<strong>ÈöèÊú∫Ê£ÆÊûó</strong>Ôºå <strong>Ê∑±Â∫¶Â≠¶‰π†</strong>Á≠â„ÄÇ</p>
<h2 id="tools">Tools</h2>
<p>ÁõÆÂâçÔºåÊúâÂ§öÁßç‰∏ìÈó®Áî®‰∫éÁóÖÊØíÈâ¥ÂÆöÁöÑÂ∑•ÂÖ∑ÔºåÊØè‰∏™Â∑•ÂÖ∑ÈÉΩÊúâÂÖ∂Áã¨ÁâπÁöÑ‰ºòÂäøÂíåÈÄÇÁî®Âú∫ÊôØ„ÄÇ‰ª•‰∏ãÊòØ‰∏Ä‰∫õÂ∏∏Áî®ÁöÑÁóÖÊØíÈâ¥ÂÆöÂ∑•ÂÖ∑ÁöÑÁÆÄË¶Å‰ªãÁªçÔºö</p>
<h3 id="genomad">geNomad</h3>
<img src="images/genomad.png" title=""/>
<p>geNomad ÊòØ‰∏ÄÁßçÁî®‰∫éËØÜÂà´ÂíåÂàÜÁ±ªÁóÖÊØíÂ∫èÂàóÁöÑÂ∑•ÂÖ∑Ôºå2023Âπ¥9ÊúàÂèëË°®Âú®Nature Biotechnology‰∏ä[1]„ÄÇÂÆÉÁªìÂêà‰∫ÜÂ∫èÂàóÊØîÂØπÂíåÊú∫Âô®Â≠¶‰π†ÊñπÊ≥ïÔºåËÉΩÂ§üÈ´òÊïàÂú∞‰ªéÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠ÊèêÂèñÁóÖÊØíÂ∫èÂàó„ÄÇgeNomad ÊìÖÈïøÂ§ÑÁêÜÂ§çÊùÇÂíåÈ´òÂô™Â£∞ÁöÑÁéØÂ¢ÉÊï∞ÊçÆÔºåÈÄÇÁî®‰∫éÂ§öÊ†∑ÂåñÁöÑÂÆèÂü∫Âõ†ÁªÑÊ†∑Êú¨„ÄÇgeNomadËøêË°åÈÄüÂ∫¶Âø´ÔºåÂèØÂØπÂ§ßÂûãÊï∞ÊçÆÈõÜËøõË°åÂàÜÊûê„ÄÇ</p>
<p>ËΩØ‰ª∂Ôºö<a class="link" href="https://github.com/apcamargo/genomad"  target="_blank" rel="noopener"
    >https://github.com/apcamargo/genomad</a></p>
<p>ÊïôÁ®ãÔºö<a class="link" href="https://portal.nersc.gov/genomad/"  target="_blank" rel="noopener"
    >https://portal.nersc.gov/genomad/</a></p>
<h4 id="installation">Installation</h4>
<ul>
<li>ÂÆâË£ÖËΩØ‰ª∂</li>
</ul>
<p>ÂèØ‰ª•‰ΩøÁî®ÈÄöÁî®ÂåÖÁÆ°ÁêÜÂô®Ôºàmamba Êàñ condaÔºâÊàñÁâπÂÆö‰∫é Python ÁöÑÂåÖÁÆ°ÁêÜÂô®ÔºàpipÔºâÂú®ËÆ°ÁÆóÊú∫‰∏≠ÂÆâË£Ö geNomad„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># mamba</span>
</span></span><span class="line"><span class="cl">mamba create -n genomad -c conda-forge -c bioconda genomad
</span></span><span class="line"><span class="cl">mamba activate genomad
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># conda</span>
</span></span><span class="line"><span class="cl">conda create -n genomad -c conda-forge -c bioconda genomad
</span></span><span class="line"><span class="cl">conda activate genomad
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># pip</span>
</span></span><span class="line"><span class="cl">pip install genomad
</span></span></code></pre></td></tr></table>
</div>
</div><p>Conda Âíå Mamba Â∞ÜÂÆâË£Ö geNomad ÊâÄÈúÄÁöÑ Python ‰æùËµñÈ°πÂíåÁ¨¨‰∏âÊñπËΩØ‰ª∂„ÄÇÂ¶ÇÊûú‰ΩøÁî® pip Êàñ pipx ÂÆâË£Ö geNomadÔºåË¶ÅÂ∞Ü MMseqs2 Âíå ARAGORN Ê∑ªÂä†Âà∞ÁéØÂ¢ÉÂèòÈáèÈáå„ÄÇ
MMseqs2 Âíå ARAGORNÊòØÊàëÂ∏∏Áî®ÁöÑËΩØ‰ª∂ÔºåÊâÄ‰ª•ÊàëÁõ¥Êé•Áî®pipÂÆâË£ÖÔºåÊ≥®ÊÑèMMseqs2ÊúÄÂ•ΩÊòØÊúÄÊñ∞ÁâàÔºå‰∏çÁÑ∂ÂèØËÉΩÊúâÁÇπÈóÆÈ¢ò„ÄÇ</p>
<ul>
<li>‰∏ãËΩΩÊï∞ÊçÆÂ∫ì
geNomad ‰æùËµñ‰∫é‰∏Ä‰∏™ÂåÖÂê´Áî®‰∫éÂØπÂ∫èÂàóËøõË°åÂàÜÁ±ªÁöÑÊ†áËÆ∞ÁöÑÊ¶ÇÂÜµ„ÄÅÂÆÉ‰ª¨ÁöÑÂàÜÁ±ª‰ø°ÊÅØ„ÄÅÂÆÉ‰ª¨ÁöÑÂäüËÉΩÊ≥®ÈáäÁ≠âÁöÑÊï∞ÊçÆÂ∫ìÔºö</li>
</ul>
<p>ÂèØ‰ª•ÈÄöËøáËΩØ‰ª∂‰∏ãËΩΩÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">genomad download-database .
</span></span></code></pre></td></tr></table>
</div>
</div><p>‰πüÂèØ‰ª•Ëá™Ë°å‰∏ãËΩΩÂπ∂Ëß£ÂéãÔºàÊàëËØï‰∫ÜËøô‰∏™Âø´ÔºâÔºåÁΩëÂùÄÔºöhttps://portal.nersc.gov/genomad/<strong>data</strong>/genomad_db_v1.7.tar.gz ÔºåÊàëÊîæÂú®‰∫Ü~/db/genomad_db/genomad_dbÁõÆÂΩïÔºåËß£ÂéãÂêéÂ§ßÊ¶Ç1.4G„ÄÇ</p>
<h4 id="example">Example</h4>
<p>geNomadÈÄöËøáÈ°∫Â∫èÊâßË°å‰∏ÄÁ≥ªÂàóÊ®°ÂùóÊù•Â∑•‰ΩúÔºàËßÅ‰∏ãÂàóPipelineÔºâÔºå‰ΩÜ‰πüÊèê‰æõ‰∫Ü‰∏Ä‰∏™Êñπ‰æøÁöÑ‰∏ÄÈîÆÂºèÂëΩ‰ª§ÔºåÂèØ‰ª•‰∏ÄÊ¨°ÊÄßÊâßË°åÊï¥‰∏™ÊµÅÁ®ã„ÄÇ</p>
<p>Á§∫‰æãÔºö‰ΩøÁî®ËÇ∫ÁÇéÂÖãÈõ∑‰ºØÊ∞èËèåÂü∫Âõ†ÁªÑÔºàGCF_009025895.1Ôºâ‰Ωú‰∏∫ËæìÂÖ•„ÄÇgeNomadÈÄÇÁî®‰∫éÂàÜÁ¶ªÂü∫Âõ†ÁªÑ„ÄÅÂÆèÂü∫Âõ†ÁªÑÂíåÂÆèËΩ¨ÂΩïÁªÑ‰∏≠ÁöÑÁóÖÊØíÂ∫èÂàó„ÄÇ</p>
<p>ÊâßË°ågeNomadÁöÑÂëΩ‰ª§ÁªìÊûÑÂ¶Ç‰∏ãÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">genomad end-to-end <span class="o">[</span>OPTIONS<span class="o">]</span> INPUT OUTPUT DATABASE
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÊâÄ‰ª•Ôºå‰∏∫‰∫ÜËøêË°åÂÆåÊï¥ÁöÑgeNomadÊµÅÁ®ãÔºàend-to-endÂëΩ‰ª§ÔºâÔºåÊàë‰ª¨Â∞ÜÊ†∏Ëã∑ÈÖ∏FASTAÊñá‰ª∂ÔºàGCF_009025895.1.fna.gzÔºâÂíåÊï∞ÊçÆÂ∫ìÔºàgenomad_dbÔºâ‰Ωú‰∏∫ËæìÂÖ•ÔºåÊâßË°å‰ª•‰∏ãÂëΩ‰ª§Ôºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#‰∏ãËΩΩgenome</span>
</span></span><span class="line"><span class="cl">wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/009/025/895/GCF_009025895.1_ASM902589v1/GCF_009025895.1_ASM902589v1_genomic.fna.gz -O GCF_009025895.1.fna.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">genomad end-to-end --cleanup --splits <span class="m">8</span> GCF_009025895.1.fna.gz genomad_output ~/db/genomad_db/genomad_db
</span></span></code></pre></td></tr></table>
</div>
</div><p>Ê≥®ÊÑèÔºåËæìÂÖ•ÁöÑFASTAÊñá‰ª∂ÊòØÂéãÁº©ÁöÑ„ÄÇgeNomadÊîØÊåÅ.gz„ÄÅ.bz2Êàñ.xzÊ†ºÂºèÁöÑÂéãÁº©Êñá‰ª∂„ÄÇÁªìÊûúÂ∞ÜÂÜôÂÖ•<code>genomad_output</code>ÁõÆÂΩï‰∏≠Ôºå<code>--splits 8</code>ÂèÇÊï∞ÔºöÂ∞ÜÊêúÁ¥¢ÂàÜÊàê8‰∏™Âùó‰ª•ËäÇÁúÅÂÜÖÂ≠òÁ©∫Èó¥„ÄÇÂ¶ÇÊûú‰Ω†Âú®Â§ßÊúçÂä°Âô®‰∏äËøêË°ågeNomadÔºåÂèØËÉΩ‰∏çÈúÄË¶ÅÂàÜÂâ≤ÊêúÁ¥¢ÔºåËøôÊ†∑ÂèØ‰ª•ÊèêÈ´òÊâßË°åÈÄüÂ∫¶„ÄÇ<code>--cleanup</code>‰ºöÂº∫Âà∂geNomadÂà†Èô§Âú®ÊâßË°åËøáÁ®ã‰∏≠ÁîüÊàêÁöÑ‰∏≠Èó¥Êñá‰ª∂„ÄÇ</p>
<p>ÊàëËØï‰∫Ü‰∏Ä‰∏ãÔºå‰∏çÂ∏¶<code>--splits 8</code>ÂèÇÊï∞Êó∂ÔºåÂÜÖÂ≠òÊ∂àËÄó20GÔºåËÄóÊó∂300s„ÄÇÂ¶ÇÊûúÊÉ≥Ë¶ÅËøõ‰∏ÄÊ≠•Âä†Âø´ÈÄüÂ∫¶ÔºåÂèØ‰ª•ËÄÉËôëÔºö</p>
<ul>
<li>‰ΩøÁî®ÂêàÈÄÇÁöÑ <code>--threads</code>ÔºåÈªòËÆ§ÊòØ64Ôºå‰ΩÜ‰∏ÄËà¨ÂèØËÉΩÊ≤°ÊúâËøô‰πàÂ§ö„ÄÇ</li>
<li>‰ΩøÁî® <code>--disable-nn-classification</code> ÈÄâÈ°πÁ¶ÅÁî®Âü∫‰∫éÁ•ûÁªèÁΩëÁªúÁöÑÂàÜÁ±ªÔºåËøô‰πüÂ∞ÜÁ¶ÅÁî®ÂàÜÊï∞ËÅöÂêàÂπ∂Âº∫Âà∂ geNomad ‰ªÖ‰æùËµñÂü∫‰∫éÊ†áËÆ∞ÁöÑÂàÜÁ±ªÂô®„ÄÇÂèØËÉΩÂØπ geNomad ÁöÑÂàÜÁ±ªÊÄßËÉΩ‰∫ßÁîüË¥üÈù¢ÂΩ±Âìç„ÄÇ</li>
<li>‰ΩøÁî® <code>--sensitivity</code> ÂèÇÊï∞Èôç‰Ωé MMseqs2 ÊêúÁ¥¢ÁöÑÁÅµÊïèÂ∫¶ÔºåËØ•ÊêúÁ¥¢Â∞ÜÊ†áËÆ∞ÂàÜÈÖçÁªôÂü∫Âõ†„ÄÇËøôÂ∞Ü‰ΩøÊ≥®ÈáäÊ®°ÂùóÊõ¥Âø´Ôºå‰ΩÜ‰πü‰ºöÂáèÂ∞ëÂàÜÈÖçÁªôÊ†áËÆ∞ÁöÑÂü∫Âõ†Êï∞Èáè„ÄÇÂèØËÉΩÂØπ geNomad ÁöÑÂàÜÁ±ªÊÄßËÉΩ‰∫ßÁîüË¥üÈù¢ÂΩ±Âìç„ÄÇ</li>
</ul>
<p><strong>ÊéßÂà∂ÂàÜÁ±ª‰∏•Ê†ºÊÄß</strong></p>
<p>ÈªòËÆ§ÊÉÖÂÜµ‰∏ãÔºågeNomadÂ∫îÁî®‰∏ÄÁ≥ªÂàóÂàÜÁ±ªÂêéËøáÊª§Âô®‰ª•ÁßªÈô§ÂèØËÉΩÁöÑÂÅáÈò≥ÊÄß„ÄÇ‰æãÂ¶ÇÔºåÂ∫èÂàóÈúÄË¶ÅÂÖ∑ÊúâËá≥Â∞ë0.7ÁöÑË¥®Á≤íÊàñÁóÖÊØíÂæóÂàÜÔºå‰∏îÈïøÂ∫¶Â∞è‰∫é2500 bpÁöÑÂ∫èÂàóÈúÄË¶ÅÁºñÁ†ÅËá≥Â∞ë‰∏Ä‰∏™Ê†áÂøóÂü∫Âõ†„ÄÇÂ¶ÇÊûú‰Ω†ÊÉ≥Á¶ÅÁî®Ëøô‰∫õËøáÊª§Âô®ÔºåÂèØ‰ª•Âú®ÂëΩ‰ª§‰∏≠Ê∑ªÂä†<code>--relaxed</code>Ê†áÂøó„ÄÇÁõ∏ÂèçÔºåÂ¶ÇÊûú‰Ω†Â∏åÊúõÂàÜÁ±ªÊõ¥Âä†‰øùÂÆàÔºåÂèØ‰ª•‰ΩøÁî®<code>--conservative</code>Ê†áÂøó„ÄÇËøôÂ∞Ü‰ΩøÂàÜÁ±ªÂêéËøáÊª§Âô®Êõ¥Âä†‰∏•Ê†ºÔºåÈò≤Ê≠¢Ê≤°ÊúâÂº∫ÊúâÂäõÊîØÊåÅÁöÑÂ∫èÂàóË¢´ÂàÜÁ±ª‰∏∫Ë¥®Á≤íÊàñÁóÖÊØí„ÄÇ</p>
<p><strong>Output</strong></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">genomad_output
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_aggregated_classification
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_aggregated_classification.log
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_annotate
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_annotate.log
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_find_proviruses
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_find_proviruses.log
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_marker_classification
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_marker_classification.log
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_nn_classification
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_nn_classification.log
</span></span><span class="line"><span class="cl">‚îú‚îÄ‚îÄ GCF_009025895.1_summary
</span></span><span class="line"><span class="cl">‚ï∞‚îÄ‚îÄ GCF_009025895.1_summary.log
</span></span></code></pre></td></tr></table>
</div>
</div><p><prefix>_summary ÁõÆÂΩïÂåÖÂê´ÊÄªÁªìÊï¥‰∏™ÁÆ°ÈÅìÁîüÊàêÁöÑÁªìÊûúÁöÑÊñá‰ª∂„ÄÇÂ¶ÇÊûúÂè™ÊÉ≥Ë¶ÅËæìÂÖ•‰∏≠ËØÜÂà´Âá∫ÁöÑË¥®Á≤íÂíåÁóÖÊØíÁöÑÂàóË°®ÔºåËøô‰∏™ÊòØÊúÄÂÖ≥ÈîÆÁöÑ„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">genomad_output
</span></span><span class="line"><span class="cl">‚ï∞‚îÄ‚îÄ GCF_009025895.1_summary
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_plasmid.fna
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_plasmid_genes.tsv
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_plasmid_proteins.faa
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_plasmid_summary.tsv
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_summary.json
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_virus.fna
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_virus_genes.tsv
</span></span><span class="line"><span class="cl">    ‚îú‚îÄ‚îÄ GCF_009025895.1_virus_proteins.faa
</span></span><span class="line"><span class="cl">    ‚ï∞‚îÄ‚îÄ GCF_009025895.1_virus_summary.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><p>GCF_009025895.1_virus_summary.tsv:</p>
<p>ËØ•Ë°®Ê†ºÊñá‰ª∂ÂàóÂá∫‰∫ÜgeNomadÂú®ËæìÂÖ•Êï∞ÊçÆ‰∏≠ÊâæÂà∞ÁöÑÊâÄÊúâÁóÖÊØíÔºåÂπ∂Êèê‰æõ‰∫Ü‰∏Ä‰∫õÊñπ‰æøÁöÑ‰ø°ÊÅØ„ÄÇÊØèÂàóÂåÖÂê´ÁöÑÂÜÖÂÆπÂ¶Ç‰∏ãÔºö</p>
<ul>
<li><strong>seq_name</strong>ÔºöËæìÂÖ•FASTAÊñá‰ª∂‰∏≠Â∫èÂàóÁöÑÊ†áËØÜÁ¨¶„ÄÇÂÜÖÊ∫êÁóÖÊØí‰ºöÊúâ‰ª•‰∏ãÂëΩÂêçÊ†ºÂºèÔºö<code>&lt;sequence_identifier&gt;|provirus_&lt;start_coordinate&gt;_&lt;end_coordinate&gt;</code>„ÄÇ</li>
<li><strong>length</strong>ÔºöÂ∫èÂàóÔºàÊàñÂÜÖÊ∫êÁóÖÊØíÔºâÁöÑÈïøÂ∫¶„ÄÇ</li>
<li><strong>topology</strong>ÔºöÁóÖÊØíÂ∫èÂàóÁöÑÊãìÊâëÁªìÊûÑ„ÄÇÂèØËÉΩÁöÑÂÄºÂåÖÊã¨ÔºöÊó†Êú´Á´ØÈáçÂ§çÔºàNo terminal repeatsÔºâ„ÄÅÁõ¥Êé•Êú´Á´ØÈáçÂ§çÔºàDTRÔºâ„ÄÅÂÄíÁΩÆÊú´Á´ØÈáçÂ§çÔºàITRÔºâÊàñÂÜÖÊ∫êÁóÖÊØíÔºàProvirusÔºâ„ÄÇ</li>
<li><strong>coordinates</strong>ÔºöÂÜÖÊ∫êÁóÖÊØíÂú®ÂÆø‰∏ªÂ∫èÂàó‰∏≠ÁöÑ1Á¥¢ÂºïÂùêÊ†á„ÄÇÂØπ‰∫éÊú™È¢ÑÊµã‰∏∫ÂÜÖÊ∫êÁóÖÊØíÁöÑÁóÖÊØíÔºåÊ≠§ÂÄº‰∏∫NA„ÄÇ</li>
<li><strong>n_genes</strong>ÔºöÂ∫èÂàó‰∏≠ÁºñÁ†ÅÁöÑÂü∫Âõ†Êï∞Èáè„ÄÇ</li>
<li><strong>genetic_code</strong>ÔºöÈ¢ÑÊµãÁöÑÈÅó‰º†ÂØÜÁ†Å„ÄÇÂèØËÉΩÁöÑÂÄºÂåÖÊã¨Ôºö11ÔºàÁªÜËèåÂíåÂè§ËèåÁöÑÊ†áÂáÜÂØÜÁ†ÅÔºâ„ÄÅ4ÔºàÈáçÊñ∞ÁºñÁ†ÅÁöÑTGAÁªàÊ≠¢ÂØÜÁ†ÅÔºâÊàñ15ÔºàÈáçÊñ∞ÁºñÁ†ÅÁöÑTAGÁªàÊ≠¢ÂØÜÁ†ÅÔºâ„ÄÇ</li>
<li><strong>virus_score</strong>ÔºögeNomadÂØπÂ∫èÂàóÊòØÁóÖÊØíÁöÑÁΩÆ‰ø°Â∫¶ÊµãÈáè„ÄÇÂæóÂàÜÊé•Ëøë1.0ÁöÑÂ∫èÂàóÊõ¥ÊúâÂèØËÉΩÊòØÁóÖÊØí„ÄÇ</li>
<li><strong>fdr</strong>ÔºöÂàÜÁ±ªÁöÑ‰º∞ËÆ°ÂÅáÂèëÁé∞ÁéáÔºàFDRÔºâÔºåÂç≥ËØ•Ë°å‰πãÂâçÂ∫èÂàó‰∏≠ÁöÑÈ¢ÑÊúüÂÅáÈò≥ÊÄßÊØî‰æã„ÄÇ‰º∞ËÆ°FDRÈúÄË¶ÅÂæóÂàÜÊ†°ÂáÜÔºåÈªòËÆ§ÂÖ≥Èó≠ÔºåÂõ†Ê≠§Êú¨‰æã‰∏≠Ê≠§ÂàóÂ∞Ü‰ªÖÂåÖÂê´NAÂÄº„ÄÇ</li>
<li><strong>n_hallmarks</strong>ÔºöÂåπÈÖçgeNomadÊ†áÂøóÂü∫Âõ†ÁöÑÂü∫Âõ†Êï∞Èáè„ÄÇÊ†áÂøóÂü∫Âõ†‰∏éÁóÖÊØíÂäüËÉΩÁõ∏ÂÖ≥ÔºåÂÖ∂Â≠òÂú®Âº∫ÁÉàË°®ÊòéËØ•Â∫èÂàóÁ°ÆÂÆûÊòØÁóÖÊØí„ÄÇ</li>
<li><strong>marker_enrichment</strong>ÔºöË°®Á§∫Â∫èÂàó‰∏≠ÁóÖÊØíÊ†áÂøóÂü∫Âõ†ÊÄªÂØåÈõÜÁöÑÂæóÂàÜ„ÄÇÈöèÁùÄÂ∫èÂàó‰∏≠ÁóÖÊØíÊ†áÂøóÂü∫Âõ†Êï∞ÈáèÁöÑÂ¢ûÂä†ÔºåËØ•ÂÄº‰ºöÂ¢ûÂä†ÔºåÂÖ∑ÊúâÂ§ö‰∏™Ê†áÂøóÂü∫Âõ†ÁöÑÂ∫èÂàóÂæóÂàÜËæÉÈ´ò„ÄÇÊüìËâ≤‰ΩìÂíåË¥®Á≤íÊ†áÂøóÂü∫Âõ†‰ºöÈôç‰ΩéÂæóÂàÜ„ÄÇ</li>
<li><strong>taxonomy</strong>ÔºöÁóÖÊØíÂü∫Âõ†ÁªÑÁöÑÂàÜÁ±ª„ÄÇÂàÜÁ±ª‰æùÊçÆICTVÁöÑVMRÁ¨¨19Âè∑ÂÜÖÂÆπ„ÄÇÁóÖÊØíÂèØ‰ª•ÂàÜÁ±ªÂà∞ÁßëÁ∫ßÔºå‰ΩÜ‰∏çËÉΩÂàÜÁ±ªÂà∞ÂÖ∑‰ΩìÁöÑÂ±ûÊàñÁßç„ÄÇÂàÜÁ±ª‰ª•Âõ∫ÂÆöÊï∞ÈáèÁöÑÂ≠óÊÆµÔºàÂØπÂ∫îÂàÜÁ±ªÁ≠âÁ∫ßÔºâË°®Á§∫ÔºåÁî®ÂàÜÂè∑ÂàÜÈöîÔºåÁ©∫Â≠óÊÆµÁïôÁ©∫„ÄÇ</li>
</ul>
<p>Ê≠§Â§ñÔºåÂú®ÊâßË°åËøáÁ®ã‰∏≠ÔºågeNomad ‰ΩøÁî®ÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÂíåÁóÖÊØíÁâπÂºÇÊÄßÊ†áËÆ∞Êï∞ÊçÆÂ∫ìÂØπËæìÂÖ•Â∫èÂàóÁºñÁ†ÅÁöÑÂü∫Âõ†ËøõË°åÊ≥®Èáä„ÄÇ <prefix>_virus_genes.tsv Êñá‰ª∂ÊÄªÁªì‰∫ÜÂ∑≤ËØÜÂà´ÁóÖÊØíÁºñÁ†ÅÁöÑÂü∫Âõ†ÁöÑÊ≥®Èáä„ÄÇ</p>
<p>GCF_009025895.1_virus_genes.tsvÔºåÊñá‰ª∂‰∏≠ÁöÑÂàóÂ¶Ç‰∏ãÔºö</p>
<ul>
<li><strong>gene</strong>ÔºöÂü∫Âõ†ÁöÑÊ†áËØÜÁ¨¶Ôºà&lt;sequence_name&gt;_&lt;gene_number&gt;Ôºâ„ÄÇÈÄöÂ∏∏Âü∫Âõ†ÁºñÂè∑‰ªé1ÂºÄÂßãÔºàÂ∫èÂàó‰∏≠ÁöÑÁ¨¨‰∏Ä‰∏™Âü∫Âõ†Ôºâ„ÄÇÁÑ∂ËÄåÔºåÁî±‰∫éÂÜÖÊ∫êÂô¨Ëèå‰ΩìÊï¥ÂêàÂú®ÂÆø‰∏ªÊüìËâ≤‰Ωì‰∏≠Èó¥ÔºåÂü∫Âõ†ÁºñÂè∑ÂèØËÉΩ‰ºöÂõ†ÂÖ∂Âú®ÊüìËâ≤‰Ωì‰∏≠ÁöÑ‰ΩçÁΩÆËÄå‰∏çÂêå„ÄÇ</li>
<li><strong>start</strong>ÔºöÂü∫Âõ†ÁöÑËµ∑ÂßãÂùêÊ†áÔºà1Á¥¢ÂºïÔºâ„ÄÇ</li>
<li><strong>end</strong>ÔºöÂü∫Âõ†ÁöÑÁªàÊ≠¢ÂùêÊ†áÔºà1Á¥¢ÂºïÔºâ„ÄÇ</li>
<li><strong>length</strong>ÔºöÂü∫Âõ†‰ΩçÁÇπÁöÑÈïøÂ∫¶Ôºà‰ª•Á¢±Âü∫ÂØπ‰∏∫Âçï‰ΩçÔºâ„ÄÇ</li>
<li><strong>strand</strong>ÔºöÁºñÁ†ÅÂü∫Âõ†ÁöÑÈìæ„ÄÇÂèØ‰ª•ÊòØ1ÔºàÊ≠£ÈìæÔºâÊàñ-1ÔºàË¥üÈìæÔºâ„ÄÇ</li>
<li><strong>gc_content</strong>ÔºöÂü∫Âõ†‰ΩçÁÇπÁöÑGCÂê´Èáè„ÄÇ</li>
<li><strong>genetic_code</strong>ÔºöÈ¢ÑÊµãÁöÑÈÅó‰º†ÂØÜÁ†ÅÔºàËØ¶ËßÅÊëòË¶ÅÊñá‰ª∂Ëß£ÈáäÔºâ„ÄÇ</li>
<li><strong>rbs_motif</strong>ÔºöÊ£ÄÊµãÂà∞ÁöÑÊ†∏Á≥ñ‰ΩìÁªìÂêà‰ΩçÁÇπÁöÑÂü∫Â∫è„ÄÇ</li>
<li><strong>marker</strong>ÔºöÊúÄ‰Ω≥ÂåπÈÖçÁöÑgeNomadÊ†áËÆ∞„ÄÇÂ¶ÇÊûúËØ•Âü∫Âõ†‰∏çÂåπÈÖç‰ªª‰ΩïÊ†áËÆ∞ÔºåÂàôÂÄº‰∏∫NA„ÄÇ</li>
<li><strong>evalue</strong>ÔºöÂü∫Âõ†ÁºñÁ†ÅÁöÑËõãÁôΩË¥®‰∏éÊúÄ‰Ω≥ÂåπÈÖçÁöÑgeNomadÊ†áËÆ∞‰πãÈó¥ÊØîÂØπÁöÑEÂÄº„ÄÇ</li>
<li><strong>bitscore</strong>ÔºöÂü∫Âõ†ÁºñÁ†ÅÁöÑËõãÁôΩË¥®‰∏éÊúÄ‰Ω≥ÂåπÈÖçÁöÑgeNomadÊ†áËÆ∞‰πãÈó¥ÊØîÂØπÁöÑBitscoreÂÄº„ÄÇ</li>
<li><strong>uscg</strong>ÔºöÂàÜÈÖçÁªôËØ•Âü∫Âõ†ÁöÑÊ†áËÆ∞ÊòØÂê¶ÂØπÂ∫î‰∫éÈÄöÁî®ÂçïÊã∑Ë¥ùÂü∫Âõ†ÔºàUCSGÔºåÂ¶ÇBUSCO v5‰∏≠ÂÆö‰πâÔºâ„ÄÇËøô‰∫õÂü∫Âõ†È¢ÑÊúüÂú®ÊüìËâ≤‰Ωì‰∏≠ÂèëÁé∞ÔºåÂú®Ë¥®Á≤íÂíåÁóÖÊØí‰∏≠ÂæàÂ∞ëËßÅ„ÄÇÂèØ‰ª•ÊòØ1ÔºàÂü∫Âõ†ÊòØUSCGÔºâÊàñ0ÔºàÂü∫Âõ†‰∏çÊòØUSCGÔºâ„ÄÇ</li>
<li><strong>plasmid_hallmark</strong>ÔºöÂàÜÈÖçÁªôËØ•Âü∫Âõ†ÁöÑÊ†áËÆ∞ÊòØÂê¶‰ª£Ë°®Ë¥®Á≤íÊ†áÂøóÂü∫Âõ†„ÄÇ</li>
<li><strong>virus_hallmark</strong>ÔºöÂàÜÈÖçÁªôËØ•Âü∫Âõ†ÁöÑÊ†áËÆ∞ÊòØÂê¶‰ª£Ë°®ÁóÖÊØíÊ†áÂøóÂü∫Âõ†„ÄÇ</li>
<li><strong>taxid</strong>ÔºöÂàÜÈÖçÁªôËØ•Âü∫Âõ†ÁöÑÊ†áËÆ∞ÁöÑÂàÜÁ±ªÊ†áËØÜÁ¨¶ÔºàÊ≠§Â≠óÊÆµ‰∏ªË¶ÅÁî®‰∫égeNomadÂÜÖÈÉ®Ôºâ„ÄÇ</li>
<li><strong>taxname</strong>Ôºö‰∏éÂàÜÈÖçÁöÑgeNomadÊ†áËÆ∞Áõ∏ÂÖ≥ÁöÑÂàÜÁ±ªÂêçÁß∞„ÄÇÂú®Ê≠§Á§∫‰æã‰∏≠ÔºåÂèØ‰ª•ÁúãÂà∞ÊâÄÊúâÊ≥®ÈáäÁöÑËõãÁôΩË¥®ÈÉΩÂÖ∑ÊúâCaudoviricetesÁöÑÁâπÂæÅÔºàËøô‰πüÊòØÂÜÖÊ∫êÁóÖÊØíË¢´ÂàÜÈÖçÂà∞ËØ•Á±ªÁöÑÂéüÂõ†Ôºâ„ÄÇ</li>
<li><strong>annotation_conjscan</strong>ÔºöÂ¶ÇÊûúÂåπÈÖçÂü∫Âõ†ÁöÑÊ†áËÆ∞ÊòØ‰∏éÊé•Âêà‰ΩúÁî®Áõ∏ÂÖ≥ÁöÑÂü∫Âõ†ÔºàÂ¶ÇCONJscan‰∏≠ÂÆö‰πâÔºâÔºåÊ≠§Â≠óÊÆµÂ∞ÜÊòæÁ§∫ÂàÜÈÖçÁªôÊ†áËÆ∞ÁöÑCONJscanËÆøÈóÆÂè∑„ÄÇ</li>
<li><strong>annotation_amr</strong>ÔºöÂ¶ÇÊûúÂåπÈÖçÂü∫Âõ†ÁöÑÊ†áËÆ∞Ë¢´Ê≥®Èáä‰∏∫ÂÖ∑ÊúâÊäóÂæÆÁîüÁâ©ËÄêËçØÊÄßÔºàAMRÔºâÂäüËÉΩÔºàÂ¶ÇNCBIfam-AMRFinder‰∏≠ÂÆö‰πâÔºâÔºåÊ≠§Â≠óÊÆµÂ∞ÜÊòæÁ§∫ÂàÜÈÖçÁªôÊ†áËÆ∞ÁöÑNCBIfamËÆøÈóÆÂè∑„ÄÇ</li>
<li><strong>annotation_accessions</strong>Ôºö‰∏Ä‰∫õgeNomadÊ†áËÆ∞ÂÖ∑ÊúâÂäüËÉΩÊ≥®Èáä„ÄÇÊ≠§ÂàóÂëäËØâÊÇ®Âì™‰∫õPfam„ÄÅTIGRFAM„ÄÅCOGÂíåKEGGÊù°ÁõÆÂàÜÈÖçÁªô‰∫ÜÊ†áËÆ∞„ÄÇ</li>
<li><strong>annotation_description</strong>ÔºöÊèèËø∞ÂàÜÈÖçÁªôÊ†áËÆ∞ÂäüËÉΩÁöÑÊñáÊú¨„ÄÇ</li>
</ul>
<p>Ë¥®Á≤íÈâ¥ÂÆöÁõ∏ÂÖ≥ÁöÑÊï∞ÊçÆÂèØ‰ª•Âú® <prefix>_plasmid_summary.tsv„ÄÅ<prefix>_genes.tsv„ÄÅ<prefix>_plasmid.fna Âíå <prefix><em>plasmid</em> Proteins.faa Êñá‰ª∂‰∏≠ÊâæÂà∞„ÄÇ</p>
<h4 id="pipeline">Pipeline</h4>
<p>geNomadÁöÑÂÖ∑‰ΩìÂ∑•‰ΩúÊµÅÁ®ãÂ¶Ç‰∏ãÔºö</p>
<img src="images/genomad_pipe.png" title=""/>
<p>ÂØπ‰∫éÂ§ßÂ§öÊï∞ÊÉÖÂÜµÔºå‰ΩøÁî® <code>genomad end-to-end</code> Â∞±Ë∂≥Â§ü‰∫Ü„ÄÇÁÑ∂ËÄåÔºå‰∫ÜËß£ÊâßË°åÂÆåÊï¥ÁÆ°ÈÅìÊó∂Ê∂âÂèäÁöÑÂÖ∑‰ΩìÊµÅÁ®ãÈùûÂ∏∏ÈáçË¶Å„ÄÇ
ËøôÈáåÂØπÊØè‰∏™Ê®°ÂùóÁöÑÂäüËÉΩËøõË°åËß£ÈáäÔºå‰∫ÜËß£Ëøô‰∫õÂäüËÉΩÂ∞ÜÂ∏ÆÂä©ÊéåÊè° geNomad Â¶Ç‰ΩïÂ§ÑÁêÜËæìÂÖ•Â∫èÂàó‰ª•ËØÜÂà´Ë¥®Á≤íÂíåÁóÖÊØí„ÄÇ</p>
<ol>
<li>annotate</li>
</ol>
<p><code>annotate</code>Ê®°ÂùóÊúâ‰∏§‰∏™‰∏ªË¶ÅÂäüËÉΩÔºö‰ΩøÁî® Pyrodigal-gv È¢ÑÊµãËæìÂÖ•Â∫èÂàó‰∏≠ÁöÑÂü∫Âõ†ÔºåÂπ∂‰ΩøÁî® MMseqs2 Â∞ÜËøô‰∫õÈ¢ÑÊµãÂü∫Âõ†ÂàÜÈÖçÁªôÊù•Ëá™ 227,897 ‰∏™ÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÊàñÁóÖÊØíÁâπÊúâÂõæË∞±ÁöÑÊï∞ÊçÆÈõÜ‰∏≠ÁöÑÊ†áËÆ∞ËõãÁôΩÂÆ∂Êóè„ÄÇËØ•Ê†áËÆ∞Êï∞ÊçÆÈõÜÊèê‰æõ‰∫ÜÂÖ®Èù¢ÁöÑÂÖÉÊï∞ÊçÆÔºåÂèØ‰ª•Â∏ÆÂä©‰∏ãÊ∏∏ÁªìÊûúÁöÑËß£Èáä„ÄÇËøôÂåÖÊã¨Ôºö</p>
<ul>
<li>ÈÄöËøá Pfam„ÄÅCOG„ÄÅTIGRFAM Âíå KEGG Orthology ÁßçË¥®ËøõË°åÂäüËÉΩÊ≥®Èáä„ÄÇ</li>
<li>Ê†áÂøóÂü∫Âõ†ÔºåÊ∂âÂèäÂÖ≥ÈîÆÁöÑË¥®Á≤íÊàñÁóÖÊØíÂäüËÉΩ„ÄÇ</li>
<li>Êé•ÂêàÂü∫Âõ†ÔºåÈÄöËøáCONJscanÂä†ÂÖ•„ÄÇ</li>
<li>ÊäóËèåÁ¥†ÊäóÊÄßÂü∫Âõ†ÔºåÈÄöËøá AMRFinderËé∑Âæó„ÄÇ</li>
<li>‰ΩøÁî® BUSCO Èâ¥ÂÆöÁöÑÈÄöÁî®ÂçïÊã∑Ë¥ùÂü∫Âõ† (USCG) ÈÄöÂ∏∏Â≠òÂú®‰∫éÊüìËâ≤‰Ωì‰∏≠Ôºå‰ΩÜÂú®Ë¥®Á≤íÂíåÁóÖÊØí‰∏≠ÂæàÂ∞ëËßÅ„ÄÇ</li>
<li>ÈÄöËøá‰ΩøÁî® ICTV ÁöÑ VMR ÁºñÂè∑ 19 Ë∞±Á≥ªËøõË°åÁóÖÊØíÂàÜÁ±ª„ÄÇ</li>
</ul>
<p>Ê≥®ÈáäÊ®°ÂùóÁîüÊàê‰∏§‰∏™‰∏ªË¶ÅËæìÂá∫ÔºöËæìÂÖ•Â∫èÂàóÁöÑÂàÜÁ±ªÂàÜÈÖçÔºàÂèØ‰ª•Âú®Ê≠§Â§ÑÊâæÂà∞ geNomad Â¶Ç‰ΩïÂ∞ÜÂ∫èÂàóÂàÜÈÖçÁªôÁóÖÊØíÂàÜÁ±ªÁæ§ÁöÑËØ¥ÊòéÔºâÂíåÂü∫Âõ†Á∫ßÊ≥®ÈáäÔºàÂ¶ÇÂø´ÈÄüÂÖ•Èó®Á§∫‰æã‰∏≠ÊâÄÁ§∫Ôºâ„ÄÇËøô‰∫õËæìÂá∫Ë¢´ <code>find-proviruses</code>, <code>marker-classification</code>, <code>summary</code>‰ΩøÁî®„ÄÇ</p>
<ol start="2">
<li>find-proviruses</li>
</ol>
<p><code>find-proviruses</code> Ê®°ÂùóÊó®Âú®ËØÜÂà´ÂÆø‰∏ªÂ∫èÂàóÂÜÖÁöÑÂéüÁóÖÊØíÂå∫Âüü„ÄÇ
‰∏∫‰∫ÜÂÆûÁé∞Ëøô‰∏ÄÁõÆÊ†áÔºåÂÆÉ‰ΩøÁî®Êù°‰ª∂ÈöèÊú∫Âú∫ (CRF) Ê®°ÂûãÔºåËØ•Ê®°ÂûãÈááÁî®Ê≥®ÈáäÊ®°ÂùóÁîüÊàêÁöÑÂü∫Âõ†Ê≥®ÈáäÔºåÂπ∂ÂàíÂàÜÂØåÂê´ÁóÖÊØíÁâπÂºÇÊÄßÊ†áËÆ∞ÁöÑÂå∫ÂüüÔºåÂë®Âõ¥ÁéØÁªïÁùÄÂÆø‰∏ªÁâπÂºÇÊÄßÊ†áËÆ∞„ÄÇ
‰∏∫‰∫ÜÁªÜÂåñÂéüÁóÖÊØíÁöÑËæπÁïåÔºågeNomad Âà©Áî®‰∫ÜÂô¨Ëèå‰ΩìÈÄöÂ∏∏Âú® tRNA ÊóÅËæπÊï¥Âêà‰ª•ÂèäÊï¥ÂêàÈÖ∂ÈÄöÂ∏∏‰Ωç‰∫éÊï¥ÂêàÂô¨Ëèå‰ΩìËæπÁºòÁöÑ‰∫ãÂÆû„ÄÇ
ËøôÊòØÈÄöËøáÂª∂‰º∏ËæπÁºòÁõ¥Âà∞Âà∞ËææÈÇªËøëÁöÑ tRNAÔºàÁî® ARAGORN ËØÜÂà´ÔºâÂíå/ÊàñÊï¥ÂêàÈÖ∂ÔºàÁî® MMseqs2 ËØÜÂà´ÔºâÊù•ÂÆûÁé∞ÁöÑ„ÄÇ</p>
<ol start="3">
<li>marker-classification</li>
</ol>
<p>geNomad ‰∏≠ÁöÑ<code>marker-classification</code>Ê®°ÂùóÊó®Âú®Ê†πÊçÆÊ†áËÆ∞ÂÜÖÂÆπÂ∞ÜÂ∫èÂàóÂàÜÁ±ª‰∏∫ÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÊàñÁóÖÊØí„ÄÇ‰∏∫‰∫ÜÂÆûÁé∞Ëøô‰∏ÄÁõÆÊ†áÔºåËØ•Ê®°ÂùóËé∑ÂèñÂü∫Âõ†Ê≥®ÈáäÂπ∂ËÆ°ÁÆó‰∏ÄÁªÑÊï∞Â≠óÁâπÂæÅÔºåËøô‰∫õÁâπÂæÅÊèèËø∞‰∫ÜÈúÄË¶ÅÂàÜÁ±ªÁöÑÂ∫èÂàóÁöÑÂü∫Âõ†ÁªìÊûÑÂíåÊ†áËÆ∞ÂÜÖÂÆπ„ÄÇËøô‰∫õÁâπÂæÅÂåÖÊã¨Âü∫Âõ†ÂØÜÂ∫¶‰ª•ÂèäÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÂíåÁóÖÊØíÊ†áËÆ∞ÁöÑÈ¢ëÁéá„ÄÇ</p>
<p>ÁÑ∂ÂêéÔºå<code>marker-classification</code>Â∞ÜËøô‰∫õÁâπÂæÅÊèê‰æõÁªô‰ΩøÁî® XGBoost ËÆ≠ÁªÉÁöÑÊ†ëÈõÜÊàêÂàÜÁ±ªÁÆóÊ≥ïÔºåËØ•ÁÆóÊ≥ï‰∏∫ÊØè‰∏™Â∫èÂàóÁîüÊàê‰∏â‰∏™ÂàÜÊï∞„ÄÇËøô‰∫õÂàÜÊï∞‰ª£Ë°®Ê®°ÂûãÂØπËØ•Â∫èÂàó‰ª£Ë°®ÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÊàñÁóÖÊØíÁöÑÁΩÆ‰ø°Â∫¶„ÄÇ</p>
<ol start="4">
<li>nn-classification</li>
</ol>
<p><code>nn-classification</code>Ê®°ÂùóËøòÂ∞ÜËæìÂÖ•Â∫èÂàóÂàÜÁ±ª‰∏∫ÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÊàñÁóÖÊØíÔºåÁ±ª‰ºº‰∫éÊ†áËÆ∞ÂàÜÁ±ªÊ®°Âùó„ÄÇÁÑ∂ËÄåÔºå‰∏éÂêéËÄÖ‰∏çÂêåÁöÑÊòØÔºåÂÆÉ‰∏ç‰æùËµñ‰∫éÊ†áËÆ∞‰ø°ÊÅØ„ÄÇÁõ∏ÂèçÔºåÂÆÉ‰ΩøÁî®Á•ûÁªèÁΩëÁªúÁõ¥Êé•Â§ÑÁêÜÊ†∏Ëã∑ÈÖ∏Â∫èÂàó„ÄÇÊ†∏Ëã∑ÈÖ∏Â∫èÂàóÈ¶ñÂÖàË¢´ÁºñÁ†ÅÊàêÊï∞Â≠óÁü©ÈòµÔºåÁÑ∂ÂêéËæìÂÖ• IGLOO Á•ûÁªèÁΩëÁªú„ÄÇËØ•ÁΩëÁªúËÉΩÂ§üÊ£ÄÊµãÂå∫ÂàÜÊüìËâ≤‰Ωì„ÄÅË¥®Á≤íÂíåÁóÖÊØíÁöÑÂ∫èÂàóÁâπÂæÅ„ÄÇÊúÄÂêéÔºåËØ•Ê®°ÂùóÁîüÊàêÂàÜÁ±ªÁöÑÁΩÆ‰ø°Â∫¶ÂàÜÊï∞„ÄÇ</p>
<ol start="5">
<li>aggregated-classification</li>
</ol>
<p><code>aggregated-classification</code>Ê®°ÂùóÁªìÂêà‰∫Ü<code>marker-classification</code>Âíå<code>nn-classification</code>ÁöÑËæìÂá∫Ôºå‰ª•‰∫ßÁîü‰∏ÄÁªÑÂà©Áî®‰∏§‰∏™ÂàÜÁ±ªÂô®ÁöÑ‰ºòÂäøÁöÑÂàÜÊï∞„ÄÇ‰∏∫‰∫ÜÂÆûÁé∞Ëøô‰∏ÄÁõÆÊ†áÔºåÂÆÉÈááÁî®‰∫Ü‰∏ÄÁßçÊ≥®ÊÑèÂäõÊú∫Âà∂ÔºåÂØπÊØè‰∏™ÂàÜÁ±ªÂô®ÁöÑË¥°ÁåÆËøõË°åÂä†ÊùÉÔºå‰ΩøÂæóÊ†áËÆ∞ÂàÜÁ±ªÁöÑË¥°ÁåÆ‰∏éÂàÜÈÖçÁªôÊ†áËÆ∞ÁöÑÂü∫Âõ†ÁöÑÊØî‰æãÊàêÊØî‰æãÂú∞Â¢ûÂä†„ÄÇ</p>
<ol start="6">
<li>score-calibration</li>
</ol>
<p>Ê†áËÆ∞ÂàÜÁ±ª„ÄÅÁ•ûÁªèÁΩëÁªúÂàÜÁ±ªÂíåËÅöÂêàÂàÜÁ±ªÁîüÊàêÁöÑÂàÜÊï∞Ë°®ÊòéËøô‰∫õÊ®°ÂûãÂØπÂÖ∂È¢ÑÊµãÁöÑÁΩÆ‰ø°Â∫¶ÔºåÂÄºË∂äÈ´òÂàôË°®ÊòéÁΩÆ‰ø°Â∫¶Ë∂äÈ´ò„ÄÇÁÑ∂ËÄåÔºåËøô‰∫õÂÄºÂπ∂‰∏çÁ≠â‰∫éÂÆûÈôÖÊ¶ÇÁéá„ÄÇ‰æãÂ¶ÇÔºåÊú™Ê†°ÂáÜÁóÖÊØíÂæóÂàÜ‰∏∫ 0.87 ÁöÑÂ∫èÂàóÊ≤°Êúâ 87% ÁöÑÂèØËÉΩÊÄßÊòØÁóÖÊØí„ÄÇ
<code>score-calibration</code>ÊòØ‰∏Ä‰∏™ÂèØÈÄâÊ®°ÂùóÔºåÂÆÉÂ∞ÜÂÖàÂâçÊ®°Âùó‰∫ßÁîüÁöÑÂéüÂßãÂàÜÊï∞ËΩ¨Êç¢‰∏∫‰º∞ËÆ°Ê¶ÇÁéá„ÄÇËøôÁ°Æ‰øù‰∫ÜÊ†°ÂáÜÁóÖÊØíÂæóÂàÜ‰∏∫ 0.87 ÁöÑÂ∫èÂàóÊàê‰∏∫ÁóÖÊØíÁöÑÊ¶ÇÁéáÊé•Ëøë 87%„ÄÇË¶ÅÂêØÁî®ÂàÜÊï∞Ê†°ÂáÜ‰ΩøÁî® <code>--enable-score-calibration</code> ÂèÇÊï∞„ÄÇ</p>
<ol start="7">
<li>summary</li>
</ol>
<p><code>summary</code>Ê®°ÂùóÂÖ∑Êúâ‰∏â‰∏™‰∏ªË¶ÅÂäüËÉΩÔºö(1) Ê†πÊçÆÂêÑÁßçÊ†áÂáÜËøáÊª§Â∫èÂàóÔºå‰∏∫Áî®Êà∑Êèê‰æõÊúÄÂèØÈù†ÁöÑÈ¢ÑÊµãÔºàÂú®Ê≠§Â§ÑÈòÖËØªÊúâÂÖ≥ËøáÊª§ËøáÁ®ãÁöÑÊõ¥Â§ö‰ø°ÊÅØÔºâÔºå
(2) ÊÄªÁªìÊâÄÊúâÂÖàÂâçÊ®°Âùó‰∏∫Â∑≤ËØÜÂà´Ë¥®Á≤íÁîüÊàêÁöÑÊï∞ÊçÆÂíå(3) ÁºñÂÜôÂåÖÂê´Â∑≤ËØÜÂà´Ë¥®Á≤íÂíåÁóÖÊØíÁöÑÊ†∏Ëã∑ÈÖ∏ÂíåËõãÁôΩË¥®Â∫èÂàóÁöÑ FASTA Êñá‰ª∂ÔºåÂπ∂ÈôÑÊúâÂü∫Âõ†Ê≥®ÈáäÊñá‰ª∂„ÄÇ</p>
<h3 id="deepvirfinder">DeepVirFinder</h3>
<img src="images/deepvirfinder.png" title=""/>
<p>DeepVirFinder ‰ΩøÁî®Ê∑±Â∫¶Â≠¶‰π†ÊñπÊ≥ïÈ¢ÑÊµãÁóÖÊØíÂ∫èÂàó„ÄÇËØ•ÊñπÊ≥ïÂØπÁü≠ÁóÖÊØíÂ∫èÂàóÂÖ∑ÊúâËâØÂ•ΩÁöÑÈ¢ÑÊµãÂáÜÁ°ÆÊÄßÔºåÂõ†Ê≠§ÂèØÁî®‰∫éÈ¢ÑÊµãÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆ‰∏≠ÁöÑÂ∫èÂàóÔºå‰∫é2020Âπ¥ÂèëË°®Âú®quantitative biology‰∏ä[2]„ÄÇ
Áõ∏ÊØîÂü∫‰∫ék-merÁöÑÊñπÊ≥ïVirFinderÔºåDeepVirFinder‰ΩøÁî®Âç∑ÁßØÁ•ûÁªèÁΩëÁªúÔºàCNNÔºâÊòæËëóÊèêÈ´ò‰∫ÜÈ¢ÑÊµãÂáÜÁ°ÆÊÄß„ÄÇCNNÂèØ‰ª•Ëá™Âä®‰ªéÁóÖÊØíÂíåÂéüÊ†∏ÁîüÁâ©Â∫èÂàó‰∏≠Â≠¶‰π†Âü∫Âõ†ÁªÑÊ®°ÂºèÔºåÂπ∂Âü∫‰∫éÊâÄÂ≠¶ÁöÑÂü∫Âõ†ÁªÑÊ®°ÂºèÊûÑÂª∫È¢ÑÊµãÊ®°Âûã„ÄÇÂ≠¶Âà∞ÁöÑÊ®°Âºè‰ª•Â§ßÂ∞è‰∏∫4√ókÁöÑÊùÉÈáçÁü©ÈòµÂΩ¢ÂºèË°®Á§∫ÔºåÁ±ª‰ºº‰∫é‰ΩçÁΩÆÊùÉÈáçÁü©ÈòµÔºàPWMÔºâ„ÄÇËøôÁßçÁÅµÊ¥ªÁöÑCNNÊ®°ÂûãÂú®ÁóÖÊØíÂ∫èÂàóÈ¢ÑÊµãÈóÆÈ¢ò‰∏ä‰ºò‰∫ék-merÊ®°Âûã„ÄÇ</p>
<p>ËΩØ‰ª∂Ôºö<a class="link" href="https://github.com/jessieren/DeepVirFinder"  target="_blank" rel="noopener"
    >https://github.com/jessieren/DeepVirFinder</a></p>
<h4 id="installation-1">Installation</h4>
<p>DeepVirFinder ÈúÄË¶ÅPython 3.6ÔºåÂπ∂ÂÆâË£Önumpy„ÄÅtheano„ÄÅkeras„ÄÅscikit-learnÂíåBiopythonÁ≠âÂåÖ„ÄÇÊé®Ëçê‰ΩøÁî®MinicondaÂÆâË£ÖÊâÄÊúâ‰æùËµñÈ°π„ÄÇ</p>
<p>ÊàëÂ∑≤ÁªèÊúâ‰∫Üpython3.5Âíåpython3.7‰∏§‰∏™ÂÅöÊ∑±Â∫¶Â≠¶‰π†ÁöÑÁéØÂ¢ÉÔºåÊú¨Êù•ÊÉ≥ÁùÄÂÅ∑ÊáíÁõ¥Êé•Áî®Ôºå‰ΩÜÊòØÂèëÁé∞ÈÉΩ‰∏çÊîØÊåÅÔºåÊâÄ‰ª•Âè™ËÉΩÂÜçÂàõÂª∫‰∏Ä‰∏™python3.6ÁöÑÁéØÂ¢ÉÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">conda create --name dvf <span class="nv">python</span><span class="o">=</span>3.6 numpy <span class="nv">theano</span><span class="o">=</span>1.0.3 <span class="nv">keras</span><span class="o">=</span>2.2.4 scikit-learn Biopython h5py
</span></span><span class="line"><span class="cl"><span class="nb">source</span> activate dvf
</span></span></code></pre></td></tr></table>
</div>
</div><p>‰∏ãËΩΩËΩØ‰ª∂ÂåÖÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">git clone https://github.com/jessieren/DeepVirFinder
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> DeepVirFinder
</span></span><span class="line"><span class="cl">python dvf.py -h
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># ÂèØ‰ª•Ê∑ªÂä†ÁéØÂ¢ÉÂèòÈáèÔºåÊñπ‰æøË∞ÉÁî®</span>
</span></span><span class="line"><span class="cl">chmod +x dvf.py
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;1d&#39;</span> dvf.py
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;1i #!/miniconda3/envs/dvf/bin/python&#39;</span> dvf.py
</span></span><span class="line"><span class="cl">ln -s ./dvf.py ~/miniconda3/envs/dvf/bin/
</span></span><span class="line"><span class="cl">ln -s ~/biosoft/DeepVirFinder/models ~/miniconda3/envs/dvf/bin/models
</span></span><span class="line"><span class="cl">dvf.py -h
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="example-1">Example</h4>
<p>DeepVirFinderÁöÑËæìÂÖ•ÊòØÂåÖÂê´Ë¶ÅÈ¢ÑÊµãÂ∫èÂàóÁöÑfastaÊñá‰ª∂ÔºåËæìÂá∫ÊòØÂåÖÂê´ÊØè‰∏™ËæìÂÖ•Â∫èÂàóÁöÑÈ¢ÑÊµãÂæóÂàÜÂíåpÂÄºÁöÑ.txtÊñá‰ª∂„ÄÇÊõ¥È´òÁöÑÂæóÂàÜÊàñÊõ¥‰ΩéÁöÑpÂÄºË°®ÊòéÊõ¥È´òÁöÑÁóÖÊØíÂ∫èÂàóÂèØËÉΩÊÄß„ÄÇ</p>
<p>ÈªòËÆ§ÊÉÖÂÜµ‰∏ãÔºåËæìÂá∫Êñá‰ª∂‰∏éËæìÂÖ•Êñá‰ª∂Âú®Âêå‰∏ÄÁõÆÂΩï„ÄÇÁî®Êà∑‰πüÂèØ‰ª•ÈÄöËøá[-o]ÈÄâÈ°πÊåáÂÆöËæìÂá∫ÁõÆÂΩï„ÄÇ[-l]ÈÄâÈ°πËÆæÁΩÆÊúÄÂ∞èÂ∫èÂàóÈïøÂ∫¶ÈòàÂÄºÔºå‰Ωé‰∫éÊ≠§ÈòàÂÄºÁöÑÂ∫èÂàóÂ∞Ü‰∏ç‰ºöË¢´È¢ÑÊµã„ÄÇËØ•Á®ãÂ∫èÊîØÊåÅÂπ∂Ë°åËÆ°ÁÆóÔºåÈÄöËøá[-c]ÊåáÂÆö‰ΩøÁî®ÁöÑÁ∫øÁ®ãÊï∞„ÄÇ[-m]ÈÄâÈ°πÊåáÂÆöÊ®°ÂûãÁõÆÂΩïÔºåÈªòËÆ§Ê®°ÂûãÁõÆÂΩï‰∏∫./models„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">python dvf.py [-i INPUT_FA] [-o OUTPUT_DIR] [-l CUTOFF_LEN] [-c CORE_NUM]
</span></span><span class="line"><span class="cl">  -h, --help            ÊòæÁ§∫Ê≠§Â∏ÆÂä©‰ø°ÊÅØÂπ∂ÈÄÄÂá∫
</span></span><span class="line"><span class="cl">  -i INPUT_FA, --in=INPUT_FA   ËæìÂÖ•fastaÊñá‰ª∂
</span></span><span class="line"><span class="cl">  -m MODDIR, --mod=MODDIR      Ê®°ÂûãÁõÆÂΩïÔºàÈªòËÆ§./modelsÔºâ
</span></span><span class="line"><span class="cl">  -o OUTPUT_DIR, --out=OUTPUT_DIR ËæìÂá∫ÁõÆÂΩï
</span></span><span class="line"><span class="cl">  -l CUTOFF_LEN, --len=CUTOFF_LEN  ‰ªÖÈ¢ÑÊµãÈïøÂ∫¶&gt;= L bpÁöÑÂ∫èÂàóÔºàÈªòËÆ§1Ôºâ
</span></span><span class="line"><span class="cl">  -c CORE_NUM, --core=CORE_NUM   Âπ∂Ë°åÊ†∏ÂøÉÊï∞ÔºàÈªòËÆ§1Ôºâ
</span></span></code></pre></td></tr></table>
</div>
</div><p>Êàë‰ª¨Êù•ËØïËØïÔºåÈ¢ÑÊµãtestÊñá‰ª∂‰∏≠ÁöÑcrAssphageÂü∫Âõ†ÁªÑÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">python dvf.py -i ./test/crAssphage.fa -o ./test/ -l <span class="m">300</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÊµãËØïÊ≤°ÈóÆÈ¢òÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ python dvf.py -i ./test/crAssphage.fa -o ./test/ -l <span class="m">300</span>
</span></span><span class="line"><span class="cl">WARNING <span class="o">(</span>theano.configdefaults<span class="o">)</span>: install mkl with <span class="sb">`</span>conda install mkl-service<span class="sb">`</span>: No module named <span class="s1">&#39;mkl&#39;</span>
</span></span><span class="line"><span class="cl">Using Theano backend.
</span></span><span class="line"><span class="cl">1. Loading Models.
</span></span><span class="line"><span class="cl">   model directory /share/home/jianglab/pengchen/biosoft/DeepVirFinder/models
</span></span><span class="line"><span class="cl">2. Encoding and Predicting Sequences.
</span></span><span class="line"><span class="cl">   processing line <span class="m">1</span>
</span></span><span class="line"><span class="cl">   processing line <span class="m">1389</span>
</span></span><span class="line"><span class="cl">3. Done. Thank you <span class="k">for</span> using DeepVirFinder.
</span></span><span class="line"><span class="cl">   output in ./test/crAssphage.fa_gt300bp_dvfpred.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p><strong>Output:</strong></p>
<p>crAssphage.fa_gt300bp_dvfpred.txtÔºåËæìÂá∫ÈùûÂ∏∏ÁÆÄÂçïÔºå4Âàó:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">name	len	score	pvalue
</span></span><span class="line"><span class="cl">gi|674660337|ref|NC_024711.1| Uncultured phage crAssphage, complete genome	97065	0.9978806972503662	0.004702016768638115
</span></span></code></pre></td></tr></table>
</div>
</div><p>Â¶ÇÈúÄËÆ°ÁÆóqÂÄºÔºàÂÅáÂèëÁé∞ÁéáÔºâÔºåÂèØ‰ª•‰ΩøÁî®RÂåÖ‚Äúqvalue‚ÄùËøõË°åpÂÄºÁü´Ê≠£„ÄÇ</p>
<p>ÂÖ≥‰∫éÈòàÂÄºÔºödeepvirfinderÂÆòÊñπÊé®Ëçêq-value&lt;0.01ÁöÑÂ∫èÂàóËÆ§ÂÆö‰∏∫ÁóÖÊØíÂ∫èÂàóÔºå‰πüÊúâÊñáÁåÆÔºàBiogeographic patterns and drivers of soil viromesÔºâÈááÁî® score ‚â•0.7 and P &lt; 0.05Êù•ÁïåÂÆöÁóÖÊØíÂ∫èÂàóÔºåÊ≤°ÊúâÂÖ∑‰ΩìÊ†áÂáÜ„ÄÇ</p>
<p>Â¶ÇÊûúË¶ÅËá™Â∑±ÊûÑÂª∫Êï∞ÊçÆÂ∫ìËÆ≠ÁªÉÊ®°ÂûãÔºåÂèØ‰ª•ÂèÇËÄÉËΩØ‰ª∂‰∏ªÈ°µ„ÄÇ</p>
<h3 id="virsorter2">VirSorter2</h3>
<img src="images/virsorter2.webp" title=""/>
<p>VirSorter2 ÊòØ VirSorter ÁöÑÂçáÁ∫ßÁâàÔºå‰∫é2021Âπ¥ÂèëË°®Âú®Microbiome‰∏ä[3]„ÄÇ</p>
<p>VirSorter2 ÈááÁî®Â§öÂàÜÁ±ªÂô®„ÄÅ‰∏ìÂÆ∂ÊåáÂØºÁöÑÊñπÊ≥ïÊù•Ê£ÄÊµã‰∏çÂêåÁöÑ DNA Âíå RNA ÁóÖÊØíÂü∫Âõ†ÁªÑ„ÄÇÂÆÉÂØπ‰πãÂâçÁöÑÁâàÊú¨ËøõË°å‰∫ÜÈáçÂ§ßÊõ¥Êñ∞Ôºö</p>
<p>ÂåÖÂê´Êõ¥Â§öÁóÖÊØíÔºåÂåÖÊã¨ÂèåÈìæ DNA Âô¨Ëèå‰Ωì„ÄÅÂçïÈìæ DNA ÁóÖÊØí„ÄÅRNA ÁóÖÊØí„ÄÅNCLDVÔºàÊ†∏ÁªÜËÉûÁóÖÊØíÔºâ„ÄÅÊãâÁª¥ÁóÖÊØíÁßëÔºàÁóÖÊØíÂô¨Ëèå‰ΩìÔºâÔºõ
Â∫îÁî®Êú∫Âô®Â≠¶‰π†‰ΩøÁî®Âü∫Âõ†ÁªÑÁâπÂæÅÔºàÂåÖÊã¨ÁªìÊûÑ/ÂäüËÉΩ/ÂàÜÁ±ªÊ≥®ÈáäÂíåÁóÖÊØíÊ†áÂøóÂü∫Âõ†ÔºâÊù•‰º∞ËÆ°ÁóÖÊØíÊÄßÔºõ
‰ΩøÁî®Êù•Ëá™ÂÆèÂü∫Âõ†ÁªÑÊàñÂÖ∂‰ªñÊù•Ê∫êÁöÑÈ´òË¥®ÈáèÁóÖÊØíÂü∫Âõ†ÁªÑËøõË°åËÆ≠ÁªÉ„ÄÇ</p>
<p>ËΩØ‰ª∂Ôºö<a class="link" href="https://github.com/jiarong/VirSorter2"  target="_blank" rel="noopener"
    >https://github.com/jiarong/VirSorter2</a></p>
<h4 id="installation-2">Installation</h4>
<p>ÂÆâË£ÖÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mamba create -n vs2 -c conda-forge -c bioconda <span class="nv">virsorter</span><span class="o">=</span><span class="m">2</span>
</span></span><span class="line"><span class="cl">mamba activate vs2
</span></span></code></pre></td></tr></table>
</div>
</div><p>‰∏ãËΩΩÊï∞ÊçÆÂ∫ìÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#‰ª•Èò≤‰πãÂâçÂ∞ùËØïÂ§±Ë¥•Ôºõ</span>
</span></span><span class="line"><span class="cl"><span class="c1">#Âà†Èô§-dÊåáÂÆöÁöÑÊï¥‰∏™ÁõÆÂΩï</span>
</span></span><span class="line"><span class="cl">rm -rf ./virsorter_db
</span></span><span class="line"><span class="cl"><span class="c1"># run setup</span>
</span></span><span class="line"><span class="cl">virsorter setup -d ./virsorter_db -j <span class="m">4</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#Ëã•‰∏ãËΩΩÂ§±Ë¥•‰πüÂèØÊâãÂä®‰∏ãËΩΩÔºåhttps://osf.io/v46sc/download</span>
</span></span><span class="line"><span class="cl"><span class="c1">#Ëß£ÂéãÂêéÂ∞ÜË∑ØÂæÑ‰º†Áªô--db-dir</span>
</span></span><span class="line"><span class="cl">tar -xzf db.tgz
</span></span><span class="line"><span class="cl">mv db virsorter_db
</span></span><span class="line"><span class="cl">virsorter config --init-source --db-dir<span class="o">=</span>./virsorter_db
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="example-2">Example</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">wget -O test.fa https://raw.githubusercontent.com/jiarong/VirSorter2/master/test/8seq.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">virsorter run -w test.out -i test.fa --min-length <span class="m">1500</span> -j <span class="m">4</span> all
</span></span><span class="line"><span class="cl"> 
</span></span><span class="line"><span class="cl"><span class="c1">#Âü∫‰∫é‰∏çÂêåÂæóÂàÜÔºàmin-scoreÔºâÈáçÊñ∞Âø´ÈÄüËøêË°åÁ®ãÂ∫èÔºàÂè™ËøêË°åclassifyÊ≠•È™§ÔºâÔºåÊñ∞Êñá‰ª∂Ê∑ªÂä†ÂêéÁºÄÔºàlabelÔºâ‰∏∫rerun</span>
</span></span><span class="line"><span class="cl">virsorter run -w test.out -i test.fa --include-groups <span class="s2">&#34;dsDNAphage,ssDNA&#34;</span> -j <span class="m">4</span> --min-score 0.9 --label rerun classify
</span></span><span class="line"><span class="cl"> 
</span></span><span class="line"><span class="cl"><span class="c1">#Â¢ûÂä†hmmscanÁ∫øÁ®ãÊï∞ÊèêÈ´òËøêË°åÈÄüÂ∫¶</span>
</span></span><span class="line"><span class="cl">virsorter config --set <span class="nv">HMMSEARCH_THREADS</span><span class="o">=</span><span class="m">4</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>Áî±‰∫é VirSorter2 ‰ΩøÁî®Â§ßÂûã HMM Êï∞ÊçÆÂ∫ìÔºåËøô‰∏™Â∞èÊï∞ÊçÆÈõÜÈúÄË¶ÅÂá†ÂàÜÈíüÊâçËÉΩÂÆåÊàê„ÄÇÂú®ËæìÂá∫ÁõÆÂΩïÔºàtest.outÔºâ‰∏≠Ôºå‰∏â‰∏™Êñá‰ª∂ÂæàÊúâÁî®Ôºö</p>
<ul>
<li>Final-viral-combined.faÔºöÁ°ÆÂÆöÁöÑÁóÖÊØíÂ∫èÂàó</li>
<li>Final-viral-score.tsvÔºöÂåÖÂê´ÂêÑÁªÑÁóÖÊØíÂ∫èÂàóÂæóÂàÜÁöÑË°®Ê†º‰ª•Âèä‰∏Ä‰∫õÂÖ∂‰ªñÂÖ≥ÈîÆÁâπÂæÅÔºåÂèØÁî®‰∫éËøõ‰∏ÄÊ≠•ËøáÊª§</li>
<li>Final-viral-boundary.tsvÔºöÂåÖÂê´ËæπÁïå‰ø°ÊÅØÁöÑË°®ÔºõËøôÊòØ‰∏Ä‰∏™‰∏≠Èó¥Êñá‰ª∂Ôºå1Ôºâ‰∏éÂÖ∂‰ªñ‰∏§‰∏™Êñá‰ª∂Áõ∏ÊØîÂèØËÉΩÊúâÈ¢ùÂ§ñÁöÑËÆ∞ÂΩïÔºåÂ∫îË¢´ÂøΩÁï•Ôºõ 2) ‰∏çÂåÖÊã¨Â∏¶Êúâ &lt; 2 ‰∏™Âü∫Âõ†‰ΩÜÂÖ∑Êúâ &gt;= 1 ‰∏™Ê†áÂøóÂü∫Âõ†ÁöÑÁóÖÊØíÂ∫èÂàóÔºõ 3) groupÂíåtrim_prÊòØ‰∏≠Èó¥ÁªìÊûúÔºåÂèØËÉΩ‰∏éfinal-viral-score.tsv‰∏≠ÁöÑmax_groupÂíåmax_scoreÂàÜÂà´‰∏çÂåπÈÖç</li>
</ul>
<p>Â¶ÇÊûúË¶ÅËá™Â∑±ÊûÑÂª∫Êï∞ÊçÆÂ∫ìËÆ≠ÁªÉÊ®°ÂûãÔºåÂèØ‰ª•ÂèÇËÄÉËΩØ‰ª∂‰∏ªÈ°µ„ÄÇ</p>
<h3 id="vibrant">VIBRANT</h3>
<p>VIBRANT ÊòØ‰∏Ä‰∏™ÂäüËÉΩÂº∫Â§ßÁöÑÂ∑•ÂÖ∑ÔºåÁî®‰∫éËá™Âä®ÊÅ¢Â§çÂíåÊ≥®ÈáäÁªÜËèåÂíåÂè§ÁªÜËèåÁóÖÊØí„ÄÅÁ°ÆÂÆöÂü∫Âõ†ÁªÑÂÆåÊï¥ÊÄß‰ª•Âèä‰ªéÂÆèÂü∫Âõ†ÁªÑÁªÑË£Ö‰∏≠Ë°®ÂæÅÁóÖÊØíÁæ§ËêΩÔºå‰∫é2020Âπ¥ÂèëË°®Âú®Microbiome‰∏ä[4]„ÄÇ</p>
<ul>
<li>‰ΩøÁî®ËõãÁôΩË¥®Ê≥®ÈáäÁ≠æÂêçÁöÑÁ•ûÁªèÁΩëÁªúÊú∫Âô®Â≠¶‰π†</li>
<li>ÊåáÂÆöÊñ∞È¢ñÁöÑ‚Äúv-score‚ÄùÊù•Á°ÆÂÆöÊâÄÊúâÊ≥®ÈáäÁöÑÁóÖÊØíÊÄßË¥®</li>
<li>Á°ÆÂÆöÂü∫Âõ†ÁªÑÂÆåÊï¥ÊÄß</li>
<li>ÈÄöËøá‰ª£Ë∞¢ÂàÜÊûêË°®ÂæÅÁóÖÊØíÁæ§ËêΩÂäüËÉΩ</li>
<li>Èâ¥ÂÆöËæÖÂä©‰ª£Ë∞¢Âü∫Âõ† (AMG)</li>
<li>‰ªéÂÆø‰∏ªÊîØÊû∂‰∏äÂàáÈô§Êï¥ÂêàÁöÑÁóÖÊØíÂü∫Âõ†ÁªÑ</li>
<li>Âú®‰∏çÂêåÁöÑÁéØÂ¢É‰∏≠Ë°®Áé∞ËâØÂ•Ω</li>
<li>ÊÅ¢Â§çÊñ∞È¢ñ‰∏î‰∏∞ÂØåÁöÑÁóÖÊØíÂü∫Âõ†ÁªÑ</li>
<li>‰∏ì‰∏∫ dsDNA„ÄÅssDNA Âíå RNA ÁóÖÊØíËÄåËÆæËÆ°</li>
</ul>
<p>VIBRANT ‰ΩøÁî®‰∏â‰∏™Êï∞ÊçÆÂ∫ìÊù•ËØÜÂà´ÁóÖÊØíÂπ∂Ë°®ÂæÅÁóÖÊØíÁªÑ‰ª£Ë∞¢ÊΩúÂäõÔºö</p>
<ul>
<li>KEGGÔºà‰∏âÊúàÂèëÂ∏ÉÔºâÔºöhttps://www.genome.jp/kegg/ÔºàFTPÔºöftp://ftp.genome.jp/pub/db/kofam/archives/2019-03-20/Ôºâ</li>
<li>Pfam (v32)Ôºöhttps://pfam.xfam.orgÔºàFTPÔºöftp://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam32.0/Ôºâ</li>
<li>VOGÔºàÁâàÊú¨ 94ÔºâÔºöhttp://vogdb.org/ÔºàFTPÔºöhttp://fileshare.csb.univie.ac.at/vog/vog94/Ôºâ</li>
</ul>
<p>ÂÆâË£ÖÂíåÁ§∫‰æãÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-gdscript3" data-lang="gdscript3"><span class="line"><span class="cl"><span class="n">conda</span> <span class="n">create</span> <span class="o">-</span><span class="n">n</span> <span class="n">vibrant</span> <span class="n">python</span><span class="o">=</span><span class="mi">3</span> <span class="n">pandas</span> <span class="n">numpy</span><span class="o">=</span><span class="mf">1.19</span> <span class="n">scikit</span><span class="o">-</span><span class="n">learn</span><span class="o">=</span><span class="mf">0.21</span><span class="o">.</span><span class="mi">3</span> <span class="n">prodigal</span> <span class="n">hmmer</span> <span class="n">biopython</span> <span class="n">matplotlib</span> <span class="n">seaborn</span>
</span></span><span class="line"><span class="cl"><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">vibrant</span><span class="o">==</span><span class="mf">1.2</span><span class="o">.</span><span class="mi">1</span>
</span></span><span class="line"><span class="cl"><span class="n">download</span><span class="o">-</span><span class="n">db</span><span class="o">.</span><span class="n">sh</span> <span class="o">~/</span><span class="n">db</span><span class="o">/</span><span class="n">VIBRANT</span><span class="o">/</span> <span class="c1">#‰∏ãËΩΩÊï∞ÊçÆÂ∫ì</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="o">~/</span><span class="n">miniconda3</span><span class="o">/</span><span class="n">envs</span><span class="o">/</span><span class="n">vibrant</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">VIBRANT_run</span><span class="o">.</span><span class="n">py</span>
</span></span><span class="line"><span class="cl"><span class="o">~/</span><span class="n">miniconda3</span><span class="o">/</span><span class="n">envs</span><span class="o">/</span><span class="n">vibrant</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">VIBRANT_run</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="n">contig</span><span class="o">.</span><span class="n">fa</span> <span class="o">-</span><span class="n">t</span> <span class="mi">180</span> <span class="o">-</span><span class="n">folder</span> <span class="p">{</span><span class="n">out_dir</span><span class="p">}</span> <span class="o">-</span><span class="n">l</span> <span class="mi">1000</span> <span class="o">-</span><span class="n">d</span> <span class="o">~/</span><span class="n">db</span><span class="o">/</span><span class="n">VIBRANT</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span> <span class="o">-</span><span class="n">m</span> <span class="o">~/</span><span class="n">db</span><span class="o">/</span><span class="n">VIBRANT</span><span class="o">/</span><span class="n">files</span><span class="o">/</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="others">Others</h3>
<p>ÂÖ∂‰ªñËøòÊúâ‰∏Ä‰∫õÁ±ª‰ººÁöÑÂ∑•ÂÖ∑Ôºå‰ΩÜÊàëÊ≤°Êúâ‰∏Ä‰∏ÄÂÆâË£ÖÊµãËØïÔºåÂ¶ÇÔºö</p>
<ul>
<li>PPR-Meta ‰ΩøÁî®Êú∫Âô®Â≠¶‰π†ÊñπÊ≥ïÊù•È¢ÑÊµãÁóÖÊØíÂíåÂéüÊ†∏ÁîüÁâ©ÁöÑÂ∫èÂàó„ÄÇÈÄöËøáËÆ≠ÁªÉÊîØÊåÅÂêëÈáèÊú∫ÔºàSVMÔºâÊ®°ÂûãÔºåPPR-Meta ËÉΩÂ§üÈ´òÊïàÂú∞Âå∫ÂàÜÁóÖÊØíÂíåÈùûÁóÖÊØíÂ∫èÂàó„ÄÇËØ•Â∑•ÂÖ∑ÈÄÇÁî®‰∫éÂ§ßËßÑÊ®°ÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆÁöÑÂø´ÈÄüÂ§ÑÁêÜÂíåÂàÜÊûê„ÄÇ</li>
<li>viralVerify ÊòØ‰∏Ä‰∏™‰∏ìÈó®Áî®‰∫éÁóÖÊØíÂ∫èÂàóÈ™åËØÅÁöÑÂ∑•ÂÖ∑„ÄÇÂÆÉÁªìÂêà‰∫ÜÂ§öÁßçÂàÜÊûêÊñπÊ≥ïÔºåÂåÖÊã¨Â∫èÂàóÊØîÂØπ„ÄÅÁâπÂæÅÂàÜÊûêÂíåÊú∫Âô®Â≠¶‰π†ÔºåÊù•È™åËØÅÂíåÊ≥®ÈáäÁóÖÊØíÂ∫èÂàó„ÄÇviralVerify Êèê‰æõ‰∫ÜËØ¶ÁªÜÁöÑÊä•ÂëäÔºåÂåÖÊã¨ÁóÖÊØíÂ∫èÂàóÁöÑË¶ÜÁõñÂ∫¶„ÄÅÂü∫Âõ†ÁªÑÁªÑÊàêÂíåÂäüËÉΩÊ≥®Èáä„ÄÇ</li>
<li>MetaPhinder ‰ΩøÁî®Â∫èÂàóÊØîÂØπÊñπÊ≥ïÔºåÈÄöËøá‰∏éÁóÖÊØíÊï∞ÊçÆÂ∫ìÊØîÂØπÔºåËØÜÂà´ÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁöÑÁóÖÊØíÂ∫èÂàó„ÄÇËØ•Â∑•ÂÖ∑ÈÄÇÁî®‰∫éÂø´ÈÄüËØÜÂà´Â∑≤Áü•ÁóÖÊØíÂ∫èÂàóÔºåÂπ∂ËÉΩÊèê‰æõËØ¶ÁªÜÁöÑÂàÜÁ±ª‰ø°ÊÅØ„ÄÇ</li>
</ul>
<p>ÊúÄËøëËøòÁúãÂà∞‰∫Ü‰∏Ä‰∫õÊØîËæÉÊñ∞ÁöÑÂ∑•ÂÖ∑ÊñáÁ´†ÂèØ‰ª•Â∏ÆÂä©ÁóÖÊØíÂàÜÊûêÔºåÂ¶ÇÔºö</p>
<ul>
<li>
<p>COBRAÔºàContig Overlap Based Re-AssemblyÔºâÊòØ‰∏ÄÁßçÁîüÁâ©‰ø°ÊÅØÂ≠¶Â∑•ÂÖ∑[5]ÔºåÂèØ‰ªéÁü≠ÂèåÁ´ØËØªÈïøÁöÑÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁªÑË£ÖÂá∫Êõ¥È´òË¥®ÈáèÁöÑÁóÖÊØíÂü∫Âõ†ÁªÑ„ÄÇ COBRA ÊòØÁî® Python ÁºñÂÜôÁöÑ„ÄÇËøÑ‰ªä‰∏∫Ê≠¢ÔºåCOBRA ‰ªÖÂú®Êù•Ëá™ metaSPAdes„ÄÅIDBA_UD Âíå MEGAHIT ÁöÑÁªÑË£ÖcontigsÂíåÊîØÊû∂‰∏äËøõË°å‰∫ÜÊµãËØï„ÄÇ</p>
</li>
<li>
<p>bio-transformers ËØÅÊòéËõãÁôΩË¥®ËØ≠Ë®ÄÊ®°ÂûãÂèØ‰ª•ÊçïËé∑ÂéüÊ†∏ÁóÖÊØíËõãÁôΩË¥®ÂäüËÉΩ[6]Ôºå‰ªéËÄå‰ΩøÁóÖÊØíÂ∫èÂàóÁ©∫Èó¥ÁöÑÊñ∞ÈÉ®ÂàÜËÉΩÂ§üË¢´ÂàÜÈÖçÂÖ∑ÊúâÁîüÁâ©Â≠¶ÊÑè‰πâÁöÑÊ†áÁ≠æ„ÄÇËõãÁôΩË¥®ËØ≠Ë®ÄÊ®°ÂûãÂ¢ûÂº∫‰∫ÜÁóÖÊØíËõãÁôΩË¥®ÁöÑËøúÁ®ãÂêåÊ∫êÊÄßÊ£ÄÊµãÔºå‰Ωú‰∏∫Áé∞ÊúâÊñπÊ≥ïÁöÑÊúâÁî®Ë°•ÂÖÖ„ÄÇ</p>
</li>
</ul>
<h2 id="quality-assessment">Quality assessment</h2>
<h3 id="checkv">CheckV</h3>
<p>CheckV ÊòØ‰∏Ä‰∏™ÂÆåÂÖ®Ëá™Âä®ÂåñÁöÑÂëΩ‰ª§Ë°åÁÆ°ÈÅìÔºåÁî®‰∫éËØÑ‰º∞ÂçïcontigÁóÖÊØíÂü∫Âõ†ÁªÑÁöÑË¥®ÈáèÔºåÂåÖÊã¨ËØÜÂà´ÈõÜÊàêÂéüÁóÖÊØíÁöÑÂÆø‰∏ªÊ±°Êüì„ÄÅ‰º∞ËÆ°Âü∫Âõ†ÁªÑÁâáÊÆµÁöÑÂÆåÊï¥ÊÄß‰ª•ÂèäËØÜÂà´Â∞ÅÈó≠Âü∫Âõ†ÁªÑÔºå‰∫é2017Âπ¥ÂèëË°®Âú®Nature Biotechnology[7]„ÄÇ</p>
<p>ËΩØ‰ª∂Ôºö<a class="link" href="https://bitbucket.org/berkeleylab/checkv/src/master/"  target="_blank" rel="noopener"
    >https://bitbucket.org/berkeleylab/checkv/src/master/</a></p>
<h4 id="installation-3">Installation</h4>
<ul>
<li>ÂÆâË£ÖËΩØ‰ª∂</li>
</ul>
<p>ÂèØ‰ª•‰ΩøÁî®ÈÄöÁî®ÂåÖÁÆ°ÁêÜÂô®Ôºàmamba Êàñ condaÔºâÊàñÁâπÂÆö‰∫é Python ÁöÑÂåÖÁÆ°ÁêÜÂô®ÔºàpipÔºâÂú®ËÆ°ÁÆóÊú∫‰∏≠ÂÆâË£Ö checkv„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># mamba</span>
</span></span><span class="line"><span class="cl">mamba install -c conda-forge -c bioconda <span class="nv">checkv</span><span class="o">=</span>1.0.1
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># conda</span>
</span></span><span class="line"><span class="cl">conda install -c conda-forge -c bioconda <span class="nv">checkv</span><span class="o">=</span>1.0.1
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># pip</span>
</span></span><span class="line"><span class="cl">pip install checkv
</span></span></code></pre></td></tr></table>
</div>
</div><p>Âõ†‰∏∫Ë∑ü‰∏äÈù¢ËÆ≤ÁöÑgenomadÁöÑ‰æùËµñÈ°πÂü∫Êú¨‰∏ÄËá¥ÔºåÊâÄ‰ª•ÊàëÁõ¥Êé•Áî®pipÂÆâË£ÖÂú®Âêå‰∏ÄÁéØÂ¢É‰∏≠Â∞±ÂèØ‰ª•Áî®‰∫Ü„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ checkv end_to_end -h
</span></span><span class="line"><span class="cl">usage: checkv end_to_end &lt;input&gt; &lt;output&gt; <span class="o">[</span>options<span class="o">]</span>
</span></span><span class="line"><span class="cl">positional arguments:
</span></span><span class="line"><span class="cl">  input         ‰ª•FASTAÊ†ºÂºèËæìÂÖ•Ê†∏Ëã∑ÈÖ∏Â∫èÂàóÔºàÊîØÊåÅ.gzÔºå.bz2Âíå.xzÊñá‰ª∂Ôºâ
</span></span><span class="line"><span class="cl">  output        ËæìÂá∫ÁõÆÂΩï
</span></span><span class="line"><span class="cl"> 
</span></span><span class="line"><span class="cl">optional arguments:
</span></span><span class="line"><span class="cl">  -h, --help    ÊòæÁ§∫Ê≠§Â∏ÆÂä©Ê∂àÊÅØÂπ∂ÈÄÄÂá∫
</span></span><span class="line"><span class="cl">  -d PATH       ÂºïÁî®Êï∞ÊçÆÂ∫ìË∑ØÂæÑ„ÄÇÈªòËÆ§ÊÉÖÂÜµ‰∏ãÔºå‰ΩøÁî® CHECKVDB ÁéØÂ¢ÉÂèòÈáè„ÄêÊ∑ªÂä†ÂêéÂèØ‰∏çÂÜô„Äë
</span></span><span class="line"><span class="cl">  --remove_tmp  ‰ªéËæìÂá∫ÁõÆÂΩï‰∏≠Âà†Èô§‰∏≠Èó¥Êñá‰ª∂
</span></span><span class="line"><span class="cl">  -t INT        Áî®‰∫éProdigalÂíåDIAMONDÁöÑÁ∫øÁ®ãÊï∞
</span></span><span class="line"><span class="cl">  --restart     Ë¶ÜÁõñÁé∞ÊúâÁöÑ‰∏≠Èó¥Êñá‰ª∂„ÄÇÈªòËÆ§ÊÉÖÂÜµ‰∏ãÔºåCheckV Âú®Á®ãÂ∫è‰∏≠Êñ≠ÁöÑÂú∞ÊñπÁªßÁª≠
</span></span><span class="line"><span class="cl">  --quiet       Á¶ÅÊ≠¢ËÆ∞ÂΩïÊ∂àÊÅØ
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl">programs:
</span></span><span class="line"><span class="cl">    end_to_end          ËøêË°åÂÆåÊï¥ÁöÑÁÆ°ÈÅì‰ª•‰º∞ËÆ°ÂÆåÊï¥ÊÄßÔºåÊ±°ÊüìÂπ∂ËØÜÂà´Â∞ÅÈó≠ÁöÑÂü∫Âõ†ÁªÑ
</span></span><span class="line"><span class="cl">    contamination       ËØÜÂà´Âπ∂Ê∂àÈô§ÈõÜÊàêÂâçÁóÖÊØí‰∏äÁöÑÂÆø‰∏ªÊ±°Êüì
</span></span><span class="line"><span class="cl">    completeness        ‰º∞ËÆ°Âü∫Âõ†ÁªÑÁâáÊÆµÁöÑÂÆåÊï¥ÊÄß
</span></span><span class="line"><span class="cl">    complete_genomes    Ê†πÊçÆÊú´Á´ØÈáçÂ§çÂ∫èÂàóÂíå‰æßÁøºÂÆø‰∏ªÂå∫ÂüüËØÜÂà´ÂÆåÊï¥Âü∫Âõ†ÁªÑ
</span></span><span class="line"><span class="cl">    quality_summary     Ë∑®Ê®°ÂùóÊ±áÊÄªÁªìÊûú
</span></span><span class="line"><span class="cl">    download_database   ‰∏ãËΩΩÊúÄÊñ∞ÁâàÊú¨ÁöÑCheckVÊï∞ÊçÆÂ∫ì
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>‰∏ãËΩΩÊï∞ÊçÆÂ∫ì
geNomad ‰æùËµñ‰∫é‰∏Ä‰∏™ÂåÖÂê´Áî®‰∫éÂØπÂ∫èÂàóËøõË°åÂàÜÁ±ªÁöÑÊ†áËÆ∞ÁöÑÊ¶ÇÂÜµ„ÄÅÂÆÉ‰ª¨ÁöÑÂàÜÁ±ª‰ø°ÊÅØ„ÄÅÂÆÉ‰ª¨ÁöÑÂäüËÉΩÊ≥®ÈáäÁ≠âÁöÑÊï∞ÊçÆÂ∫ìÔºö</li>
</ul>
<p>ÂèØ‰ª•ÈÄöËøáËΩØ‰ª∂‰∏ãËΩΩÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">checkv download_database ./
</span></span></code></pre></td></tr></table>
</div>
</div><p>‰πüÂèØ‰ª•Ëá™Ë°å‰∏ãËΩΩÂπ∂Ëß£ÂéãÔºàÊàëËØï‰∫ÜËøô‰∏™Âø´ÔºâÔºåËß£ÂéãÂêéÂ§ßÊ¶Ç5.4GÔºåÊàëÊîæÂú®~/db/genomad_db/checkv-db-v1.0„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#Êï∞ÊçÆÂ∫ì‰∏ãËΩΩÔºàÊâãÂä®Ôºâ</span>
</span></span><span class="line"><span class="cl">wget https://portal.nersc.gov/CheckV/checkv-db-v1.0.tar.gz
</span></span><span class="line"><span class="cl">tar zxvf checkv-db-v1.0.tar.gz
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="example-3">Example</h4>
<p>Êàë‰ª¨‰ΩøÁî®‰∏äÈù¢genomadËøêË°åÂæóÂà∞ÁöÑGCF_009025895.1_ASM902589v1_genomic_virus.fnaÊñá‰ª∂Êù•ÊµãËØïÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">checkv end_to_end GCF_009025895.1_ASM902589v1_genomic_virus.fna checkv_out <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -t <span class="m">4</span> -d ~/db/genomad_db/checkv-db-v1.0 --remove_tmp
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">CheckV v1.0.3: contamination
</span></span><span class="line"><span class="cl"><span class="o">[</span>1/8<span class="o">]</span> Reading database info...
</span></span><span class="line"><span class="cl"><span class="o">[</span>2/8<span class="o">]</span> Reading genome info...
</span></span><span class="line"><span class="cl"><span class="o">[</span>3/8<span class="o">]</span> Skipping gene calling...
</span></span><span class="line"><span class="cl"><span class="o">[</span>4/8<span class="o">]</span> Reading gene info...
</span></span><span class="line"><span class="cl"><span class="o">[</span>5/8<span class="o">]</span> Skipping hmmsearch...
</span></span><span class="line"><span class="cl"><span class="o">[</span>6/8<span class="o">]</span> Annotating genes...
</span></span><span class="line"><span class="cl"><span class="o">[</span>7/8<span class="o">]</span> Identifying host regions...
</span></span><span class="line"><span class="cl"><span class="o">[</span>8/8<span class="o">]</span> Writing results...
</span></span><span class="line"><span class="cl">Run time: 0.77 seconds
</span></span><span class="line"><span class="cl">Peak mem: 0.08 GB
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">CheckV v1.0.3: completeness
</span></span><span class="line"><span class="cl"><span class="o">[</span>1/8<span class="o">]</span> Skipping gene calling...
</span></span><span class="line"><span class="cl"><span class="o">[</span>2/8<span class="o">]</span> Initializing queries and database...
</span></span><span class="line"><span class="cl"><span class="o">[</span>3/8<span class="o">]</span> Running DIAMOND blastp search...
</span></span><span class="line"><span class="cl"><span class="o">[</span>4/8<span class="o">]</span> Computing AAI...
</span></span><span class="line"><span class="cl"><span class="o">[</span>5/8<span class="o">]</span> Running AAI based completeness estimation...
</span></span><span class="line"><span class="cl"><span class="o">[</span>6/8<span class="o">]</span> Running HMM based completeness estimation...
</span></span><span class="line"><span class="cl"><span class="o">[</span>7/8<span class="o">]</span> Determining genome copy number...
</span></span><span class="line"><span class="cl"><span class="o">[</span>8/8<span class="o">]</span> Writing results...
</span></span><span class="line"><span class="cl">Run time: 9.69 seconds
</span></span><span class="line"><span class="cl">Peak mem: 1.12 GB
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">CheckV v1.0.3: complete_genomes
</span></span><span class="line"><span class="cl"><span class="o">[</span>1/7<span class="o">]</span> Reading input sequences...
</span></span><span class="line"><span class="cl"><span class="o">[</span>2/7<span class="o">]</span> Finding <span class="nb">complete</span> proviruses...
</span></span><span class="line"><span class="cl"><span class="o">[</span>3/7<span class="o">]</span> Finding direct/inverted terminal repeats...
</span></span><span class="line"><span class="cl"><span class="o">[</span>4/7<span class="o">]</span> Filtering terminal repeats...
</span></span><span class="line"><span class="cl"><span class="o">[</span>5/7<span class="o">]</span> Checking genome <span class="k">for</span> completeness...
</span></span><span class="line"><span class="cl"><span class="o">[</span>6/7<span class="o">]</span> Checking genome <span class="k">for</span> large duplications...
</span></span><span class="line"><span class="cl"><span class="o">[</span>7/7<span class="o">]</span> Writing results...
</span></span><span class="line"><span class="cl">Run time: 0.02 seconds
</span></span><span class="line"><span class="cl">Peak mem: 1.12 GB
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">CheckV v1.0.3: quality_summary
</span></span><span class="line"><span class="cl"><span class="o">[</span>1/6<span class="o">]</span> Reading input sequences...
</span></span><span class="line"><span class="cl"><span class="o">[</span>2/6<span class="o">]</span> Reading results from contamination module...
</span></span><span class="line"><span class="cl"><span class="o">[</span>3/6<span class="o">]</span> Reading results from completeness module...
</span></span><span class="line"><span class="cl"><span class="o">[</span>4/6<span class="o">]</span> Reading results from <span class="nb">complete</span> genomes module...
</span></span><span class="line"><span class="cl"><span class="o">[</span>5/6<span class="o">]</span> Classifying contigs into quality tiers...
</span></span><span class="line"><span class="cl"><span class="o">[</span>6/6<span class="o">]</span> Writing results...
</span></span><span class="line"><span class="cl">Run time: 2.42 seconds
</span></span><span class="line"><span class="cl">Peak mem: 1.12 GB
</span></span></code></pre></td></tr></table>
</div>
</div><p><strong>Output</strong></p>
<ul>
<li>quality_summary.tsv
ËøôÂåÖÂê´‰∏â‰∏™‰∏ªË¶ÅÊ®°ÂùóÁöÑÁªºÂêàÁªìÊûúÔºåÂ∫îËØ•ÊòØÊâÄÁî®ÁöÑ‰∏ªË¶ÅËæìÂá∫Ôºö</li>
</ul>
<div class="table-wrapper"><table>
  <thead>
      <tr>
          <th style="text-align: right">contig_id</th>
          <th style="text-align: right">contig_length</th>
          <th style="text-align: left">provirus</th>
          <th style="text-align: right">proviral_length</th>
          <th style="text-align: right">gene_count</th>
          <th style="text-align: right">viral_genes</th>
          <th style="text-align: right">host_genes</th>
          <th style="text-align: left">checkv_quality</th>
          <th style="text-align: left">miuvig_quality</th>
          <th style="text-align: right">completeness</th>
          <th style="text-align: left">completeness_method</th>
          <th style="text-align: left">complete_genome_type</th>
          <th style="text-align: right">contamination</th>
          <th style="text-align: right">kmer_freq</th>
          <th style="text-align: left">warnings</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: right">1</td>
          <td style="text-align: right">5325</td>
          <td style="text-align: left">No</td>
          <td style="text-align: right">NA</td>
          <td style="text-align: right">11</td>
          <td style="text-align: right">0</td>
          <td style="text-align: right">2</td>
          <td style="text-align: left">Not-determined</td>
          <td style="text-align: left">Genome-fragment</td>
          <td style="text-align: right">NA</td>
          <td style="text-align: left">NA</td>
          <td style="text-align: left">NA</td>
          <td style="text-align: right">0.0</td>
          <td style="text-align: right">1.00</td>
          <td style="text-align: left">no viral genes detected</td>
      </tr>
      <tr>
          <td style="text-align: right">2</td>
          <td style="text-align: right">41803</td>
          <td style="text-align: left">No</td>
          <td style="text-align: right">NA</td>
          <td style="text-align: right">72</td>
          <td style="text-align: right">27</td>
          <td style="text-align: right">1</td>
          <td style="text-align: left">Low-quality</td>
          <td style="text-align: left">Genome-fragment</td>
          <td style="text-align: right">21.99</td>
          <td style="text-align: left">AAI-based (medium-confidence)</td>
          <td style="text-align: left">NA</td>
          <td style="text-align: right">0.0</td>
          <td style="text-align: right">1.00</td>
          <td style="text-align: left">flagged DTR</td>
      </tr>
      <tr>
          <td style="text-align: right">3</td>
          <td style="text-align: right">38254</td>
          <td style="text-align: left">Yes</td>
          <td style="text-align: right">36072</td>
          <td style="text-align: right">54</td>
          <td style="text-align: right">23</td>
          <td style="text-align: right">2</td>
          <td style="text-align: left">Medium-quality</td>
          <td style="text-align: left">Genome-fragment</td>
          <td style="text-align: right">80.30</td>
          <td style="text-align: left">HMM-based (lower-bound)</td>
          <td style="text-align: left">NA</td>
          <td style="text-align: right">5.7</td>
          <td style="text-align: right">1.00</td>
          <td style="text-align: left">NA</td>
      </tr>
      <tr>
          <td style="text-align: right">4</td>
          <td style="text-align: right">67622</td>
          <td style="text-align: left">No</td>
          <td style="text-align: right">NA</td>
          <td style="text-align: right">143</td>
          <td style="text-align: right">25</td>
          <td style="text-align: right">0</td>
          <td style="text-align: left">High-quality</td>
          <td style="text-align: left">High-quality</td>
          <td style="text-align: right">100.00</td>
          <td style="text-align: left">AAI-based (high-confidence)</td>
          <td style="text-align: left">NA</td>
          <td style="text-align: right">0.0</td>
          <td style="text-align: right">1.76</td>
          <td style="text-align: left">high kmer_freq</td>
      </tr>
      <tr>
          <td style="text-align: right">5</td>
          <td style="text-align: right">98051</td>
          <td style="text-align: left">No</td>
          <td style="text-align: right">NA</td>
          <td style="text-align: right">158</td>
          <td style="text-align: right">27</td>
          <td style="text-align: right">1</td>
          <td style="text-align: left">Complete</td>
          <td style="text-align: left">High-quality</td>
          <td style="text-align: right">100.00</td>
          <td style="text-align: left">AAI-based (high-confidence)</td>
          <td style="text-align: left">DTR</td>
          <td style="text-align: right">0.0</td>
          <td style="text-align: right">1.00</td>
          <td style="text-align: left">NA</td>
      </tr>
  </tbody>
</table></div>
<p>Âú®‰∏äÈù¢ÁöÑÁ§∫‰æã‰∏≠ÔºåÊúâ 5 ‰∏™ÁóÖÊØícontigÁöÑÁªìÊûúÔºö</p>
<ul>
<li>
<p>Á¨¨‰∏Ä‰∏™ 5325 bp contigÊ≤°ÊúâÂÆåÊï¥ÊÄßÈ¢ÑÊµãÔºåËøôÁî±‚Äúcheckv_quality‚ÄùÂ≠óÊÆµÁöÑ‚ÄúÊú™Á°ÆÂÆö‚ÄùÊåáÁ§∫„ÄÇËØ•contig‰πüÊ≤°ÊúâÈâ¥ÂÆöÂá∫ÁóÖÊØíÂü∫Âõ†ÔºåÂõ†Ê≠§ÂÆÉÁîöËá≥ÊúâÂèØËÉΩ‰∏çÊòØÁóÖÊØí„ÄÇ</p>
</li>
<li>
<p>Á¨¨‰∫å‰∏™ 41803 bp contigË¢´ÂΩíÁ±ª‰∏∫‚Äú‰ΩéË¥®Èáè‚ÄùÔºåÂõ†‰∏∫ÂÖ∂ÂÆåÊï¥ÊÄß&lt;50%„ÄÇËøôÊòØÂü∫‰∫é‚ÄúAAI‚ÄùÊñπÊ≥ïÁöÑ‰º∞ËÆ°„ÄÇËØ∑Ê≥®ÊÑèÔºåquality_summary.tsv Êñá‰ª∂‰∏≠‰ªÖÊä•ÂëäÈ´òÊàñ‰∏≠ÁΩÆ‰ø°Â∫¶‰º∞ËÆ°ÂÄº„ÄÇÊÇ®ÂèØ‰ª•Êü•Áúã‚Äúcompleteness.tsv‚Äù‰∫ÜËß£Êõ¥Â§öËØ¶ÁªÜ‰ø°ÊÅØ„ÄÇËØ•contigÂÖ∑Êúâ DTRÔºå‰ΩÜÁî±‰∫éÊüêÁßçÂéüÂõ†Ë¢´Ê†áËÆ∞ÔºàÊúâÂÖ≥ËØ¶ÁªÜ‰ø°ÊÅØÔºåËØ∑ÂèÇÈòÖcomplete_genomes.tsvÔºâ</p>
</li>
<li>
<p>Á¨¨‰∏â‰∏™contigË¢´ËÆ§‰∏∫ÊòØ‚Äú‰∏≠Á≠âË¥®Èáè‚ÄùÔºåÂõ†‰∏∫ÂÖ∂ÂÆåÊï¥ÊÄß‰º∞ËÆ°‰∏∫ 80%ÔºàÂü∫‰∫é‚ÄúHMM‚ÄùÊñπÊ≥ïÔºâ„ÄÇËøôÊÑèÂë≥ÁùÄÂÆÉÂ§™Êñ∞È¢ñ‰∫ÜÔºåÊó†Ê≥ïÂü∫‰∫é AAI Êù•‰º∞ËÆ°ÂÆåÊï¥ÊÄßÔºå‰ΩÜ‰∏é CheckV ÂèÇËÄÉÂü∫Âõ†ÁªÑÂÖ±‰∫´ HMM„ÄÇËØ∑Ê≥®ÊÑèÔºåËØ•ÂÄº‰ª£Ë°®‰∏Ä‰∏™‰∏ãÈôêÔºàÊÑèÂë≥ÁùÄÁúüÂÆûÁöÑÂÆåÊï¥ÊÄßÂèØËÉΩÈ´ò‰∫é‰ΩÜ‰∏ç‰Ωé‰∫éËØ•ÂÄºÔºâ„ÄÇËØ∑Ê≥®ÊÑèÔºåËØ•contig‰πüË¢´ÂΩíÁ±ª‰∏∫ÂéüÁóÖÊØí„ÄÇ</p>
</li>
<li>
<p>Á¨¨Âõõ‰∏™contigÂü∫‰∫é &gt;90% ÁöÑÂÆåÊï¥ÊÄßË¢´ÂΩíÁ±ª‰∏∫È´òË¥®Èáè„ÄÇ‰ΩÜËØ∑Ê≥®ÊÑèÔºå‚Äúkmer_freq‚ÄùÁöÑÂÄº‰∏∫ 1.7„ÄÇËøôË°®ÊòéÁóÖÊØíÂü∫Âõ†ÁªÑÂú®contig‰∏≠Â§öÊ¨°Âá∫Áé∞„ÄÇËøô‰∫õÊÉÖÂÜµÁõ∏ÂΩìÁΩïËßÅÔºå‰ΩÜ‰ªçÈúÄË≠¶ÊÉï„ÄÇ</p>
</li>
<li>
<p>Ê†πÊçÆÁõ¥Êé•Êú´Á´ØÈáçÂ§ç (DTR) ÁöÑÂ≠òÂú®ÔºåÁ¨¨‰∫î‰∏™contigË¢´ÂàÜÁ±ª‰∏∫ÂÆåÊï¥ÔºåÂπ∂‰∏îÊ†πÊçÆ AAI ÊñπÊ≥ïÂÖ∑Êúâ 100% ÂÆåÊï¥ÊÄß„ÄÇËØ•Â∫èÂàóÂèØ‰ª•ÊîæÂøÉÂú∞ËßÜ‰∏∫ÂÆåÊï¥ÁöÑÂü∫Âõ†ÁªÑ„ÄÇ</p>
</li>
<li>
<p>contamination.tsv
Â¶Ç‰Ωï‰º∞ËÆ°Ê±°ÊüìÁöÑËØ¶ÁªÜÊ¶ÇËø∞</p>
</li>
<li>
<p>completeness.tsv
ÂÖ≥‰∫éÂ¶Ç‰Ωï‰º∞ËÆ°ÂÆåÊï¥ÊÄßÁöÑËØ¶ÁªÜÊ¶ÇËø∞</p>
</li>
<li>
<p>complete_genomes.tsv
Â∑≤Á°ÆÂÆöÁöÑÂÅáÂÆöÂÆåÊï¥Âü∫Âõ†ÁªÑÁöÑËØ¶ÁªÜÊ¶ÇËø∞</p>
</li>
</ul>
<h4 id="pipeline-1">Pipeline</h4>
<img src="images/checkV.png" title=""/>
<p>CheckVÁöÑÂÖ∑‰ΩìÂ∑•‰ΩúÊµÅÁ®ãÂ¶Ç‰∏ãÔºö</p>
<p>A: ÂéªÈô§ÂÆø‰∏ªÊ±°Êüì</p>
<ol>
<li>Âü∫Âõ†È¶ñÂÖàÂü∫‰∫é‰∏éËá™ÂÆö‰πâHMMÊï∞ÊçÆÂ∫ìÁöÑÊØîËæÉË¢´Ê≥®Èáä‰∏∫ÁóÖÊØíÊàñÂæÆÁîüÁâ©„ÄÇ</li>
<li>CheckVÊâ´ÊèècontigÔºà‰ªé5&rsquo;Âà∞3&rsquo;ÔºâÔºåÊØîËæÉÂü∫Âõ†Ê≥®ÈáäÂíåÁõ∏ÈÇªÂü∫Âõ†Á™óÂè£‰πãÈó¥ÁöÑGCÂê´Èáè„ÄÇ</li>
<li>Ëøô‰∫õ‰ø°ÊÅØÁî®‰∫éËÆ°ÁÆóÊØè‰∏™Âü∫Âõ†Èó¥‰ΩçÁΩÆÁöÑÂàÜÊï∞Âπ∂ËØÜÂà´ÂÆø‰∏ª-ÁóÖÊØíÊñ≠ÁÇπ„ÄÇ</li>
<li>ËøôÁßçÊñπÊ≥ïÊúÄÈÄÇÂêà‰∏ªË¶Å‰∏∫ÁóÖÊØíÁöÑcontig„ÄÇ</li>
</ol>
<p>B: ‰º∞ËÆ°Âü∫Âõ†ÁªÑÂÆåÊï¥ÊÄß</p>
<ol>
<li>ËõãÁôΩË¥®È¶ñÂÖà‰ΩøÁî®AAIÔºàÂπ≥ÂùáÊ∞®Âü∫ÈÖ∏Ë∫´‰ªΩÔºâ‰∏éCheckVÂü∫Âõ†ÁªÑÊï∞ÊçÆÂ∫ìËøõË°åÊØîËæÉ„ÄÇ</li>
<li>Âú®ËØÜÂà´Âà∞ÊúÄ‰Ω≥ÂåπÈÖçÂêéÔºåÂÆåÊï¥ÊÄß‰Ωú‰∏∫contigÈïøÂ∫¶ÔºàÊàñprovirusÁöÑÁóÖÊØíÂå∫ÂüüÈïøÂ∫¶Ôºâ‰∏éÂåπÈÖçÂèÇËÄÉÈïøÂ∫¶ÁöÑÊØîÁéáËÆ°ÁÆó„ÄÇ</li>
<li>Ê†πÊçÆÊØîÂØπÁöÑÂº∫Â∫¶Êä•ÂëäÁΩÆ‰ø°Ê∞¥Âπ≥„ÄÇ</li>
<li>ÈÄöÂ∏∏ÔºåÈ´òÂíå‰∏≠Á≠âÁΩÆ‰ø°Ê∞¥Âπ≥ÁöÑ‰º∞ËÆ°ÈùûÂ∏∏ÂáÜÁ°Æ„ÄÇ</li>
<li>Â¶ÇÊûúÁóÖÊØíÂü∫Âõ†ÁªÑÊ≤°Êúâ‰∏éCheckVÊï∞ÊçÆÂ∫ìÁöÑÂØÜÂàáÂåπÈÖçÔºåCheckVÂü∫‰∫écontig‰∏äËØÜÂà´ÁöÑÁóÖÊØíHMM‰º∞ËÆ°ÂÆåÊï¥ÊÄß„ÄÇ</li>
<li>Âü∫‰∫éÊâæÂà∞ÁöÑHMMÔºåCheckVËøîÂõûÂü∫Âõ†ÁªÑÂÆåÊï¥ÊÄßÁöÑ‰º∞ËÆ°ËåÉÂõ¥Ôºà‰æãÂ¶ÇÔºå35%Âà∞60%ÂÆåÊï¥ÊÄßÔºâÔºåË°®Á§∫Âü∫‰∫éÂÖ∑ÊúâÁõ∏ÂêåÁóÖÊØíHMMÁöÑÂèÇËÄÉÂü∫Âõ†ÁªÑÈïøÂ∫¶ÂàÜÂ∏ÉÁöÑ90%ÁΩÆ‰ø°Âå∫Èó¥„ÄÇ</li>
</ol>
<p>C: È¢ÑÊµãÂ∞ÅÈó≠Âü∫Âõ†ÁªÑ</p>
<ol>
<li>Áõ¥Êé•Êú´Á´ØÈáçÂ§çÔºàDTRsÔºâÔºöÂú®contigÁöÑËµ∑ÂßãÂíåÁªìÊùüÂ§ÑÊúâ&gt;20bpÁöÑÈáçÂ§çÂ∫èÂàóÔºåËøôÊòØÊúÄÂèØ‰ø°ÁöÑÊ†áÂøóÔºåÂèØËÉΩË°®Á§∫ÁéØÁä∂Âü∫Âõ†ÁªÑÊàñ‰ªéÁéØÁä∂Ê®°ÊùøÂ§çÂà∂ÁöÑÁ∫øÊÄßÂü∫Âõ†ÁªÑ„ÄÇ</li>
<li>ProvirusÔºöÂú®5&rsquo;Âíå3&rsquo;Êú´Á´ØÊúâÈ¢ÑÊµãÁöÑÂÆø‰∏ªËæπÁïåÁöÑÁóÖÊØíÂå∫ÂüüÔºàÂèÇËßÅÈÉ®ÂàÜAÔºâ„ÄÇ
<ul>
<li>Ê≥®ÊÑèÔºöÂ¶ÇÊûúÂÆø‰∏ªÂå∫ÂüüÂ∑≤Ë¢´ÁßªÈô§ÔºàÂ¶Ç‰ΩøÁî®VIBRANTÊàñVirSorterÔºâÔºåCheckVÂ∞Ü‰∏ç‰ºöÊ£ÄÊµãÂà∞provirus„ÄÇ</li>
</ul>
</li>
<li>ÂÄí‰ΩçÊú´Á´ØÈáçÂ§çÔºàITRsÔºâÔºöÂú®contigÁöÑËµ∑ÂßãÂíåÁªìÊùüÂ§ÑÊúâ&gt;20bpÁöÑÈáçÂ§çÂ∫èÂàóÔºà3&rsquo;ÈáçÂ§çÂÄíÁΩÆÔºâÔºåËøôÊòØÊúÄ‰∏çÂèØ‰ø°ÁöÑÊ†áÂøó„ÄÇ</li>
<li>ÂØπ‰∫é‰∏äËø∞ÊâÄÊúâÊñπÊ≥ïÔºåCheckVËøò‰ºöÊ£ÄÊü•contigÊòØÂê¶Â§ßËá¥Á¨¶ÂêàÂü∫‰∫é‰º∞ËÆ°ÂÆåÊï¥ÊÄßÁöÑÊ≠£Á°ÆÂ∫èÂàóÈïøÂ∫¶ÔºõËøôÂæàÈáçË¶ÅÔºåÂõ†‰∏∫Êú´Á´ØÈáçÂ§çÂèØËÉΩ‰ª£Ë°®ÂÆèÂü∫Âõ†ÁªÑÁªÑË£ÖÁöÑ‰º™ÂΩ±„ÄÇ</li>
</ol>
<p>D: ÊÄªÁªìË¥®Èáè</p>
<ol>
<li>Âü∫‰∫éA-CÁöÑÁªìÊûúÔºåCheckVÁîüÊàêÊä•ÂëäÊñá‰ª∂ÔºåÂπ∂Â∞ÜÊü•ËØ¢contigÂàÜÈÖçÂà∞‰∫î‰∏™Ë¥®ÈáèÂ±ÇÊ¨°‰πã‰∏ÄÔºà‰∏éMIUViGË¥®ÈáèÂ±ÇÊ¨°‰∏ÄËá¥Âπ∂Êâ©Â±ïÔºâÔºö
<ul>
<li>ÂÆåÊï¥ÔºàÂèÇËßÅÈÉ®ÂàÜCÔºâ</li>
<li>È´òË¥®ÈáèÔºà&gt;90%ÂÆåÊï¥ÊÄßÔºâ</li>
<li>‰∏≠Á≠âË¥®ÈáèÔºà50-90%ÂÆåÊï¥ÊÄßÔºâ</li>
<li>‰ΩéË¥®ÈáèÔºà&lt;50%ÂÆåÊï¥ÊÄßÔºâ</li>
<li>Êú™Á°ÆÂÆöË¥®Èáè</li>
</ul>
</li>
</ol>
<p>ÂèØ‰ª•ÂàÜÊ≠•‰ΩøÁî®checkVÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">checkv contamination input_file.fna output_directory -t <span class="m">16</span>
</span></span><span class="line"><span class="cl">checkv completeness input_file.fna output_directory -t <span class="m">16</span>
</span></span><span class="line"><span class="cl">checkv complete_genomes input_file.fna output_directory
</span></span><span class="line"><span class="cl">checkv quality_summary input_file.fna output_directory
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="application">Application</h2>
<p>Ëøô‰∫õ‰∏ìÈó®ÁöÑÁóÖÊØíÈâ¥ÂÆöÂ∑•ÂÖ∑ÂêÑÊúâ‰ºòÂä£ÔºåÁ†îÁ©∂ËÄÖÂèØ‰ª•Ê†πÊçÆÂÖ∑‰ΩìÁöÑÁ†îÁ©∂ÈúÄÊ±ÇÂíåÊï∞ÊçÆÁâπÂæÅÈÄâÊã©ÂêàÈÄÇÁöÑÂ∑•ÂÖ∑„ÄÇÈÄöËøáÁªìÂêàÂ§öÁßçÊñπÊ≥ïÔºåÂèØ‰ª•ÊèêÈ´òÁóÖÊØíÂ∫èÂàóÈâ¥ÂÆöÁöÑÂáÜÁ°ÆÊÄßÂíåÊïàÁéáÔºå‰∏∫Ê∑±ÂÖ•ÁêÜËß£ÁóÖÊØíÂú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑËßíËâ≤ÂíåÂäüËÉΩÊèê‰æõÊúâÂäõÊîØÊåÅ„ÄÇ</p>
<h3 id="biogeographic-patterns-and-drivers-of-soil-viromes">Biogeographic patterns and drivers of soil viromes</h3>
<img src="images/soil_virome.png" title=""/>
<p>ËøôÊòØÈ©¨ÊñåËÄÅÂ∏àÂõ¢Èòü2024ÂèëË°®‰∫éNature Ecology &amp; EvolutionÁöÑ‰∏ÄÁØáÊñáÁ´†[8]„ÄÇ</p>
<p>ÁóÖÊØíÂØπ‰∫éÂ°ëÈÄ†ÂúüÂ£§ÂæÆÁîüÁâ©ÂäüËÉΩÂíåÁîüÊÄÅÁ≥ªÁªüËá≥ÂÖ≥ÈáçË¶Å„ÄÇÁÑ∂ËÄåÔºåÂØπÂúüÂ£§ÁóÖÊØíÁªÑÁöÑÁ†îÁ©∂Âú®Á©∫Èó¥Â∞∫Â∫¶ÂíåÁîüÁâ©Áæ§ËêΩË¶ÜÁõñÊñπÈù¢ÈÉΩÂèóÂà∞ÈôêÂà∂„ÄÇ</p>
<p>Âú®ËøôÈáåÔºå‰ΩøÁî®ÂÖ®ÁêÉÂúüÂ£§ÁóÖÊØíÁªÑÊï∞ÊçÆÈõÜ (GSV) ÂÖ®Èù¢ÁªºÂêà‰∫ÜÂúüÂ£§ÁóÖÊØíÁªÑÁîüÁâ©Âú∞ÁêÜÊ®°ÂºèÔºåÂàÜÊûê‰∫ÜÂÖ®ÁêÉ 1,824 ‰∏™ÂúüÂ£§ÂÆèÂü∫Âõ†ÁªÑÔºåÂèëÁé∞‰∫Ü 80,750 ‰∏™ DNA ÁóÖÊØíÁöÑÈÉ®ÂàÜÂü∫Âõ†ÁªÑÔºåÂÖ∂‰∏≠ 96.7% Êú™ÂàÜÁ±ª„ÄÇ
‰∏çÂêåÁîüÁâ©Áæ§ËêΩÁöÑÂúüÂ£§ÁóÖÊØíÂ§öÊ†∑ÊÄßÂíåÁæ§ËêΩÁªìÊûÑÁöÑÁîüÁâ©Âú∞ÁêÜÂ≠¶ÂêÑ‰∏çÁõ∏Âêå„ÄÇÊúâË∂£ÁöÑÊòØÔºåÁóÖÊØíÁöÑÂ§öÊ†∑ÊÄß‰∏éÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄßÂπ∂‰∏çÁõ∏Á¨¶ÔºåÁõ∏ÂèçÔºåÊ£ÆÊûóÂíåÁÅåÊú®‰∏õÂúüÂ£§‰∏≠ÁöÑÁóÖÊØíÂ§öÊ†∑ÊÄßËæÉ‰Ωé„ÄÇ</p>
<p>È¢ÑÊµãÁöÑÂúüÂ£§ÁóÖÊØíÂ§öÊ†∑ÊÄßÂõæË∞±Ëøõ‰∏ÄÊ≠•ËØÅÂÆû‰∫ÜÂúüÂ£§Ë¥®Âú∞ÂíåÊ∞¥ÂàÜÊù°‰ª∂ÊòØÂΩ±ÂìçÂ§öÊ†∑ÊÄßÁöÑÂÖ≥ÈîÆÂõ†Á¥†ÔºåÊè≠Á§∫‰∫ÜÊΩÆÊπøÂíåÂçäÊπøÊ∂¶Âú∞Âå∫Êõ¥È´òÁöÑÂ§öÊ†∑ÊÄß„ÄÇÊ≠§Â§ñÔºå‰∫åÈ°πÂºèÂ∫¶ÂàÜÂ∏ÉÊ®°ÂºèË°®ÊòéÂúüÂ£§ÁóÖÊØíÁöÑÈöèÊú∫ÂÖ±Áé∞Ê®°Âºè„ÄÇËøô‰∫õÂèëÁé∞ÂØπ‰∫éÈòêÊòéÂúüÂ£§ÁóÖÊØíÁîüÊÄÅÂ≠¶‰ª•ÂèäÂ∞ÜÁóÖÊØíÂÖ®Èù¢Á∫≥ÂÖ•ÂúüÂ£§ÁîüÊÄÅÁ≥ªÁªüÊ®°ÂûãËá≥ÂÖ≥ÈáçË¶Å„ÄÇ</p>
<p>‰ªñ‰ª¨Âú®ËøôÁØáÊñáÁ´†ÈáåÁî®ÁöÑÈâ¥ÂÆöÁóÖÊØíÁöÑÊñπÊ≥ïÊòØÔºö</p>
<p>ÁªÑË£ÖÁöÑcontigsÈÄöËøá VirSorter (v.1.0.5)ÊØîÂØπ NCBI ÁóÖÊØíÂèÇËÄÉÂ∫èÂàóÊï∞ÊçÆÂ∫ì (Refseq) (<a class="link" href="https://www.ncbi.nlm.nih.gov/genome/viruses/"  target="_blank" rel="noopener"
    >www.ncbi.nlm.nih.gov/genome/viruses/</a>) Ôºõ‰ΩøÁî®VIBRANT (v.1.2.1) Âíå DeepVirFinder (v.1.0)ÔºåÊà™Ê≠¢ÈïøÂ∫¶‰∏∫ 1,000bp„ÄÇÊ≥®Èáä‰∏∫ VirSorter Á±ªÂà´ 1„ÄÅ2„ÄÅ4 Âíå 5 ÁöÑcontigsÔºåÊàñ DeepVirFinder ËØÑÂàÜ‚â•0.7 ‰∏î P &lt; 0.05 ÁöÑcontigsÔºåË¢´ËßÜ‰∏∫ÂÅáÂÆöÁöÑÁóÖÊØícontigs„ÄÇÈÄöËøáÂ∞ÜÊï∞ÊçÆ‰∏é VIBRANT ÁöÑÊï∞ÊçÆÁõ∏ÁªìÂêàÔºåCAT (v.5.0.3) Áî®‰∫éËøõ‰∏ÄÊ≠•‰º∞ËÆ°Êåâ VirSorter Á±ªÂà´ 3 Âíå 6 ÊéíÂ∫èÁöÑcontigsÔºåÊàñ DeepVirFinder ÂæóÂàÜ‰∏∫ 0.7-0.9 ‰∏î P&lt; 0.05 ÁöÑcontigs„ÄÇÂÖ∑Êúâ &gt;40% ÂºÄÊîæÈòÖËØªÊ°ÜÊ≥®Èáä‰∏∫ÁªÜËèå„ÄÅÂè§ÁªÜËèåÊàñÁúüÊ†∏ÁîüÁâ©ÁöÑcontigsË¢´ËßÜ‰∏∫ÈùûÁóÖÊØícontigs„ÄÇÂèñÂá∫‚â•5 kbÊàñ‚â•1.5 kbÁöÑcontigsÂíåÁéØÁä∂ÁâáÊÆµËøõË°åËøõ‰∏ÄÊ≠•Á†îÁ©∂ÔºõÈÄöËøá BLAST (v.2.11.0) Èâ¥ÂÆöÁöÑ Œ¶x174 DNA Ë¢´ÊâãÂä®ÂéªÈô§„ÄÇ</p>
<h3 id="a-global-atlas-of-soil-viruses">A global atlas of soil viruses</h3>
<img src="images/soil_atlats.png" title=""/>
<p>Ëøô‰πüÊòØ‰∏ÄÁØá2024Âπ¥ÂèëË°®‰∫éNature microbiologyÁöÑÊñáÁ´†[9]Ôºö</p>
<p>ÂéÜÂè≤‰∏äË¢´ÂæÆÁîüÁâ©ÁîüÊÄÅÂ≠¶ÂÆ∂ÂøΩËßÜÁöÑÂúüÂ£§ÁóÖÊØíÁé∞Âú®Ë¢´ËÆ§‰∏∫ÂØπÂÖ®ÁêÉÁîüÁâ©Âú∞ÁêÉÂåñÂ≠¶Âæ™ÁéØËá≥ÂÖ≥ÈáçË¶Å„ÄÇÁÑ∂ËÄåÔºåÊàë‰ª¨ÂØπÂÖ∂ÂÖ®ÁêÉÂàÜÂ∏É„ÄÅÊ¥ªÂä®‰ª•Âèä‰∏éÂúüÂ£§ÂæÆÁîüÁâ©ÁªÑÁõ∏‰∫í‰ΩúÁî®ÁöÑ‰∫ÜËß£‰ªçÁÑ∂ÊúâÈôê„ÄÇ
Âú®ËøôÈáåÂ±ïÁ§∫‰∫ÜÂÖ®ÁêÉÂúüÂ£§ÁóÖÊØíÂõæË∞±ÔºåËøôÊòØ‰∏Ä‰∏™ÁªºÂêàÊï∞ÊçÆÈõÜÔºåÁî± 2,953 ‰∏™ÂÖàÂâçÊµãÂ∫èÁöÑÂúüÂ£§ÂÆèÂü∫Âõ†ÁªÑÁºñËØëËÄåÊàêÔºåÁî± 616,935 ‰∏™Êú™ÂüπÂÖªÁöÑÁóÖÊØíÂü∫Âõ†ÁªÑÂíå 38,508 ‰∏™Áã¨ÁâπÁöÑÁóÖÊØíÊìç‰ΩúÂàÜÁ±ªÂçïÂÖÉÁªÑÊàê„ÄÇ
ÂÖ®ÁêÉÂúüÂ£§ÁóÖÊØíÂõæË∞±ÁöÑÁ®ÄÁñèÊõ≤Á∫øË°®ÊòéÔºåÂ§ßÂ§öÊï∞ÂúüÂ£§ÁóÖÊØíÂ§öÊ†∑ÊÄß‰ªçÊú™Ë¢´Êé¢Á¥¢ÔºåÊ†∑Êú¨Èó¥È´òÁ©∫Èó¥Âë®ËΩ¨ÁéáÂíå‰ΩéÂÖ±‰∫´ÁóÖÊØíÊìç‰ΩúÂàÜÁ±ªÂçïÂÖÉÁéáËøõ‰∏ÄÊ≠•Âº∫Ë∞É‰∫ÜËøô‰∏ÄÁÇπ„ÄÇ
ÈÄöËøáÊ£ÄÊü•‰∏éÁîüÁâ©Âú∞ÁêÉÂåñÂ≠¶ÂäüËÉΩÁõ∏ÂÖ≥ÁöÑÂü∫Âõ†ÔºåËøòËØÅÊòé‰∫ÜÁóÖÊØíÂΩ±ÂìçÂúüÂ£§Á¢≥ÂíåÂÖªÂàÜÂæ™ÁéØÁöÑÊΩúÂäõ„ÄÇËøôÈ°πÁ†îÁ©∂‰ª£Ë°®‰∫ÜÂúüÂ£§ÁóÖÊØíÂ§öÊ†∑ÊÄßÁöÑÂπøÊ≥õÁâπÂæÅÔºåÂπ∂‰∏∫Âà∂ÂÆöÊúâÂÖ≥ÁóÖÊØíÂúàÂú®ÂúüÂ£§ÂæÆÁîüÁâ©ÁªÑÂíåÂÖ®ÁêÉÁîüÁâ©Âú∞ÁêÉÂåñÂ≠¶‰∏≠ÁöÑ‰ΩúÁî®ÁöÑÂèØÊ£ÄÈ™åÂÅáËÆæÂ•†ÂÆö‰∫ÜÂü∫Á°Ä„ÄÇ</p>
<p>ËøôÁØáÊñáÁ´†‰∏≠Èâ¥ÂÆöÁóÖÊØíÁöÑÊñπÊ≥ïÊòØÔºö</p>
<p>‰ΩøÁî® IMG/VR v3 ÁóÖÊØíËØÜÂà´ÁÆ°ÈÅìÁöÑ‰øÆÊîπÁâàÊú¨ÂØπÁóÖÊØícontigsËøõË°å‰∫ÜÂàùÊ≠•ËØÜÂà´„ÄÇËØ•ÁÆ°ÈÅìÊ†πÊçÆ Pfam Êï∞ÊçÆÂ∫ì Âíå VirFinder ‰∏≠ 23,841 ‰∏™ÁóÖÊØíËõãÁôΩÂÆ∂Êóè„ÄÅ16,260 ‰∏™ÂæÆÁîüÁâ©Êù•Ê∫êËõãÁôΩÂÆ∂ÊóèÁöÑÂ≠òÂú®Êù•ËØÜÂà´ÁóÖÊØíÔºå‰ª•ËØÜÂà´Ëá≥Â∞ë 1 kb ÈïøÁöÑcontigs‰∏≠ÁöÑÂÅáÂÆöÁóÖÊØíÂü∫Âõ†ÁªÑ„ÄÇ</p>
<p>Âú®Êú¨Á†îÁ©∂ËøáÁ®ã‰∏≠ÔºåÂèëÂ∏É‰∫Ü geNomad v1.3.3ÔºåËøôÊòØ‰∏ÄÁßçÂÖ∑ÊúâÊîπËøõÁöÑÂàÜÁ±ªÊÄßËÉΩÁöÑÁóÖÊØíËØÜÂà´Â∑•ÂÖ∑ÔºåÂπ∂Â∞ÜÂÖ∂Á∫≥ÂÖ•ÊµÅÁ®ã‰∏≠Ôºå‰ª•ÊèêÈ´òÈ¢ÑÊµãÁΩÆ‰ø°Â∫¶Âπ∂ÊâßË°åÂàÜÁ±ªÂàÜÈÖç„ÄÇ
‰ΩøÁî® CheckVv1.0.1ÔºàÊï∞ÊçÆÂ∫ìÁâàÊú¨ 1.5Ôºâ Ëøõ‰∏ÄÊ≠•Â§ÑÁêÜÈ¢ÑÊµãÁöÑÁóÖÊØíÂ∫èÂàóÔºå‰ª•ËØÑ‰º∞ÁóÖÊØíÂü∫Âõ†ÁªÑÁöÑË¥®Èáè„ÄÇ
Áî±‰∫éÊú¨Á†îÁ©∂ÈáçÁÇπÂÖ≥Ê≥®ÈùûÊï¥ÂêàÁóÖÊØíÂü∫Âõ†ÁªÑÔºåÂõ†Ê≠§Ë¢´ geNomad Êàñ CheckV Ê†áËÆ∞‰∏∫ÂéüÁóÖÊØíÁöÑcontigsË¢´‰∏¢ÂºÉ„ÄÇ
‰ªéÂâ©‰ΩôÁöÑcontigs‰∏≠Ôºå‰ΩøÁî®‰ª•‰∏ãËßÑÂàôÈÄâÊã©ÁóÖÊØíÂü∫Âõ†ÁªÑÔºö</p>
<ol>
<li>Ëá≥Â∞ë 1 kb ÁöÑcontigsÔºå‰∏é CheckVÊï∞ÊçÆÂ∫ì‰∏≠ÁöÑÂü∫Âõ†ÁªÑÂÖ∑ÊúâÈ´òÂ∫¶Áõ∏‰ººÊÄßÔºàÂç≥ÂÖ∑ÊúâÈ´òË¥®ÈáèÊàñ‰∏≠Á≠âÂÆåÊï¥ÊÄß‰º∞ËÆ°ÔºâÊàñÂåÖÂê´Ëá™Âä®ÈÄâÊã©Áõ¥Êé•Êú´Á´ØÈáçÂ§çÔºõ</li>
<li>Èïø‰∫é 10 kb ÁöÑcontigsÈúÄË¶ÅÂÖ∑ÊúâÈ´ò‰∫é 0.8 ÁöÑ geNomad ÁóÖÊØíËØÑÂàÜÔºåÂπ∂‰∏îÁºñÁ†Å‰∏ÄÁßçÁóÖÊØíÊ†áÂøóÔºà‰æãÂ¶ÇÔºåÁªàÊ≠¢ÈÖ∂„ÄÅË°£Â£≥ËõãÁôΩ„ÄÅÈó®Êà∑ËõãÁôΩÁ≠âÔºâÔºàÁî± geNomad Á°ÆÂÆöÔºâÔºåÊàñËÄÖÂÖ∑ÊúâgNomad ÁóÖÊØíÊ†áËÆ∞Ëá≥Â∞ë‰∏∫ 5.0Ôºõ</li>
<li>Áü≠‰∫é10kbÂíåÈïø‰∫é5kbÁöÑcontigsÈúÄË¶ÅÂÖ∑ÊúâÈ´ò‰∫é0.9ÁöÑgeNomadÁóÖÊØíËØÑÂàÜÔºåÁºñÁ†ÅËá≥Â∞ë‰∏ÄÁßçÁóÖÊØíÊ†áÂøóÂπ∂‰∏îÂÖ∑ÊúâÈ´ò‰∫é2.0ÁöÑÁóÖÊØíÊ†áËÆ∞ÂØåÈõÜ„ÄÇ
Ëøô‰∫ßÁîü‰∫Ü 49,649 ‰∏™ÁóÖÊØícontigsÔºåÂ∞ÜÂÖ∂Áî®‰∫é‰∏ãÊ∏∏ÂàÜÊûê„ÄÇ</li>
</ol>
<p>ÁóÖÊØíÂü∫Âõ†ÁªÑÊåâÁÖß MIUViG ÊåáÂçóËÅöÁ±ªÊàê vOTUÔºàÂπ≥ÂùáÊ†∏Ëã∑ÈÖ∏Âêå‰∏ÄÊÄß‰∏∫ 95%ÔºåÊØîÂØπÂàÜÊï∞‰∏∫ 85%Ôºâ„ÄÇÁÆÄËÄåË®Ä‰πãÔºåËøõË°å‰∫ÜÂÖ®ÂØπÂÖ® BLASTÔºàv2.13.0+Ôºå&rsquo;-task megablast -evalue 1e-5 -max_target_seqs 20000&rsquo;ÔºâÊêúÁ¥¢Êù•‰º∞ËÆ°ÊàêÂØπÂπ≥ÂùáÊ†∏Ëã∑ÈÖ∏Ë∫´‰ªΩÂíåÊØîÂØπÂàÜÊï∞ (AF)„ÄÇÂπ∂ÈááÁî® pyLeidenÂØπÂü∫Âõ†ÁªÑËøõË°åËÅöÁ±ªÔºå‰ΩøÁî®Âõæ‰Ωú‰∏∫ËæìÂÖ•ÔºåÂÖ∂‰∏≠Êª°Ë∂≥ MIUViG Ê†áÂáÜÁöÑÂü∫Âõ†ÁªÑÂØπÈÄöËøáËæπËøûÊé•„ÄÇ</p>
<p>‰ΩøÁî® DIAMOND ËøõË°åËõãÁôΩË¥®ÊØîÂØπÔºå‰ΩøÁî® Markov Cluster Process ËøõË°åËÅöÁ±ªÔºåÂ∞ÜÁóÖÊØíÊåâÂ§ßËá¥Â±ûÊ∞¥Âπ≥Ôºà40% Âπ≥ÂùáÊ∞®Âü∫ÈÖ∏Âêå‰∏ÄÊÄßÔºå20% ÂÖ±‰∫´Âü∫Âõ†ÔºâÂíåÁßëÊ∞¥Âπ≥Ôºà20% Âπ≥ÂùáÊ∞®Âü∫ÈÖ∏Âêå‰∏ÄÊÄßÔºå10% ÂÖ±‰∫´Âü∫Âõ†ÔºâËÅöÁ±ªËøõË°åÂàÜÁªÑ„ÄÇÈÄöËøá‰∏éÂÖàÂâçÊèèËø∞ÁöÑÊù•Ëá™ NCBI GenBank Âíå MAG ÁöÑ 160 ‰∏á‰∏™ÁªÜËèåÂíåÂè§ËèåÂü∫Âõ†ÁªÑÁöÑ CRISPR Èó¥ÈöîÂå∫Êï∞ÊçÆÂ∫ìÁõ∏ÂåπÈÖçÔºåÂ∞ÜÁóÖÊØíÂ∫èÂàóÂàÜÈÖçÁªôÂÅáÂÆöÁöÑÂÆø‰∏ªÔºàÁªÜËèåÂíåÂè§ËèåÔºâÂàÜÁ±ªÁæ§„ÄÇ</p>
<h3 id="bacteriophage-in-1300-year-old-human-palaeofaeces">Bacteriophage in 1300-year-old human palaeofaeces</h3>
<img src="images/palaeofaeces.png" title=""/>
<p>ËøôÊòØ2024Âπ¥Âú®NC‰∏äÂèëË°®ÁöÑÁ†îÁ©∂[10]Ôºö</p>
<p>Âô¨Ëèå‰ΩìË¢´ÂπøÊ≥õËÆ§‰∏∫ÊòØÂø´ÈÄüËøõÂåñÁöÑÁîüÁâ©ÂÆû‰Ωì„ÄÇÁÑ∂ËÄåÔºåÂÖ≥‰∫éÂè§‰ª£Âô¨Ëèå‰ΩìÁöÑÁü•ËØÜÊòØÊúâÈôêÁöÑ„ÄÇÂú®ËøôÈáåÔºåÊàë‰ª¨ÂàÜÊûê‰∫ÜÂÖàÂâç‰ªéÂè§‰ª£Âè§Á≤™‰æøÂíå‰∫∫Á±ªËÇ†ÈÅìÂÜÖÂÆπÁâ©Ê†∑Êú¨‰∏≠ÁîüÊàêÁöÑ DNA Â∫èÂàóÊï∞ÊçÆÈõÜÔºåÂπ∂Èâ¥ÂÆö‰∫Ü‰∏éÂΩì‰ªäÁöÑ Mushuvirus mushuÔºà‰∏ÄÁßçÊÑüÊüìËÇ†ÈÅìÂÖ±ÁîüÁªÜËèåÁöÑÁóÖÊØíÔºâÂá†‰πéÁõ∏ÂêåÁöÑÂè§‰ª£Âô¨Ëèå‰ΩìÂü∫Âõ†ÁªÑ„ÄÇ
Âü∫Âõ†ÁªÑÁöÑ DNA Êçü‰º§Ê®°Âºè‰∏éÂÖ∂Âè§‰ª£Ëµ∑Ê∫ê‰∏ÄËá¥ÔºåÂ∞ΩÁÆ°ÁªèËøá 1300 Âπ¥ÁöÑËøõÂåñÔºåÂè§‰ª£Êú®È°ªÁóÖÊØíÂü∫Âõ†ÁªÑ‰∏éÁé∞‰ª£ÂØπÂ∫îÁâ©ÂÖ∑Êúâ 97.7% ÁöÑÊ†∏Ëã∑ÈÖ∏Âêå‰∏ÄÊÄßÔºåË°®ÊòéÂéüÂô¨Ëèå‰Ωì‰∏éÂÖ∂ÂÆø‰∏ª‰πãÈó¥Â≠òÂú®ÈïøÊúüÂÖ≥Á≥ª„ÄÇÊ≠§Â§ñÔºåËøòÈáçÂª∫Âπ∂È™åËØÅ‰∫ÜËøáÂéª 5300 Âπ¥ÁöÑ 297 ‰∏™ÂÖ∂‰ªñÂô¨Ëèå‰ΩìÂü∫Âõ†ÁªÑÔºåÂåÖÊã¨Â±û‰∫éÊú™Áü•ÂÆ∂ÊóèÁöÑÂô¨Ëèå‰ΩìÂü∫Âõ†ÁªÑ„ÄÇ
Á†îÁ©∂ÁªìÊûúËØÅÊòé‰∫ÜÈáçÂª∫Âè§‰ª£Âô¨Ëèå‰ΩìÂü∫Âõ†ÁªÑÂ∫èÂàóÁöÑÂèØË°åÊÄßÔºå‰ªéËÄåÊâ©Â§ß‰∫ÜÂ∑≤Áü•ÁöÑÁóÖÊØíÂúàÔºåÂπ∂‰∏∫Ë∑®Ë∂äÊï∞ÂçÉÂπ¥ÁöÑÂô¨Ëèå‰Ωì-ÁªÜËèåÁõ∏‰∫í‰ΩúÁî®Êèê‰æõ‰∫ÜËßÅËß£„ÄÇ</p>
<p>ÁóÖÊØícontigsÈâ¥ÂÆöÔºö</p>
<p>‰ΩøÁî®‰∏âÁßçÊú∫Âô®Â≠¶‰π†Â∑•ÂÖ∑Êù•ËØÜÂà´ÁóÖÊØícontigs„ÄÇÁ¨¨‰∏Ä‰∏™ÊòØ Jaeger v.1.1.0ÔºåËøôÊòØ‰∏ÄÁßçÊ∑±Â∫¶Â≠¶‰π†Ê®°ÂûãÔºåÂèØÂü∫‰∫éËá™Âä®ÁªÑÊàêÁâπÂæÅÊèêÂèñÊù•ËØÜÂà´ÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁöÑÂô¨Ëèå‰ΩìÂü∫Âõ†ÁªÑÂ∫èÂàó (<a class="link" href="https://github.com/Yasas1994/Jaeger"  target="_blank" rel="noopener"
    >https://github.com/Yasas1994/Jaeger</a>)„ÄÇÁ¨¨‰∫å‰∏™ÂíåÁ¨¨‰∏â‰∏™ÊòØ VIBRANT v.1.2.124 Âíå VirSorter2 v.2.2.325ÔºåÂÆÉ‰ª¨‰æùËµñ‰∫éÂàÜÊûê‰ª£Ë°®‰∏éÈ¢ÑÊµãËõãÁôΩË¥®Áõ∏‰ºº‰ΩÜÂ∫îÁî®‰∏çÂêåÂàÜÁ±ªÂô®ÂíåÂèÇËÄÉÊï∞ÊçÆÂ∫ìÁöÑ‰øùÂÆàÂÆ∂ÊóèÂíå/ÊàñÁªìÊûÑÂüüÁöÑ HMM ÂõæË∞±„ÄÇ Jaeger Âíå VIBRANT ‰ΩøÁî®ÈªòËÆ§ËÆæÁΩÆËøêË°å„ÄÇÂØπ‰∫é VirSorter2ÔºåÊàë‰ª¨‰ΩøÁî®‰∫Ü‰ΩçÁΩÆÂèÇÊï∞‚Äú&ndash;include-groups dsDNAphage,NCLDV,ssDNA,lavidaviridae all‚Äù„ÄÇËøõ‰∏ÄÊ≠•ÂàÜÊûê‰∫ÜËá≥Â∞ë‰∏§ÁßçÂ∑•ÂÖ∑ÂàÜÁ±ª‰∏∫ÁóÖÊØíÁöÑcontigs„ÄÇ</p>
<h2 id="references">References</h2>
<ol>
<li>Camargo, A.P., Roux, S., Schulz, F. et al. Identification of mobile genetic elements with geNomad. Nat Biotechnol (2023). <a class="link" href="https://doi.org/10.1038/s41587-023-01953-y"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41587-023-01953-y</a></li>
<li>Ren, J., Song, K., Deng, C., Ahlgren, N.A., Fuhrman, J.A., Li, Y., Xie, X., Poplin, R. and Sun, F. (2020), Identifying viruses from metagenomic data using deep learning. Quantitative Biology, 8: 64-77. <a class="link" href="https://doi.org/10.1007/s40484-019-0187-4"  target="_blank" rel="noopener"
    >https://doi.org/10.1007/s40484-019-0187-4</a></li>
<li>Guo, J., Bolduc, B., Zayed, A.A. et al. VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses. Microbiome 9, 37 (2021). <a class="link" href="https://doi.org/10.1186/s40168-020-00990-y"  target="_blank" rel="noopener"
    >https://doi.org/10.1186/s40168-020-00990-y</a></li>
<li>Kieft, K., Zhou, Z. &amp; Anantharaman, K. VIBRANT: automated recovery, annotation and curation of microbial viruses, and evaluation of viral community function from genomic sequences. Microbiome 8, 90 (2020). <a class="link" href="https://doi.org/10.1186/s40168-020-00867-0"  target="_blank" rel="noopener"
    >https://doi.org/10.1186/s40168-020-00867-0</a></li>
<li>Chen, L., Banfield, J.F. COBRA improves the completeness and contiguity of viral genomes assembled from metagenomes. Nat Microbiol 9, 737‚Äì750 (2024). <a class="link" href="https://doi.org/10.1038/s41564-023-01598-2"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41564-023-01598-2</a></li>
<li>Flamholz, Z.N., Biller, S.J. &amp; Kelly, L. Large language models improve annotation of prokaryotic viral proteins. Nat Microbiol 9, 537‚Äì549 (2024). <a class="link" href="https://doi.org/10.1038/s41564-023-01584-8"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41564-023-01584-8</a></li>
<li>Nayfach, S., Camargo, A.P., Schulz, F. et al. CheckV assesses the quality and completeness of metagenome-assembled viral genomes. Nat Biotechnol 39, 578‚Äì585 (2021). <a class="link" href="https://doi.org/10.1038/s41587-020-00774-7"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41587-020-00774-7</a></li>
<li>Ma, B., Wang, Y., Zhao, K. et al. Biogeographic patterns and drivers of soil viromes. Nat Ecol Evol 8, 717‚Äì728 (2024). <a class="link" href="https://doi.org/10.1038/s41559-024-02347-2"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41559-024-02347-2</a></li>
<li>Graham, E.B., Camargo, A.P., Wu, R. et al. A global atlas of soil viruses reveals unexplored biodiversity and potential biogeochemical impacts. Nat Microbiol (2024). <a class="link" href="https://doi.org/10.1038/s41564-024-01686-x"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41564-024-01686-x</a></li>
<li>Rozwalak, P., Barylski, J., Wijesekara, Y. et al. Ultraconserved bacteriophage genome sequence identified in 1300-year-old human palaeofaeces. Nat Commun 15, 495 (2024). <a class="link" href="https://doi.org/10.1038/s41467-023-44370-0"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41467-023-44370-0</a></li>
</ol>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/microbiome/">Microbiome</a>
        
            <a href="/Cao-blog/tags/virus/">Virus</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/virus-host/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/virus-host/images/IPEV.78086c186ccdcca626f22273564f07d4_hu_df4169bfbc6d92eb.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post ÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÂÆø‰∏ªÈ¢ÑÊµã"
                        data-key="virus-host" 
                        data-hash="md5-eAhsGGzNzKYm8iJzVk8H1A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">ÂÆèÂü∫Âõ†ÁªÑ‰∏≠ÁóÖÊØíÂ∫èÂàóÁöÑÂÆø‰∏ªÈ¢ÑÊµã</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/virrep/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/virrep/images/workflow.96d665cd6f8105e8fd4ac804152698e2_hu_e61849de4631fb72.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post VirRep: ‰∫∫Á±ªËÇ†ÈÅìÂæÆÁîüÁâ©ÁªÑËØÜÂà´ÁóÖÊØíÊñ∞ÊñπÊ≥ï"
                        data-key="virrep" 
                        data-hash="md5-ltZlzW&#43;BBej9SsgEFSaY4g==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">VirRep: ‰∫∫Á±ªËÇ†ÈÅìÂæÆÁîüÁâ©ÁªÑËØÜÂà´ÁóÖÊØíÊñ∞ÊñπÊ≥ï</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà4Ôºâ"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà4Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-3/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-3/images/fig1.20d52f738dfd506b0754bda8d4b0ead0_hu_d6f6d5c8a568f26d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà3Ôºâ"
                        data-key="25-nature-3" 
                        data-hash="md5-INUvc439UGsHVL2o1LDq0A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà3Ôºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-2/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-2/images/2-1-2.f13dbba185acaee28c931cc543f77916_hu_94453c680a4ffc60.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà2Ôºâ"
                        data-key="25-nature-2" 
                        data-hash="md5-8T27oYWsruKMkxzFQ/d5Fg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25Âπ¥ÊúÄÊñ∞ÁöÑ‰∏Ä‰∫õNatureÁéØÂ¢ÉÂæÆÁîüÁâ©Á†îÁ©∂ÊñáÁ´†Ôºà2Ôºâ</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
