<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='这篇文章将介绍一个较新的binning流程：BASALT及其具体使用方法。'>
<title>宏基因组分箱（binning）｜3. BASALT优化binning流程</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/binning3/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='宏基因组分箱（binning）｜3. BASALT优化binning流程'>
<meta property='og:description' content='这篇文章将介绍一个较新的binning流程：BASALT及其具体使用方法。'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/binning3/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='binning' /><meta property='article:tag' content='microbiome' /><meta property='article:published_time' content='2025-02-18T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2025-02-18T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/binning3/images/workflow.png' />
<meta name="twitter:title" content="宏基因组分箱（binning）｜3. BASALT优化binning流程">
<meta name="twitter:description" content="这篇文章将介绍一个较新的binning流程：BASALT及其具体使用方法。"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/binning3/images/workflow.png' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">🐛🐶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/142776729'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://asa12138.github.io/'
                        target="_blank"
                        title="Homepage"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-world-pin"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M20.972 11.291a9 9 0 1 0 -8.322 9.686" /><path d="M3.6 9h16.8" /><path d="M3.6 15h8.9" /><path d="M11.5 3a17 17 0 0 0 0 18" /><path d="M12.5 3a16.986 16.986 0 0 1 2.578 9.018" /><path d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z" /><path d="M19 18v.01" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com/Asa12138_'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >中文</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a>
      <ol>
        <li><a href="#功能模块">功能模块</a></li>
        <li><a href="#效果评估">效果评估</a>
          <ol>
            <li><a href="#利用cami数据集评估basalt的性能">利用CAMI数据集评估BASALT的性能</a></li>
            <li><a href="#与其他主流软件的比较">与其他主流软件的比较</a></li>
            <li><a href="#真实数据集评估">真实数据集评估</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#使用方法">使用方法</a>
      <ol>
        <li><a href="#安装">安装</a>
          <ol>
            <li><a href="#测试数据">测试数据</a></li>
          </ol>
        </li>
        <li><a href="#命令参数">命令参数</a></li>
        <li><a href="#具体示例">具体示例</a>
          <ol>
            <li><a href="#项目背景">项目背景</a></li>
            <li><a href="#组装策略">组装策略</a></li>
          </ol>
        </li>
        <li><a href="#注意事项">注意事项</a>
          <ol>
            <li><a href="#输入输出">输入输出</a></li>
            <li><a href="#运行速度">运行速度</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/binning3/">
                <img src="/Cao-blog/p/binning3/images/workflow_hu_dcdf49645952a935.png"
                        srcset="/Cao-blog/p/binning3/images/workflow_hu_dcdf49645952a935.png 800w, /Cao-blog/p/binning3/images/workflow_hu_c7870ebf5f1a7759.png 1600w"
                        width="800" 
                        height="858" 
                        loading="lazy"
                        alt="Featured image of post 宏基因组分箱（binning）｜3. BASALT优化binning流程" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/binning3/">宏基因组分箱（binning）｜3. BASALT优化binning流程</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            这篇文章将介绍一个较新的binning流程：BASALT及其具体使用方法。
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Feb 18, 2025</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    3 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                版权归Cao's blog所有，转载请注明出处
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>在微生物组学研究中，基因组binning（分箱，即将测序数据中的reads或contigs分类到不同的基因组中）是一个关键步骤。发表于2018年的MetaWRAP一直没有重大更新，这篇文章将介绍一个较新的binning流程：BASALT及其具体使用方法。</p>
<p>BASALT （Binning Across a Series of AssembLies Toolkit）是一个高效的基因组binning工具，旨在处理复杂的微生物组数据，特别是在多个组装（assemblies）之间进行binning时表现出色。它通过整合多个组装结果，提高了binning的准确性和完整性，尤其适用于处理高度复杂的微生物群落数据。</p>
<p>BASALT的文章于2024年发表于Nature Communications，软件主页：<a class="link" href="https://github.com/EMBL-PKU/BASALT"  target="_blank" rel="noopener"
    >https://github.com/EMBL-PKU/BASALT</a>。以下是BASALT的主要工作流程模块：</p>
<img src="images/workflow.png" title=""/>
<h3 id="功能模块">功能模块</h3>
<p>BASALT由四个模块组成：<strong>自动分箱（Automated Binning）</strong>、<strong>分箱选择（Bin Selection）</strong>、<strong>优化（Refinement）<strong>和</strong>缺口填充（Gap Filling）</strong>。以下是各模块的详细工作流程：</p>
<ol>
<li>
<p><strong>自动分箱模块</strong><br>
首先，通过多个分箱工具（每个工具使用不同的阈值）将组装的contigs分类到不同的bin中，生成初始的binsets。接着，通过识别contig ID，将包含相似contig的bin合并为混合bin（hybrid bins）。然后，基于平均核苷酸一致性（ANI）对这些混合binsets进行分组，并通过覆盖度估计（coverage estimation）识别每组中的核心序列（Core sequence identification, CSI），用于比较组内每对bin中的内部序列（inliers）。随后，使用神经网络算法识别冗余bin，并保留非冗余bin用于后续的<strong>离群值去除（Outlier removal, OR）</strong>。</p>
</li>
<li>
<p><strong>分箱选择模块</strong><br>
在优化模块中，结合四核苷酸频率（Tetranucleotide Frequency, TNF）和覆盖度相关系数（Coverage Correlation Coefficient, CCC），通过多维内部四分位距（Multidimensional Internal Quartile Range, IQR）检测来识别离群序列（outlier sequences）。接着，通过序列检索步骤（sequence retrieval），利用双端测序（Pair-End, PE）或长读长追踪（Long-Read Tracking）将未分箱的内部序列（un-binned inliers）从短读长（SR）和长读长（LR）contig池中连接并重新迭代，从而生成优化后的bin。成功映射到优化bin的reads会进一步进行抛光（polishing），生成抛光后的bin。</p>
</li>
<li>
<p><strong>优化模块</strong><br>
在优化模块的最后一步，通过受限的重叠-布局-一致性程序（restrained Overlap–Layout–Consensus program, rOLC），将优化/抛光后的bin与相应的冗余bin进行重叠，然后在短读长（SRS）和长读长（LRS）数据的基础上进行重新组装。重新组装的bin会进一步抛光，并经过另一轮rOLC处理，最终生成最终的bin-set。</p>
</li>
<li>
<p><strong>缺口填充模块</strong><br>
在缺口填充模块中，长读长测序数据（+LRS）可以用于补充和完善bin中的缺口区域，进一步提高bin的完整性和准确性。</p>
</li>
</ol>
<h3 id="效果评估">效果评估</h3>
<h4 id="利用cami数据集评估basalt的性能">利用CAMI数据集评估BASALT的性能</h4>
<p>采用Critical Assessment of Metagenome Interpretation （CAMI）中的CAMI-high数据。使用二代+三代数据混装产生的组装文件，BASALT可以恢复其中的392个满足完整度 ≥ 50，污染度 ≤ 10的MAGs。经过基因组选择、优化和空缺填补每一个模块处理后，所产生的MAGs无论从完整度（Completeness）、污染度（Contamination）和总质量（完整度-5*污染度）上都有很大的提升。</p>
<img src="images/compare1.png" title=""/>
<h4 id="与其他主流软件的比较">与其他主流软件的比较</h4>
<p>在MAGs（宏基因组组装基因组）的评估中，采用了更为严格的质量标准，即<strong>质量值（完整度 - 5 × 污染度）≥ 50</strong>。通过对比VAMB、DASTool、metaWRAP和BASALT四个软件的处理结果，发现BASALT的表现显著优于其他工具。具体而言，BASALT比VAMB、DASTool和metaWRAP分别多获得了<strong>63.7%</strong>、<strong>61.5%<strong>和</strong>33.3%<strong>的MAGs，其中包括</strong>69个独有的MAGs</strong>（图a）。</p>
<p>在四个软件共同获得的<strong>168个MAGs</strong>中，BASALT生成的基因组质量也显著高于其他软件（图b）。两两比较的结果进一步显示，BASALT生成的MAGs质量普遍优于其他工具。具体来说，BASALT生成的<strong>质量更好的MAGs数量</strong>分别是VAMB的<strong>9.6倍</strong>、DASTool的<strong>14.6倍</strong>和metaWRAP的<strong>6.1倍</strong>（图c）。</p>
<p>此外，在高质量MAGs（质量值 &gt; 90）的获取方面，BASALT的表现同样突出。与VAMB、DASTool和metaWRAP相比，BASALT获得的高质量MAGs数量分别是它们的<strong>2.28倍</strong>、<strong>3.59倍</strong>和<strong>2.02倍</strong>（图d）。</p>
<img src="images/compare2.png" title=""/>
<h4 id="真实数据集评估">真实数据集评估</h4>
<p>除了测试数据集外，作者团队还选用了十个真实数据集，包括<strong>SRS（短读长测序）</strong>、<strong>SRS+LRS（短读长+长读长测序）<strong>以及</strong>LRS（HiFi长读长测序）<strong>数据，涵盖了盐湖沉积物、人类肠道、海水、南极土壤、活性污泥（AS）、鸡肠道、羊肠道、热泉、厌氧反应器（AD）等多种环境类型。以高度复杂的盐湖沉积物微生物群落样本为例（仅使用二代测序数据），BASALT比metaWRAP多获得了</strong>30%的MAGs</strong>，尤其在恢复低丰度MAGs方面表现出显著优势（图a）。在功能注释分析中，BASALT比metaWRAP多获得了47.6%的非冗余开放式阅读框（ORF）序列（图b），使得BASALT恢复的MAGs在功能模块上更加完整（图c）。</p>
<img src="images/compare3.png" title=""/>
<p>从盐湖沉积物样品中，BASALT共获得了<strong>557个MAGs</strong>，涵盖了<strong>54个门</strong>；而metaWRAP仅识别了<strong>392个MAGs</strong>，涵盖<strong>45个门</strong>，且这些门均被BASALT检测到（图5）。此外，BASALT独特地识别出了metaWRAP未能发现的<strong>9个门</strong>，其中包括<strong>21个纲水平上的细菌分支</strong>和<strong>2个目水平上的古菌分支</strong>（红色高亮分支）。这些独特的分支为挖掘复杂环境中未被探索的微生物及其功能元件提供了重要线索，进一步拓展了对微生物多样性和功能潜力的理解。</p>
<img src="images/example1.png" title="" width="60%"/>
<h2 id="使用方法">使用方法</h2>
<h3 id="安装">安装</h3>
<p>由于需要安装许多依赖项，推荐使用conda安装和管理BASALT，命令耗时较长，需要耐心等待：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 我这里安装到~/biosoft/目录</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/biosoft/
</span></span><span class="line"><span class="cl">git clone https://github.com/EMBL-PKU/BASALT.git
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> BASALT
</span></span><span class="line"><span class="cl">conda env create -n BASALT --file basalt_env.yml
</span></span></code></pre></td></tr></table>
</div>
</div><p>然后要授权BASALT脚本。一般来说conda BASALT环境位于conda安装的子目录envs中。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">chmod -R <span class="m">777</span> ~/miniconda3/envs/BASALT/bin/*
</span></span></code></pre></td></tr></table>
</div>
</div><p>然后下载神经网络训练模型，可以直接用1脚本下载，如果网络不好可以手动下载（或者在浏览器下载好再ftp上传到服务器）：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 1.脚本下载训练模型</span>
</span></span><span class="line"><span class="cl">python BASALT_models_download.py
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 2.手动下载训练模型</span>
</span></span><span class="line"><span class="cl">wget https://figshare.com/ndownloader/files/41093033
</span></span><span class="line"><span class="cl">mv <span class="m">41093033</span> BASALT.zip
</span></span><span class="line"><span class="cl">mv BASALT.zip ~/.cache
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/.cache
</span></span><span class="line"><span class="cl">unzip BASALT.zip
</span></span></code></pre></td></tr></table>
</div>
</div><p>后续一般会用到checkM，所以也最好按照<a class="link" href="../binning2/#%e8%bd%af%e4%bb%b6%e6%95%b0%e6%8d%ae%e5%87%86%e5%a4%87" >上一篇文章</a>下载配置一下相应数据库：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 设置CheckM数据存储位置</span>
</span></span><span class="line"><span class="cl">checkm data setRoot ~/db/checkm
</span></span></code></pre></td></tr></table>
</div>
</div><p>除上述conda方法外，还可以使用脚本直接安装或是利用singularity运行等方法，可参考<a class="link" href="https://github.com/EMBL-PKU/BASALT"  target="_blank" rel="noopener"
    >https://github.com/EMBL-PKU/BASALT</a>。</p>
<h4 id="测试数据">测试数据</h4>
<p>作者团队准备了测试文件以测试BASALT是否成功安装并且可以顺利运行。从<a class="link" href="https://figshare.com/articles/dataset/BASALT_demo_files/22323424"  target="_blank" rel="noopener"
    >https://figshare.com/articles/dataset/BASALT_demo_files/22323424</a>下载测试文件（共三个：Data.tar.gz、Final_bestbinset.tar.gz、basalt.sh），然后执行下面的命令运行程序（该过程耗时大约六小时）：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">unzip 22323424.zip
</span></span><span class="line"><span class="cl">tar zxvf Data.tar.gz
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> Data/
</span></span><span class="line"><span class="cl">BASALT -a assembly.fa -s sample1.R1.fq,sample1.R2.fq -l sample1.nanopore.fq -t <span class="m">16</span> -m <span class="m">300</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="命令参数">命令参数</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span><span class="lnt">50
</span><span class="lnt">51
</span><span class="lnt">52
</span><span class="lnt">53
</span><span class="lnt">54
</span><span class="lnt">55
</span><span class="lnt">56
</span><span class="lnt">57
</span><span class="lnt">58
</span><span class="lnt">59
</span><span class="lnt">60
</span><span class="lnt">61
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-mysql" data-lang="mysql"><span class="line"><span class="cl"><span class="err">用法</span><span class="p">:</span><span class="w"> </span><span class="n">BASALT</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">h</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">a</span><span class="w"> </span><span class="n">ASSEMBLIES</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">s</span><span class="w"> </span><span class="n">SR_DATASETS</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">l</span><span class="w"> </span><span class="n">LONG_DATASETS</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">-</span><span class="n">hf</span><span class="w"> </span><span class="n">HIFI_DATASETS</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">c</span><span class="w"> </span><span class="n">HI_C_DATASET</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">t</span><span class="w"> </span><span class="n">THREADS</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">m</span><span class="w"> </span><span class="n">RAM</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">-</span><span class="n">e</span><span class="w"> </span><span class="n">EXTRA_BINNER</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">-</span><span class="n">qc</span><span class="w"> </span><span class="n">QUALITY_CHECK</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">--</span><span class="n">min</span><span class="o">-</span><span class="n">cpn</span><span class="w"> </span><span class="n">MIN_COMPLETENESS</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">--</span><span class="n">max</span><span class="o">-</span><span class="n">ctn</span><span class="w"> </span><span class="n">MAX_CONTAMINATION</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">--</span><span class="n">mode</span><span class="w"> </span><span class="n">RUNNING_MODE</span><span class="p">]</span><span class="w"> </span><span class="p">[</span><span class="o">--</span><span class="n">module</span><span class="w"> </span><span class="n">FUNCTIONAL_MODULE</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">--</span><span class="n">autopara</span><span class="w"> </span><span class="n">AUTOBINING_PARAMETERS</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">              </span><span class="p">[</span><span class="o">--</span><span class="n">refinepara</span><span class="w"> </span><span class="n">REFINEMENT_PARAMTER</span><span class="p">]</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">h</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">help</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">显示帮助信息并退出。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">a</span><span class="w"> </span><span class="n">ASSEMBLIES</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">assemblies</span><span class="w"> </span><span class="n">ASSEMBLIES</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">组装序列文件列表，例如：</span><span class="o">`</span><span class="n">as1</span><span class="p">.</span><span class="n">fa</span><span class="p">,</span><span class="n">as2</span><span class="p">.</span><span class="n">fa</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">s</span><span class="w"> </span><span class="n">SR_DATASETS</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">shortreads</span><span class="w"> </span><span class="n">SR_DATASETS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">双端测序数据列表，例如：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">r1_1</span><span class="p">.</span><span class="n">fq</span><span class="p">,</span><span class="n">r1_2</span><span class="p">.</span><span class="n">fq</span><span class="o">/</span><span class="n">r2_1</span><span class="p">.</span><span class="n">fq</span><span class="p">,</span><span class="n">r2_2</span><span class="p">.</span><span class="n">fq</span><span class="o">`</span><span class="err">（双端数据需使用</span><span class="o">`/`</span><span class="err">分隔）。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">注意，这里不是随便什么路径，要求一定要在同一个文件夹下。</span><span class="w">
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">l</span><span class="w"> </span><span class="n">LONG_DATASETS</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">longreads</span><span class="w"> </span><span class="n">LONG_DATASETS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">包含</span><span class="n">ONT和PacBio长读长数据</span><span class="err">（不包括</span><span class="n">HiFi数据</span><span class="err">）。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`</span><span class="n">lr1</span><span class="p">.</span><span class="n">fq</span><span class="p">,</span><span class="n">lr2</span><span class="p">.</span><span class="n">fq</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">hf</span><span class="w"> </span><span class="n">HIFI_DATASETS</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">hifi</span><span class="w"> </span><span class="n">HIFI_DATASETS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="n">HiFi测序数据</span><span class="err">，例如：</span><span class="o">`</span><span class="n">hf1</span><span class="p">.</span><span class="n">fq</span><span class="p">,</span><span class="n">hf2</span><span class="p">.</span><span class="n">fq</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">c</span><span class="w"> </span><span class="n">HI_C_DATASET</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">HIC</span><span class="w"> </span><span class="n">HI_C_DATASET</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="n">Hi</span><span class="o">-</span><span class="n">C测序数据列表</span><span class="err">，例如：</span><span class="o">`</span><span class="n">hc1</span><span class="p">.</span><span class="n">fq</span><span class="p">,</span><span class="n">hc2</span><span class="p">.</span><span class="n">fq</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">t</span><span class="w"> </span><span class="n">THREADS</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">threads</span><span class="w"> </span><span class="n">THREADS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">线程数，例如：</span><span class="o">`</span><span class="mi">64</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">m</span><span class="w"> </span><span class="n">RAM</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">ram</span><span class="w"> </span><span class="n">RAM</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">内存大小，最低建议</span><span class="o">`</span><span class="mi">32</span><span class="n">G</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">e</span><span class="w"> </span><span class="n">EXTRA_BINNER</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">extra_binner</span><span class="w"> </span><span class="n">EXTRA_BINNER</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">额外的分箱工具：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">m</span><span class="o">`</span><span class="err">：</span><span class="n">MetaBinner</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">v</span><span class="o">`</span><span class="err">：</span><span class="n">Vamb</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`-</span><span class="n">e</span><span class="w"> </span><span class="n">m</span><span class="o">`</span><span class="err">表示在默认使用</span><span class="n">MetaBAT2</span><span class="err">、</span><span class="n">MaxBin2和CONCOCT的基础上</span><span class="err">，额外使用</span><span class="n">MetaBinner进行分箱</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">-</span><span class="n">qc</span><span class="w"> </span><span class="n">QUALITY_CHECK</span><span class="p">,</span><span class="w"> </span><span class="o">--</span><span class="n">quality</span><span class="o">-</span><span class="k">check</span><span class="w"> </span><span class="n">QUALITY_CHECK</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">选择</span><span class="n">CheckM版本</span><span class="err">，默认</span><span class="o">`</span><span class="n">checkm</span><span class="o">`</span><span class="err">，可指定</span><span class="o">`</span><span class="n">checkm2</span><span class="o">`</span><span class="err">，例如：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`-</span><span class="n">qc</span><span class="w"> </span><span class="n">checkm2</span><span class="o">`</span><span class="err">用于质量检查。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">min</span><span class="o">-</span><span class="n">cpn</span><span class="w"> </span><span class="n">MIN_COMPLETENESS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">过滤保留</span><span class="n">bin的最小完整度</span><span class="err">（默认</span><span class="o">`</span><span class="mi">35</span><span class="o">`</span><span class="err">）。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">max</span><span class="o">-</span><span class="n">ctn</span><span class="w"> </span><span class="n">MAX_CONTAMINATION</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">过滤保留</span><span class="n">bin的最大污染度</span><span class="err">（默认</span><span class="o">`</span><span class="mi">20</span><span class="o">`</span><span class="err">）。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">mode</span><span class="w"> </span><span class="n">RUNNING_MODE</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">运行模式：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">new</span><span class="o">`</span><span class="err">：新建项目</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="k">continue</span><span class="o">`</span><span class="err">：继续运行</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`--</span><span class="n">mode</span><span class="w"> </span><span class="k">continue</span><span class="o">`</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="o">`--</span><span class="n">mode</span><span class="w"> </span><span class="n">new</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">module</span><span class="w"> </span><span class="n">FUNCTIONAL_MODULE</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">运行的功能模块（默认运行所有模块）：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">autobinning</span><span class="o">`</span><span class="err">：自动分箱</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">refinement</span><span class="o">`</span><span class="err">：分箱优化</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">reassembly</span><span class="o">`</span><span class="err">：重新组装</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`--</span><span class="n">module</span><span class="w"> </span><span class="n">reassembly</span><span class="o">`</span><span class="w"> </span><span class="err">仅运行重新组装模块。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">autopara</span><span class="w"> </span><span class="n">AUTOBINING_PARAMETERS</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">自动分箱模式：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">more</span><span class="o">-</span><span class="k">sensitive</span><span class="o">`</span><span class="err">（默认）</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="k">sensitive</span><span class="o">`</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">quick</span><span class="o">`</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`--</span><span class="n">autopara</span><span class="w"> </span><span class="k">sensitive</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">  </span><span class="o">--</span><span class="n">refinepara</span><span class="w"> </span><span class="n">REFINEMENT_PARAMTER</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">分箱优化参数：</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">deep</span><span class="o">`</span><span class="err">（默认）</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="o">`</span><span class="n">quick</span><span class="o">`</span><span class="w">  
</span></span></span><span class="line"><span class="cl"><span class="w">                        </span><span class="err">例如：</span><span class="o">`--</span><span class="n">refinepara</span><span class="w"> </span><span class="n">quick</span><span class="o">`</span><span class="err">。</span><span class="w">  
</span></span></span></code></pre></td></tr></table>
</div>
</div><p>BASALT支持单一组装文件或多组装文件进行binning；在样品的物种非常相似，但群落中物种丰度不同的情况下，如时间序列样品，多组装文件可大幅度提高binning的效果，但也可能会导致运算时间过长，但作者推荐用多组装文件模式进行分箱分析。如果运算资源不足，推荐使用多组数据进行合并组装的组装文件进行后续的分箱分析。</p>
<ol>
<li>单独使用二代测序数据进行 binning 及修正，比较常用</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -a as1.fa,as2.fa,as3.fa -s ds1_r1.fq,d1_r2.fq/d2_r1.fq,d2_r2.fq -t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>使用二代测序数据及三代测序数据进行 binning 及修正</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -a as1.fa,as2.fa,as3.fa -s ds1_r1.fq,d1_r2.fq/d2_r1.fq,d2_r2.fq -l lr1.fq,lr2.fq -t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>使用 HiFi 数据及二代测序数据进行 binning 及修正</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -a as1.fa -hf hifi1.fq -s ds1_r1.fq,d1_r2.fq -t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="4">
<li>使用 BASALT 进行基因组的获取，并融合已有的基因组数据</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -a as1.fa,as2.fa,as3.fa -s ds1_r1.fq,d1_r2.fq/d2_r1.fq,d2_r2.fq -d m_binset -t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="5">
<li>使用 BASALT 进行外源数据基因组的去冗余</li>
</ol>
<p><strong>(a) 数据输入（Data feeding）</strong></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -s sample1.R1.fq,sample1.R2.fq/sample2.R1.fq,sample2.R2.fq -d mbin,dbin -t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>该命令会生成 <code>Date_feeded</code> 文件夹，其中包含：</p>
<ul>
<li>修改 index 后的组装文件，如 <code>500_mbin.fa</code></li>
<li>基因组文件夹，如 <code>500_mbin.fa_BestBinsSet</code></li>
<li>覆盖率文件，如 <code>Coverage_matrix_for_binning_500_mbin.fa.txt</code></li>
<li>修改格式后的 reads 文件</li>
</ul>
<p><strong>(b) 基因组去冗余（De-replication）</strong></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -b 500_mbin.fa_BestBinsSet,501_dbin.fa_BestBinsSet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-c Coverage_matrix_for_binning_500_mbin.fa.txt,Coverage_matrix_for_binning_501_mbin.fa.txt <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-a 500_mbin.fa,501_mbin.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-s sample1.R1.fq,sample1.R2.fq/sample2.R1.fq,sample2.R2.fq -t <span class="m">60</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>该命令输入多个基因组文件夹、组装文件、reads 和覆盖率文件，最终生成去冗余后的基因组文件夹 <code>BestBinset</code>。</p>
<ol start="6">
<li>使用 BASALT 进行基因组的修正</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -r BestBinset <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-c Coverage_matrix_for_binning_500_mbin.fa.txt,Coverage_matrix_for_binning_501_mbin.fa.txt <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-a 500_mbin.fa,501_mbin.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-s PE_r1_sample1.R1.fq,PE_r2_sample1.R2.fq/PE_r1_sample2.R1.fq,PE_r2_sample2.R2.fq -t <span class="m">60</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>该命令会生成 <code>BestBinset_outlier_refined</code> 文件夹，存储优化后的基因组。</p>
<h3 id="具体示例">具体示例</h3>
<img src="images/640.webp" title=""/>
<h4 id="项目背景">项目背景</h4>
<p>在鸟类生长的不同阶段，采集了若干种鸟类的粪便样本，通过宏基因组二代测序分析它们的微生物组。每个样本根据鸟类不同的生长周期进行了采集，并使用 <strong>metaspades</strong> 软件对数据进行单个样本的组装。在得到单个样本的组装结果后，进一步对同种鸟类在同一生长周期的多个平行样本的数据进行合并组装，以便获取更丰富的群落信息，辅助后续的分箱处理。</p>
<h4 id="组装策略">组装策略</h4>
<ol>
<li><strong>单个样本组装</strong>：使用 <strong>metaspades</strong> 对每个鸟类在不同生长周期的粪便样本进行单样本组装。</li>
<li><strong>多样本合并组装</strong>：将同一生长周期、同种鸟类的多个平行样本进行合并组装，以提高组装的完整性和准确性。</li>
<li><strong>分箱处理</strong>：在完成组装后，进行基于二代测序数据的分箱，获取更准确的基因组信息。</li>
</ol>
<p>以A鸟样本为例</p>
<p>样本数据：</p>
<ul>
<li>
<p><strong>A类鸟阶段1样本</strong>：</p>
<ul>
<li>sampleA_s1_1_R1.fq, sampleA_s1_1_R2.fq</li>
<li>sampleA_s1_2_R1.fq, sampleA_s1_2_R2.fq</li>
<li>sampleA_s1_3_R1.fq, sampleA_s1_3_R2.fq</li>
</ul>
</li>
<li>
<p><strong>A类鸟阶段2样本</strong>：</p>
<ul>
<li>sampleA_s2_1_R1.fq, sampleA_s2_1_R2.fq</li>
<li>sampleA_s2_2_R1.fq, sampleA_s2_2_R2.fq</li>
<li>sampleA_s2_3_R1.fq, sampleA_s2_3_R2.fq</li>
</ul>
</li>
<li>
<p><strong>A类鸟阶段3样本</strong>：</p>
<ul>
<li>sampleA_s3_1_R1.fq, sampleA_s3_1_R2.fq</li>
<li>sampleA_s3_2_R1.fq, sampleA_s3_2_R2.fq</li>
<li>sampleA_s3_3_R1.fq, sampleA_s3_3_R2.fq</li>
</ul>
</li>
</ul>
<p>组装结果：</p>
<ul>
<li><strong>阶段1单样品组装结果</strong>：
<ul>
<li>sampleA_s1_1_assembly.fa, sampleA_s1_2_assembly.fa, sampleA_s1_3_assembly.fa</li>
</ul>
</li>
<li><strong>阶段2单样品组装结果</strong>：
<ul>
<li>sampleA_s2_1_assembly.fa, sampleA_s2_2_assembly.fa, sampleA_s2_3_assembly.fa</li>
</ul>
</li>
<li><strong>阶段3单样品组装结果</strong>：
<ul>
<li>sampleA_s3_1_assembly.fa, sampleA_s3_2_assembly.fa, sampleA_s3_3_assembly.fa</li>
</ul>
</li>
<li><strong>阶段1多样品合并组装结果</strong>：
<ul>
<li>sampleA_s1_assembly.fa</li>
</ul>
</li>
<li><strong>阶段2多样品合并组装结果</strong>：
<ul>
<li>sampleA_s2_assembly.fa</li>
</ul>
</li>
<li><strong>阶段3多样品合并组装结果</strong>：
<ul>
<li>sampleA_s3_assembly.fa</li>
</ul>
</li>
</ul>
<p>分箱命令：</p>
<ul>
<li><strong>阶段1样品分箱命令</strong>：</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">BASALT -a sampleA_s1_assembly.fa, sampleA_s1_1_assembly.fa,sampleA_s1_2_assembly.fa,sampleA_s1_3_assembly.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-s sampleA_s1_1_R1.fq,sampleA_s1_1_R2.fq/sampleA_s1_2_R1.fq,sampleA_s1_2_R2.fq/sampleA_s1_3_R1.fq,sampleA_s1_3_R2.fq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-t <span class="m">60</span> -m <span class="m">250</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="注意事项">注意事项</h3>
<h4 id="输入输出">输入输出</h4>
<p>BASALT的程序输入 -a,-s等参数不是任意文件的路径，而是要把这些文件放置到脚本运行同一文件夹下，因为&rsquo;/&lsquo;已经被用作文件之间的分隔符了。并且BASALT还不能设定输出路径，在输出时候自动将结果及重要过程文件输出至当前工作路径下，建议用户将需要分析的文件单独建立文件夹存放和运行BASALT，以免重复输出覆盖原有结果。用起来有点麻烦。</p>
<h4 id="运行速度">运行速度</h4>
<p>在使用单个组装文件时，BASALT的完整流程耗时通常比metaWRAP长，尤其随着样品复杂度的增加，耗时差异会更加显著。然而，文章中也提到，即便不运行gap filling模块，BASALT在基因组质量和数量上通常优于metaWRAP。实际上，在多个组装文件输入模式下，BASALT的效率反而高于metaWRAP，因为BASALT仅需运行一次，且其单独组装、合并组装和去冗余的流程有助于大大提高非冗余基因组的产出。如果用户对基因组的深度挖掘需求不高，且需要加速分箱过程，以下几种策略可能会有所帮助：</p>
<ol>
<li><strong>使用MetaBAT2 + Semibin2作为初始Binners</strong>：这可以显著加快auto-binning的过程。</li>
<li><strong>只运行auto-binning、bin selection和refinement模块</strong>：跳过gap filling模块可以有效减少运行时间。</li>
<li><strong>仅使用合并组装文件</strong>：在同时输入多个样品时，使用单一的合并组装文件（例如：sampleA1A2A3_assembly.fa）可以减少auto-binning的计算时间，但也可能会减少基因组的产出量。</li>
</ol>
<h2 id="references">References</h2>
<ol>
<li>Qiu, Z., Yuan, L., Lian, CA. et al. BASALT refines binning from metagenomic data and increases resolution of genome-resolved metagenomic analysis. Nat Commun 15, 2179 (2024). <a class="link" href="https://doi.org/10.1038/s41467-024-46539-7"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41467-024-46539-7</a></li>
<li><a class="link" href="https://github.com/EMBL-PKU/BASALT"  target="_blank" rel="noopener"
    >https://github.com/EMBL-PKU/BASALT</a></li>
<li><a class="link" href="https://mp.weixin.qq.com/s/TlxaiQGjoiE3c0Onqatu8A"  target="_blank" rel="noopener"
    >https://mp.weixin.qq.com/s/TlxaiQGjoiE3c0Onqatu8A</a></li>
</ol>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/binning/">Binning</a>
        
            <a href="/Cao-blog/tags/microbiome/">Microbiome</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/binning2/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/binning2/images/metawrap.97b98d34d6d47fb1beaba090bdeee79f_hu_a369a8bdfe123e4a.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 宏基因组分箱（binning）｜2. MetaWRAP实战深入binning"
                        data-key="binning2" 
                        data-hash="md5-l7mNNNbUf7G&#43;q6CQve7nnw==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">宏基因组分箱（binning）｜2. MetaWRAP实战深入binning</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/binning/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/binning/images/binning.5c920b33c9c2dc6daa4357b5aa5a31fc_hu_ab7bca35f351d1cb.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 宏基因组分箱（binning）｜1.Metabat实战了解binning"
                        data-key="binning" 
                        data-hash="md5-XJILM8nC3G2qQ1e1qlox/A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">宏基因组分箱（binning）｜1.Metabat实战了解binning</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（4）"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（4）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-3/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-3/images/fig1.20d52f738dfd506b0754bda8d4b0ead0_hu_d6f6d5c8a568f26d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（3）"
                        data-key="25-nature-3" 
                        data-hash="md5-INUvc439UGsHVL2o1LDq0A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（3）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-2/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-2/images/2-1-2.f13dbba185acaee28c931cc543f77916_hu_94453c680a4ffc60.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（2）"
                        data-key="25-nature-2" 
                        data-hash="md5-8T27oYWsruKMkxzFQ/d5Fg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（2）</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
