<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='了解细菌在各种环境中的生长动态对于各类研究都很重要，本文主要介绍从宏基因组数据推断细菌生长速率的方法和简要应用。'>
<title>从宏基因组量化细菌生长动态</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/growth/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='从宏基因组量化细菌生长动态'>
<meta property='og:description' content='了解细菌在各种环境中的生长动态对于各类研究都很重要，本文主要介绍从宏基因组数据推断细菌生长速率的方法和简要应用。'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/growth/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='microbiome' /><meta property='article:tag' content='growth' /><meta property='article:published_time' content='2024-07-15T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-07-15T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/growth/images/PTR.jpeg' />
<meta name="twitter:title" content="从宏基因组量化细菌生长动态">
<meta name="twitter:description" content="了解细菌在各种环境中的生长动态对于各类研究都很重要，本文主要介绍从宏基因组数据推断细菌生长速率的方法和简要应用。"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/growth/images/PTR.jpeg' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">🐛🐶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/472862038'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/JB-Cao'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >中文</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduciton">Introduciton</a></li>
    <li><a href="#methods">Methods</a>
      <ol>
        <li><a href="#ptr">PTR</a></li>
        <li><a href="#irep">iRep</a></li>
        <li><a href="#demic">DEMIC</a></li>
        <li><a href="#coptr">CoPTR</a></li>
      </ol>
    </li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/growth/">
                <img src="/Cao-blog/p/growth/images/PTR_hu_345dfed7c65ff3b3.jpeg"
                        srcset="/Cao-blog/p/growth/images/PTR_hu_345dfed7c65ff3b3.jpeg 800w, /Cao-blog/p/growth/images/PTR_hu_71256b8ebba67ecb.jpeg 1600w"
                        width="800" 
                        height="651" 
                        loading="lazy"
                        alt="Featured image of post 从宏基因组量化细菌生长动态" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/growth/">从宏基因组量化细菌生长动态</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            了解细菌在各种环境中的生长动态对于各类研究都很重要，本文主要介绍从宏基因组数据推断细菌生长速率的方法和简要应用。
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Jul 15, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    4 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                版权归Cao's blog所有，转载请注明出处
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduciton">Introduciton</h2>
<p>了解细菌在各种环境中的生长动态对于人类健康和环境监测等广泛领域至关重要。传统研究细菌生长的方法往往依赖于培养技术，这不仅耗时，而且对易培养的物种有偏向。然而，随着宏基因组测序技术的兴起，我们现在可以直接从环境样本中探讨整个微生物群落的生长动态。</p>
<p>宏基因组学通过测序样本中所有微生物的集体基因组，提供了微生物群落的全面快照。这种方法使研究人员能够绕过培养步骤，捕捉细菌在自然栖息地中的多样性和动态。通过分析宏基因组数据，我们可以推断细菌的生长速率、种群变化和群落相互作用。</p>
<p>本文主要介绍从宏基因组数据推断细菌生长速率的方法和简要应用。</p>
<h2 id="methods">Methods</h2>
<h3 id="ptr">PTR</h3>
<p>Science 2015: <a class="link" href="https://www.science.org/doi/10.1126/science.aac4812"  target="_blank" rel="noopener"
    >https://www.science.org/doi/10.1126/science.aac4812</a></p>
<p>宏基因组测序增加了我们对微生物组在健康和疾病中的作用的理解，但它只提供了高度动态的生态系统的快照。在这里，作者表明不同微生物基因组的宏基因组测序读数覆盖模式包含单个波谷和单个波峰，后者与细菌复制起点一致。此外，峰和谷之间的测序覆盖率提供了物种生长速率的定量测量。</p>
<img src="images/PTR.jpeg" title=""/>
<p>原理：是否可以通过检查跨细菌基因组的测序读数覆盖模式来从单个宏基因组样本中探测微生物群的生长动态。除了少数例子 之外，大多数细菌都具有单个圆形染色体，它从单个固定起点向单个终点双向复制。因此，在DNA复制过程中，已经被复制叉通过的区域将有两个拷贝，而尚未复制的区域将有一个拷贝。</p>
<p>总而言之，DNA 区域的拷贝数越接近复制起点，该区域的拷贝数就会越高；反之，该区域越接近复制终点，该区域的拷贝数就会越低。因此，复制起点附近的DNA拷贝数与复制终点附近的DNA拷贝数之间的比率，称之为峰谷比（Peak-to-Trough Ratio，PTR），应该反映细菌群体的生长速度。在较高的生长速率下，更大比例的细胞进行 DNA 复制，并且每个细胞中存在更活跃的复制叉。这导致近Origin DNA 和近 Terminus DNA 之间的比率高于 1:1，从而提供群落增长率的定量读数。</p>
<p>该文章在体外和体内、不同生长条件下以及复杂的细菌群落中证明了这一点。对于几种细菌物种，峰谷覆盖率（而非相对丰度）与炎症性肠病和 II 型糖尿病的表现相关。</p>
<p>文章提供的pipeline：<a class="link" href="http://genie.weizmann.ac.il/software/bac_growth.html"  target="_blank" rel="noopener"
    >http://genie.weizmann.ac.il/software/bac_growth.html</a> ，该网页已经打不开了😂。</p>
<h3 id="irep">iRep</h3>
<p>Nature Biotechnology 2016：<a class="link" href="https://www.nature.com/articles/nbt.3704"  target="_blank" rel="noopener"
    >https://www.nature.com/articles/nbt.3704</a></p>
<p>独立于培养物的微生物组研究增加了对微生物群落的复杂性和代谢潜力的理解。然而，为了了解单个微生物组成员对群落功能的贡献，确定哪些细菌正在积极复制非常重要。
作者开发了一种算法 iRep，它使用草稿质量的基因组序列和单时间点宏基因组测序来推断微生物群体的复制率。该算法根据从单个复制起点进行双向基因组复制产生的测序覆盖趋势来计算复制指数 (index of replication, iRep)。
作者应用这种方法来证明人类婴儿服用抗生素后微生物复制率增加。文章还表明，未培养的、与地下水相关的候选门辐射细菌很少在经历地球化学重大变化的地下群落中快速复制。该方法可以应用于任何基因组解析的微生物组研究，以跟踪生物体对不同条件的反应，识别活跃增长的种群并测量复制率以用于建模研究。</p>
<img src="images/iRep.webp" title=""/>
<p>软件：<a class="link" href="https://github.com/christophertbrown/iRep"  target="_blank" rel="noopener"
    >https://github.com/christophertbrown/iRep</a>，亲测可用</p>
<p>iRep 是一种通过单时间点宏基因组测序和草图质量基因组确定细菌复制率的方法。</p>
<ol>
<li>
<p><strong>安装</strong>:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">pip install iRep
</span></span></code></pre></td></tr></table>
</div>
</div><p>注意：iRep 需要 Python 3。依赖项有lmfit，numpy，scipy，pandas，seaborn，matplotlib，cPickle。</p>
</li>
<li>
<p><strong>脚本</strong>:</p>
<ul>
<li><code>iRep</code>: 使用草稿质量的基因组序列测量复制速率</li>
<li><code>iRep_filter.py</code>: 合并和/或过滤 iRep 输出</li>
<li><code>bPTR</code>: 使用完整的基因组序列测量复制速率（修改自 Korem 等人，Science 2015）</li>
<li><code>gc_skew.py</code>: 计算完整基因组序列的 GC 偏斜</li>
</ul>
</li>
<li>
<p><strong>示例用法</strong>:</p>
<ul>
<li>
<p>iRep:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">iRep -f sample_data/l_gasseri.fna -s sample_data/l_gasseri*sam -o test.iRep
</span></span></code></pre></td></tr></table>
</div>
</div><p>注意：为节省时间，可以同时分析多个基因组：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">iRep -f genome1.fna genome2.fna -s sample_data/l_gasseri*sam -o test.iRep
</span></span></code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>bPTR:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">bPTR -f sample_data/l_gasseri.fna -s sample_data/l_gasseri*sam -o test.bPTR.tsv -plot test.bPTR.pdf -m coverage
</span></span></code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>GC 偏斜:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gc_skew.py -f sample_data/l_gasseri.fna
</span></span></code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
<li>
<p><strong>使用说明</strong>:
运行 iRep 和 bPTR 需要每个目标微生物的基因组序列。每个程序都需要单独的 FASTA 文件作为输入。bPTR 需要完整（封闭）的基因组序列，而 iRep 需要高质量的草稿基因组（≥75% 完整性，≤175 个片段/Mbp 序列，≤2% 污染）。iRep 只使用长度≥5 Kbp 的 scaffold，因此基因组完整性估算需要基于通过此阈值的 scaffold。两种方法在基因组从研究样本中组装或已知系统中存在具有高度相似基因组序列的微生物时最为准确。第二组输入是 SAM 格式的映射文件，这些文件应通过使用 Bowtie2 将 DNA 测序读取映射到组装的基因组上生成。提供每个样本生成的 SAM 文件路径。</p>
</li>
</ol>
<p>仅通过所有基因组和映射质量要求的 iRep 值应被视为有效。默认阈值：min cov. = 5，min wins. = 0.98，min r² = 0.9，max fragments/Mbp = 175，GC 校正 min r² = 0.0。</p>
<p>请注意，高水平的菌株变异会影响结果。</p>
<p>iRep 和 bPTR.py 输出包括结果表格（.tsv 文件）和 PDF，显示用于计算复制速率的基因组覆盖测量。
包含许多未映射读取的大型 SAM 文件解析速度较慢，可以使用 <code>shrinksam</code> 快速过滤 SAM 文件中的未映射读取。</p>
<img src="images/iRep_res.png" title=""/>
<p>如果有 &ldquo;ImportError: No module named &lsquo;_tkinter&rsquo;&rdquo; 错误。可以通过修改 bash MPLBACKEND 变量修复：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">export</span> <span class="nv">MPLBACKEND</span><span class="o">=</span><span class="s2">&#34;agg&#34;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>重要提示：
iRep 和 bPTR 都需要有序的 SAM 文件，这些文件可以使用 Bowtie2 的 <code>--reorder</code> 标志生成。有序的 SAM 文件中，表示一组配对读取的两个读取一个接一个排列，这对于这些脚本基于映射质量过滤读取（见 <code>-mm</code> 选项）是必需的。如果 SAM 文件未排序，请确保重新运行 Bowtie 并带上 <code>--reorder</code> 标志，或使用 <code>--sort</code> 选项排序 SAM 文件；否则，脚本在计算覆盖率时会错误选择要包含的映射读取。<code>--sort</code> 选项使用 Unix 排序重新排列 SAM 文件。此选项仅在使用映射质量截止时适用；然而，使用质量截止是推荐的，因为它有助于防止偏离目标的读取映射。</p>
<h3 id="demic">DEMIC</h3>
<p>Nature Methods 2018: <a class="link" href="https://www.nature.com/articles/s41592-018-0182-0"  target="_blank" rel="noopener"
    >https://www.nature.com/articles/s41592-018-0182-0</a></p>
<p>对没有完整基因组序列的物种的微生物生长动态进行准确定量在生物学上很重要，但在宏基因组学中在计算上具有挑战性。在这里，作者提出了微生物群落的动态估计器（DEMIC；https://sourceforge.net/projects/demic/），这是一种基于重叠群和覆盖值的多样本算法，用于推断重叠群与复制起点的相对距离，并得出准确比较样品之间的细菌生长率。作者使用多个合成和真实数据集展示了 DEMIC 对于各种样本大小和组装质量的稳健性能。</p>
<p>软件：<a class="link" href="https://sourceforge.net/projects/demic/"  target="_blank" rel="noopener"
    >https://sourceforge.net/projects/demic/</a>，perl报错未能解决</p>
<p>安装，下载压缩包解压即可：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">wget -c https://master.dl.sourceforge.net/project/demic/DEMIC_v1.0.2.zip?viasf<span class="o">=</span><span class="m">1</span> -O DEMIC_v1.0.2.zip
</span></span><span class="line"><span class="cl">tar -jxvf DEMIC_v1.0.2.zip
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> DEMIC_v1.0.2
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 下载测试数据</span>
</span></span><span class="line"><span class="cl">wget -c https://sourceforge.net/projects/demic/files/test_data.zip?viasf<span class="o">=</span><span class="m">1</span> -O test_data.zip
</span></span><span class="line"><span class="cl">unzip test_data.zip
</span></span></code></pre></td></tr></table>
</div>
</div><p>DEMIC是一种计算工具，用于比较宏基因组样本中细菌的生长速率，基于推测的从复制起点到contig相对距离及其覆盖度。</p>
<p>如有任何关于DEMIC的问题，可以发邮件给工具作者：gy.james@163.com</p>
<ol>
<li>命令和参数</li>
</ol>
<p>如何运行DEMIC：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">perl DEMIC.pl -S /input/sam_dir/ -F /input/fa_dir/ -O /output/dir/
</span></span></code></pre></td></tr></table>
</div>
</div><p>DEMIC的参数如下：</p>
<ul>
<li><code>-S, --sam_dir</code>：输入SAM文件目录（由bowtie2生成）（必需）</li>
<li><code>-F, --fasta_dir</code>：输入FASTA文件目录，每个文件对应一个物种（必需）</li>
<li><code>-O, --out_dir</code>：输出文件目录（默认：./）</li>
<li><code>-W, --window_size</code>：计算覆盖度的窗口大小（默认：5000）</li>
<li><code>-D, --window_step</code>：计算覆盖度的窗口步长（默认：100）</li>
<li><code>-M, --mapq_cutoff</code>：计算覆盖度时的映射质量阈值（默认：5）</li>
<li><code>-L, --mapl_cutoff</code>：计算覆盖度时的映射长度阈值（默认：50）</li>
<li><code>-R, --max_mismatch_ratio</code>：每个读取作为命中的最大错配率（默认：0.03）</li>
<li><code>-G, --log</code>：输出日志文件名（可选）</li>
<li><code>-H, --help</code>：显示帮助信息</li>
<li><code>-T, --thread_num</code>：设置并行运行的线程数（默认：1）</li>
<li><code>-Q, --quiet</code>：运行时保持安静</li>
<li><code>-A, --output_all</code>：运行后保留临时文件（需要更多磁盘空间）</li>
</ul>
<ol start="2">
<li>使用DEMIC的准备工作</li>
</ol>
<p>(1) 共组装
不同样本中的宏基因组读取共同组装。请使用现有算法如MegaHit和Ray-Meta。例如，使用MegaHit从三个样本的宏基因组读取生成contigs：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">cat sample1.fastq sample2.fastq sample3.fastq &gt; samples.fastq
</span></span><span class="line"><span class="cl">megahit -m <span class="m">100000000000</span> -r samples.fastq -o MEGAHIT_assembly
</span></span></code></pre></td></tr></table>
</div>
</div><p>(2) 比对
将宏基因组读取比对到contigs。请使用BowTie2。SAM文件需要用samtools排序。例如，使用以下命令比对sample1的读取并输出到目录SAM/：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">bowtie2-build MEGAHIT_assembly/final.contigs.fa MEGAHIT_assembly/final.contigs.fa
</span></span><span class="line"><span class="cl">bowtie2 -q -x MEGAHIT_assembly/final.contigs.fa -U sample1.fastq -S SAM/sample1.sam
</span></span></code></pre></td></tr></table>
</div>
</div><p>排序比对并输出到SAM_sorted/sample1_sort.sam：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">samtools view -bS SAM/sample1.sam <span class="p">|</span> samtools sort - SAM/sample1_sort
</span></span><span class="line"><span class="cl">samtools view -h SAM/sample1_sort.bam &gt; SAM_sorted/sample1_sort.sam
</span></span></code></pre></td></tr></table>
</div>
</div><p>请为所有样本的.sam文件执行此操作。</p>
<p>(3) 分类
将contigs聚类到组。请使用现有算法如MaxBin和MetaBAT。例如，使用MaxBin将contig聚类输出到maxbin/：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">run_MaxBin.pl -contig MEGAHIT_assembly/final.contigs.fa -out maxbin -reads_list ./reads.list
</span></span></code></pre></td></tr></table>
</div>
</div><p>（<code>./reads.list</code>包含上述三个.fastq文件的路径）</p>
<p>以上步骤是基于组装的宏基因组数据分析中常用的步骤。</p>
<p>DEMIC需要目录maxbin/和SAM_sorted/。</p>
<ol start="3">
<li>运行DEMIC示例</li>
</ol>
<p>(1) 确保已安装并行Perl 5.10.1或更高版本，R 3.3.1或更高版本，并使用Linux或Mac OS X操作系统。</p>
<p>(2) 在R中安装“lme4”和“FactoMineR”包：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;lme4&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;FactoMineR&#34;</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>(3) 在终端进入DEMIC目录，并将测试数据移动到该目录：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> /path/to/DEMIC_v1.0.2/
</span></span><span class="line"><span class="cl">mv /path/to/test_data/ ./
</span></span></code></pre></td></tr></table>
</div>
</div><p>测试数据包括来自PTRC研究的三个样本的测序数据比对，这些数据格式化为目录SAM_sorted/中的已排序.sam。目录maxbin/中包括两个contig聚类，分别对应C. rodentium和E. faecalis。</p>
<p>(4) 运行DEMIC命令：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">perl DEMIC.pl -S test_data/SAM_sorted/ -F test_data/maxbin/ -O DEMIC/
</span></span></code></pre></td></tr></table>
</div>
</div><p>！！暂时会报错，且没有具体报错原因</p>
<p>(5) DEMIC将在大约1分钟内完成工作，并在目录DEMIC/中生成输出文件。</p>
<p>在该目录中，可以查看文件<code>all_PTR.txt</code>的详细信息：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ cat DEMIC/all_PTR.txt
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">                Sample1 Sample2 Sample3
</span></span><span class="line"><span class="cl">ContigCluster1  1.5549  2.3313  1.8395
</span></span><span class="line"><span class="cl">ContigCluster2  2.2430  1.7119  2.2856
</span></span></code></pre></td></tr></table>
</div>
</div><p>输出文件易于理解。例如，C. rodentium（ContigCluster1）在Sample2中的生长速率最高。</p>
<ol start="4">
<li>常见问题</li>
</ol>
<p>(1) 如何在DEMIC中使用多线程？
DEMIC支持多线程。使用多个CPU或内核时，请使用参数<code>-T</code>指定线程数。虽然优化了DEMIC的RAM使用，但使用多线程仍会略微增加RAM使用。建议选择不超过SAM文件数量和contig聚类数量的线程数。</p>
<p>(2) 如何设计实验和分析流程以更好地使用DEMIC？
虽然DEMIC适用于各种条件，但当实验和分析流程设计得更好时，它的性能可能更佳。</p>
<ul>
<li><strong>样本数量</strong>：DEMIC基于多个样本，对于包含六个或更多样本的物种，其准确性更高。建议以合理和可比的深度测序超过六个样本。不同样本可以来自对照组、重复实验、不同个体或条件。</li>
<li><strong>组装器</strong>：组装方法对binning方法和DEMIC的性能很重要。能够为不同深度和物种生成相对较长contig且错误较少的组装器更为优选。在测试中使用了MEGAHIT，表现良好。</li>
<li><strong>分类器</strong>：尽管DEMIC可以通过去除潜在污染物纯化contig聚类，但其仍受binning质量（如完整性）的影响。对于低或中等复杂度的微生物群落，建议使用MaxBin。</li>
</ul>
<h3 id="coptr">CoPTR</h3>
<p>Genome Research 2022: <a class="link" href="https://genome.cshlp.org/content/32/3/558.full"  target="_blank" rel="noopener"
    >https://genome.cshlp.org/content/32/3/558.full</a></p>
<p>细菌基因组的测序覆盖模式（通过峰谷比（PTR）总结）已被证明可以准确反映微生物的生长速率，揭示微生物动力学和宿主-微生物相互作用的新方面。在这里，作者介绍Compute PTR (CoPTR)：一种用于从完整的参考基因组和组装中计算PTR的工具。</p>
<p>使用简单和复杂群落中的生长实验的模拟和数据，作者表明 CoPTR 比当前最先进的技术更准确，同时还提供了更多的总体 PTR 估计。作者进一步发展了一种理论，正式化了 PTR 的生物学解释。作者使用 2935 个物种的参考数据库，将 CoPTR 应用于来自 106 名炎症性肠病患者的 1304 个宏基因组样本的病例对照研究。作者表明，增长率是个性化的，仅与相对丰度松散相关，并且与疾病状态相关。最后，作者展示了如何将 PTR 与相对丰度和代谢组学相结合来研究它们对微生物组的影响。</p>
<img src="images/F1.large.jpg" title=""/>
<p>软件：<a class="link" href="https://github.com/tyjo/coptr"  target="_blank" rel="noopener"
    >https://github.com/tyjo/coptr</a></p>
<p>CoPTR 是一种工具，用于根据完整参考基因组和草图组件的宏基因组测序数据计算峰谷比。 CoPTR 将来自多个宏基因组样本和参考数据库的 fastq 文件作为输入。它输出样本中每个物种的 log2 PTR。</p>
<p>快速使用：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 项目下载</span>
</span></span><span class="line"><span class="cl">git clone https://github.com/tyjo/coptr
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> coptr
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 测试数据</span>
</span></span><span class="line"><span class="cl">wget https://dl.dropboxusercontent.com/s/wrsxjimr6l96lcq/example-data.tar.gz
</span></span><span class="line"><span class="cl">tar -xzvf example-data.tar.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 设置环境：</span>
</span></span><span class="line"><span class="cl">conda env create -f coptr.yml
</span></span><span class="line"><span class="cl">conda activate coptr
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 软件安装</span>
</span></span><span class="line"><span class="cl">pip install .
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 索引参考数据库：</span>
</span></span><span class="line"><span class="cl">coptr index example-data/ref-db example-data/ref-db/example-index
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 地图显示与参考相反：</span>
</span></span><span class="line"><span class="cl">coptr map example-data/ref-db/example-index example-data/fastq example-data/bam
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 提取读取位置：</span>
</span></span><span class="line"><span class="cl">coptr extract example-data/bam example-data/coverage-maps
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#估计ptrs：</span>
</span></span><span class="line"><span class="cl"><span class="c1">#请注意，最小读取标志是可选的。我们建议使用默认设置（5000次读取）。</span>
</span></span><span class="line"><span class="cl">coptr estimate example-data/coverage-maps out.csv --min-reads <span class="m">2500</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Output</span>
</span></span><span class="line"><span class="cl">cat out.csv
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">log2<span class="o">(</span>PTR<span class="o">)</span>:genome_id/sample_id,ERR969281,ERR969282,ERR969283,ERR969285,ERR969286,ERR969428,ERR969429,ERR969430
</span></span><span class="line"><span class="cl">l-gasseri-ref,,,,,,1.1840987863325785,1.1945539660363145,1.2879271469720541
</span></span><span class="line"><span class="cl">e-coli-mag,1.2078623467253466,1.0375575947553943,0.9433005522894075,0.759132363901812,0.7846476652840171,,,
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="references">References</h2>
<ol>
<li>Tal Korem et al., Growth dynamics of gut microbiota in health and disease inferred from single metagenomic samples.Science349,1101-1106 (2015). DOI:10.1126/science.aac4812</li>
<li>Brown, C., Olm, M., Thomas, B. et al. Measurement of bacterial replication rates in microbial communities. Nat Biotechnol 34, 1256–1263 (2016). <a class="link" href="https://doi.org/10.1038/nbt.3704"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/nbt.3704</a></li>
<li>Gao, Y., Li, H. Quantifying and comparing bacterial growth dynamics in multiple metagenomic samples. Nat Methods 15, 1041–1044 (2018). <a class="link" href="https://doi.org/10.1038/s41592-018-0182-0"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41592-018-0182-0</a></li>
<li>Tyler A. Joseph et al., Accurate and robust inference of microbial growth dynamics from metagenomic sequencing reveals personalized growth rates. Genome Res. 2022. 32: 558-568 (2022). doi:
10.1101/gr.275533.121</li>
</ol>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/microbiome/">Microbiome</a>
        
            <a href="/Cao-blog/tags/growth/">Growth</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（4）"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（4）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-3/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-3/images/fig1.20d52f738dfd506b0754bda8d4b0ead0_hu_d6f6d5c8a568f26d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（3）"
                        data-key="25-nature-3" 
                        data-hash="md5-INUvc439UGsHVL2o1LDq0A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（3）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-2/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-2/images/2-1-2.f13dbba185acaee28c931cc543f77916_hu_94453c680a4ffc60.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（2）"
                        data-key="25-nature-2" 
                        data-hash="md5-8T27oYWsruKMkxzFQ/d5Fg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（2）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature/images/1-1-1.7ec2e930cb5bff1118bfef6c33cf66bd_hu_e11798ac155eaa9a.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（1）"
                        data-key="25-nature" 
                        data-hash="md5-fsLpMMtb/xEYv&#43;9sM89mvQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（1）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/indoor/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/indoor/images/ind3.f88b22cb27b61a7f362968c78b715b40_hu_3690543364814c81.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 室内微生物组相关研究"
                        data-key="indoor" 
                        data-hash="md5-&#43;Isiyye2Gn82KWjHi3FbQA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">室内微生物组相关研究</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
