<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='基因组规模代谢模型（GEM）是通过整合基因组数据、代谢途径和生化反应信息，模拟和分析生物体代谢网络的一种强大工具，这里简单介绍一下'>
<title>从宏基因组构建基因组规模代谢模型（GEM）</title>

<link rel='canonical' href='/p/gem/'>

<link rel="stylesheet" href="/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='从宏基因组构建基因组规模代谢模型（GEM）'>
<meta property='og:description' content='基因组规模代谢模型（GEM）是通过整合基因组数据、代谢途径和生化反应信息，模拟和分析生物体代谢网络的一种强大工具，这里简单介绍一下'>
<meta property='og:url' content='/p/gem/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='GEM' /><meta property='article:tag' content='microbiome' /><meta property='article:published_time' content='2024-07-17T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-07-17T00:00:00&#43;00:00'/><meta property='og:image' content='/p/gem/images/carveme.jpeg' />
<meta name="twitter:title" content="从宏基因组构建基因组规模代谢模型（GEM）">
<meta name="twitter:description" content="基因组规模代谢模型（GEM）是通过整合基因组数据、代谢途径和生化反应信息，模拟和分析生物体代谢网络的一种强大工具，这里简单介绍一下"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='/p/gem/images/carveme.jpeg' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/">
                
                    
                    
                    
                        
                        <img src="/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">🐛🐶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/142776729'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://asa12138.github.io/'
                        target="_blank"
                        title="Homepage"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-world-pin"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M20.972 11.291a9 9 0 1 0 -8.322 9.686" /><path d="M3.6 9h16.8" /><path d="M3.6 15h8.9" /><path d="M11.5 3a17 17 0 0 0 0 18" /><path d="M12.5 3a16.986 16.986 0 0 1 2.578 9.018" /><path d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z" /><path d="M19 18v.01" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com/Asa12138_'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="/" selected>English</option>
                        
                            <option value="/zh-cn/" >中文</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#methods">Methods</a>
      <ol>
        <li><a href="#metagem">metaGEM</a></li>
        <li><a href="#carveme">CarveMe</a>
          <ol>
            <li><a href="#安装">安装</a></li>
            <li><a href="#使用">使用</a></li>
          </ol>
        </li>
        <li><a href="#memote">MEMOTE</a>
          <ol>
            <li><a href="#安装-1">安装</a></li>
            <li><a href="#使用-1">使用</a></li>
          </ol>
        </li>
        <li><a href="#smetana">SMETANA</a>
          <ol>
            <li><a href="#安装-2">安装</a></li>
            <li><a href="#使用-2">使用</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#应用">应用</a></li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/p/gem/">
                <img src="/p/gem/images/carveme_hu_c1b5d7ca5872ddd.jpeg"
                        srcset="/p/gem/images/carveme_hu_c1b5d7ca5872ddd.jpeg 800w, /p/gem/images/carveme_hu_767aa4beaa08c7e.jpeg 1600w"
                        width="800" 
                        height="441" 
                        loading="lazy"
                        alt="Featured image of post 从宏基因组构建基因组规模代谢模型（GEM）" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/p/gem/">从宏基因组构建基因组规模代谢模型（GEM）</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            基因组规模代谢模型（GEM）是通过整合基因组数据、代谢途径和生化反应信息，模拟和分析生物体代谢网络的一种强大工具，这里简单介绍一下
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Jul 17, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    4 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                版权归Cao's blog所有，转载请注明出处
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>基因组规模代谢模型（Genome-scale metabolic model，GEM）是通过整合基因组数据、代谢途径和生化反应信息，模拟和分析生物体代谢网络的一种强大工具。GEM不仅能帮助我们理解生物体的代谢功能和代谢网络的复杂性，还可以用于预测基因敲除效应、代谢工程、药物开发等多种应用。我们应该可以从宏基因组组装的MAGs出发来做GEM建模。</p>
<p>多个基因组规模的代谢重建软件平台已经开发出来并且正在不断更新。这些工具已广泛应用于重建数百种微生物的代谢模型，从重要的人类病原体到工业相关物种。2019年，Mendoza等人对以下7个基因组规模重建软件平台进行了系统评估，文章发表在genome biology上[1]，每个工具在不同的方面显示出优势和不足。</p>
<img src="images/GEM_table.png" title=""/>
19年有另一篇Genome biology回顾了当前重建的 GEM 并讨论了它们的应用[2]，包括化学品和材料生产的菌株开发、病原体药物靶向、酶功能预测、泛反应组分析、多个细胞或生物体之间的相互作用建模以及了解人类疾病。其中还介绍了以下参考GEM数据库：
<img src="images/GEM_table2.png" title=""/>
<p>本文将通过metaGEM 流程中的几个软件工具，介绍从宏基因组数据做 GEM 建模的原理的步骤。</p>
<h2 id="methods">Methods</h2>
<h3 id="metagem">metaGEM</h3>
<img src="images/metaGEM.png" title=""/>
<p><a class="link" href="https://github.com/franciscozorrilla/metaGEM/"  target="_blank" rel="noopener"
    >https://github.com/franciscozorrilla/metaGEM/</a></p>
<p>metaGEM 是一个 Snakemake 工作流程[3]，集成了一系列现有的生物信息学和代谢建模工具，旨在预测微生物组细菌群落内的代谢相互作用。根据整个宏基因组鸟枪数据集，重建宏基因组组装基因组 (MAG)，然后将其转换为基因组规模代谢模型 (GEM) 以进行计算机模拟。其他输出包括丰度估计、分类分配、增长率估计、全基因组分析和真核 MAG 鉴定。</p>
<p>很明显，绿色方框内的MAG生成模块是我们之前已经介绍过的<a class="link" href="https://asa-blog.netlify.app/p/binning2/"  target="_blank" rel="noopener"
    >MetaWRAP实战深入binning</a>；橙色方框是鉴定真核MAG模块；蓝色方框是MAG注释模块。这几步和GEM建模没有直接的关系，所以不在这里描述。关键的是粉色方框内的GEM建模模块。metaGEM 使用了以下几个工具：CarveMe，MEMOTE和SMETANA。</p>
<p>如果按照metaGEM教程来安装这个流程会比较麻烦，需要conda安装大量的软件，metaWRAP一个流程就很难装了😂。他这里强行建了三个环境，还把mamba装在非base环境中，容易把我们已有的环境搞乱。</p>
<p>所以我们假定我们已经有组装好的MAG开始分析么，关注最重要的粉色模块即可，分别介绍CarveMe，SMETANA和MEMOTE三个软件。</p>
<h3 id="carveme">CarveMe</h3>
<img src="images/carveme.jpeg" title=""/>
<p>CarveMe 是一个基于 Python 的工具，用于基因组规模的代谢模型重建[4]。输入为 MAGs 的蛋白文件，
通过 BiGG 数据库构建一个通用的代谢模型，然后通用模型根据基因组文件转换为特定模型，输出构建好的代谢模型(GEMs)，是一个自上而下的流程。</p>
<p>主页：<a class="link" href="https://github.com/cdanielmachado/carveme"  target="_blank" rel="noopener"
    >https://github.com/cdanielmachado/carveme</a></p>
<h4 id="安装">安装</h4>
<ol>
<li>使用pip包管理器可以轻松安装CarveMe：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">pip install carveme
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>另外，还需手动安装diamond序列比对工具：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">conda install -c bioconda diamond
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>安装优化求解器：推荐安装IBM CPLEX或Gurobi，并获取相应的学术许可证（演示版无效），或者用4的替代。
安装CPLEX后，还需安装CPLEX的Python API。</li>
</ol>
<p>Cplex是IBM公司开发的一款商业版的优化引擎，当然也有免费版，只不过免费版的有规模限制，不能求解规模过大的问题。这里可以去<a class="link" href="https://community.ibm.com/community/user/ai-datascience/blogs/xavier-nodet1/2020/07/09/cplex-free-for-students"  target="_blank" rel="noopener"
    >申请学术教育版</a>，但我们学校的邮箱居然不可以申请😭。</p>
<p>然后我试了一下Gurobi：<a class="link" href="https://www.gurobi.com/downloads/gurobi-software/"  target="_blank" rel="noopener"
    >https://www.gurobi.com/downloads/gurobi-software/</a>，用学校的邮箱申请免费试用，可以参考<a class="link" href="https://blog.csdn.net/weixin_46530492/article/details/134396504"  target="_blank" rel="noopener"
    >https://blog.csdn.net/weixin_46530492/article/details/134396504</a>。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 下载解压</span>
</span></span><span class="line"><span class="cl">tar xvfz gurobi11.0.2_linux64.tar.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 设置环境变量：</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export GUROBI_HOME=~/biosoft/gurobi1103/linux64/&#39;</span> &gt;&gt;~/.bashrc      
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export PATH=$PATH:~/biosoft/gurobi1103/linux64/bin&#39;</span> &gt;&gt;~/.bashrc
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export LD_LIBRARY_PATH=~/biosoft/gurobi1103/linux64/lib:$LD_LIBRARY_PATH&#39;</span> &gt;&gt;~/.bashrc
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#安装gurobipy</span>
</span></span><span class="line"><span class="cl">python -m pip install gurobipy
</span></span></code></pre></td></tr></table>
</div>
</div><p>申请之后会运行一行命令得到gurobi.lic激活。</p>
<ol start="4">
<li>作为替代方案，可以使用开源的SCIP求解器，但速度较慢（每次执行至少需10分钟）：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">conda install -c conda-forge pyscipopt
</span></span></code></pre></td></tr></table>
</div>
</div><p>执行时选择正确的求解器，默认是cplex，可以指定gurobi和scip：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve <span class="o">[</span>ARGS<span class="o">]</span> --solver scip
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="使用">使用</h4>
<ol start="5">
<li>构建模型：CarveMe提供了一个非常简单的命令行接口来构建模型。最基本的用法是：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 我们去下载一个E.coli的蛋白序列做测试</span>
</span></span><span class="line"><span class="cl">wget -c https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_protein.faa.gz
</span></span><span class="line"><span class="cl">gunzip GCF_000005845.2_ASM584v2_protein.faa.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">carve GCF_000005845.2_ASM584v2_protein.faa
</span></span></code></pre></td></tr></table>
</div>
</div><p>这将从基因组文件构建一个全基因组代谢模型，试了一下，用SCIP求解器要11min，gurobi花了2min，400MB内存。</p>
<p>默认情况下，CarveMe期望输入的是蛋白质FASTA文件。或者，你也可以提供DNA序列：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve --dna genome.fna
</span></span></code></pre></td></tr></table>
</div>
</div><p>注意，不支持原始基因组文件，FASTA文件必须分成单独的基因。</p>
<ol start="6">
<li>可以为输出文件指定不同的名称或目录：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa --output model.xml
</span></span></code></pre></td></tr></table>
</div>
</div><p>如果你想生成一个压缩的SBML文件，只需更改扩展名（libSBML自动支持）：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa -o model.xml.gz
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="7">
<li>如果不是自己提供基因组数据，你也可以提供一个NCBI RefSeq访问代码。这将自动下载序列并构建模型：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve --refseq GCF_000005845.2 -o ecoli_k12_mg1655.xml
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="8">
<li>如果你已经下载了多个基因组序列，你可以运行递归模式，在一次调用中构建多个模型。这将启动多个并行进程，如果你在多核CPU或计算集群中运行CarveMe，可以减少整体计算时间：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve -r myfolder/*.faa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#与-o结合以更改输出文件夹：</span>
</span></span><span class="line"><span class="cl">carve -r myfolder/*.faa -o mymodels/
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="9">
<li>缺口填补：CarveMe尝试仅从遗传证据预测一个生物体的摄取和分泌能力，并且会生成一个不进行特定培养基缺口填补的模拟就绪模型。然而，有些情况下你希望确保模型能够在一个或多个实验验证的培养基中再现生长。例如，你可以确保模型在M9和LB培养基中再现生长：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa --gapfill M9,LB
</span></span></code></pre></td></tr></table>
</div>
</div><p>请参阅高级用法部分了解如何提供你自己的培养基成分。如果你已经有了一个模型，只是想填补缺口，你可以使用gapfill实用函数：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">gapfill model.xml -m M9 -o new_model.xml
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="10">
<li>请注意，使用CarveMe生成的模型不会初始化任何培养基成分。你可以通过设置交换反应的流量边界来定义生物体的生长环境，以匹配相应的培养基成分。或者，你可以告诉CarveMe希望模型附带一个预定义的培养基成分：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa --init M9
</span></span></code></pre></td></tr></table>
</div>
</div><p>注意，这不会填补模型的缺口，只会定义模拟目的的外部环境。要同时填补缺口并初始化模型以用于所需的培养基，你必须结合两个标志：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">carve genome.faa -g M9 -I M9
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="11">
<li>微生物群落：CarveMe允许从单一物种模型生成微生物群落模型。最基本的用法是：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">merge_community organism_1.xml organism_2.xml ... organism_N.xml -o community.xml
</span></span><span class="line"><span class="cl"><span class="c1"># 或更简单地：</span>
</span></span><span class="line"><span class="cl">merge_community *.xml -o community.xml
</span></span></code></pre></td></tr></table>
</div>
</div><p>这将生成一个包含每个生物体分配到自己隔室的群落的SBML文件，并生成一个通用的群落生物量方程。你可以像使用任何普通的约束模型一样将合并的模型导入任何模拟工具，并应用不同类型的模拟方法（FBA，FVA等）。你可以像单物种重建过程中一样用预定义的培养基初始化群落：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">merge_community <span class="o">[</span>input files<span class="o">]</span> -I M9
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="memote">MEMOTE</h3>
<p>主页：<a class="link" href="https://github.com/opencobra/memote"  target="_blank" rel="noopener"
    >https://github.com/opencobra/memote</a></p>
<p>MEMOTE全称基因组规模代谢模型测试套件，目标是推动代谢模型构建社区实现两个重大转变：</p>
<ol>
<li>模型应进行版本控制，以便可以跟踪并在必要时恢复更改。理想情况下，模型应通过公共存储库（如GitHub）提供，使其他研究人员能够检查、共享和贡献模型。</li>
<li>模型应达到某些标准和最低功能，以造福社区和研究。</li>
</ol>
<p>MEMOTE工具执行以下四个子功能：</p>
<ol>
<li>创建模型的git存储库框架。</li>
<li>通过测试套件运行当前模型，代表社区标准。</li>
<li>生成详细报告，以视觉上吸引人的方式展示测试结果。</li>
<li>重新计算代谢模型版本控制历史的测试统计数据。</li>
</ol>
<p>生成的存储库可以轻松集成到持续集成测试提供者（如Travis CI）中，这意味着每次将模型更改推送到GitHub时，测试套件将自动运行，并通过GitHub页面查看报告。</p>
<h4 id="安装-1">安装</h4>
<ol>
<li>稳定版本：要安装memote，请在终端中运行以下命令：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ pip install memote
</span></span></code></pre></td></tr></table>
</div>
</div><p>这是安装memote的首选方法，因为它将始终安装最新的稳定版本。</p>
<ol start="2">
<li>从源码安装：memote的源码可以从Github仓库下载。可以克隆公共仓库：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ git clone https://github.com/opencobra/memote.git
</span></span></code></pre></td></tr></table>
</div>
</div><p>或者下载压缩包：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ curl -OL https://github.com/opencobra/memote/archive/master.zip
</span></span></code></pre></td></tr></table>
</div>
</div><p>下载源码文件后，可以使用以下命令安装：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ pip install .
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="使用-1">使用</h4>
<p>安装后，memote 可以通过三种不同的方式使用：</p>
<ol>
<li>作为临时模型评估的基准测试工具。当使用 memote 对模型进行基准测试时，测试会运行一次，并生成一份描述现状的报告。</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">memote report snapshot --filename <span class="s2">&#34;report.html&#34;</span> GCF_000005845.2_ASM584v2_protein.xml
</span></span></code></pre></td></tr></table>
</div>
</div><img src="images/ReportSnapshot.webp" title=""/>
<p>虽然建议使用 html 报告，但频繁用户可能希望跳过 html 报告的生成，而仅使用原始控制台输出来获得快速概览。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">memote run path/to/model.xml
</span></span></code></pre></td></tr></table>
</div>
</div><p>在控制台中，您首先会看到所有测试的列表及其 pytest 状态，读数为“通过”、“失败”、“跳过”或“错误”，然后是一个简短的回溯，显示某些测试失败的原因：</p>
<ol start="2">
<li>同样，可以测试多个模型并生成比较差异的报告。</li>
</ol>
<p>要比较两个（或更多）模型的性能，请在终端中运行以下命令：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">memote report diff path/to/model1.xml path/to/model2.xml <span class="o">[</span>path/to/model3.xml ...<span class="o">]</span>
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>作为一个自动化测试套件，memote 有助于跟踪版本控制存储库中的增量模型更改，并且可以根据需要启用连续测试和报告。这个我暂时用不到</li>
</ol>
<h3 id="smetana">SMETANA</h3>
<p>物种代谢相互作用分析 (SMETANA) 是一个基于 Python 的命令行工具，用于分析微生物群落。
它以微生物群落（来自 SBML 格式的基因组规模代谢模型的集合）作为输入，并计算描述群落成员之间交叉喂养相互作用的潜力的几个指标[5]。</p>
<p>主页：<a class="link" href="https://github.com/SBRG/smetana"  target="_blank" rel="noopener"
    >https://github.com/SBRG/smetana</a></p>
<h4 id="安装-2">安装</h4>
<p>SMETANA 目前支持 Python 2.7 和 Python 3.6，可用于所有主要操作系统。</p>
<p>可以使用 pip 包管理器轻松安装它：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ pip install smetana
</span></span></code></pre></td></tr></table>
</div>
</div><p>这也会自动安装其他依赖项：</p>
<ul>
<li>framed</li>
<li>pandas</li>
</ul>
<p>此外，必须安装受支持的求解器之一：</p>
<ul>
<li>CPLEX</li>
<li>Gurobi</li>
</ul>
<p>请注意，虽然这两种求解器均受支持，但 SMETANA 主要针对 CPLEX 进行了优化。</p>
<h4 id="使用-2">使用</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ smetana -h
</span></span><span class="line"><span class="cl">用法: smetana <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>-c COMMUNITIES.TSV<span class="o">]</span> <span class="o">[</span>-o OUTPUT<span class="o">]</span> <span class="o">[</span>--flavor FLAVOR<span class="o">]</span> <span class="o">[</span>-m MEDIA<span class="o">]</span> <span class="o">[</span>--mediadb MEDIADB<span class="o">]</span> <span class="o">[</span>-g <span class="p">|</span> -d <span class="p">|</span> -a ABIOTIC <span class="p">|</span> -b BIOTIC<span class="o">]</span>
</span></span><span class="line"><span class="cl">               <span class="o">[</span>-p P<span class="o">]</span> <span class="o">[</span>-n N<span class="o">]</span> <span class="o">[</span>-v<span class="o">]</span> <span class="o">[</span>-z<span class="o">]</span> <span class="o">[</span>--solver SOLVER<span class="o">]</span> <span class="o">[</span>--molweight<span class="o">]</span> <span class="o">[</span>--exclude EXCLUDE<span class="o">]</span> <span class="o">[</span>--no-coupling<span class="o">]</span>
</span></span><span class="line"><span class="cl">               MODELS <span class="o">[</span>MODELS ...<span class="o">]</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">计算单一或多个微生物群落的SMETANA得分。
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">位置参数:
</span></span><span class="line"><span class="cl">  MODELS
</span></span><span class="line"><span class="cl">                        多个单一物种模型（一个或多个文件）。
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">                        你可以使用通配符，例如：models/*.xml，并选择用引号保护，以避免自动bash扩展（对于长列表，这会更快）：<span class="s2">&#34;models/*.xml&#34;</span>。
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">可选参数:
</span></span><span class="line"><span class="cl">  -h, --help            显示帮助信息并退出
</span></span><span class="line"><span class="cl">  -c COMMUNITIES.TSV, --communities COMMUNITIES.TSV
</span></span><span class="line"><span class="cl">                        为多个（子）群落运行SMETANA。
</span></span><span class="line"><span class="cl">                        群落必须在一个两列的制表符分隔文件中指定，包含群落和生物体标识符。
</span></span><span class="line"><span class="cl">                        生物体标识符应与SBML文件中的文件名匹配（无扩展名）。
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">                        示例:
</span></span><span class="line"><span class="cl">                            community1	organism1
</span></span><span class="line"><span class="cl">                            community1	organism2
</span></span><span class="line"><span class="cl">                            community2	organism1
</span></span><span class="line"><span class="cl">                            community2	organism3
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  -o OUTPUT, --output OUTPUT
</span></span><span class="line"><span class="cl">                        输出文件前缀。
</span></span><span class="line"><span class="cl">  --flavor FLAVOR       输入文件的预期SBML风格（cobra或fbc2）。
</span></span><span class="line"><span class="cl">  -m MEDIA, --media MEDIA
</span></span><span class="line"><span class="cl">                        为指定培养基运行SMETANA（逗号分隔）。
</span></span><span class="line"><span class="cl">  --mediadb MEDIADB     培养基数据库文件。
</span></span><span class="line"><span class="cl">  -g, --global          运行全局分析（MIP/MRO），更快。
</span></span><span class="line"><span class="cl">  -d, --detailed        运行详细的SMETANA分析，较慢。
</span></span><span class="line"><span class="cl">  -a ABIOTIC, --abiotic ABIOTIC
</span></span><span class="line"><span class="cl">                        使用给定的化合物列表测试无生物扰动。
</span></span><span class="line"><span class="cl">  -b BIOTIC, --biotic BIOTIC
</span></span><span class="line"><span class="cl">                        使用给定的物种列表测试生物扰动。
</span></span><span class="line"><span class="cl">  -p P                  同时扰动的成分数量（默认: 1）。
</span></span><span class="line"><span class="cl">  -n N
</span></span><span class="line"><span class="cl">                        每个群落的随机扰动实验次数（默认: 1）。
</span></span><span class="line"><span class="cl">                        <span class="nv">选择n</span> <span class="o">=</span> 0将精确测试所有单一物种/化合物扰动。
</span></span><span class="line"><span class="cl">  -v, --verbose         切换到详细模式。
</span></span><span class="line"><span class="cl">  -z, --zeros           包括得分为零的条目。
</span></span><span class="line"><span class="cl">  --solver SOLVER       更改默认求解器（当前选项: <span class="s1">&#39;gurobi&#39;</span>, <span class="s1">&#39;cplex&#39;</span>）。
</span></span><span class="line"><span class="cl">  --molweight           使用分子量最小化（推荐）。
</span></span><span class="line"><span class="cl">  --exclude EXCLUDE     从计算中排除的化合物列表（例如：无机化合物）。
</span></span><span class="line"><span class="cl">  --no-coupling         不计算物种耦合得分。
</span></span></code></pre></td></tr></table>
</div>
</div><ol>
<li>单一群落
要在单一群落上运行SMETANA，只需提供一组以SBML格式表示的单一物种模型：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ smetana species1.xml species2.xml ...
</span></span><span class="line"><span class="cl">$ smetana *.xml
</span></span></code></pre></td></tr></table>
</div>
</div><p>尝试一下：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">smetana GCF_000005845.2_ASM584v2_protein.xml GCF_015291705.1_ASM1529170v1_protein.xml --solver gurobi
</span></span></code></pre></td></tr></table>
</div>
</div><p>两个物种1分钟就算完了，结果：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">community	medium	size	mip	mro
</span></span><span class="line"><span class="cl">all	complete	2	4	0.6428571428571429
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>多个群落
可以在单行代码中对多个群落运行SMETANA。为此，需要创建一个描述每个群落组成的表格。该表应为长格式，使用制表符分隔，包含两列：群落ID和物种ID。后者应与SBML文件名（不带扩展名）匹配。例如：</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">community1    organism1
</span></span><span class="line"><span class="cl">community1    organism2
</span></span><span class="line"><span class="cl">community2    organism1
</span></span><span class="line"><span class="cl">community2    organism3
</span></span></code></pre></td></tr></table>
</div>
</div><p>创建此文件后，按以下方式调用SMETANA：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ smetana *.xml -c communities.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>培养基成分
默认情况下，SMETANA将在完全培养基成分上模拟每个群落。要在不同培养基上测试群落，提供培养基列表和包含培养基成分的文件。可以在单个命令中指定多个培养基，SMETANA将依次运行每个培养基。</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">$ smetana *.xml -m M9,LB --mediadb library.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><p>如何创建自己的库文件，请查看此示例。</p>
<ol start="4">
<li>全局模式与详细模式
SMETANA有两种主要运行模式：</li>
</ol>
<ul>
<li>全局模式 (-g, &ndash;global): 运行MIP/MRO，速度更快，推荐用于分析多个群落。</li>
<li>详细模式 (-d, &ndash;detailed): 计算所有物种间相互作用（速度较慢），详细信息请参阅算法。</li>
</ul>
<p>全局模式SMETANA 包括两种分析社区全局属性的算法：
MRO（代谢资源重叠）：计算物种对相同代谢物的竞争程度。
MIP（代谢相互作用潜力）：计算物种可以共享多少代谢物以减少对外部资源的依赖。</p>
<p>详细模式SMETANA 包含多个分数来表征社区中的物种互动。这包括：
SCS（物种耦合评分）：衡量一个物种在其他物种存在的情况下生存的依赖性
MUS（代谢物摄取分数）：衡量一个物种生存所需摄取代谢物的频率
MPS（代谢物生产得分）：衡量一个物种产生代谢物的能力
SMETANA：个体 smetana 评分是上述 3 个评分的组合，它给出了交叉喂养相互作用的确定性度量（物种 A 从物种 B 接收代谢物 X）。</p>
<h2 id="应用">应用</h2>
<img src="images/41587_2022_1628_Fig1_HTML.webp" title=""/>
<p>2023年的一篇Nature Biotechnology，对7,302 种人类微生物的基因组规模代谢重建，用于个性化医疗[6]。</p>
<p>人类微生物群影响多种常用处方药的功效和安全性。设计结合微生物代谢的精准医学方法需要应变和分子解析的、可扩展的计算模型。在这里，通过大幅扩展的版本扩展了之前人类肠道微生物基因组规模代谢重建的资源。</p>
<p>AGORA2（通过重建和分析组装肠道生物体，版本 2）包含 7,302 种菌株，包括 98 种药物的菌株解析药物降解和生物转化能力，并根据比较基因组学和文献检索进行了广泛的策划。微生物重建在三个独立组装的实验数据集上表现非常好，准确度为 0.72 至 0.84，超越了其他重建资源，并以 0.81 的准确度预测了已知的微生物药物转化。</p>
<p>作者证明，AGORA2 通过预测 616 名结直肠癌患者和对照组肠道微生物组的药物转化潜力，实现了个性化、菌株解析的建模，这些微生物组在个体之间存在很大差异，并与年龄、性别、体重指数和疾病阶段相关。 AGORA2 作为人类微生物组的知识库，为宿主与微生物组代谢相互作用的个性化、预测性分析铺平了道路。</p>
<h2 id="references">References</h2>
<ol>
<li>Mendoza, S.N., Olivier, B.G., Molenaar, D. et al. A systematic assessment of current genome-scale metabolic reconstruction tools. Genome Biol 20, 158 (2019). <a class="link" href="https://doi.org/10.1186/s13059-019-1769-1"  target="_blank" rel="noopener"
    >https://doi.org/10.1186/s13059-019-1769-1</a></li>
<li>Gu, C., Kim, G.B., Kim, W.J. et al. Current status and applications of genome-scale metabolic models. Genome Biol 20, 121 (2019). <a class="link" href="https://doi.org/10.1186/s13059-019-1730-3"  target="_blank" rel="noopener"
    >https://doi.org/10.1186/s13059-019-1730-3</a></li>
<li>metaGEM: reconstruction of genome scale metabolic models directly from metagenomes
Francisco Zorrilla, Filip Buric, Kiran R Patil, Aleksej Zelezniak
Nucleic Acids Research, 2021; gkab815, <a class="link" href="https://doi.org/10.1093/nar/gkab815"  target="_blank" rel="noopener"
    >https://doi.org/10.1093/nar/gkab815</a></li>
<li>D. Machado et al, &ldquo;Fast automated reconstruction of genome-scale metabolic models for microbial species and communities&rdquo;, Nucleic Acids Research, gky537, 2018. doi: <a class="link" href="https://doi.org/10.1093/nar/gky537"  target="_blank" rel="noopener"
    >https://doi.org/10.1093/nar/gky537</a></li>
<li>Aleksej Zelezniak et al, Metabolic dependencies drive species co-occurrence in diverse microbial communities, PNAS, May 4, 2015, 112 (20) 6449-6454, <a class="link" href="https://doi.org/10.1073/pnas.1421834112"  target="_blank" rel="noopener"
    >https://doi.org/10.1073/pnas.1421834112</a></li>
<li>Heinken, A., Hertel, J., Acharya, G. et al. Genome-scale metabolic reconstruction of 7,302 human microorganisms for personalized medicine. Nat Biotechnol 41, 1320–1331 (2023). <a class="link" href="https://doi.org/10.1038/s41587-022-01628-0"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41587-022-01628-0</a></li>
</ol>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/tags/gem/">GEM</a>
        
            <a href="/tags/microbiome/">Microbiome</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（4）"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（4）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/25-nature-3/">
        
        
            <div class="article-image">
                <img src="/p/25-nature-3/images/fig1.20d52f738dfd506b0754bda8d4b0ead0_hu_d6f6d5c8a568f26d.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（3）"
                        data-key="25-nature-3" 
                        data-hash="md5-INUvc439UGsHVL2o1LDq0A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（3）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/25-nature-2/">
        
        
            <div class="article-image">
                <img src="/p/25-nature-2/images/2-1-2.f13dbba185acaee28c931cc543f77916_hu_94453c680a4ffc60.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（2）"
                        data-key="25-nature-2" 
                        data-hash="md5-8T27oYWsruKMkxzFQ/d5Fg==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（2）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/25-nature/">
        
        
            <div class="article-image">
                <img src="/p/25-nature/images/1-1-1.7ec2e930cb5bff1118bfef6c33cf66bd_hu_e11798ac155eaa9a.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（1）"
                        data-key="25-nature" 
                        data-hash="md5-fsLpMMtb/xEYv&#43;9sM89mvQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（1）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/p/indoor/">
        
        
            <div class="article-image">
                <img src="/p/indoor/images/ind3.f88b22cb27b61a7f362968c78b715b40_hu_3690543364814c81.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 室内微生物组相关研究"
                        data-key="indoor" 
                        data-hash="md5-&#43;Isiyye2Gn82KWjHi3FbQA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">室内微生物组相关研究</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
