<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='病毒的分类注释一直是个难题，PhaGCN2/vConTACT2基于病毒基因组间共享的蛋白质簇（PCs）构建的基因共享网络来进行分类注释，这里简要介绍一下用法。'>
<title>使用PhaGCN2/vConTACT2进行病毒分类注释</title>

<link rel='canonical' href='https://jb-cao.github.io/Cao-blog/p/vcontact2/'>

<link rel="stylesheet" href="/Cao-blog/scss/style.min.f0ecc11a39cd68bcc1eeb399f153637c1d516d2f399c93e65d25e7606d35ca1e.css"><meta property='og:title' content='使用PhaGCN2/vConTACT2进行病毒分类注释'>
<meta property='og:description' content='病毒的分类注释一直是个难题，PhaGCN2/vConTACT2基于病毒基因组间共享的蛋白质簇（PCs）构建的基因共享网络来进行分类注释，这里简要介绍一下用法。'>
<meta property='og:url' content='https://jb-cao.github.io/Cao-blog/p/vcontact2/'>
<meta property='og:site_name' content='Cao&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='virus' /><meta property='article:published_time' content='2024-08-14T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2024-08-14T00:00:00&#43;00:00'/><meta property='og:image' content='https://jb-cao.github.io/Cao-blog/p/vcontact2/images/PhaGCN2.jpeg' />
<meta name="twitter:title" content="使用PhaGCN2/vConTACT2进行病毒分类注释">
<meta name="twitter:description" content="病毒的分类注释一直是个难题，PhaGCN2/vConTACT2基于病毒基因组间共享的蛋白质簇（PCs）构建的基因共享网络来进行分类注释，这里简要介绍一下用法。"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='https://jb-cao.github.io/Cao-blog/p/vcontact2/images/PhaGCN2.jpeg' />
    <link rel="shortcut icon" href="/favicon/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png">
<link rel="manifest" href="favicon/site.webmanifest">
<link rel="shortcut icon" href="favicon/android-chrome-512x512.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/Cao-blog/">
                
                    
                    
                    
                        
                        <img src="/Cao-blog/img/guodong1_hu_72aa525a63157f1a.jpg" width="300"
                            height="300" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">🐛🐶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/Cao-blog">Cao&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://space.bilibili.com/142776729'
                        target="_blank"
                        title="Bilibili"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-brand-bilibili"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 10a4 4 0 0 1 4 -4h10a4 4 0 0 1 4 4v6a4 4 0 0 1 -4 4h-10a4 4 0 0 1 -4 -4v-6z" /><path d="M8 3l2 3" /><path d="M16 3l-2 3" /><path d="M9 13v-2" /><path d="M15 11v2" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://asa12138.github.io/'
                        target="_blank"
                        title="Homepage"
                        rel="me"
                    >
                        
                        
                            <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-world-pin"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M20.972 11.291a9 9 0 1 0 -8.322 9.686" /><path d="M3.6 9h16.8" /><path d="M3.6 15h8.9" /><path d="M11.5 3a17 17 0 0 0 0 18" /><path d="M12.5 3a16.986 16.986 0 0 1 2.578 9.018" /><path d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z" /><path d="M19 18v.01" /></svg>
                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com/Asa12138_'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/Cao-blog/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/mindmap/' >
                
                
                
                    <svg  xmlns="http://www.w3.org/2000/svg"  width="24"  height="24"  viewBox="0 0 24 24"  fill="none"  stroke="currentColor"  stroke-width="2"  stroke-linecap="round"  stroke-linejoin="round"  class="icon icon-tabler icons-tabler-outline icon-tabler-sitemap"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z" /><path d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1" /><path d="M12 9l0 3" /></svg>
                
                <span>Mindmap</span>
            </a>
        </li>
        
        
        <li >
            <a href='/Cao-blog/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="https://jb-cao.github.io/Cao-blog/" selected>English</option>
                        
                            <option value="https://jb-cao.github.io/Cao-blog/zh-cn/" >中文</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#vcontact2">vConTACT2</a>
      <ol>
        <li><a href="#安装">安装</a></li>
        <li><a href="#运行">运行</a>
          <ol>
            <li><a href="#主要参数">主要参数：</a></li>
            <li><a href="#输入形式">输入形式：</a></li>
            <li><a href="#测试数据">测试数据：</a></li>
          </ol>
        </li>
        <li><a href="#输出">输出</a>
          <ol>
            <li><a href="#genome_by_genome_overviewcsv">genome_by_genome_overview.csv</a></li>
            <li><a href="#c1ntw">c1.ntw</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#phagcn2">PhaGCN2</a>
      <ol>
        <li><a href="#安装-1">安装</a></li>
        <li><a href="#运行-1">运行</a>
          <ol>
            <li><a href="#示例">示例：</a></li>
          </ol>
        </li>
      </ol>
    </li>
    <li><a href="#references">References</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="has-image main-article">
    <header class="article-header">
        <div class="article-image">
            <a href="/Cao-blog/p/vcontact2/">
                <img src="/Cao-blog/p/vcontact2/images/PhaGCN2_hu_36ff472aafa69a1b.jpeg"
                        srcset="/Cao-blog/p/vcontact2/images/PhaGCN2_hu_36ff472aafa69a1b.jpeg 800w, /Cao-blog/p/vcontact2/images/PhaGCN2_hu_2cb45decefdf48f4.jpeg 1600w"
                        width="800" 
                        height="265" 
                        loading="lazy"
                        alt="Featured image of post 使用PhaGCN2/vConTACT2进行病毒分类注释" />
                
            </a>
        </div>
    

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/Cao-blog/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/Cao-blog/p/vcontact2/">使用PhaGCN2/vConTACT2进行病毒分类注释</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            病毒的分类注释一直是个难题，PhaGCN2/vConTACT2基于病毒基因组间共享的蛋白质簇（PCs）构建的基因共享网络来进行分类注释，这里简要介绍一下用法。
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Aug 14, 2024</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    4 minute read
                </time>
            </div>
        
        
        

           <div>
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



              <time class="article-words">
                版权归Cao's blog所有，转载请注明出处
              </time>
           </div>
        
    </footer>
    
    
    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>在微生物群落的研究中，分类和注释数量庞大的未培养古细菌和细菌病毒一直是一个难题，主要原因是缺乏统一的分类框架。</p>
<p>目前，用于病毒分类的多种基于基因组的方法已经被提出，主要集中在细菌、古细菌和真核生物病毒的分类上。</p>
<ol>
<li>
<p><strong>蛋白序列比较</strong>：早期方法使用完整基因组的蛋白序列进行配对比较，生成“噬菌体蛋白质树”。但因“镶嵌性”特征影响分类界限，这一方法未被广泛采用。</p>
</li>
<li>
<p><strong>共享基因的比例</strong>：另一些方法则基于共享基因的比例和同源基因的百分比定义病毒分类。然而，由于病毒进化模式变化大，这些方法未能有效处理所有已知的病毒群体。</p>
</li>
<li>
<p><strong>GBDP</strong>：基于基因组Blast距离的系统发育算法，如VICTOR工具，结合系统发育和聚类方法进行病毒分类。该方法的可扩展性有限，且对缺乏参考基因组的病毒分类效果有限。</p>
</li>
<li>
<p><strong>基因共享网络</strong>：基于病毒基因组间共享的蛋白质簇（PCs）构建的基因共享网络，已被证明与ICTV认可的分类基本一致。vConTACT v.1.0应用该方法实现了75%的一致性，并在全球海洋、淡水和土壤研究中用于大规模病毒分类。</p>
</li>
</ol>
<p>尽管这些方法各有优缺点，但基因共享网络逐渐成为病毒分类的重要工具，尤其在缺乏参考基因组的情况下发挥了关键作用。</p>
<h2 id="vcontact2">vConTACT2</h2>
<p>vConTACT v.2.0 提供了一种可扩展的网络化解决方案，通过分析全基因组共享的基因模式，结合层次聚类和置信评分，实现病毒分类。它不仅能够成功复现已有的病毒分类，还可以高效处理大型宏基因组数据集，成为病毒分类的强大工具，适用于各种环境中的病毒研究。文章于2019年发表在Nature Biotechnology上[1]。</p>
<img src="images/vcontact2.webp" title=""/>
在网络为基础的基因组分类算法中，相关的基因组通过多个边紧密连接。vConTACT v.2.0使用了一种新的两步聚类算法，以提高分类的准确性和鲁棒性。首先，基于蛋白质簇（PC）的马尔可夫聚类算法 (MCL) 被用于初步分类。然后，使用ClusterONE替代MCL以定义病毒簇（VC），并通过层次聚类进一步细分网络问题区域。此外，v.2.0 引入了拓扑和分类基础的置信度分数，以帮助识别和优化分类任务。
<p>软件主页：<a class="link" href="https://github.com/Hocnonsense/vcontact2"  target="_blank" rel="noopener"
    >https://github.com/Hocnonsense/vcontact2</a></p>
<blockquote>
<p>注意：由于资助结束，vConTACT2 不再被积极开发，尽管作者仍在修复错误并解决出现的其他问题。</p></blockquote>
<h3 id="安装">安装</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 注意使用python3.7而不是直接指定python=3，现在python版本太新了会出问题</span>
</span></span><span class="line"><span class="cl">conda create --name vContact2 <span class="nv">python</span><span class="o">=</span>3.7
</span></span><span class="line"><span class="cl"><span class="nb">source</span> activate vContact2
</span></span><span class="line"><span class="cl">conda install -y -c bioconda vcontact2 mcl blast diamond
</span></span></code></pre></td></tr></table>
</div>
</div><p>再安装一下依赖的cluster_one：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># 下载聚类软件，移动到conda/bin路径 (可使用win下载代替)</span>
</span></span><span class="line"><span class="cl">wget -c http://www.paccanarolab.org/static_content/clusterone/cluster_one-1.0.jar
</span></span><span class="line"><span class="cl">java -jar cluster_one-1.0.jar -h
</span></span><span class="line"><span class="cl">chmod +x ./cluster_one-1.0.jar
</span></span><span class="line"><span class="cl">mv ./cluster_one-1.0.jar ~/miniconda3/bin/
</span></span></code></pre></td></tr></table>
</div>
</div><p>不然会报错：<code>ERROR:vcontact2: Could not find ClusterONE java file.</code></p>
<p>vContact2自带数据库，在<code>~/miniconda3/envs/vContact2/lib/python3.7/site-packages/vcontact2/data/</code>目录下。</p>
<h3 id="运行">运行</h3>
<h4 id="主要参数">主要参数：</h4>
<ul>
<li><strong>-r RAW_PROTEINS, &ndash;raw-proteins RAW_PROTEINS</strong>：输入FASTA格式的蛋白质文件。如果同时提供<code>--proteins-fn</code>，vConTACT将从蛋白质簇(PC)生成前开始处理。</li>
<li><strong>&ndash;rel-mode {BLASTP, Diamond, MMSeqs2}</strong>：选择用于创建蛋白质相似性边文件的方法。默认使用Diamond。</li>
<li><strong>-b BLAST_FP, &ndash;blast-fp BLAST_FP</strong>：输入BLAST结果文件(CSV或TSV格式)，用于生成蛋白质簇。提供此选项时，将从PC生成步骤开始。</li>
<li><strong>-p PROTEINS_FP, &ndash;proteins-fp PROTEINS_FP</strong>：提供一个文件，将蛋白质名称与基因组名称链接。如果提供了<code>--blast-fp</code>，则从PC生成步骤开始。</li>
<li><strong>&ndash;db {None, ProkaryoticViralRefSeq85-ICTV, &hellip;}</strong>：选择参考数据库用于去新蛋白质簇生成。默认使用<code>ProkaryoticViralRefSeq85-ICTV</code>。</li>
<li><strong>&ndash;pcs-mode {ClusterONE, MCL}</strong>：选择用于蛋白质簇生成的算法，默认为MCL。</li>
<li><strong>&ndash;vcs-mode {ClusterONE, MCL}</strong>：选择用于病毒簇生成的算法，默认为ClusterONE。</li>
<li><strong>&ndash;c1-bin CLUSTER_ONE</strong>：指定ClusterONE可执行文件的路径。</li>
<li><strong>&ndash;blastp-bin BLASTP_FP</strong>：指定BLASTP可执行文件的路径。</li>
<li><strong>&ndash;diamond-bin DIAMOND_FP</strong>：指定DIAMOND可执行文件的路径。</li>
<li><strong>-o OUTPUT_DIR, &ndash;output-dir OUTPUT_DIR</strong>：指定输出目录，默认<code>vContact_Output</code>。</li>
<li><strong>-t THREADS, &ndash;threads THREADS</strong>：设置使用的CPU数量，默认32个。</li>
</ul>
<h4 id="输入形式">输入形式：</h4>
<ol>
<li>FASTA格式的氨基酸文件。</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">&gt;ref|NP_039777.1| ORF B-251 [Sulfolobus spindle-shaped virus 1]
</span></span><span class="line"><span class="cl">MVRNMKMKKSNEWLWLGTKIINAHKTNGFESAIIFGKQGTGKTTYALKVAKEVYQRLGHE
</span></span><span class="line"><span class="cl">PDKAWELALDSLFFELKDALRIMKIFRQNDRTIPIIIFDDAGIWLQKYLWYKEEMIKFYR
</span></span><span class="line"><span class="cl">IYNIIRNIVSGVIFTTPSPNDIAFYVREKGWKLIMITRNGRQPDGTPKAVAKIAVNKITI
</span></span><span class="line"><span class="cl">IKGKITNKMKWRTVDDYTVKLPDWVYKEYVERRKVYEEKLLEELDEVLDSDNKTENPSNP
</span></span><span class="line"><span class="cl">SLLTKIDDVTR
</span></span><span class="line"><span class="cl">&gt;ref|NP_039778.1| ORF D-335 [Sulfolobus spindle-shaped virus 1]
</span></span><span class="line"><span class="cl">MTKDKTRYKYGDYILRERKGRYYVYKLEYENGEVKERYVGPLADVVESYLKMKLGVVGDT
</span></span><span class="line"><span class="cl">PLQADPPGFEPGTSGSGGGKEGTERRKIALVANLRQYATDGNIKAFYDYLMNERGISEKT
</span></span><span class="line"><span class="cl">AKDYINAISKPYKETRDAQKAYRLFARFLASRNIIHDEFADKILKAVKVKKANADIYIPT
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>一个“基因到基因组”的映射文件，采用tsv（制表符）或csv（逗号）分隔格式。</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">protein_id,contig_id,keywords
</span></span><span class="line"><span class="cl">ref|NP_039777.1|,Sulfolobus spindle-shaped virus 1,ORF B-251
</span></span><span class="line"><span class="cl">ref|NP_039778.1|,Sulfolobus spindle-shaped virus 1,ORF D-335
</span></span><span class="line"><span class="cl">ref|NP_039779.1|,Sulfolobus spindle-shaped virus 1,ORF E-54
</span></span><span class="line"><span class="cl">ref|NP_039780.1|,Sulfolobus spindle-shaped virus 1,ORF F-92
</span></span><span class="line"><span class="cl">ref|NP_039781.1|,Sulfolobus spindle-shaped virus 1,ORF D-244
</span></span><span class="line"><span class="cl">ref|NP_039782.1|,Sulfolobus spindle-shaped virus 1,ORF E-178
</span></span><span class="line"><span class="cl">ref|NP_039783.1|,Sulfolobus spindle-shaped virus 1,ORF F-93
</span></span><span class="line"><span class="cl">ref|NP_039784.1|,Sulfolobus spindle-shaped virus 1,ORF E-51
</span></span><span class="line"><span class="cl">ref|NP_039785.1|,Sulfolobus spindle-shaped virus 1,ORF E-96
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="测试数据">测试数据：</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">git clone https://github.com/Hocnonsense/vcontact2.git
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> vcontact2/test_data
</span></span><span class="line"><span class="cl">vcontact2 --raw-proteins test_data/VIRSorter_genomes.faa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --proteins-fp test_data/VIRSorter_genomes_g2g.csv <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --db <span class="s1">&#39;ProkaryoticViralRefSeq211-Merged&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --output-dir vConTACT2_Results
</span></span></code></pre></td></tr></table>
</div>
</div><p>运行时间还挺久的，输入文件有11个病毒contig，共433个gene。给了4核16G内存，结果跑了6485s，平均占用10.6GB内存。</p>
<p>应该要给多一点线程。</p>
<h3 id="输出">输出</h3>
<p>vConTACT2 生成了很多输出文件，其中大多数是临时或中间文件，对一般用户没有用处。最重要的文件是网络和注释文件。</p>
<h4 id="genome_by_genome_overviewcsv">genome_by_genome_overview.csv</h4>
<p>包含参考基因组的所有分类信息，以及所有聚类信息（初始 VC (VC_22)、精炼 VC (VC_22_1)）、置信度指标和杂项分数。</p>
<p>一个重要的注意事项是，用户序列不包含分类信息。这意味着每个用户都需要找到他们感兴趣的基因组并检查参考基因组是否位于同一 VC 中。</p>
<ul>
<li>如果用户基因组与参考基因组位于同一 VC 子簇内，则用户基因组属于同一属的可能性非常高。</li>
<li>如果用户基因组与参考基因组位于同一 VC 但不在同一子簇中，则两个基因组很可能在大致属亚科水平上相关。</li>
<li>如果同一 VC 或 VC 子簇中没有参考基因组，那么它们很可能在属水平上根本不相关。也就是说它们有可能在更高的分类水平上相关（亚科、科、目）。</li>
</ul>
<p>每列名称：</p>
<ul>
<li>1	Genome	基因组/序列名</li>
<li>2-4	Order/Family/Genus	目/科/属</li>
<li>5	preVC	初始病毒聚类</li>
<li>6	VC Status	病毒聚类状态</li>
<li>7	VC	病毒聚类</li>
<li>8	VC Size	病毒聚类数</li>
<li>9	Quality	质量值</li>
<li>10	Adjusted P-value	调整后的P值</li>
<li>11	VC Avg Distance	病毒聚类平均距离</li>
<li>12	Topology Confidence Score	拓扑学置信度</li>
<li>13	Genus Confidence Score	属级分类置信度</li>
<li>14-16	VC Orders/Families/Genra	病毒聚类目/科/属数</li>
</ul>
<h4 id="c1ntw">c1.ntw</h4>
<p>包含高于显着性阈值的所有基因组对的源/目标/边缘权重信息，该显着性阈值由这两个基因组共享 N 个基因的概率确定。该文件中的最小值必须大于最小显着性阈值（默认值：1）。</p>
<p>要在 Gephi 或 Cytoscape 中创建网络图形，用户需要将此文件导入到他们喜欢的程序中。
导入后，用户可以添加一个“注释文件” ，可以是genome_by_genome_overview.csv。
每个基因组的注释信息将被添加到网络中的每个节点/基因组。然后，用户可以通过注释文件中的任意属性对网络图形进行着色。</p>
<blockquote>
<p>注意：很多时候，用户会注意到他们的基因组连接到网络中的另一个（可能是参考）基因组，但这两个基因组不会位于同一个 VC 子簇中，甚至不会位于同一个 VC 中。这并不意味着它们没有相关性，只是意味着它们的基因共享比例不够大，无法属于同一属。他们在亚科或科层面上可能有很大的关联。</p></blockquote>
<p>可在<a class="link" href="https://dx.doi.org/10.17504/protocols.io.x5xfq7n"  target="_blank" rel="noopener"
    >https://dx.doi.org/10.17504/protocols.io.x5xfq7n</a>找到详细的说明。</p>
<img src="images/bbcza.png" title=""/>
<h2 id="phagcn2">PhaGCN2</h2>
<img src="images/PhaGCN2.jpeg" title=""/>
<p>PhaGCN2是一种可以快速分类病毒序列的工具，能够在科水平上进行分类，并支持所有科之间关系的可视化。文章于2022年发表在Briefings in Bioinformatics上[2]。</p>
<p>PhaGCN2的性能与其他先进的病毒分类工具（如vConTACT2、CAT和VPF-Class）进行比较显示，PhaGCN2在病毒分类的精度和召回率上大大提高，并且显著增加了可分类病毒序列的数量。它将全球海洋病毒组数据库（Global Ocean Virome）中的可分类序列数量提高了四倍，并且能够对超过90%的肠道噬菌体数据库（Gut Phage Database）进行分类。PhaGCN2使得国际病毒分类委员会数据库（ICTV）的大规模、自动化扩展成为可能。</p>
<p>软件主页：<a class="link" href="https://github.com/KennthShang/PhaGCN2.0"  target="_blank" rel="noopener"
    >https://github.com/KennthShang/PhaGCN2.0</a></p>
<h3 id="安装-1">安装</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">git clone https://github.com/KennthShang/PhaGCN2.0.git
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> PhaGCN2.0
</span></span><span class="line"><span class="cl">rm supplementary<span class="se">\ </span>file/ __pycache__/ pred/ final_prediction.csv -rf
</span></span><span class="line"><span class="cl">vi run_KnowledgeGraph.py <span class="c1"># 把第169行注释掉，因为不需要再建一次数据库。</span>
</span></span><span class="line"><span class="cl">conda env create -f environment.yaml -n phagcn2
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 准备数据库</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> database
</span></span><span class="line"><span class="cl">tar -zxvf ALL_protein.tar.gz
</span></span><span class="line"><span class="cl">diamond makedb --in ALL_protein.fasta -d database.dmnd
</span></span><span class="line"><span class="cl">diamond blastp --sensitive -d database.dmnd -q ALL_protein.fasta -o database.self-diamond.tab
</span></span><span class="line"><span class="cl">awk <span class="s1">&#39;$1!=$2 {{print $1,$2,$11}}&#39;</span> database.self-diamond.tab &gt; database.self-diamond.tab.abc
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ..
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="运行-1">运行</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="cl">$ python run_Speed_up.py -h
</span></span><span class="line"><span class="cl">usage: run_Speed_up.py [-h] [--contigs CONTIGS] [--len LEN]
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">manual to this script
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional arguments:
</span></span><span class="line"><span class="cl">  -h, --help         show this help message and exit
</span></span><span class="line"><span class="cl">  --contigs CONTIGS
</span></span><span class="line"><span class="cl">  --len LEN
</span></span></code></pre></td></tr></table>
</div>
</div><p>该程序有两个参数：</p>
<ul>
<li><code>--contigs</code> 是 contigs 文件的路径。</li>
<li><code>--len</code> 是您要预测的重叠群的长度。正如我们的论文所示，随着重叠群长度的增加，查全率和查准率也会增加。我们建议您根据需要选择合适的长度。默认长度为 8000bp。支持的最短长度为1700bp。输出文件为final_prediction.csv。此 csv 文件中有三列：“contig_name、median_file_name、prediction”。</li>
</ul>
<h4 id="示例">示例：</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">conda activate phagcn2
</span></span><span class="line"><span class="cl"><span class="nb">export</span> <span class="nv">MKL_SERVICE_FORCE_INTEL</span><span class="o">=</span><span class="m">1</span> <span class="c1">#要设置一下这个</span>
</span></span><span class="line"><span class="cl">python run_Speed_up.py --contigs contigs.fa --len <span class="m">8000</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p><strong>注意，该程序没有指定输出路程，会在当前目录生成，每次重跑会覆盖之前的内容😂，而且因为它的环境路径也不是绝对的，所以只能在PhaGCN2.0目录下跑，所以也不能同时跑多个任务</strong>，作者暂时也没有修改这个问题：<a class="link" href="https://github.com/KennthShang/PhaGCN2.0/issues/10"  target="_blank" rel="noopener"
    >https://github.com/KennthShang/PhaGCN2.0/issues/10</a>。</p>
<p>所以最好在运行该程序之前先切换到相应的输出目录（把运行文件全部拷贝到该目录），或者每次运行完把结果移动到输出目录。</p>
<p>每次运行完把结果再移动到输出目录的策略还是有不足，会导致无法同时运行多个任务，所以选择前者。</p>
<p>我们可以看一下run_Speed_up.py具体的几个步骤：</p>
<ol>
<li>diamond，blastp 数据库准备，这里只需要运行一遍就好了，不知道为啥要放在run_Speed_up.py里每次都运行，至少还需要几十分钟呢。</li>
<li>把我们输入的contigs分割为每1000条序列一个子文件，过滤掉小于8000的序列，放在Split_files/下。</li>
<li>循环每个子文件：
<ol>
<li>把子文件mv到input/下，运行run_CNN.py，会用到CNN_Classifier/目录，得到Cyber_data/contig.F</li>
<li>运行run_KnowledgeGraph.py，生成中间文件single_contig/，all_proteins/，network/，输出到out/和Cyber_data/</li>
<li>运行run_GCN.py，得到prediction.csv，</li>
<li>整理子文件输出，放在pred/，删除中间文件</li>
</ol>
</li>
<li>合并所有子文件输出，运行run_network.py</li>
</ol>
<p>所以还是选择自己写一个pipeline来跑吧，方便指定输出位置以及同时跑多个任务，不用run_Speed_up.py：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span><span class="lnt">50
</span><span class="lnt">51
</span><span class="lnt">52
</span><span class="lnt">53
</span><span class="lnt">54
</span><span class="lnt">55
</span><span class="lnt">56
</span><span class="lnt">57
</span><span class="lnt">58
</span><span class="lnt">59
</span><span class="lnt">60
</span><span class="lnt">61
</span><span class="lnt">62
</span><span class="lnt">63
</span><span class="lnt">64
</span><span class="lnt">65
</span><span class="lnt">66
</span><span class="lnt">67
</span><span class="lnt">68
</span><span class="lnt">69
</span><span class="lnt">70
</span><span class="lnt">71
</span><span class="lnt">72
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="c1"># 打印帮助信息</span>
</span></span><span class="line"><span class="cl">usage<span class="o">()</span> <span class="o">{</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Usage: </span><span class="nv">$0</span><span class="s2"> -p &lt;phaGCN_dir&gt; -i &lt;input_file&gt; -o &lt;output_dir&gt;&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nb">exit</span> <span class="m">1</span>
</span></span><span class="line"><span class="cl"><span class="o">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 解析命令行参数</span>
</span></span><span class="line"><span class="cl"><span class="k">while</span> <span class="nb">getopts</span> <span class="s2">&#34;:p:i:o:&#34;</span> opt<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="k">case</span> <span class="s2">&#34;</span><span class="si">${</span><span class="nv">opt</span><span class="si">}</span><span class="s2">&#34;</span> in
</span></span><span class="line"><span class="cl">        p<span class="o">)</span>
</span></span><span class="line"><span class="cl">            <span class="nv">phaGCN_dir</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">            <span class="p">;;</span>
</span></span><span class="line"><span class="cl">        i<span class="o">)</span>
</span></span><span class="line"><span class="cl">            <span class="nv">input</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">            <span class="p">;;</span>
</span></span><span class="line"><span class="cl">        o<span class="o">)</span>
</span></span><span class="line"><span class="cl">            <span class="nv">output</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">            <span class="p">;;</span>
</span></span><span class="line"><span class="cl">        *<span class="o">)</span>
</span></span><span class="line"><span class="cl">            usage
</span></span><span class="line"><span class="cl">            <span class="p">;;</span>
</span></span><span class="line"><span class="cl">    <span class="k">esac</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 检查是否提供了所有参数</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -z <span class="s2">&#34;</span><span class="si">${</span><span class="nv">phaGCN_dir</span><span class="si">}</span><span class="s2">&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> -z <span class="s2">&#34;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&#34;</span> <span class="o">]</span> <span class="o">||</span> <span class="o">[</span> -z <span class="s2">&#34;</span><span class="si">${</span><span class="nv">output</span><span class="si">}</span><span class="s2">&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">    usage
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 将路径转换为绝对路径</span>
</span></span><span class="line"><span class="cl"><span class="nv">phaGCN_dir</span><span class="o">=</span><span class="k">$(</span><span class="nb">cd</span> <span class="s2">&#34;</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="nv">$phaGCN_dir</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span> <span class="o">&amp;&amp;</span> <span class="nb">pwd</span><span class="k">)</span>/<span class="k">$(</span>basename <span class="s2">&#34;</span><span class="nv">$phaGCN_dir</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="nv">input</span><span class="o">=</span><span class="k">$(</span><span class="nb">cd</span> <span class="s2">&#34;</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="nv">$input</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span> <span class="o">&amp;&amp;</span> <span class="nb">pwd</span><span class="k">)</span>/<span class="k">$(</span>basename <span class="s2">&#34;</span><span class="nv">$input</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="nv">output</span><span class="o">=</span><span class="k">$(</span><span class="nb">cd</span> <span class="s2">&#34;</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="nv">$output</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span> <span class="o">&amp;&amp;</span> <span class="nb">pwd</span><span class="k">)</span>/<span class="k">$(</span>basename <span class="s2">&#34;</span><span class="nv">$output</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 检查输出目录是否存在且非空</span>
</span></span><span class="line"><span class="cl"><span class="k">if</span> <span class="o">[</span> -d <span class="s2">&#34;</span><span class="nv">$output</span><span class="s2">&#34;</span> <span class="o">]</span> <span class="o">&amp;&amp;</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="k">$(</span>ls -A <span class="nv">$output</span><span class="k">)</span><span class="s2">&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="s2">&#34;Error: Output directory </span><span class="nv">$output</span><span class="s2"> already exists and is not empty.&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="nb">exit</span> <span class="m">1</span>
</span></span><span class="line"><span class="cl"><span class="k">fi</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 创建输出目录并进入</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="s2">&#34;</span><span class="nv">$output</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> <span class="s2">&#34;</span><span class="nv">$output</span><span class="s2">&#34;</span> <span class="o">||</span> <span class="nb">exit</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 复制 Python 脚本和 C 相关内容</span>
</span></span><span class="line"><span class="cl">cp <span class="s2">&#34;</span><span class="si">${</span><span class="nv">phaGCN_dir</span><span class="si">}</span><span class="s2">&#34;</span>/*.py ./
</span></span><span class="line"><span class="cl">cp -r <span class="s2">&#34;</span><span class="si">${</span><span class="nv">phaGCN_dir</span><span class="si">}</span><span class="s2">/C&#34;</span>* ./
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 创建 database 目录的符号链接</span>
</span></span><span class="line"><span class="cl">ln -s <span class="s2">&#34;</span><span class="si">${</span><span class="nv">phaGCN_dir</span><span class="si">}</span><span class="s2">/database/&#34;</span> ./
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 创建 input 目录并复制输入文件</span>
</span></span><span class="line"><span class="cl">mkdir input/
</span></span><span class="line"><span class="cl">cp <span class="s2">&#34;</span><span class="nv">$input</span><span class="s2">&#34;</span> input/
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 运行各个 Python 脚本</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;Running CNN...&#34;</span>
</span></span><span class="line"><span class="cl">python run_CNN.py
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;Running KnowledgeGraph...&#34;</span>
</span></span><span class="line"><span class="cl">mkdir network
</span></span><span class="line"><span class="cl">python run_KnowledgeGraph.py
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;Running GCN...&#34;</span>
</span></span><span class="line"><span class="cl">python run_GCN.py
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;All tasks completed.&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 删除拷贝过来的脚本和目录</span>
</span></span><span class="line"><span class="cl">rm -rf *.py C* database
</span></span></code></pre></td></tr></table>
</div>
</div><p>把上面内容拷贝到名为 run_phagcn的文件，注意修改~/biosoft/PhaGCN2.0为自己目录：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">vi run_phagcn
</span></span><span class="line"><span class="cl">chmod +x run_phagcn
</span></span><span class="line"><span class="cl"><span class="c1"># 链接到环境变量目录</span>
</span></span><span class="line"><span class="cl">ln -s ~/biosoft/PhaGCN2.0/run_phagcn ~/miniconda3/envs/phagcn2/bin/
</span></span></code></pre></td></tr></table>
</div>
</div><p>这样就可以在任意目录运行该程序并指定输出了。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/work/test/phagcn2_test
</span></span><span class="line"><span class="cl"><span class="c1"># 注意input不要是contigs.fa，不然会覆盖</span>
</span></span><span class="line"><span class="cl">cp ~/biosoft/PhaGCN2.0/contigs.fa ./test_contigs.fa
</span></span><span class="line"><span class="cl">run_phagcn -p ~/biosoft/PhaGCN2.0/ -i test_contigs.fa -o ~/work/test/phagcn2_test/output
</span></span></code></pre></td></tr></table>
</div>
</div><p>结果就是prediction.csv文件和network/phage_0.ntw文件，network/phage_0.ntw可以类似上述方法画一个网络图。</p>
<h2 id="references">References</h2>
<ol>
<li>Bin Jang, H., Bolduc, B., Zablocki, O. et al. Taxonomic assignment of uncultivated prokaryotic virus genomes is enabled by gene-sharing networks. Nat Biotechnol 37, 632–639 (2019). <a class="link" href="https://doi.org/10.1038/s41587-019-0100-8"  target="_blank" rel="noopener"
    >https://doi.org/10.1038/s41587-019-0100-8</a></li>
<li>Jing-Zhe Jiang, Wen-Guang Yuan, Jiayu Shang, Ying-Hui Shi, Li-Ling Yang, Min Liu, Peng Zhu, Tao Jin, Yanni Sun, Li-Hong Yuan, Virus classification for viral genomic fragments using PhaGCN2, Briefings in Bioinformatics, 2022;, bbac505, <a class="link" href="https://doi.org/10.1093/bib/bbac505"  target="_blank" rel="noopener"
    >https://doi.org/10.1093/bib/bbac505</a></li>
</ol>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/Cao-blog/tags/virus/">Virus</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/Cao-blog/p/virus-host/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/virus-host/images/IPEV.78086c186ccdcca626f22273564f07d4_hu_df4169bfbc6d92eb.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 宏基因组中病毒序列的宿主预测"
                        data-key="virus-host" 
                        data-hash="md5-eAhsGGzNzKYm8iJzVk8H1A==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">宏基因组中病毒序列的宿主预测</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/Cao-blog/p/env-virus/">
        
        

        <div class="article-details">
            <h2 class="article-title">近期的一些环境病毒组研究（2024.10）</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/virrep/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/virrep/images/workflow.96d665cd6f8105e8fd4ac804152698e2_hu_e61849de4631fb72.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post VirRep: 人类肠道微生物组识别病毒新方法"
                        data-key="virrep" 
                        data-hash="md5-ltZlzW&#43;BBej9SsgEFSaY4g==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">VirRep: 人类肠道微生物组识别病毒新方法</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/virus/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/virus/images/genomad.7524574c14ac60e91704fe391b0e1659_hu_cbc71fd967078866.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 从宏基因组中鉴定病毒序列"
                        data-key="virus" 
                        data-hash="md5-dSRXTBSsYOkXBP45Gw4WWQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">从宏基因组中鉴定病毒序列</h2>
        </div>
    </a>
</article>

            
                
<article class="has-image">
    <a href="/Cao-blog/p/25-nature-4/">
        
        
            <div class="article-image">
                <img src="/Cao-blog/p/25-nature-4/images/1-2.7f04df50146143db7f38e3c60bbebaa9_hu_4ebeb6bc608a4e31.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post 25年最新的一些Nature环境微生物研究文章（4）"
                        data-key="25-nature-4" 
                        data-hash="md5-fwTfUBRhQ9t/OOPGC766qQ==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">25年最新的一些Nature环境微生物研究文章（4）</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog2"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2025-6 - 
        
        2025 Cao&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: 17837411716@163.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/Cao-blog/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
